{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4ca7e700",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "\n",
    "from bio_mcp.cache.load import load_biotools, load_galaxy_singularity\n",
    "from bio_mcp.mcp.server import _search_containers, _describe_container"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03258a05",
   "metadata": {},
   "source": [
    "## Loading snapshots\n",
    "\n",
    "Galaxy containers from the CVMFS and tool metadata from e.g. biotools are indexed by lowercased tool names so they can be mapped together."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e4034b2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "bt = load_biotools(Path(\"/home/ubuntu/bio-mcp/data/toolfinder_meta.yaml\"))\n",
    "gx = load_galaxy_singularity(\n",
    "    Path(\"/home/ubuntu/bio-mcp/data/galaxy_singularity_cache.json\")\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "376b6ea0",
   "metadata": {},
   "source": [
    "## `search_containers`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dad7a5a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'found': True,\n",
       "  'tool_id': 'samtools',\n",
       "  'containers': [{'entry_name': 'samtools:1.3--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--1',\n",
       "    'size_bytes': 5083136,\n",
       "    'mtime': 1567233286.0},\n",
       "   {'entry_name': 'samtools:0.1.19--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--2',\n",
       "    'size_bytes': 8941568,\n",
       "    'mtime': 1567233201.0},\n",
       "   {'entry_name': 'samtools:0.1.15--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.15--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.15--0',\n",
       "    'size_bytes': 15171584,\n",
       "    'mtime': 1566672582.0},\n",
       "   {'entry_name': 'samtools:1.7--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.7--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--0',\n",
       "    'size_bytes': 24928256,\n",
       "    'mtime': 1567233243.0},\n",
       "   {'entry_name': 'samtools:1.5--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.5--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--2',\n",
       "    'size_bytes': 16613376,\n",
       "    'mtime': 1566672724.0},\n",
       "   {'entry_name': 'samtools:1.2.rglab--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.2.rglab--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2.rglab--0',\n",
       "    'size_bytes': 15081472,\n",
       "    'mtime': 1566672663.0},\n",
       "   {'entry_name': 'samtools:1.9--h91753b0_4',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h91753b0_4',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_4',\n",
       "    'size_bytes': 26267648,\n",
       "    'mtime': 1567233524.0},\n",
       "   {'entry_name': 'samtools:1.6--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--0',\n",
       "    'size_bytes': 24866816,\n",
       "    'mtime': 1566672741.0},\n",
       "   {'entry_name': 'samtools:1.4--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.4--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.4--0',\n",
       "    'size_bytes': 10760192,\n",
       "    'mtime': 1566672698.0},\n",
       "   {'entry_name': 'samtools:1.9--h8571acd_11',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h8571acd_11',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8571acd_11',\n",
       "    'size_bytes': 27271168,\n",
       "    'mtime': 1566672911.0},\n",
       "   {'entry_name': 'samtools:1.9--h8571acd_10',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h8571acd_10',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8571acd_10',\n",
       "    'size_bytes': 27267072,\n",
       "    'mtime': 1567233320.0},\n",
       "   {'entry_name': 'samtools:1.3--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--2',\n",
       "    'size_bytes': 6459392,\n",
       "    'mtime': 1566672674.0},\n",
       "   {'entry_name': 'samtools:0.1.12--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.12--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--0',\n",
       "    'size_bytes': 15060992,\n",
       "    'mtime': 1567233334.0},\n",
       "   {'entry_name': 'samtools:1.8--h46bd0b3_5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.8--h46bd0b3_5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--h46bd0b3_5',\n",
       "    'size_bytes': 23281664,\n",
       "    'mtime': 1566672843.0},\n",
       "   {'entry_name': 'samtools:1.9--h91753b0_8',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h91753b0_8',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_8',\n",
       "    'size_bytes': 26439680,\n",
       "    'mtime': 1566672944.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h94a8ba4_6',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h94a8ba4_6',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h94a8ba4_6',\n",
       "    'size_bytes': 19910656,\n",
       "    'mtime': 1566672611.0},\n",
       "   {'entry_name': 'samtools:1.6--he673b24_3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--he673b24_3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--he673b24_3',\n",
       "    'size_bytes': 27086848,\n",
       "    'mtime': 1566672790.0},\n",
       "   {'entry_name': 'samtools:1.9--h10a08f8_12',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h10a08f8_12',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h10a08f8_12',\n",
       "    'size_bytes': 30167040,\n",
       "    'mtime': 1567745001.0},\n",
       "   {'entry_name': 'samtools:1.6--h02bfda8_2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h02bfda8_2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h02bfda8_2',\n",
       "    'size_bytes': 26087424,\n",
       "    'mtime': 1566672757.0},\n",
       "   {'entry_name': 'samtools:1.1--hf89b575_4',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--hf89b575_4',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--hf89b575_4',\n",
       "    'size_bytes': 18546688,\n",
       "    'mtime': 1567233430.0},\n",
       "   {'entry_name': 'samtools:1.1--h94a8ba4_3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--h94a8ba4_3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h94a8ba4_3',\n",
       "    'size_bytes': 16830464,\n",
       "    'mtime': 1566672650.0},\n",
       "   {'entry_name': 'samtools:1.5--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.5--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--1',\n",
       "    'size_bytes': 16617472,\n",
       "    'mtime': 1567233548.0},\n",
       "   {'entry_name': 'samtools:1.7--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.7--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--2',\n",
       "    'size_bytes': 27439104,\n",
       "    'mtime': 1566672807.0},\n",
       "   {'entry_name': 'samtools:1.0--hdd8ed8b_2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.0--hdd8ed8b_2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--hdd8ed8b_2',\n",
       "    'size_bytes': 12582912,\n",
       "    'mtime': 1566672636.0},\n",
       "   {'entry_name': 'samtools:0.1.14--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.14--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.14--1',\n",
       "    'size_bytes': 15147008,\n",
       "    'mtime': 1566672568.0},\n",
       "   {'entry_name': 'samtools:1.9--h91753b0_3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h91753b0_3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_3',\n",
       "    'size_bytes': 26267648,\n",
       "    'mtime': 1567233567.0},\n",
       "   {'entry_name': 'samtools:0.1.19--3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--3',\n",
       "    'size_bytes': 17596416,\n",
       "    'mtime': 1566672596.0},\n",
       "   {'entry_name': 'samtools:1.9--h91753b0_5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h91753b0_5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_5',\n",
       "    'size_bytes': 26267648,\n",
       "    'mtime': 1567233504.0},\n",
       "   {'entry_name': 'samtools:1.9--h43f6869_9',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h43f6869_9',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h43f6869_9',\n",
       "    'size_bytes': 26439680,\n",
       "    'mtime': 1566672860.0},\n",
       "   {'entry_name': 'samtools:1.9--h91753b0_2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h91753b0_2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_2',\n",
       "    'size_bytes': 26267648,\n",
       "    'mtime': 1567233589.0},\n",
       "   {'entry_name': 'samtools:1.0--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.0--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--0',\n",
       "    'size_bytes': 4820992,\n",
       "    'mtime': 1567233382.0},\n",
       "   {'entry_name': 'samtools:1.9--h57cc563_6',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h57cc563_6',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h57cc563_6',\n",
       "    'size_bytes': 26267648,\n",
       "    'mtime': 1567233220.0},\n",
       "   {'entry_name': 'samtools:0.1.12--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.12--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--2',\n",
       "    'size_bytes': 15990784,\n",
       "    'mtime': 1566672541.0},\n",
       "   {'entry_name': 'samtools:0.1.13--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.13--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.13--0',\n",
       "    'size_bytes': 15122432,\n",
       "    'mtime': 1567233279.0},\n",
       "   {'entry_name': 'samtools:1.5--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.5--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--0',\n",
       "    'size_bytes': 11829248,\n",
       "    'mtime': 1567233535.0},\n",
       "   {'entry_name': 'samtools:1.3.1--3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--3',\n",
       "    'size_bytes': 10768384,\n",
       "    'mtime': 1567233404.0},\n",
       "   {'entry_name': 'samtools:0.1.14--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.14--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.14--0',\n",
       "    'size_bytes': 15138816,\n",
       "    'mtime': 1567233485.0},\n",
       "   {'entry_name': 'samtools:1.3.1--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--2',\n",
       "    'size_bytes': 11583488,\n",
       "    'mtime': 1567233415.0},\n",
       "   {'entry_name': 'samtools:1.9--h46bd0b3_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h46bd0b3_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h46bd0b3_0',\n",
       "    'size_bytes': 23293952,\n",
       "    'mtime': 1566672876.0},\n",
       "   {'entry_name': 'samtools:1.4.1--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.4.1--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.4.1--0',\n",
       "    'size_bytes': 10764288,\n",
       "    'mtime': 1566672709.0},\n",
       "   {'entry_name': 'samtools:0.1.13--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.13--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.13--1',\n",
       "    'size_bytes': 15130624,\n",
       "    'mtime': 1566672555.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h94a8ba4_5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h94a8ba4_5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h94a8ba4_5',\n",
       "    'size_bytes': 17698816,\n",
       "    'mtime': 1567233300.0},\n",
       "   {'entry_name': 'samtools:1.3.1--5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--5',\n",
       "    'size_bytes': 10108928,\n",
       "    'mtime': 1566672686.0},\n",
       "   {'entry_name': 'samtools:1.3.1--4',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--4',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--4',\n",
       "    'size_bytes': 10768384,\n",
       "    'mtime': 1567233393.0},\n",
       "   {'entry_name': 'samtools:1.0--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.0--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--1',\n",
       "    'size_bytes': 6144000,\n",
       "    'mtime': 1566672622.0},\n",
       "   {'entry_name': 'samtools:1.6--h4da6232_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h4da6232_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h4da6232_1',\n",
       "    'size_bytes': 21774336,\n",
       "    'mtime': 1566672772.0},\n",
       "   {'entry_name': 'samtools:1.6--h244ad75_4',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h244ad75_4',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h244ad75_4',\n",
       "    'size_bytes': 28876800,\n",
       "    'mtime': 1567233453.0},\n",
       "   {'entry_name': 'samtools:1.7--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.7--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--1',\n",
       "    'size_bytes': 25194496,\n",
       "    'mtime': 1567233266.0},\n",
       "   {'entry_name': 'samtools:1.8--4',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.8--4',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--4',\n",
       "    'size_bytes': 27516928,\n",
       "    'mtime': 1566672825.0},\n",
       "   {'entry_name': 'samtools:1.9--h8ee4bcc_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h8ee4bcc_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8ee4bcc_1',\n",
       "    'size_bytes': 25468928,\n",
       "    'mtime': 1566672928.0},\n",
       "   {'entry_name': 'samtools:1.9--h57cc563_7',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.9--h57cc563_7',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h57cc563_7',\n",
       "    'size_bytes': 26439680,\n",
       "    'mtime': 1566672894.0},\n",
       "   {'entry_name': 'samtools:1.8--3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.8--3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--3',\n",
       "    'size_bytes': 27443200,\n",
       "    'mtime': 1567233355.0},\n",
       "   {'entry_name': 'samtools:1.8--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.8--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--2',\n",
       "    'size_bytes': 27410432,\n",
       "    'mtime': 1567233376.0},\n",
       "   {'entry_name': 'samtools:0.1.12--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.12--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--1',\n",
       "    'size_bytes': 15982592,\n",
       "    'mtime': 1567233469.0},\n",
       "   {'entry_name': 'samtools:1.2-0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.2-0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2-0',\n",
       "    'size_bytes': 15081472,\n",
       "    'mtime': 1578458984.0},\n",
       "   {'entry_name': 'samtools:1.3--h0592bc0_3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h0592bc0_3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h0592bc0_3',\n",
       "    'size_bytes': 28184576,\n",
       "    'mtime': 1587102580.0},\n",
       "   {'entry_name': 'samtools:1.6--h244ad75_5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h244ad75_5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h244ad75_5',\n",
       "    'size_bytes': 28868608,\n",
       "    'mtime': 1572240316.0},\n",
       "   {'entry_name': 'samtools:1.10--h9402c20_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.10--h9402c20_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_0',\n",
       "    'size_bytes': 30883840,\n",
       "    'mtime': 1575780266.0},\n",
       "   {'entry_name': 'samtools:1.2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2',\n",
       "    'size_bytes': 4673567,\n",
       "    'mtime': 1580725809.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h0592bc0_6',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h0592bc0_6',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h0592bc0_6',\n",
       "    'size_bytes': 28028928,\n",
       "    'mtime': 1573620952.0},\n",
       "   {'entry_name': 'samtools:1.10--h9402c20_2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.10--h9402c20_2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_2',\n",
       "    'size_bytes': 30732288,\n",
       "    'mtime': 1577335041.0},\n",
       "   {'entry_name': 'samtools:1.10--h9402c20_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.10--h9402c20_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_1',\n",
       "    'size_bytes': 30748672,\n",
       "    'mtime': 1576385579.0},\n",
       "   {'entry_name': 'samtools:1.6--h9dace67_6',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h9dace67_6',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h9dace67_6',\n",
       "    'size_bytes': 29175808,\n",
       "    'mtime': 1593840974.0},\n",
       "   {'entry_name': 'samtools:0.1.19--hf89b575_7',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--hf89b575_7',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--hf89b575_7',\n",
       "    'size_bytes': 21643264,\n",
       "    'mtime': 1593840928.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h80b0bb3_7',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h80b0bb3_7',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h80b0bb3_7',\n",
       "    'size_bytes': 28295168,\n",
       "    'mtime': 1593840958.0},\n",
       "   {'entry_name': 'samtools:1.3--h80b0bb3_4',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h80b0bb3_4',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h80b0bb3_4',\n",
       "    'size_bytes': 28282880,\n",
       "    'mtime': 1593840943.0},\n",
       "   {'entry_name': 'samtools:1.11--h6270b1f_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.11--h6270b1f_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.11--h6270b1f_0',\n",
       "    'size_bytes': 30932992,\n",
       "    'mtime': 1601962088.0},\n",
       "   {'entry_name': 'samtools:1.1--hfb9b9cc_5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--hfb9b9cc_5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--hfb9b9cc_5',\n",
       "    'size_bytes': 18018304,\n",
       "    'mtime': 1598945503.0},\n",
       "   {'entry_name': 'samtools:0.1.19--hfb9b9cc_8',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--hfb9b9cc_8',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--hfb9b9cc_8',\n",
       "    'size_bytes': 21094400,\n",
       "    'mtime': 1598945491.0},\n",
       "   {'entry_name': 'samtools:0.1.18--hfb9b9cc_10',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.18--hfb9b9cc_10',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--hfb9b9cc_10',\n",
       "    'size_bytes': 13987840,\n",
       "    'mtime': 1606105551.0},\n",
       "   {'entry_name': 'samtools:1.10--h2e538c0_3',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.10--h2e538c0_3',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h2e538c0_3',\n",
       "    'size_bytes': 30826496,\n",
       "    'mtime': 1598945521.0},\n",
       "   {'entry_name': 'samtools:0.1.16',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.16',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16',\n",
       "    'size_bytes': 4284447,\n",
       "    'mtime': 1611774443.0},\n",
       "   {'entry_name': 'samtools:1.12--hd5e65b6_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.12--hd5e65b6_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.12--hd5e65b6_0',\n",
       "    'size_bytes': 31371264,\n",
       "    'mtime': 1616020387.0},\n",
       "   {'entry_name': 'samtools:1.6--hb116620_7',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--hb116620_7',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hb116620_7',\n",
       "    'size_bytes': 30642176,\n",
       "    'mtime': 1616939365.0},\n",
       "   {'entry_name': 'samtools:1.12--h9aed4be_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.12--h9aed4be_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.12--h9aed4be_1',\n",
       "    'size_bytes': 32980992,\n",
       "    'mtime': 1616939316.0},\n",
       "   {'entry_name': 'samtools:0.1.18--h270b39a_11',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.18--h270b39a_11',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h270b39a_11',\n",
       "    'size_bytes': 15671296,\n",
       "    'mtime': 1616939277.0},\n",
       "   {'entry_name': 'samtools:1.1--h270b39a_6',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--h270b39a_6',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h270b39a_6',\n",
       "    'size_bytes': 15421440,\n",
       "    'mtime': 1616939299.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h1b8c3c0_8',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h1b8c3c0_8',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h1b8c3c0_8',\n",
       "    'size_bytes': 29818880,\n",
       "    'mtime': 1616939348.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h270b39a_9',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h270b39a_9',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h270b39a_9',\n",
       "    'size_bytes': 18968576,\n",
       "    'mtime': 1616939289.0},\n",
       "   {'entry_name': 'samtools:1.3--h1b8c3c0_5',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h1b8c3c0_5',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h1b8c3c0_5',\n",
       "    'size_bytes': 29806592,\n",
       "    'mtime': 1616939332.0},\n",
       "   {'entry_name': 'samtools:1.13--h8c37831_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.13--h8c37831_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.13--h8c37831_0',\n",
       "    'size_bytes': 33050624,\n",
       "    'mtime': 1625893518.0},\n",
       "   {'entry_name': 'samtools:1.2--0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.2--0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--0',\n",
       "    'size_bytes': 15081472,\n",
       "    'mtime': 1630560056.0},\n",
       "   {'entry_name': 'samtools:1.14--hb421002_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.14--hb421002_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.14--hb421002_0',\n",
       "    'size_bytes': 25956352,\n",
       "    'mtime': 1635222951.0},\n",
       "   {'entry_name': 'samtools:1.15--h3843a85_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.15--h3843a85_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15--h3843a85_0',\n",
       "    'size_bytes': 25894912,\n",
       "    'mtime': 1645591369.0},\n",
       "   {'entry_name': 'samtools:1.6--h3f2fef4_8',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h3f2fef4_8',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h3f2fef4_8',\n",
       "    'size_bytes': 24383488,\n",
       "    'mtime': 1645591414.0},\n",
       "   {'entry_name': 'samtools:1.1--h20b1175_7',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--h20b1175_7',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h20b1175_7',\n",
       "    'size_bytes': 8036352,\n",
       "    'mtime': 1645591355.0},\n",
       "   {'entry_name': 'samtools:1.3--h41a57b0_6',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h41a57b0_6',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h41a57b0_6',\n",
       "    'size_bytes': 23552000,\n",
       "    'mtime': 1645591384.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h41a57b0_9',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h41a57b0_9',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h41a57b0_9',\n",
       "    'size_bytes': 23564288,\n",
       "    'mtime': 1645591398.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h9071d68_10',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h9071d68_10',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h9071d68_10',\n",
       "    'size_bytes': 22724608,\n",
       "    'mtime': 1645849601.0},\n",
       "   {'entry_name': 'samtools:1.6--hcd7b337_9',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--hcd7b337_9',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hcd7b337_9',\n",
       "    'size_bytes': 23539712,\n",
       "    'mtime': 1645849616.0},\n",
       "   {'entry_name': 'samtools:1.3--h9071d68_7',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h9071d68_7',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h9071d68_7',\n",
       "    'size_bytes': 22712320,\n",
       "    'mtime': 1645849587.0},\n",
       "   {'entry_name': 'samtools:1.15--h1170115_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.15--h1170115_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15--h1170115_1',\n",
       "    'size_bytes': 25931776,\n",
       "    'mtime': 1645849573.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h20b1175_10',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h20b1175_10',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h20b1175_10',\n",
       "    'size_bytes': 11431936,\n",
       "    'mtime': 1646281497.0},\n",
       "   {'entry_name': 'samtools:0.1.18--h20b1175_12',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.18--h20b1175_12',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h20b1175_12',\n",
       "    'size_bytes': 8241152,\n",
       "    'mtime': 1646281490.0},\n",
       "   {'entry_name': 'samtools:1.15.1--h1170115_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.15.1--h1170115_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15.1--h1170115_0',\n",
       "    'size_bytes': 22114304,\n",
       "    'mtime': 1649394526.0},\n",
       "   {'entry_name': 'samtools:1.15.1--h6899075_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.15.1--h6899075_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15.1--h6899075_1',\n",
       "    'size_bytes': 22642688,\n",
       "    'mtime': 1663479189.0},\n",
       "   {'entry_name': 'samtools:1.16.1--h6899075_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.16.1--h6899075_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_0',\n",
       "    'size_bytes': 22659072,\n",
       "    'mtime': 1664862117.0},\n",
       "   {'entry_name': 'samtools:1.16.1--h6899075_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.16.1--h6899075_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_1',\n",
       "    'size_bytes': 22667264,\n",
       "    'mtime': 1665466977.0},\n",
       "   {'entry_name': 'samtools:1.16.1--h00cdaf9_2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.16.1--h00cdaf9_2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h00cdaf9_2',\n",
       "    'size_bytes': 22605824,\n",
       "    'mtime': 1678241797.0},\n",
       "   {'entry_name': 'samtools:1.17--h00cdaf9_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.17--h00cdaf9_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--h00cdaf9_0',\n",
       "    'size_bytes': 22634496,\n",
       "    'mtime': 1680973648.0},\n",
       "   {'entry_name': 'samtools:1.1--h50ea8bc_8',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--h50ea8bc_8',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h50ea8bc_8',\n",
       "    'size_bytes': 8138752,\n",
       "    'mtime': 1684329537.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h50ea8bc_11',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h50ea8bc_11',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h50ea8bc_11',\n",
       "    'size_bytes': 11509760,\n",
       "    'mtime': 1684329532.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h0cf4675_11',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h0cf4675_11',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h0cf4675_11',\n",
       "    'size_bytes': 20373504,\n",
       "    'mtime': 1684329568.0},\n",
       "   {'entry_name': 'samtools:1.17--hd87286a_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.17--hd87286a_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--hd87286a_1',\n",
       "    'size_bytes': 24088576,\n",
       "    'mtime': 1684329547.0},\n",
       "   {'entry_name': 'samtools:0.1.18--h50ea8bc_13',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.18--h50ea8bc_13',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h50ea8bc_13',\n",
       "    'size_bytes': 8347648,\n",
       "    'mtime': 1684329525.0},\n",
       "   {'entry_name': 'samtools:1.3--h0cf4675_8',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h0cf4675_8',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h0cf4675_8',\n",
       "    'size_bytes': 20361216,\n",
       "    'mtime': 1684329558.0},\n",
       "   {'entry_name': 'samtools:1.6--hc3601fc_10',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--hc3601fc_10',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hc3601fc_10',\n",
       "    'size_bytes': 21295104,\n",
       "    'mtime': 1684329578.0},\n",
       "   {'entry_name': 'samtools:1.17--hd87286a_2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.17--hd87286a_2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--hd87286a_2',\n",
       "    'size_bytes': 24993792,\n",
       "    'mtime': 1695796914.0},\n",
       "   {'entry_name': 'samtools:1.18--hd87286a_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.18--hd87286a_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.18--hd87286a_0',\n",
       "    'size_bytes': 25001984,\n",
       "    'mtime': 1696489712.0},\n",
       "   {'entry_name': 'samtools:1.18--h50ea8bc_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.18--h50ea8bc_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.18--h50ea8bc_1',\n",
       "    'size_bytes': 25010176,\n",
       "    'mtime': 1696575535.0},\n",
       "   {'entry_name': 'samtools:1.19--h50ea8bc_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.19--h50ea8bc_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19--h50ea8bc_0',\n",
       "    'size_bytes': 25284608,\n",
       "    'mtime': 1702449613.0},\n",
       "   {'entry_name': 'samtools:1.19.1--h50ea8bc_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.19.1--h50ea8bc_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.1--h50ea8bc_0',\n",
       "    'size_bytes': 25288704,\n",
       "    'mtime': 1705973293.0},\n",
       "   {'entry_name': 'samtools:1.19.2--h50ea8bc_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.19.2--h50ea8bc_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.2--h50ea8bc_0',\n",
       "    'size_bytes': 25288704,\n",
       "    'mtime': 1706697298.0},\n",
       "   {'entry_name': 'samtools:1.19.2--h50ea8bc_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.19.2--h50ea8bc_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.2--h50ea8bc_1',\n",
       "    'size_bytes': 25300992,\n",
       "    'mtime': 1709688756.0},\n",
       "   {'entry_name': 'samtools:1.20--h50ea8bc_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.20--h50ea8bc_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.20--h50ea8bc_0',\n",
       "    'size_bytes': 25362432,\n",
       "    'mtime': 1713230880.0},\n",
       "   {'entry_name': 'samtools:1.20--h50ea8bc_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.20--h50ea8bc_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.20--h50ea8bc_1',\n",
       "    'size_bytes': 25563136,\n",
       "    'mtime': 1720661568.0},\n",
       "   {'entry_name': 'samtools:1.21--h50ea8bc_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.21--h50ea8bc_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.21--h50ea8bc_0',\n",
       "    'size_bytes': 25726976,\n",
       "    'mtime': 1726192714.0},\n",
       "   {'entry_name': 'samtools:0.1.18--h96c455f_14',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.18--h96c455f_14',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h96c455f_14',\n",
       "    'size_bytes': 8093696,\n",
       "    'mtime': 1733317525.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h96c455f_12',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h96c455f_12',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_12',\n",
       "    'size_bytes': 11276288,\n",
       "    'mtime': 1733317535.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h96c455f_13',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h96c455f_13',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_13',\n",
       "    'size_bytes': 11264000,\n",
       "    'mtime': 1733404440.0},\n",
       "   {'entry_name': 'samtools:1.3.1--h60f3df9_12',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3.1--h60f3df9_12',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h60f3df9_12',\n",
       "    'size_bytes': 21680128,\n",
       "    'mtime': 1734542986.0},\n",
       "   {'entry_name': 'samtools:1.3--h60f3df9_9',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.3--h60f3df9_9',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h60f3df9_9',\n",
       "    'size_bytes': 21671936,\n",
       "    'mtime': 1734542973.0},\n",
       "   {'entry_name': 'samtools:1.6--h5fe306e_11',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h5fe306e_11',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_11',\n",
       "    'size_bytes': 22810624,\n",
       "    'mtime': 1734543000.0},\n",
       "   {'entry_name': 'samtools:1.21--h96c455f_1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.21--h96c455f_1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.21--h96c455f_1',\n",
       "    'size_bytes': 25513984,\n",
       "    'mtime': 1734542957.0},\n",
       "   {'entry_name': 'samtools:1.1--h96c455f_9',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.1--h96c455f_9',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h96c455f_9',\n",
       "    'size_bytes': 7876608,\n",
       "    'mtime': 1734542942.0},\n",
       "   {'entry_name': 'samtools:1.6--h5fe306e_12',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h5fe306e_12',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_12',\n",
       "    'size_bytes': 23175168,\n",
       "    'mtime': 1747070874.0},\n",
       "   {'entry_name': 'samtools:1.22--h96c455f_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.22--h96c455f_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.22--h96c455f_0',\n",
       "    'size_bytes': 25956352,\n",
       "    'mtime': 1748917695.0},\n",
       "   {'entry_name': 'samtools:1.22.1--h96c455f_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.22.1--h96c455f_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.22.1--h96c455f_0',\n",
       "    'size_bytes': 25968640,\n",
       "    'mtime': 1752544645.0},\n",
       "   {'entry_name': 'samtools:0.1.16--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.16--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16--1',\n",
       "    'size_bytes': 9445376,\n",
       "    'mtime': 1752570120.0},\n",
       "   {'entry_name': 'samtools:1.2--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.2--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--1',\n",
       "    'size_bytes': 9842688,\n",
       "    'mtime': 1752578804.0},\n",
       "   {'entry_name': 'samtools:0.1.16--2',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.16--2',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16--2',\n",
       "    'size_bytes': 5496832,\n",
       "    'mtime': 1752856458.0},\n",
       "   {'entry_name': 'samtools:1.2--0--1',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.2--0--1',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--0--1',\n",
       "    'size_bytes': 16252928,\n",
       "    'mtime': 1752895373.0},\n",
       "   {'entry_name': 'samtools:0.1.19--h96c455f_14',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '0.1.19--h96c455f_14',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_14',\n",
       "    'size_bytes': 12816384,\n",
       "    'mtime': 1754098487.0},\n",
       "   {'entry_name': 'samtools:1.6--h5fe306e_13',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.6--h5fe306e_13',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_13',\n",
       "    'size_bytes': 22544384,\n",
       "    'mtime': 1758595447.0},\n",
       "   {'entry_name': 'samtools:1.23--h96c455f_0',\n",
       "    'tool_name': 'samtools',\n",
       "    'tag': '1.23--h96c455f_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.23--h96c455f_0',\n",
       "    'size_bytes': 29085696,\n",
       "    'mtime': 1765939608.0}]}]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_search_containers(\"samtools\", gx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa4624d5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'found': False, 'tool_id': 'fakebot', 'containers': []},\n",
       " {'found': True,\n",
       "  'tool_id': 'ggplot',\n",
       "  'containers': [{'entry_name': 'ggplot:0.6.8--py36_0',\n",
       "    'tool_name': 'ggplot',\n",
       "    'tag': '0.6.8--py36_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/ggplot:0.6.8--py36_0',\n",
       "    'size_bytes': 477306880,\n",
       "    'mtime': 1565920723.0}]}]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_search_containers([\"fakebot\", \"ggplot\"], gx)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41516606",
   "metadata": {},
   "source": [
    "## `describe_container()`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2f97f338",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'found': True,\n",
       " 'tool_id': 'ggplot',\n",
       " 'containers': [{'entry_name': 'ggplot:0.6.8--py36_0',\n",
       "   'tool_name': 'ggplot',\n",
       "   'tag': '0.6.8--py36_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/ggplot:0.6.8--py36_0',\n",
       "   'size_bytes': 477306880,\n",
       "   'mtime': 1565920723.0}],\n",
       " 'metadata': {},\n",
       " 'versions': [{'tag': '0.6.8--py36_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/ggplot:0.6.8--py36_0',\n",
       "   'size_bytes': 477306880,\n",
       "   'mtime': 1565920723.0,\n",
       "   'entry_name': 'ggplot:0.6.8--py36_0'}]}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_describe_container(gx, bt, \"ggplot\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fd78d312",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'found': True,\n",
       " 'tool_id': 'samtools',\n",
       " 'containers': [{'entry_name': 'samtools:1.3--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--1',\n",
       "   'size_bytes': 5083136,\n",
       "   'mtime': 1567233286.0},\n",
       "  {'entry_name': 'samtools:0.1.19--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--2',\n",
       "   'size_bytes': 8941568,\n",
       "   'mtime': 1567233201.0},\n",
       "  {'entry_name': 'samtools:0.1.15--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.15--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.15--0',\n",
       "   'size_bytes': 15171584,\n",
       "   'mtime': 1566672582.0},\n",
       "  {'entry_name': 'samtools:1.7--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.7--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--0',\n",
       "   'size_bytes': 24928256,\n",
       "   'mtime': 1567233243.0},\n",
       "  {'entry_name': 'samtools:1.5--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.5--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--2',\n",
       "   'size_bytes': 16613376,\n",
       "   'mtime': 1566672724.0},\n",
       "  {'entry_name': 'samtools:1.2.rglab--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.2.rglab--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2.rglab--0',\n",
       "   'size_bytes': 15081472,\n",
       "   'mtime': 1566672663.0},\n",
       "  {'entry_name': 'samtools:1.9--h91753b0_4',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h91753b0_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_4',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233524.0},\n",
       "  {'entry_name': 'samtools:1.6--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--0',\n",
       "   'size_bytes': 24866816,\n",
       "   'mtime': 1566672741.0},\n",
       "  {'entry_name': 'samtools:1.4--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.4--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.4--0',\n",
       "   'size_bytes': 10760192,\n",
       "   'mtime': 1566672698.0},\n",
       "  {'entry_name': 'samtools:1.9--h8571acd_11',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h8571acd_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8571acd_11',\n",
       "   'size_bytes': 27271168,\n",
       "   'mtime': 1566672911.0},\n",
       "  {'entry_name': 'samtools:1.9--h8571acd_10',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h8571acd_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8571acd_10',\n",
       "   'size_bytes': 27267072,\n",
       "   'mtime': 1567233320.0},\n",
       "  {'entry_name': 'samtools:1.3--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--2',\n",
       "   'size_bytes': 6459392,\n",
       "   'mtime': 1566672674.0},\n",
       "  {'entry_name': 'samtools:0.1.12--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.12--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--0',\n",
       "   'size_bytes': 15060992,\n",
       "   'mtime': 1567233334.0},\n",
       "  {'entry_name': 'samtools:1.8--h46bd0b3_5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.8--h46bd0b3_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--h46bd0b3_5',\n",
       "   'size_bytes': 23281664,\n",
       "   'mtime': 1566672843.0},\n",
       "  {'entry_name': 'samtools:1.9--h91753b0_8',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h91753b0_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_8',\n",
       "   'size_bytes': 26439680,\n",
       "   'mtime': 1566672944.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h94a8ba4_6',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h94a8ba4_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h94a8ba4_6',\n",
       "   'size_bytes': 19910656,\n",
       "   'mtime': 1566672611.0},\n",
       "  {'entry_name': 'samtools:1.6--he673b24_3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--he673b24_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--he673b24_3',\n",
       "   'size_bytes': 27086848,\n",
       "   'mtime': 1566672790.0},\n",
       "  {'entry_name': 'samtools:1.9--h10a08f8_12',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h10a08f8_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h10a08f8_12',\n",
       "   'size_bytes': 30167040,\n",
       "   'mtime': 1567745001.0},\n",
       "  {'entry_name': 'samtools:1.6--h02bfda8_2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h02bfda8_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h02bfda8_2',\n",
       "   'size_bytes': 26087424,\n",
       "   'mtime': 1566672757.0},\n",
       "  {'entry_name': 'samtools:1.1--hf89b575_4',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--hf89b575_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--hf89b575_4',\n",
       "   'size_bytes': 18546688,\n",
       "   'mtime': 1567233430.0},\n",
       "  {'entry_name': 'samtools:1.1--h94a8ba4_3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--h94a8ba4_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h94a8ba4_3',\n",
       "   'size_bytes': 16830464,\n",
       "   'mtime': 1566672650.0},\n",
       "  {'entry_name': 'samtools:1.5--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.5--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--1',\n",
       "   'size_bytes': 16617472,\n",
       "   'mtime': 1567233548.0},\n",
       "  {'entry_name': 'samtools:1.7--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.7--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--2',\n",
       "   'size_bytes': 27439104,\n",
       "   'mtime': 1566672807.0},\n",
       "  {'entry_name': 'samtools:1.0--hdd8ed8b_2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.0--hdd8ed8b_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--hdd8ed8b_2',\n",
       "   'size_bytes': 12582912,\n",
       "   'mtime': 1566672636.0},\n",
       "  {'entry_name': 'samtools:0.1.14--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.14--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.14--1',\n",
       "   'size_bytes': 15147008,\n",
       "   'mtime': 1566672568.0},\n",
       "  {'entry_name': 'samtools:1.9--h91753b0_3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h91753b0_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_3',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233567.0},\n",
       "  {'entry_name': 'samtools:0.1.19--3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--3',\n",
       "   'size_bytes': 17596416,\n",
       "   'mtime': 1566672596.0},\n",
       "  {'entry_name': 'samtools:1.9--h91753b0_5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h91753b0_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_5',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233504.0},\n",
       "  {'entry_name': 'samtools:1.9--h43f6869_9',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h43f6869_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h43f6869_9',\n",
       "   'size_bytes': 26439680,\n",
       "   'mtime': 1566672860.0},\n",
       "  {'entry_name': 'samtools:1.9--h91753b0_2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h91753b0_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_2',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233589.0},\n",
       "  {'entry_name': 'samtools:1.0--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.0--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--0',\n",
       "   'size_bytes': 4820992,\n",
       "   'mtime': 1567233382.0},\n",
       "  {'entry_name': 'samtools:1.9--h57cc563_6',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h57cc563_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h57cc563_6',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233220.0},\n",
       "  {'entry_name': 'samtools:0.1.12--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.12--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--2',\n",
       "   'size_bytes': 15990784,\n",
       "   'mtime': 1566672541.0},\n",
       "  {'entry_name': 'samtools:0.1.13--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.13--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.13--0',\n",
       "   'size_bytes': 15122432,\n",
       "   'mtime': 1567233279.0},\n",
       "  {'entry_name': 'samtools:1.5--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.5--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--0',\n",
       "   'size_bytes': 11829248,\n",
       "   'mtime': 1567233535.0},\n",
       "  {'entry_name': 'samtools:1.3.1--3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--3',\n",
       "   'size_bytes': 10768384,\n",
       "   'mtime': 1567233404.0},\n",
       "  {'entry_name': 'samtools:0.1.14--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.14--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.14--0',\n",
       "   'size_bytes': 15138816,\n",
       "   'mtime': 1567233485.0},\n",
       "  {'entry_name': 'samtools:1.3.1--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--2',\n",
       "   'size_bytes': 11583488,\n",
       "   'mtime': 1567233415.0},\n",
       "  {'entry_name': 'samtools:1.9--h46bd0b3_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h46bd0b3_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h46bd0b3_0',\n",
       "   'size_bytes': 23293952,\n",
       "   'mtime': 1566672876.0},\n",
       "  {'entry_name': 'samtools:1.4.1--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.4.1--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.4.1--0',\n",
       "   'size_bytes': 10764288,\n",
       "   'mtime': 1566672709.0},\n",
       "  {'entry_name': 'samtools:0.1.13--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.13--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.13--1',\n",
       "   'size_bytes': 15130624,\n",
       "   'mtime': 1566672555.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h94a8ba4_5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h94a8ba4_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h94a8ba4_5',\n",
       "   'size_bytes': 17698816,\n",
       "   'mtime': 1567233300.0},\n",
       "  {'entry_name': 'samtools:1.3.1--5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--5',\n",
       "   'size_bytes': 10108928,\n",
       "   'mtime': 1566672686.0},\n",
       "  {'entry_name': 'samtools:1.3.1--4',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--4',\n",
       "   'size_bytes': 10768384,\n",
       "   'mtime': 1567233393.0},\n",
       "  {'entry_name': 'samtools:1.0--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.0--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--1',\n",
       "   'size_bytes': 6144000,\n",
       "   'mtime': 1566672622.0},\n",
       "  {'entry_name': 'samtools:1.6--h4da6232_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h4da6232_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h4da6232_1',\n",
       "   'size_bytes': 21774336,\n",
       "   'mtime': 1566672772.0},\n",
       "  {'entry_name': 'samtools:1.6--h244ad75_4',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h244ad75_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h244ad75_4',\n",
       "   'size_bytes': 28876800,\n",
       "   'mtime': 1567233453.0},\n",
       "  {'entry_name': 'samtools:1.7--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.7--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--1',\n",
       "   'size_bytes': 25194496,\n",
       "   'mtime': 1567233266.0},\n",
       "  {'entry_name': 'samtools:1.8--4',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.8--4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--4',\n",
       "   'size_bytes': 27516928,\n",
       "   'mtime': 1566672825.0},\n",
       "  {'entry_name': 'samtools:1.9--h8ee4bcc_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h8ee4bcc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8ee4bcc_1',\n",
       "   'size_bytes': 25468928,\n",
       "   'mtime': 1566672928.0},\n",
       "  {'entry_name': 'samtools:1.9--h57cc563_7',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.9--h57cc563_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h57cc563_7',\n",
       "   'size_bytes': 26439680,\n",
       "   'mtime': 1566672894.0},\n",
       "  {'entry_name': 'samtools:1.8--3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.8--3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--3',\n",
       "   'size_bytes': 27443200,\n",
       "   'mtime': 1567233355.0},\n",
       "  {'entry_name': 'samtools:1.8--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.8--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--2',\n",
       "   'size_bytes': 27410432,\n",
       "   'mtime': 1567233376.0},\n",
       "  {'entry_name': 'samtools:0.1.12--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.12--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--1',\n",
       "   'size_bytes': 15982592,\n",
       "   'mtime': 1567233469.0},\n",
       "  {'entry_name': 'samtools:1.2-0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.2-0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2-0',\n",
       "   'size_bytes': 15081472,\n",
       "   'mtime': 1578458984.0},\n",
       "  {'entry_name': 'samtools:1.3--h0592bc0_3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h0592bc0_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h0592bc0_3',\n",
       "   'size_bytes': 28184576,\n",
       "   'mtime': 1587102580.0},\n",
       "  {'entry_name': 'samtools:1.6--h244ad75_5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h244ad75_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h244ad75_5',\n",
       "   'size_bytes': 28868608,\n",
       "   'mtime': 1572240316.0},\n",
       "  {'entry_name': 'samtools:1.10--h9402c20_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.10--h9402c20_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_0',\n",
       "   'size_bytes': 30883840,\n",
       "   'mtime': 1575780266.0},\n",
       "  {'entry_name': 'samtools:1.2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2',\n",
       "   'size_bytes': 4673567,\n",
       "   'mtime': 1580725809.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h0592bc0_6',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h0592bc0_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h0592bc0_6',\n",
       "   'size_bytes': 28028928,\n",
       "   'mtime': 1573620952.0},\n",
       "  {'entry_name': 'samtools:1.10--h9402c20_2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.10--h9402c20_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_2',\n",
       "   'size_bytes': 30732288,\n",
       "   'mtime': 1577335041.0},\n",
       "  {'entry_name': 'samtools:1.10--h9402c20_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.10--h9402c20_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_1',\n",
       "   'size_bytes': 30748672,\n",
       "   'mtime': 1576385579.0},\n",
       "  {'entry_name': 'samtools:1.6--h9dace67_6',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h9dace67_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h9dace67_6',\n",
       "   'size_bytes': 29175808,\n",
       "   'mtime': 1593840974.0},\n",
       "  {'entry_name': 'samtools:0.1.19--hf89b575_7',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--hf89b575_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--hf89b575_7',\n",
       "   'size_bytes': 21643264,\n",
       "   'mtime': 1593840928.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h80b0bb3_7',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h80b0bb3_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h80b0bb3_7',\n",
       "   'size_bytes': 28295168,\n",
       "   'mtime': 1593840958.0},\n",
       "  {'entry_name': 'samtools:1.3--h80b0bb3_4',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h80b0bb3_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h80b0bb3_4',\n",
       "   'size_bytes': 28282880,\n",
       "   'mtime': 1593840943.0},\n",
       "  {'entry_name': 'samtools:1.11--h6270b1f_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.11--h6270b1f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.11--h6270b1f_0',\n",
       "   'size_bytes': 30932992,\n",
       "   'mtime': 1601962088.0},\n",
       "  {'entry_name': 'samtools:1.1--hfb9b9cc_5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--hfb9b9cc_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--hfb9b9cc_5',\n",
       "   'size_bytes': 18018304,\n",
       "   'mtime': 1598945503.0},\n",
       "  {'entry_name': 'samtools:0.1.19--hfb9b9cc_8',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--hfb9b9cc_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--hfb9b9cc_8',\n",
       "   'size_bytes': 21094400,\n",
       "   'mtime': 1598945491.0},\n",
       "  {'entry_name': 'samtools:0.1.18--hfb9b9cc_10',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.18--hfb9b9cc_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--hfb9b9cc_10',\n",
       "   'size_bytes': 13987840,\n",
       "   'mtime': 1606105551.0},\n",
       "  {'entry_name': 'samtools:1.10--h2e538c0_3',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.10--h2e538c0_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h2e538c0_3',\n",
       "   'size_bytes': 30826496,\n",
       "   'mtime': 1598945521.0},\n",
       "  {'entry_name': 'samtools:0.1.16',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.16',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16',\n",
       "   'size_bytes': 4284447,\n",
       "   'mtime': 1611774443.0},\n",
       "  {'entry_name': 'samtools:1.12--hd5e65b6_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.12--hd5e65b6_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.12--hd5e65b6_0',\n",
       "   'size_bytes': 31371264,\n",
       "   'mtime': 1616020387.0},\n",
       "  {'entry_name': 'samtools:1.6--hb116620_7',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--hb116620_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hb116620_7',\n",
       "   'size_bytes': 30642176,\n",
       "   'mtime': 1616939365.0},\n",
       "  {'entry_name': 'samtools:1.12--h9aed4be_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.12--h9aed4be_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.12--h9aed4be_1',\n",
       "   'size_bytes': 32980992,\n",
       "   'mtime': 1616939316.0},\n",
       "  {'entry_name': 'samtools:0.1.18--h270b39a_11',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.18--h270b39a_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h270b39a_11',\n",
       "   'size_bytes': 15671296,\n",
       "   'mtime': 1616939277.0},\n",
       "  {'entry_name': 'samtools:1.1--h270b39a_6',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--h270b39a_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h270b39a_6',\n",
       "   'size_bytes': 15421440,\n",
       "   'mtime': 1616939299.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h1b8c3c0_8',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h1b8c3c0_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h1b8c3c0_8',\n",
       "   'size_bytes': 29818880,\n",
       "   'mtime': 1616939348.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h270b39a_9',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h270b39a_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h270b39a_9',\n",
       "   'size_bytes': 18968576,\n",
       "   'mtime': 1616939289.0},\n",
       "  {'entry_name': 'samtools:1.3--h1b8c3c0_5',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h1b8c3c0_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h1b8c3c0_5',\n",
       "   'size_bytes': 29806592,\n",
       "   'mtime': 1616939332.0},\n",
       "  {'entry_name': 'samtools:1.13--h8c37831_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.13--h8c37831_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.13--h8c37831_0',\n",
       "   'size_bytes': 33050624,\n",
       "   'mtime': 1625893518.0},\n",
       "  {'entry_name': 'samtools:1.2--0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.2--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--0',\n",
       "   'size_bytes': 15081472,\n",
       "   'mtime': 1630560056.0},\n",
       "  {'entry_name': 'samtools:1.14--hb421002_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.14--hb421002_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.14--hb421002_0',\n",
       "   'size_bytes': 25956352,\n",
       "   'mtime': 1635222951.0},\n",
       "  {'entry_name': 'samtools:1.15--h3843a85_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.15--h3843a85_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15--h3843a85_0',\n",
       "   'size_bytes': 25894912,\n",
       "   'mtime': 1645591369.0},\n",
       "  {'entry_name': 'samtools:1.6--h3f2fef4_8',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h3f2fef4_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h3f2fef4_8',\n",
       "   'size_bytes': 24383488,\n",
       "   'mtime': 1645591414.0},\n",
       "  {'entry_name': 'samtools:1.1--h20b1175_7',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--h20b1175_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h20b1175_7',\n",
       "   'size_bytes': 8036352,\n",
       "   'mtime': 1645591355.0},\n",
       "  {'entry_name': 'samtools:1.3--h41a57b0_6',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h41a57b0_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h41a57b0_6',\n",
       "   'size_bytes': 23552000,\n",
       "   'mtime': 1645591384.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h41a57b0_9',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h41a57b0_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h41a57b0_9',\n",
       "   'size_bytes': 23564288,\n",
       "   'mtime': 1645591398.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h9071d68_10',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h9071d68_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h9071d68_10',\n",
       "   'size_bytes': 22724608,\n",
       "   'mtime': 1645849601.0},\n",
       "  {'entry_name': 'samtools:1.6--hcd7b337_9',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--hcd7b337_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hcd7b337_9',\n",
       "   'size_bytes': 23539712,\n",
       "   'mtime': 1645849616.0},\n",
       "  {'entry_name': 'samtools:1.3--h9071d68_7',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h9071d68_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h9071d68_7',\n",
       "   'size_bytes': 22712320,\n",
       "   'mtime': 1645849587.0},\n",
       "  {'entry_name': 'samtools:1.15--h1170115_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.15--h1170115_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15--h1170115_1',\n",
       "   'size_bytes': 25931776,\n",
       "   'mtime': 1645849573.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h20b1175_10',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h20b1175_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h20b1175_10',\n",
       "   'size_bytes': 11431936,\n",
       "   'mtime': 1646281497.0},\n",
       "  {'entry_name': 'samtools:0.1.18--h20b1175_12',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.18--h20b1175_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h20b1175_12',\n",
       "   'size_bytes': 8241152,\n",
       "   'mtime': 1646281490.0},\n",
       "  {'entry_name': 'samtools:1.15.1--h1170115_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.15.1--h1170115_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15.1--h1170115_0',\n",
       "   'size_bytes': 22114304,\n",
       "   'mtime': 1649394526.0},\n",
       "  {'entry_name': 'samtools:1.15.1--h6899075_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.15.1--h6899075_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15.1--h6899075_1',\n",
       "   'size_bytes': 22642688,\n",
       "   'mtime': 1663479189.0},\n",
       "  {'entry_name': 'samtools:1.16.1--h6899075_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.16.1--h6899075_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_0',\n",
       "   'size_bytes': 22659072,\n",
       "   'mtime': 1664862117.0},\n",
       "  {'entry_name': 'samtools:1.16.1--h6899075_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.16.1--h6899075_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_1',\n",
       "   'size_bytes': 22667264,\n",
       "   'mtime': 1665466977.0},\n",
       "  {'entry_name': 'samtools:1.16.1--h00cdaf9_2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.16.1--h00cdaf9_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h00cdaf9_2',\n",
       "   'size_bytes': 22605824,\n",
       "   'mtime': 1678241797.0},\n",
       "  {'entry_name': 'samtools:1.17--h00cdaf9_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.17--h00cdaf9_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--h00cdaf9_0',\n",
       "   'size_bytes': 22634496,\n",
       "   'mtime': 1680973648.0},\n",
       "  {'entry_name': 'samtools:1.1--h50ea8bc_8',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--h50ea8bc_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h50ea8bc_8',\n",
       "   'size_bytes': 8138752,\n",
       "   'mtime': 1684329537.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h50ea8bc_11',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h50ea8bc_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h50ea8bc_11',\n",
       "   'size_bytes': 11509760,\n",
       "   'mtime': 1684329532.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h0cf4675_11',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h0cf4675_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h0cf4675_11',\n",
       "   'size_bytes': 20373504,\n",
       "   'mtime': 1684329568.0},\n",
       "  {'entry_name': 'samtools:1.17--hd87286a_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.17--hd87286a_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--hd87286a_1',\n",
       "   'size_bytes': 24088576,\n",
       "   'mtime': 1684329547.0},\n",
       "  {'entry_name': 'samtools:0.1.18--h50ea8bc_13',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.18--h50ea8bc_13',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h50ea8bc_13',\n",
       "   'size_bytes': 8347648,\n",
       "   'mtime': 1684329525.0},\n",
       "  {'entry_name': 'samtools:1.3--h0cf4675_8',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h0cf4675_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h0cf4675_8',\n",
       "   'size_bytes': 20361216,\n",
       "   'mtime': 1684329558.0},\n",
       "  {'entry_name': 'samtools:1.6--hc3601fc_10',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--hc3601fc_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hc3601fc_10',\n",
       "   'size_bytes': 21295104,\n",
       "   'mtime': 1684329578.0},\n",
       "  {'entry_name': 'samtools:1.17--hd87286a_2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.17--hd87286a_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--hd87286a_2',\n",
       "   'size_bytes': 24993792,\n",
       "   'mtime': 1695796914.0},\n",
       "  {'entry_name': 'samtools:1.18--hd87286a_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.18--hd87286a_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.18--hd87286a_0',\n",
       "   'size_bytes': 25001984,\n",
       "   'mtime': 1696489712.0},\n",
       "  {'entry_name': 'samtools:1.18--h50ea8bc_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.18--h50ea8bc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.18--h50ea8bc_1',\n",
       "   'size_bytes': 25010176,\n",
       "   'mtime': 1696575535.0},\n",
       "  {'entry_name': 'samtools:1.19--h50ea8bc_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.19--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19--h50ea8bc_0',\n",
       "   'size_bytes': 25284608,\n",
       "   'mtime': 1702449613.0},\n",
       "  {'entry_name': 'samtools:1.19.1--h50ea8bc_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.19.1--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.1--h50ea8bc_0',\n",
       "   'size_bytes': 25288704,\n",
       "   'mtime': 1705973293.0},\n",
       "  {'entry_name': 'samtools:1.19.2--h50ea8bc_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.19.2--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.2--h50ea8bc_0',\n",
       "   'size_bytes': 25288704,\n",
       "   'mtime': 1706697298.0},\n",
       "  {'entry_name': 'samtools:1.19.2--h50ea8bc_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.19.2--h50ea8bc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.2--h50ea8bc_1',\n",
       "   'size_bytes': 25300992,\n",
       "   'mtime': 1709688756.0},\n",
       "  {'entry_name': 'samtools:1.20--h50ea8bc_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.20--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.20--h50ea8bc_0',\n",
       "   'size_bytes': 25362432,\n",
       "   'mtime': 1713230880.0},\n",
       "  {'entry_name': 'samtools:1.20--h50ea8bc_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.20--h50ea8bc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.20--h50ea8bc_1',\n",
       "   'size_bytes': 25563136,\n",
       "   'mtime': 1720661568.0},\n",
       "  {'entry_name': 'samtools:1.21--h50ea8bc_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.21--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.21--h50ea8bc_0',\n",
       "   'size_bytes': 25726976,\n",
       "   'mtime': 1726192714.0},\n",
       "  {'entry_name': 'samtools:0.1.18--h96c455f_14',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.18--h96c455f_14',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h96c455f_14',\n",
       "   'size_bytes': 8093696,\n",
       "   'mtime': 1733317525.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h96c455f_12',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h96c455f_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_12',\n",
       "   'size_bytes': 11276288,\n",
       "   'mtime': 1733317535.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h96c455f_13',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h96c455f_13',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_13',\n",
       "   'size_bytes': 11264000,\n",
       "   'mtime': 1733404440.0},\n",
       "  {'entry_name': 'samtools:1.3.1--h60f3df9_12',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3.1--h60f3df9_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h60f3df9_12',\n",
       "   'size_bytes': 21680128,\n",
       "   'mtime': 1734542986.0},\n",
       "  {'entry_name': 'samtools:1.3--h60f3df9_9',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.3--h60f3df9_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h60f3df9_9',\n",
       "   'size_bytes': 21671936,\n",
       "   'mtime': 1734542973.0},\n",
       "  {'entry_name': 'samtools:1.6--h5fe306e_11',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h5fe306e_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_11',\n",
       "   'size_bytes': 22810624,\n",
       "   'mtime': 1734543000.0},\n",
       "  {'entry_name': 'samtools:1.21--h96c455f_1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.21--h96c455f_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.21--h96c455f_1',\n",
       "   'size_bytes': 25513984,\n",
       "   'mtime': 1734542957.0},\n",
       "  {'entry_name': 'samtools:1.1--h96c455f_9',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.1--h96c455f_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h96c455f_9',\n",
       "   'size_bytes': 7876608,\n",
       "   'mtime': 1734542942.0},\n",
       "  {'entry_name': 'samtools:1.6--h5fe306e_12',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h5fe306e_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_12',\n",
       "   'size_bytes': 23175168,\n",
       "   'mtime': 1747070874.0},\n",
       "  {'entry_name': 'samtools:1.22--h96c455f_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.22--h96c455f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.22--h96c455f_0',\n",
       "   'size_bytes': 25956352,\n",
       "   'mtime': 1748917695.0},\n",
       "  {'entry_name': 'samtools:1.22.1--h96c455f_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.22.1--h96c455f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.22.1--h96c455f_0',\n",
       "   'size_bytes': 25968640,\n",
       "   'mtime': 1752544645.0},\n",
       "  {'entry_name': 'samtools:0.1.16--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.16--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16--1',\n",
       "   'size_bytes': 9445376,\n",
       "   'mtime': 1752570120.0},\n",
       "  {'entry_name': 'samtools:1.2--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.2--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--1',\n",
       "   'size_bytes': 9842688,\n",
       "   'mtime': 1752578804.0},\n",
       "  {'entry_name': 'samtools:0.1.16--2',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.16--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16--2',\n",
       "   'size_bytes': 5496832,\n",
       "   'mtime': 1752856458.0},\n",
       "  {'entry_name': 'samtools:1.2--0--1',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.2--0--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--0--1',\n",
       "   'size_bytes': 16252928,\n",
       "   'mtime': 1752895373.0},\n",
       "  {'entry_name': 'samtools:0.1.19--h96c455f_14',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '0.1.19--h96c455f_14',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_14',\n",
       "   'size_bytes': 12816384,\n",
       "   'mtime': 1754098487.0},\n",
       "  {'entry_name': 'samtools:1.6--h5fe306e_13',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.6--h5fe306e_13',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_13',\n",
       "   'size_bytes': 22544384,\n",
       "   'mtime': 1758595447.0},\n",
       "  {'entry_name': 'samtools:1.23--h96c455f_0',\n",
       "   'tool_name': 'samtools',\n",
       "   'tag': '1.23--h96c455f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.23--h96c455f_0',\n",
       "   'size_bytes': 29085696,\n",
       "   'mtime': 1765939608.0}],\n",
       " 'metadata': {'biocontainers': 'samtools',\n",
       "  'biotools': 'samtools',\n",
       "  'bunya': ['1.13-gcc-10.3.0',\n",
       "   '1.13-gcc-11.3.0',\n",
       "   '1.16.1-gcc-11.3.0',\n",
       "   '1.18-gcc-12.3.0                  (D)'],\n",
       "  'description': 'SAMtools and BCFtools are widely used programs for processing and analysing high-throughput sequencing data. They include tools for file format conversion and manipulation, sorting, querying, statistics, variant calling, and effect analysis amongst other methods.',\n",
       "  'edam-inputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       "  'edam-operations': ['Data filtering',\n",
       "   'Visualisation',\n",
       "   'Data formatting',\n",
       "   'Indexing',\n",
       "   'Data parsing',\n",
       "   'Data sorting',\n",
       "   'Data editing'],\n",
       "  'edam-outputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       "  'edam-topics': ['Mapping',\n",
       "   'Sequence analysis',\n",
       "   'Sequencing',\n",
       "   'Rare diseases'],\n",
       "  'galaxy': [{'description': 'Samtools markdup: marks duplicate alignments',\n",
       "    'title': 'Samtools markdup 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_markdup%2Fsamtools_markdup%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Samtools view: - reformat, filter, or subsample SAM, BAM or CRAM',\n",
       "    'title': 'Samtools view 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_view%2Fsamtools_view%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Samtools split: BAM dataset on readgroups',\n",
       "    'title': 'Samtools split 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_split%2Fsamtools_split%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Slice: BAM by genomic regions',\n",
       "    'title': 'Slice 2.0.6',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_slice_bam%2Fsamtools_slice_bam%2F2.0.6'},\n",
       "   {'description': 'Samtools reheader: copy SAM/BAM header between datasets',\n",
       "    'title': 'Samtools reheader 2.0.6',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_reheader%2Fsamtools_reheader%2F2.0.6'},\n",
       "   {'description': 'Samtools phase: call and phase heterozygous SNPs',\n",
       "    'title': 'Samtools phase 2.0.2',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_phase%2Fsamtools_phase%2F2.0.2'},\n",
       "   {'description': 'Samtools mpileup: multi-way pileup of variants',\n",
       "    'title': 'Samtools mpileup 2.2.0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_mpileup%2Fsamtools_mpileup%2F2.2.0'},\n",
       "   {'description': 'Samtools merge: merge multiple sorted alignment files',\n",
       "    'title': 'Samtools merge 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_merge%2Fsamtools_merge%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Samtools idxstats: reports stats of the BAM index file',\n",
       "    'title': 'Samtools idxstats 2.0.8',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_idxstats%2Fsamtools_idxstats%2F2.0.8'},\n",
       "   {'description': 'Samtools fastx: extract FASTA or FASTQ from alignment files',\n",
       "    'title': 'Samtools fastx 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fastx%2Fsamtools_fastx%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Samtools depth: compute the depth at each position or region',\n",
       "    'title': 'Samtools depth 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_depth%2Fsamtools_depth%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Samtools coverage: Produces a histogram or table of coverage per chromosome',\n",
       "    'title': 'Samtools coverage 1.22+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_coverage%2Fsamtools_coverage%2F1.22%2Bgalaxy3'},\n",
       "   {'description': 'Samtools calmd: recalculate MD/NM tags',\n",
       "    'title': 'Samtools calmd 2.0.7',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_calmd%2Fsamtools_calmd%2F2.0.7'},\n",
       "   {'description': 'Samtools bedcov: calculate read depth for a set of genomic intervals',\n",
       "    'title': 'Samtools bedcov 2.0.7',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_bedcov%2Fsamtools_bedcov%2F2.0.7'},\n",
       "   {'description': 'BAM-to-SAM: convert BAM to SAM',\n",
       "    'title': 'BAM-to-SAM 2.0.7',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbam_to_sam%2Fbam_to_sam%2F2.0.7'},\n",
       "   {'description': 'Samtools stats: generate statistics for BAM dataset',\n",
       "    'title': 'Samtools stats 2.0.8',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_stats%2Fsamtools_stats%2F2.0.8'},\n",
       "   {'description': 'Samtools sort: order of storing aligned sequences',\n",
       "    'title': 'Samtools sort 2.0.8',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_sort%2Fsamtools_sort%2F2.0.8'},\n",
       "   {'description': 'SAM-to-BAM: convert SAM to BAM',\n",
       "    'title': 'SAM-to-BAM 2.1.5',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsam_to_bam%2Fsam_to_bam%2F2.1.5'},\n",
       "   {'description': 'Samtools flagstat: tabulate descriptive stats for BAM datset',\n",
       "    'title': 'Samtools flagstat 2.0.8',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_flagstat%2Fsamtools_flagstat%2F2.0.8'},\n",
       "   {'description': 'Samtools fixmate: fill mate coordinates, ISIZE and mate related flags',\n",
       "    'title': 'Samtools fixmate 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fixmate%2Fsamtools_fixmate%2F1.22%2Bgalaxy1'},\n",
       "   {'description': 'Filter SAM or BAM, output SAM or BAM: files on FLAG MAPQ RG LN or by region',\n",
       "    'title': 'Filter SAM or BAM, output SAM or BAM 1.8+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtool_filter2%2Fsamtool_filter2%2F1.8%2Bgalaxy1'},\n",
       "   {'description': 'RmDup: remove PCR duplicates',\n",
       "    'title': 'RmDup 2.0.1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_rmdup%2Fsamtools_rmdup%2F2.0.1'},\n",
       "   {'description': 'samtools BAM to CRAM: convert BAM alignments to CRAM format',\n",
       "    'title': 'samtools BAM to CRAM 1.22+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_bam_to_cram%2Fsamtools_bam_to_cram%2F1.22%2Bgalaxy1'}],\n",
       "  'homepage': 'http://www.htslib.org/',\n",
       "  'id': 'samtools',\n",
       "  'license': 'MIT',\n",
       "  'name': 'SAMTools',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': ['1.18', '1.19.2'],\n",
       "  'pawsey': ['1.15--h3843a85_0'],\n",
       "  'publications': [{'title': 'The Sequence Alignment/Map format and SAMtools',\n",
       "    'url': 'https://doi.org/10.1093/bioinformatics/btp352'},\n",
       "   {'title': 'Twelve years of SAMtools and BCFtools',\n",
       "    'url': 'https://doi.org/10.1093/gigascience/giab008'},\n",
       "   {'title': 'A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data',\n",
       "    'url': 'https://doi.org/10.1093/bioinformatics/btr509'}],\n",
       "  'resources': [{'title': nan, 'url': nan}]},\n",
       " 'versions': [{'tag': '1.3--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--1',\n",
       "   'size_bytes': 5083136,\n",
       "   'mtime': 1567233286.0,\n",
       "   'entry_name': 'samtools:1.3--1'},\n",
       "  {'tag': '0.1.19--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--2',\n",
       "   'size_bytes': 8941568,\n",
       "   'mtime': 1567233201.0,\n",
       "   'entry_name': 'samtools:0.1.19--2'},\n",
       "  {'tag': '0.1.15--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.15--0',\n",
       "   'size_bytes': 15171584,\n",
       "   'mtime': 1566672582.0,\n",
       "   'entry_name': 'samtools:0.1.15--0'},\n",
       "  {'tag': '1.7--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--0',\n",
       "   'size_bytes': 24928256,\n",
       "   'mtime': 1567233243.0,\n",
       "   'entry_name': 'samtools:1.7--0'},\n",
       "  {'tag': '1.5--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--2',\n",
       "   'size_bytes': 16613376,\n",
       "   'mtime': 1566672724.0,\n",
       "   'entry_name': 'samtools:1.5--2'},\n",
       "  {'tag': '1.2.rglab--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2.rglab--0',\n",
       "   'size_bytes': 15081472,\n",
       "   'mtime': 1566672663.0,\n",
       "   'entry_name': 'samtools:1.2.rglab--0'},\n",
       "  {'tag': '1.9--h91753b0_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_4',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233524.0,\n",
       "   'entry_name': 'samtools:1.9--h91753b0_4'},\n",
       "  {'tag': '1.6--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--0',\n",
       "   'size_bytes': 24866816,\n",
       "   'mtime': 1566672741.0,\n",
       "   'entry_name': 'samtools:1.6--0'},\n",
       "  {'tag': '1.4--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.4--0',\n",
       "   'size_bytes': 10760192,\n",
       "   'mtime': 1566672698.0,\n",
       "   'entry_name': 'samtools:1.4--0'},\n",
       "  {'tag': '1.9--h8571acd_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8571acd_11',\n",
       "   'size_bytes': 27271168,\n",
       "   'mtime': 1566672911.0,\n",
       "   'entry_name': 'samtools:1.9--h8571acd_11'},\n",
       "  {'tag': '1.9--h8571acd_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8571acd_10',\n",
       "   'size_bytes': 27267072,\n",
       "   'mtime': 1567233320.0,\n",
       "   'entry_name': 'samtools:1.9--h8571acd_10'},\n",
       "  {'tag': '1.3--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--2',\n",
       "   'size_bytes': 6459392,\n",
       "   'mtime': 1566672674.0,\n",
       "   'entry_name': 'samtools:1.3--2'},\n",
       "  {'tag': '0.1.12--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--0',\n",
       "   'size_bytes': 15060992,\n",
       "   'mtime': 1567233334.0,\n",
       "   'entry_name': 'samtools:0.1.12--0'},\n",
       "  {'tag': '1.8--h46bd0b3_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--h46bd0b3_5',\n",
       "   'size_bytes': 23281664,\n",
       "   'mtime': 1566672843.0,\n",
       "   'entry_name': 'samtools:1.8--h46bd0b3_5'},\n",
       "  {'tag': '1.9--h91753b0_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_8',\n",
       "   'size_bytes': 26439680,\n",
       "   'mtime': 1566672944.0,\n",
       "   'entry_name': 'samtools:1.9--h91753b0_8'},\n",
       "  {'tag': '0.1.19--h94a8ba4_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h94a8ba4_6',\n",
       "   'size_bytes': 19910656,\n",
       "   'mtime': 1566672611.0,\n",
       "   'entry_name': 'samtools:0.1.19--h94a8ba4_6'},\n",
       "  {'tag': '1.6--he673b24_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--he673b24_3',\n",
       "   'size_bytes': 27086848,\n",
       "   'mtime': 1566672790.0,\n",
       "   'entry_name': 'samtools:1.6--he673b24_3'},\n",
       "  {'tag': '1.9--h10a08f8_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h10a08f8_12',\n",
       "   'size_bytes': 30167040,\n",
       "   'mtime': 1567745001.0,\n",
       "   'entry_name': 'samtools:1.9--h10a08f8_12'},\n",
       "  {'tag': '1.6--h02bfda8_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h02bfda8_2',\n",
       "   'size_bytes': 26087424,\n",
       "   'mtime': 1566672757.0,\n",
       "   'entry_name': 'samtools:1.6--h02bfda8_2'},\n",
       "  {'tag': '1.1--hf89b575_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--hf89b575_4',\n",
       "   'size_bytes': 18546688,\n",
       "   'mtime': 1567233430.0,\n",
       "   'entry_name': 'samtools:1.1--hf89b575_4'},\n",
       "  {'tag': '1.1--h94a8ba4_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h94a8ba4_3',\n",
       "   'size_bytes': 16830464,\n",
       "   'mtime': 1566672650.0,\n",
       "   'entry_name': 'samtools:1.1--h94a8ba4_3'},\n",
       "  {'tag': '1.5--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--1',\n",
       "   'size_bytes': 16617472,\n",
       "   'mtime': 1567233548.0,\n",
       "   'entry_name': 'samtools:1.5--1'},\n",
       "  {'tag': '1.7--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--2',\n",
       "   'size_bytes': 27439104,\n",
       "   'mtime': 1566672807.0,\n",
       "   'entry_name': 'samtools:1.7--2'},\n",
       "  {'tag': '1.0--hdd8ed8b_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--hdd8ed8b_2',\n",
       "   'size_bytes': 12582912,\n",
       "   'mtime': 1566672636.0,\n",
       "   'entry_name': 'samtools:1.0--hdd8ed8b_2'},\n",
       "  {'tag': '0.1.14--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.14--1',\n",
       "   'size_bytes': 15147008,\n",
       "   'mtime': 1566672568.0,\n",
       "   'entry_name': 'samtools:0.1.14--1'},\n",
       "  {'tag': '1.9--h91753b0_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_3',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233567.0,\n",
       "   'entry_name': 'samtools:1.9--h91753b0_3'},\n",
       "  {'tag': '0.1.19--3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--3',\n",
       "   'size_bytes': 17596416,\n",
       "   'mtime': 1566672596.0,\n",
       "   'entry_name': 'samtools:0.1.19--3'},\n",
       "  {'tag': '1.9--h91753b0_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_5',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233504.0,\n",
       "   'entry_name': 'samtools:1.9--h91753b0_5'},\n",
       "  {'tag': '1.9--h43f6869_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h43f6869_9',\n",
       "   'size_bytes': 26439680,\n",
       "   'mtime': 1566672860.0,\n",
       "   'entry_name': 'samtools:1.9--h43f6869_9'},\n",
       "  {'tag': '1.9--h91753b0_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h91753b0_2',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233589.0,\n",
       "   'entry_name': 'samtools:1.9--h91753b0_2'},\n",
       "  {'tag': '1.0--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--0',\n",
       "   'size_bytes': 4820992,\n",
       "   'mtime': 1567233382.0,\n",
       "   'entry_name': 'samtools:1.0--0'},\n",
       "  {'tag': '1.9--h57cc563_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h57cc563_6',\n",
       "   'size_bytes': 26267648,\n",
       "   'mtime': 1567233220.0,\n",
       "   'entry_name': 'samtools:1.9--h57cc563_6'},\n",
       "  {'tag': '0.1.12--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--2',\n",
       "   'size_bytes': 15990784,\n",
       "   'mtime': 1566672541.0,\n",
       "   'entry_name': 'samtools:0.1.12--2'},\n",
       "  {'tag': '0.1.13--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.13--0',\n",
       "   'size_bytes': 15122432,\n",
       "   'mtime': 1567233279.0,\n",
       "   'entry_name': 'samtools:0.1.13--0'},\n",
       "  {'tag': '1.5--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.5--0',\n",
       "   'size_bytes': 11829248,\n",
       "   'mtime': 1567233535.0,\n",
       "   'entry_name': 'samtools:1.5--0'},\n",
       "  {'tag': '1.3.1--3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--3',\n",
       "   'size_bytes': 10768384,\n",
       "   'mtime': 1567233404.0,\n",
       "   'entry_name': 'samtools:1.3.1--3'},\n",
       "  {'tag': '0.1.14--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.14--0',\n",
       "   'size_bytes': 15138816,\n",
       "   'mtime': 1567233485.0,\n",
       "   'entry_name': 'samtools:0.1.14--0'},\n",
       "  {'tag': '1.3.1--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--2',\n",
       "   'size_bytes': 11583488,\n",
       "   'mtime': 1567233415.0,\n",
       "   'entry_name': 'samtools:1.3.1--2'},\n",
       "  {'tag': '1.9--h46bd0b3_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h46bd0b3_0',\n",
       "   'size_bytes': 23293952,\n",
       "   'mtime': 1566672876.0,\n",
       "   'entry_name': 'samtools:1.9--h46bd0b3_0'},\n",
       "  {'tag': '1.4.1--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.4.1--0',\n",
       "   'size_bytes': 10764288,\n",
       "   'mtime': 1566672709.0,\n",
       "   'entry_name': 'samtools:1.4.1--0'},\n",
       "  {'tag': '0.1.13--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.13--1',\n",
       "   'size_bytes': 15130624,\n",
       "   'mtime': 1566672555.0,\n",
       "   'entry_name': 'samtools:0.1.13--1'},\n",
       "  {'tag': '0.1.19--h94a8ba4_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h94a8ba4_5',\n",
       "   'size_bytes': 17698816,\n",
       "   'mtime': 1567233300.0,\n",
       "   'entry_name': 'samtools:0.1.19--h94a8ba4_5'},\n",
       "  {'tag': '1.3.1--5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--5',\n",
       "   'size_bytes': 10108928,\n",
       "   'mtime': 1566672686.0,\n",
       "   'entry_name': 'samtools:1.3.1--5'},\n",
       "  {'tag': '1.3.1--4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--4',\n",
       "   'size_bytes': 10768384,\n",
       "   'mtime': 1567233393.0,\n",
       "   'entry_name': 'samtools:1.3.1--4'},\n",
       "  {'tag': '1.0--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.0--1',\n",
       "   'size_bytes': 6144000,\n",
       "   'mtime': 1566672622.0,\n",
       "   'entry_name': 'samtools:1.0--1'},\n",
       "  {'tag': '1.6--h4da6232_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h4da6232_1',\n",
       "   'size_bytes': 21774336,\n",
       "   'mtime': 1566672772.0,\n",
       "   'entry_name': 'samtools:1.6--h4da6232_1'},\n",
       "  {'tag': '1.6--h244ad75_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h244ad75_4',\n",
       "   'size_bytes': 28876800,\n",
       "   'mtime': 1567233453.0,\n",
       "   'entry_name': 'samtools:1.6--h244ad75_4'},\n",
       "  {'tag': '1.7--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.7--1',\n",
       "   'size_bytes': 25194496,\n",
       "   'mtime': 1567233266.0,\n",
       "   'entry_name': 'samtools:1.7--1'},\n",
       "  {'tag': '1.8--4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--4',\n",
       "   'size_bytes': 27516928,\n",
       "   'mtime': 1566672825.0,\n",
       "   'entry_name': 'samtools:1.8--4'},\n",
       "  {'tag': '1.9--h8ee4bcc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h8ee4bcc_1',\n",
       "   'size_bytes': 25468928,\n",
       "   'mtime': 1566672928.0,\n",
       "   'entry_name': 'samtools:1.9--h8ee4bcc_1'},\n",
       "  {'tag': '1.9--h57cc563_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.9--h57cc563_7',\n",
       "   'size_bytes': 26439680,\n",
       "   'mtime': 1566672894.0,\n",
       "   'entry_name': 'samtools:1.9--h57cc563_7'},\n",
       "  {'tag': '1.8--3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--3',\n",
       "   'size_bytes': 27443200,\n",
       "   'mtime': 1567233355.0,\n",
       "   'entry_name': 'samtools:1.8--3'},\n",
       "  {'tag': '1.8--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.8--2',\n",
       "   'size_bytes': 27410432,\n",
       "   'mtime': 1567233376.0,\n",
       "   'entry_name': 'samtools:1.8--2'},\n",
       "  {'tag': '0.1.12--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.12--1',\n",
       "   'size_bytes': 15982592,\n",
       "   'mtime': 1567233469.0,\n",
       "   'entry_name': 'samtools:0.1.12--1'},\n",
       "  {'tag': '1.2-0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2-0',\n",
       "   'size_bytes': 15081472,\n",
       "   'mtime': 1578458984.0,\n",
       "   'entry_name': 'samtools:1.2-0'},\n",
       "  {'tag': '1.3--h0592bc0_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h0592bc0_3',\n",
       "   'size_bytes': 28184576,\n",
       "   'mtime': 1587102580.0,\n",
       "   'entry_name': 'samtools:1.3--h0592bc0_3'},\n",
       "  {'tag': '1.6--h244ad75_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h244ad75_5',\n",
       "   'size_bytes': 28868608,\n",
       "   'mtime': 1572240316.0,\n",
       "   'entry_name': 'samtools:1.6--h244ad75_5'},\n",
       "  {'tag': '1.10--h9402c20_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_0',\n",
       "   'size_bytes': 30883840,\n",
       "   'mtime': 1575780266.0,\n",
       "   'entry_name': 'samtools:1.10--h9402c20_0'},\n",
       "  {'tag': '1.2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2',\n",
       "   'size_bytes': 4673567,\n",
       "   'mtime': 1580725809.0,\n",
       "   'entry_name': 'samtools:1.2'},\n",
       "  {'tag': '1.3.1--h0592bc0_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h0592bc0_6',\n",
       "   'size_bytes': 28028928,\n",
       "   'mtime': 1573620952.0,\n",
       "   'entry_name': 'samtools:1.3.1--h0592bc0_6'},\n",
       "  {'tag': '1.10--h9402c20_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_2',\n",
       "   'size_bytes': 30732288,\n",
       "   'mtime': 1577335041.0,\n",
       "   'entry_name': 'samtools:1.10--h9402c20_2'},\n",
       "  {'tag': '1.10--h9402c20_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h9402c20_1',\n",
       "   'size_bytes': 30748672,\n",
       "   'mtime': 1576385579.0,\n",
       "   'entry_name': 'samtools:1.10--h9402c20_1'},\n",
       "  {'tag': '1.6--h9dace67_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h9dace67_6',\n",
       "   'size_bytes': 29175808,\n",
       "   'mtime': 1593840974.0,\n",
       "   'entry_name': 'samtools:1.6--h9dace67_6'},\n",
       "  {'tag': '0.1.19--hf89b575_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--hf89b575_7',\n",
       "   'size_bytes': 21643264,\n",
       "   'mtime': 1593840928.0,\n",
       "   'entry_name': 'samtools:0.1.19--hf89b575_7'},\n",
       "  {'tag': '1.3.1--h80b0bb3_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h80b0bb3_7',\n",
       "   'size_bytes': 28295168,\n",
       "   'mtime': 1593840958.0,\n",
       "   'entry_name': 'samtools:1.3.1--h80b0bb3_7'},\n",
       "  {'tag': '1.3--h80b0bb3_4',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h80b0bb3_4',\n",
       "   'size_bytes': 28282880,\n",
       "   'mtime': 1593840943.0,\n",
       "   'entry_name': 'samtools:1.3--h80b0bb3_4'},\n",
       "  {'tag': '1.11--h6270b1f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.11--h6270b1f_0',\n",
       "   'size_bytes': 30932992,\n",
       "   'mtime': 1601962088.0,\n",
       "   'entry_name': 'samtools:1.11--h6270b1f_0'},\n",
       "  {'tag': '1.1--hfb9b9cc_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--hfb9b9cc_5',\n",
       "   'size_bytes': 18018304,\n",
       "   'mtime': 1598945503.0,\n",
       "   'entry_name': 'samtools:1.1--hfb9b9cc_5'},\n",
       "  {'tag': '0.1.19--hfb9b9cc_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--hfb9b9cc_8',\n",
       "   'size_bytes': 21094400,\n",
       "   'mtime': 1598945491.0,\n",
       "   'entry_name': 'samtools:0.1.19--hfb9b9cc_8'},\n",
       "  {'tag': '0.1.18--hfb9b9cc_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--hfb9b9cc_10',\n",
       "   'size_bytes': 13987840,\n",
       "   'mtime': 1606105551.0,\n",
       "   'entry_name': 'samtools:0.1.18--hfb9b9cc_10'},\n",
       "  {'tag': '1.10--h2e538c0_3',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.10--h2e538c0_3',\n",
       "   'size_bytes': 30826496,\n",
       "   'mtime': 1598945521.0,\n",
       "   'entry_name': 'samtools:1.10--h2e538c0_3'},\n",
       "  {'tag': '0.1.16',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16',\n",
       "   'size_bytes': 4284447,\n",
       "   'mtime': 1611774443.0,\n",
       "   'entry_name': 'samtools:0.1.16'},\n",
       "  {'tag': '1.12--hd5e65b6_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.12--hd5e65b6_0',\n",
       "   'size_bytes': 31371264,\n",
       "   'mtime': 1616020387.0,\n",
       "   'entry_name': 'samtools:1.12--hd5e65b6_0'},\n",
       "  {'tag': '1.6--hb116620_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hb116620_7',\n",
       "   'size_bytes': 30642176,\n",
       "   'mtime': 1616939365.0,\n",
       "   'entry_name': 'samtools:1.6--hb116620_7'},\n",
       "  {'tag': '1.12--h9aed4be_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.12--h9aed4be_1',\n",
       "   'size_bytes': 32980992,\n",
       "   'mtime': 1616939316.0,\n",
       "   'entry_name': 'samtools:1.12--h9aed4be_1'},\n",
       "  {'tag': '0.1.18--h270b39a_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h270b39a_11',\n",
       "   'size_bytes': 15671296,\n",
       "   'mtime': 1616939277.0,\n",
       "   'entry_name': 'samtools:0.1.18--h270b39a_11'},\n",
       "  {'tag': '1.1--h270b39a_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h270b39a_6',\n",
       "   'size_bytes': 15421440,\n",
       "   'mtime': 1616939299.0,\n",
       "   'entry_name': 'samtools:1.1--h270b39a_6'},\n",
       "  {'tag': '1.3.1--h1b8c3c0_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h1b8c3c0_8',\n",
       "   'size_bytes': 29818880,\n",
       "   'mtime': 1616939348.0,\n",
       "   'entry_name': 'samtools:1.3.1--h1b8c3c0_8'},\n",
       "  {'tag': '0.1.19--h270b39a_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h270b39a_9',\n",
       "   'size_bytes': 18968576,\n",
       "   'mtime': 1616939289.0,\n",
       "   'entry_name': 'samtools:0.1.19--h270b39a_9'},\n",
       "  {'tag': '1.3--h1b8c3c0_5',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h1b8c3c0_5',\n",
       "   'size_bytes': 29806592,\n",
       "   'mtime': 1616939332.0,\n",
       "   'entry_name': 'samtools:1.3--h1b8c3c0_5'},\n",
       "  {'tag': '1.13--h8c37831_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.13--h8c37831_0',\n",
       "   'size_bytes': 33050624,\n",
       "   'mtime': 1625893518.0,\n",
       "   'entry_name': 'samtools:1.13--h8c37831_0'},\n",
       "  {'tag': '1.2--0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--0',\n",
       "   'size_bytes': 15081472,\n",
       "   'mtime': 1630560056.0,\n",
       "   'entry_name': 'samtools:1.2--0'},\n",
       "  {'tag': '1.14--hb421002_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.14--hb421002_0',\n",
       "   'size_bytes': 25956352,\n",
       "   'mtime': 1635222951.0,\n",
       "   'entry_name': 'samtools:1.14--hb421002_0'},\n",
       "  {'tag': '1.15--h3843a85_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15--h3843a85_0',\n",
       "   'size_bytes': 25894912,\n",
       "   'mtime': 1645591369.0,\n",
       "   'entry_name': 'samtools:1.15--h3843a85_0'},\n",
       "  {'tag': '1.6--h3f2fef4_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h3f2fef4_8',\n",
       "   'size_bytes': 24383488,\n",
       "   'mtime': 1645591414.0,\n",
       "   'entry_name': 'samtools:1.6--h3f2fef4_8'},\n",
       "  {'tag': '1.1--h20b1175_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h20b1175_7',\n",
       "   'size_bytes': 8036352,\n",
       "   'mtime': 1645591355.0,\n",
       "   'entry_name': 'samtools:1.1--h20b1175_7'},\n",
       "  {'tag': '1.3--h41a57b0_6',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h41a57b0_6',\n",
       "   'size_bytes': 23552000,\n",
       "   'mtime': 1645591384.0,\n",
       "   'entry_name': 'samtools:1.3--h41a57b0_6'},\n",
       "  {'tag': '1.3.1--h41a57b0_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h41a57b0_9',\n",
       "   'size_bytes': 23564288,\n",
       "   'mtime': 1645591398.0,\n",
       "   'entry_name': 'samtools:1.3.1--h41a57b0_9'},\n",
       "  {'tag': '1.3.1--h9071d68_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h9071d68_10',\n",
       "   'size_bytes': 22724608,\n",
       "   'mtime': 1645849601.0,\n",
       "   'entry_name': 'samtools:1.3.1--h9071d68_10'},\n",
       "  {'tag': '1.6--hcd7b337_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hcd7b337_9',\n",
       "   'size_bytes': 23539712,\n",
       "   'mtime': 1645849616.0,\n",
       "   'entry_name': 'samtools:1.6--hcd7b337_9'},\n",
       "  {'tag': '1.3--h9071d68_7',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h9071d68_7',\n",
       "   'size_bytes': 22712320,\n",
       "   'mtime': 1645849587.0,\n",
       "   'entry_name': 'samtools:1.3--h9071d68_7'},\n",
       "  {'tag': '1.15--h1170115_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15--h1170115_1',\n",
       "   'size_bytes': 25931776,\n",
       "   'mtime': 1645849573.0,\n",
       "   'entry_name': 'samtools:1.15--h1170115_1'},\n",
       "  {'tag': '0.1.19--h20b1175_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h20b1175_10',\n",
       "   'size_bytes': 11431936,\n",
       "   'mtime': 1646281497.0,\n",
       "   'entry_name': 'samtools:0.1.19--h20b1175_10'},\n",
       "  {'tag': '0.1.18--h20b1175_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h20b1175_12',\n",
       "   'size_bytes': 8241152,\n",
       "   'mtime': 1646281490.0,\n",
       "   'entry_name': 'samtools:0.1.18--h20b1175_12'},\n",
       "  {'tag': '1.15.1--h1170115_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15.1--h1170115_0',\n",
       "   'size_bytes': 22114304,\n",
       "   'mtime': 1649394526.0,\n",
       "   'entry_name': 'samtools:1.15.1--h1170115_0'},\n",
       "  {'tag': '1.15.1--h6899075_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.15.1--h6899075_1',\n",
       "   'size_bytes': 22642688,\n",
       "   'mtime': 1663479189.0,\n",
       "   'entry_name': 'samtools:1.15.1--h6899075_1'},\n",
       "  {'tag': '1.16.1--h6899075_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_0',\n",
       "   'size_bytes': 22659072,\n",
       "   'mtime': 1664862117.0,\n",
       "   'entry_name': 'samtools:1.16.1--h6899075_0'},\n",
       "  {'tag': '1.16.1--h6899075_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_1',\n",
       "   'size_bytes': 22667264,\n",
       "   'mtime': 1665466977.0,\n",
       "   'entry_name': 'samtools:1.16.1--h6899075_1'},\n",
       "  {'tag': '1.16.1--h00cdaf9_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h00cdaf9_2',\n",
       "   'size_bytes': 22605824,\n",
       "   'mtime': 1678241797.0,\n",
       "   'entry_name': 'samtools:1.16.1--h00cdaf9_2'},\n",
       "  {'tag': '1.17--h00cdaf9_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--h00cdaf9_0',\n",
       "   'size_bytes': 22634496,\n",
       "   'mtime': 1680973648.0,\n",
       "   'entry_name': 'samtools:1.17--h00cdaf9_0'},\n",
       "  {'tag': '1.1--h50ea8bc_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h50ea8bc_8',\n",
       "   'size_bytes': 8138752,\n",
       "   'mtime': 1684329537.0,\n",
       "   'entry_name': 'samtools:1.1--h50ea8bc_8'},\n",
       "  {'tag': '0.1.19--h50ea8bc_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h50ea8bc_11',\n",
       "   'size_bytes': 11509760,\n",
       "   'mtime': 1684329532.0,\n",
       "   'entry_name': 'samtools:0.1.19--h50ea8bc_11'},\n",
       "  {'tag': '1.3.1--h0cf4675_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h0cf4675_11',\n",
       "   'size_bytes': 20373504,\n",
       "   'mtime': 1684329568.0,\n",
       "   'entry_name': 'samtools:1.3.1--h0cf4675_11'},\n",
       "  {'tag': '1.17--hd87286a_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--hd87286a_1',\n",
       "   'size_bytes': 24088576,\n",
       "   'mtime': 1684329547.0,\n",
       "   'entry_name': 'samtools:1.17--hd87286a_1'},\n",
       "  {'tag': '0.1.18--h50ea8bc_13',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h50ea8bc_13',\n",
       "   'size_bytes': 8347648,\n",
       "   'mtime': 1684329525.0,\n",
       "   'entry_name': 'samtools:0.1.18--h50ea8bc_13'},\n",
       "  {'tag': '1.3--h0cf4675_8',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h0cf4675_8',\n",
       "   'size_bytes': 20361216,\n",
       "   'mtime': 1684329558.0,\n",
       "   'entry_name': 'samtools:1.3--h0cf4675_8'},\n",
       "  {'tag': '1.6--hc3601fc_10',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--hc3601fc_10',\n",
       "   'size_bytes': 21295104,\n",
       "   'mtime': 1684329578.0,\n",
       "   'entry_name': 'samtools:1.6--hc3601fc_10'},\n",
       "  {'tag': '1.17--hd87286a_2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.17--hd87286a_2',\n",
       "   'size_bytes': 24993792,\n",
       "   'mtime': 1695796914.0,\n",
       "   'entry_name': 'samtools:1.17--hd87286a_2'},\n",
       "  {'tag': '1.18--hd87286a_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.18--hd87286a_0',\n",
       "   'size_bytes': 25001984,\n",
       "   'mtime': 1696489712.0,\n",
       "   'entry_name': 'samtools:1.18--hd87286a_0'},\n",
       "  {'tag': '1.18--h50ea8bc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.18--h50ea8bc_1',\n",
       "   'size_bytes': 25010176,\n",
       "   'mtime': 1696575535.0,\n",
       "   'entry_name': 'samtools:1.18--h50ea8bc_1'},\n",
       "  {'tag': '1.19--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19--h50ea8bc_0',\n",
       "   'size_bytes': 25284608,\n",
       "   'mtime': 1702449613.0,\n",
       "   'entry_name': 'samtools:1.19--h50ea8bc_0'},\n",
       "  {'tag': '1.19.1--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.1--h50ea8bc_0',\n",
       "   'size_bytes': 25288704,\n",
       "   'mtime': 1705973293.0,\n",
       "   'entry_name': 'samtools:1.19.1--h50ea8bc_0'},\n",
       "  {'tag': '1.19.2--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.2--h50ea8bc_0',\n",
       "   'size_bytes': 25288704,\n",
       "   'mtime': 1706697298.0,\n",
       "   'entry_name': 'samtools:1.19.2--h50ea8bc_0'},\n",
       "  {'tag': '1.19.2--h50ea8bc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.19.2--h50ea8bc_1',\n",
       "   'size_bytes': 25300992,\n",
       "   'mtime': 1709688756.0,\n",
       "   'entry_name': 'samtools:1.19.2--h50ea8bc_1'},\n",
       "  {'tag': '1.20--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.20--h50ea8bc_0',\n",
       "   'size_bytes': 25362432,\n",
       "   'mtime': 1713230880.0,\n",
       "   'entry_name': 'samtools:1.20--h50ea8bc_0'},\n",
       "  {'tag': '1.20--h50ea8bc_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.20--h50ea8bc_1',\n",
       "   'size_bytes': 25563136,\n",
       "   'mtime': 1720661568.0,\n",
       "   'entry_name': 'samtools:1.20--h50ea8bc_1'},\n",
       "  {'tag': '1.21--h50ea8bc_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.21--h50ea8bc_0',\n",
       "   'size_bytes': 25726976,\n",
       "   'mtime': 1726192714.0,\n",
       "   'entry_name': 'samtools:1.21--h50ea8bc_0'},\n",
       "  {'tag': '0.1.18--h96c455f_14',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.18--h96c455f_14',\n",
       "   'size_bytes': 8093696,\n",
       "   'mtime': 1733317525.0,\n",
       "   'entry_name': 'samtools:0.1.18--h96c455f_14'},\n",
       "  {'tag': '0.1.19--h96c455f_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_12',\n",
       "   'size_bytes': 11276288,\n",
       "   'mtime': 1733317535.0,\n",
       "   'entry_name': 'samtools:0.1.19--h96c455f_12'},\n",
       "  {'tag': '0.1.19--h96c455f_13',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_13',\n",
       "   'size_bytes': 11264000,\n",
       "   'mtime': 1733404440.0,\n",
       "   'entry_name': 'samtools:0.1.19--h96c455f_13'},\n",
       "  {'tag': '1.3.1--h60f3df9_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3.1--h60f3df9_12',\n",
       "   'size_bytes': 21680128,\n",
       "   'mtime': 1734542986.0,\n",
       "   'entry_name': 'samtools:1.3.1--h60f3df9_12'},\n",
       "  {'tag': '1.3--h60f3df9_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.3--h60f3df9_9',\n",
       "   'size_bytes': 21671936,\n",
       "   'mtime': 1734542973.0,\n",
       "   'entry_name': 'samtools:1.3--h60f3df9_9'},\n",
       "  {'tag': '1.6--h5fe306e_11',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_11',\n",
       "   'size_bytes': 22810624,\n",
       "   'mtime': 1734543000.0,\n",
       "   'entry_name': 'samtools:1.6--h5fe306e_11'},\n",
       "  {'tag': '1.21--h96c455f_1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.21--h96c455f_1',\n",
       "   'size_bytes': 25513984,\n",
       "   'mtime': 1734542957.0,\n",
       "   'entry_name': 'samtools:1.21--h96c455f_1'},\n",
       "  {'tag': '1.1--h96c455f_9',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.1--h96c455f_9',\n",
       "   'size_bytes': 7876608,\n",
       "   'mtime': 1734542942.0,\n",
       "   'entry_name': 'samtools:1.1--h96c455f_9'},\n",
       "  {'tag': '1.6--h5fe306e_12',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_12',\n",
       "   'size_bytes': 23175168,\n",
       "   'mtime': 1747070874.0,\n",
       "   'entry_name': 'samtools:1.6--h5fe306e_12'},\n",
       "  {'tag': '1.22--h96c455f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.22--h96c455f_0',\n",
       "   'size_bytes': 25956352,\n",
       "   'mtime': 1748917695.0,\n",
       "   'entry_name': 'samtools:1.22--h96c455f_0'},\n",
       "  {'tag': '1.22.1--h96c455f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.22.1--h96c455f_0',\n",
       "   'size_bytes': 25968640,\n",
       "   'mtime': 1752544645.0,\n",
       "   'entry_name': 'samtools:1.22.1--h96c455f_0'},\n",
       "  {'tag': '0.1.16--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16--1',\n",
       "   'size_bytes': 9445376,\n",
       "   'mtime': 1752570120.0,\n",
       "   'entry_name': 'samtools:0.1.16--1'},\n",
       "  {'tag': '1.2--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--1',\n",
       "   'size_bytes': 9842688,\n",
       "   'mtime': 1752578804.0,\n",
       "   'entry_name': 'samtools:1.2--1'},\n",
       "  {'tag': '0.1.16--2',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.16--2',\n",
       "   'size_bytes': 5496832,\n",
       "   'mtime': 1752856458.0,\n",
       "   'entry_name': 'samtools:0.1.16--2'},\n",
       "  {'tag': '1.2--0--1',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.2--0--1',\n",
       "   'size_bytes': 16252928,\n",
       "   'mtime': 1752895373.0,\n",
       "   'entry_name': 'samtools:1.2--0--1'},\n",
       "  {'tag': '0.1.19--h96c455f_14',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:0.1.19--h96c455f_14',\n",
       "   'size_bytes': 12816384,\n",
       "   'mtime': 1754098487.0,\n",
       "   'entry_name': 'samtools:0.1.19--h96c455f_14'},\n",
       "  {'tag': '1.6--h5fe306e_13',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.6--h5fe306e_13',\n",
       "   'size_bytes': 22544384,\n",
       "   'mtime': 1758595447.0,\n",
       "   'entry_name': 'samtools:1.6--h5fe306e_13'},\n",
       "  {'tag': '1.23--h96c455f_0',\n",
       "   'path': '/cvmfs/singularity.galaxyproject.org/all/samtools:1.23--h96c455f_0',\n",
       "   'size_bytes': 29085696,\n",
       "   'mtime': 1765939608.0,\n",
       "   'entry_name': 'samtools:1.23--h96c455f_0'}]}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_describe_container(gx, bt, \"samtools\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fb888804",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'found': False,\n",
       "  'tool_id': 'fakebot',\n",
       "  'containers': [],\n",
       "  'metadata': {},\n",
       "  'versions': []},\n",
       " {'found': True,\n",
       "  'tool_id': 'ggplot',\n",
       "  'containers': [{'entry_name': 'ggplot:0.6.8--py36_0',\n",
       "    'tool_name': 'ggplot',\n",
       "    'tag': '0.6.8--py36_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/ggplot:0.6.8--py36_0',\n",
       "    'size_bytes': 477306880,\n",
       "    'mtime': 1565920723.0}],\n",
       "  'metadata': {},\n",
       "  'versions': [{'tag': '0.6.8--py36_0',\n",
       "    'path': '/cvmfs/singularity.galaxyproject.org/all/ggplot:0.6.8--py36_0',\n",
       "    'size_bytes': 477306880,\n",
       "    'mtime': 1565920723.0,\n",
       "    'entry_name': 'ggplot:0.6.8--py36_0'}]}]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_describe_container(gx, bt, [\"fakebot\", \"ggplot\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20594da4",
   "metadata": {},
   "source": [
    "## Create metadata embeddings\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "471aac57",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sentence_transformers.cross_encoder import CrossEncoder\n",
    "from bio_mcp.cache.embeddings import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5614ea19",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load a pretrained ST model\n",
    "model = CrossEncoder(\"cross-encoder/stsb-distilroberta-base\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ea24cac",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_queries = [\n",
    "    \"I have fastq files from a single cell sequencing experiment, what is the first step i can do?\",\n",
    "    \"How do I qc, clustering and integration of my pre-processed scrna data?\",\n",
    "    \"what tools are available for scrna analysis\",\n",
    "    \"How can I remove technical artifacts (e.g. identify and remove doublets from my data)\",\n",
    "    \"I have just generated scrna data, how do I analyse it for publication?\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dfef745",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create corpus of tool metadata\n",
    "meta_text = []\n",
    "for tool, _ in bt.items():\n",
    "    text = make_embedding_text(bt.get(tool))\n",
    "    meta_text.append(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80a20499",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b106c8a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rank all \n",
    "ranks = model.rank(example_queries[0], meta_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e5d9fa8",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(f\"Query: {example_queries[0]}\")\n",
    "for rank in ranks:\n",
    "    print(f\"{rank['score']:.2f}\\t{meta_text[rank['corpus_id']]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e689a88",
   "metadata": {},
   "outputs": [],
   "source": [
    "rank2 = model.rank(\"What should I use for phylogenetics\", meta_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26d3e139",
   "metadata": {},
   "outputs": [],
   "source": [
    "for rank in rank2:\n",
    "    print(f\"{rank['score']:.2f}\\t{meta_text[rank['corpus_id']]}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd256422",
   "metadata": {},
   "source": [
    "## Aliasing edams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "685306fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_yaml = load_biotools(Path(\"/home/ubuntu/cache/johan.yaml\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a7a6d8ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'biocontainers': 'samtools',\n",
       " 'biotools': 'samtools',\n",
       " 'bunya': ['1.13-gcc-10.3.0',\n",
       "  '1.13-gcc-11.3.0',\n",
       "  '1.16.1-gcc-11.3.0',\n",
       "  '1.18-gcc-12.3.0                  (D)'],\n",
       " 'description': 'SAMtools and BCFtools are widely used programs for processing and analysing high-throughput sequencing data. They include tools for file format conversion and manipulation, sorting, querying, statistics, variant calling, and effect analysis amongst other methods.',\n",
       " 'edam-inputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       " 'edam-operations': ['Data filtering',\n",
       "  'Visualisation',\n",
       "  'Data formatting',\n",
       "  'Indexing',\n",
       "  'Data parsing',\n",
       "  'Data sorting',\n",
       "  'Data editing'],\n",
       " 'edam-outputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       " 'edam-topics': ['Mapping',\n",
       "  'Sequence analysis',\n",
       "  'Sequencing',\n",
       "  'Rare diseases'],\n",
       " 'galaxy': [{'description': 'Samtools markdup: marks duplicate alignments',\n",
       "   'title': 'Samtools markdup 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_markdup%2Fsamtools_markdup%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools view: - reformat, filter, or subsample SAM, BAM or CRAM',\n",
       "   'title': 'Samtools view 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_view%2Fsamtools_view%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools split: BAM dataset on readgroups',\n",
       "   'title': 'Samtools split 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_split%2Fsamtools_split%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Slice: BAM by genomic regions',\n",
       "   'title': 'Slice 2.0.6',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_slice_bam%2Fsamtools_slice_bam%2F2.0.6'},\n",
       "  {'description': 'Samtools reheader: copy SAM/BAM header between datasets',\n",
       "   'title': 'Samtools reheader 2.0.6',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_reheader%2Fsamtools_reheader%2F2.0.6'},\n",
       "  {'description': 'Samtools phase: call and phase heterozygous SNPs',\n",
       "   'title': 'Samtools phase 2.0.2',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_phase%2Fsamtools_phase%2F2.0.2'},\n",
       "  {'description': 'Samtools mpileup: multi-way pileup of variants',\n",
       "   'title': 'Samtools mpileup 2.2.0',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_mpileup%2Fsamtools_mpileup%2F2.2.0'},\n",
       "  {'description': 'Samtools merge: merge multiple sorted alignment files',\n",
       "   'title': 'Samtools merge 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_merge%2Fsamtools_merge%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools idxstats: reports stats of the BAM index file',\n",
       "   'title': 'Samtools idxstats 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_idxstats%2Fsamtools_idxstats%2F2.0.8'},\n",
       "  {'description': 'Samtools fastx: extract FASTA or FASTQ from alignment files',\n",
       "   'title': 'Samtools fastx 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fastx%2Fsamtools_fastx%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools depth: compute the depth at each position or region',\n",
       "   'title': 'Samtools depth 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_depth%2Fsamtools_depth%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools coverage: Produces a histogram or table of coverage per chromosome',\n",
       "   'title': 'Samtools coverage 1.22+galaxy3',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_coverage%2Fsamtools_coverage%2F1.22%2Bgalaxy3'},\n",
       "  {'description': 'Samtools calmd: recalculate MD/NM tags',\n",
       "   'title': 'Samtools calmd 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_calmd%2Fsamtools_calmd%2F2.0.7'},\n",
       "  {'description': 'Samtools bedcov: calculate read depth for a set of genomic intervals',\n",
       "   'title': 'Samtools bedcov 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_bedcov%2Fsamtools_bedcov%2F2.0.7'},\n",
       "  {'description': 'BAM-to-SAM: convert BAM to SAM',\n",
       "   'title': 'BAM-to-SAM 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbam_to_sam%2Fbam_to_sam%2F2.0.7'},\n",
       "  {'description': 'Samtools stats: generate statistics for BAM dataset',\n",
       "   'title': 'Samtools stats 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_stats%2Fsamtools_stats%2F2.0.8'},\n",
       "  {'description': 'Samtools sort: order of storing aligned sequences',\n",
       "   'title': 'Samtools sort 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_sort%2Fsamtools_sort%2F2.0.8'},\n",
       "  {'description': 'SAM-to-BAM: convert SAM to BAM',\n",
       "   'title': 'SAM-to-BAM 2.1.5',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsam_to_bam%2Fsam_to_bam%2F2.1.5'},\n",
       "  {'description': 'Samtools flagstat: tabulate descriptive stats for BAM datset',\n",
       "   'title': 'Samtools flagstat 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_flagstat%2Fsamtools_flagstat%2F2.0.8'},\n",
       "  {'description': 'Samtools fixmate: fill mate coordinates, ISIZE and mate related flags',\n",
       "   'title': 'Samtools fixmate 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fixmate%2Fsamtools_fixmate%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Filter SAM or BAM, output SAM or BAM: files on FLAG MAPQ RG LN or by region',\n",
       "   'title': 'Filter SAM or BAM, output SAM or BAM 1.8+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtool_filter2%2Fsamtool_filter2%2F1.8%2Bgalaxy1'},\n",
       "  {'description': 'RmDup: remove PCR duplicates',\n",
       "   'title': 'RmDup 2.0.1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_rmdup%2Fsamtools_rmdup%2F2.0.1'},\n",
       "  {'description': 'samtools BAM to CRAM: convert BAM alignments to CRAM format',\n",
       "   'title': 'samtools BAM to CRAM 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_bam_to_cram%2Fsamtools_bam_to_cram%2F1.22%2Bgalaxy1'}],\n",
       " 'homepage': 'http://www.htslib.org/',\n",
       " 'id': 'samtools',\n",
       " 'license': 'MIT',\n",
       " 'name': 'SAMTools',\n",
       " 'nci-gadi': None,\n",
       " 'nci-if89': ['1.18', '1.19.2'],\n",
       " 'pawsey': ['1.15--h3843a85_0'],\n",
       " 'publications': [{'title': 'The Sequence Alignment/Map format and SAMtools',\n",
       "   'url': 'https://doi.org/10.1093/bioinformatics/btp352'},\n",
       "  {'title': 'Twelve years of SAMtools and BCFtools',\n",
       "   'url': 'https://doi.org/10.1093/gigascience/giab008'},\n",
       "  {'title': 'A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data',\n",
       "   'url': 'https://doi.org/10.1093/bioinformatics/btr509'}],\n",
       " 'resources': [{'title': nan, 'url': nan}]}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_yaml.get(\"samtools\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0ac64bb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "collected_data = []\n",
    "for tool_id, tool_data in all_yaml.items():\n",
    "    entry = {\n",
    "        \"id\": tool_id,\n",
    "        \"description\": tool_data.get(\"description\", \"\"),\n",
    "        \"edam_inputs\": tool_data.get(\"edam-inputs\", []),\n",
    "        \"edam_outputs\": tool_data.get(\"edam-outputs\", []),\n",
    "        \"edam_operations\": tool_data.get(\"edam-operations\", []),\n",
    "        \"edam_topics\": tool_data.get(\"edam-topics\", []),\n",
    "    }\n",
    "    collected_data.append(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "eea7fb7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'id': '3d-dna',\n",
       "  'description': '3D de novo assembly (3D-DNA) is a pipeline for de novo assembly using HiC.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Scaffolding', 'De-novo assembly', 'Genome assembly'],\n",
       "  'edam_topics': ['Sequence assembly', 'Mapping', 'DNA']},\n",
       " {'id': 'abricate',\n",
       "  'description': 'Mass screening of contigs for antimicrobial resistance or virulence genes.',\n",
       "  'edam_inputs': [{'formats': ['EMBL format', 'GenBank format', 'FASTA'],\n",
       "    'term': 'FASTA'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Antimicrobial resistance prediction'],\n",
       "  'edam_topics': ['Genomics', 'Microbiology']},\n",
       " {'id': 'abritamr',\n",
       "  'description': 'an AMR gene detection pipeline that runs AMRFinderPlus on a single (or list ) of given isolates and collates the results into a table, separating genes identified into functionally relevant groups.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Antimicrobial resistance prediction'],\n",
       "  'edam_topics': ['Microbiology',\n",
       "   'Public health and epidemiology',\n",
       "   'Infectious disease',\n",
       "   'Antimicrobial resistance']},\n",
       " {'id': 'abyss',\n",
       "  'description': 'De novo genome sequence assembler using short reads.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly', 'De-novo assembly', 'Scaffolding'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'agat',\n",
       "  'description': 'Another Gff Analysis Toolkit (AGAT)\\nSuite of tools to handle gene annotations in any GTF/GFF format.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Genome annotation'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'alignit',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'alphafold_2',\n",
       "  'description': \"AI-guided 3D structure prediction of novel proteins. AlphaFold uses neural networks to predict the tertiary (3D) structure of proteins. AlphaFold accepts an amino acid sequence as an input, which it will then will 'fold' into a 3D model.\",\n",
       "  'edam_inputs': [{'formats': ['FASTA-like (text)',\n",
       "     'plain text format (unformatted)'],\n",
       "    'term': 'plain text format (unformatted)'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Protein structure'}],\n",
       "  'edam_operations': ['Protein folding analysis',\n",
       "   'Backbone modelling',\n",
       "   'Loop modelling',\n",
       "   'Side chain modelling'],\n",
       "  'edam_topics': ['Protein folding, stability and design']},\n",
       " {'id': 'alphapickle',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'amber',\n",
       "  'description': 'The Assisted Model Building with Energy Refinement tool refers to two things: a set of molecular mechanical force fields for the simulation of biomolecules (which are in the public domain, and are used in a variety of simulation programs); and a package of molecular simulation programs which includes source code and demos.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Modelling and simulation'],\n",
       "  'edam_topics': ['Molecular modelling', 'Computational biology']},\n",
       " {'id': 'ambertools',\n",
       "  'description': 'Consists of several independently developed packages that work well by themselves, and with Amber (Assisted Model Building with Energy Refinement) itself. The suite can also be used to carry out complete (non-periodic) molecular dynamics simulations (using NAB), with generalized Born solvent models.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Molecular dynamics simulation'],\n",
       "  'edam_topics': ['Molecular modelling', 'Molecular dynamics']},\n",
       " {'id': 'anaconda',\n",
       "  'description': 'Software package specially developed for the study of genes primary structure. It uses gene sequences downloaded from public databases, as FASTA and GenBank, and it applies a set of statistical and visualization methods in different ways, to reveal information about codon context, codon usage, nucleotide repeats within open reading frames (ORFeome) and others.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Statistical inference',\n",
       "   'Nucleic acid sequence analysis',\n",
       "   'Coding region prediction'],\n",
       "  'edam_topics': ['DNA', 'Gene expression']},\n",
       " {'id': 'anndata',\n",
       "  'description': 'From https://anndata.readthedocs.io/en/latest/\\n\"Python package for handling annotated data matrices in memory and on disk, positioned between pandas and xarray.\"',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'annotatemyids',\n",
       "  'description': 'This tool can get annotation for a generic set of IDs, using the Bioconductor annotation data packages. Supported organisms are human, mouse, rat, fruit fly and zebrafish. The org.db packages that are used here are primarily based on mapping using Entrez Gene identifiers. More information on the annotation packages can be found at the Bioconductor website, for example, information on the human annotation package (org.Hs.eg.db) can be found here.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Annotation'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'antismash',\n",
       "  'description': 'Rapid genome-wide identification, annotation and analysis of secondary metabolite biosynthesis gene clusters in bacterial and fungal genomes. It integrates and cross-links with a large number of in silico secondary metabolite analysis tools that have been published earlier.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence clustering',\n",
       "   'Gene prediction',\n",
       "   'Differential gene expression analysis'],\n",
       "  'edam_topics': ['Molecular interactions, pathways and networks',\n",
       "   'Gene and protein families']},\n",
       " {'id': 'any2fasta',\n",
       "  'description': 'Convert various sequence formats to FASTA',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Formatting', 'Sequence conversion'],\n",
       "  'edam_topics': ['Data quality management', 'Genomics']},\n",
       " {'id': 'apollo',\n",
       "  'description': 'Apollo is a genome annotation viewer and editor. Apollo allows researchers to explore genomic annotations at many levels of detail, and to perform expert annotation curation, all in a graphical environment.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Editing', 'Sequence annotation', 'Genome annotation'],\n",
       "  'edam_topics': ['Data submission, annotation and curation',\n",
       "   'Gene transcripts',\n",
       "   'Genomics',\n",
       "   'Mapping',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'aragorn',\n",
       "  'description': 'ARAGORN detects tRNA, mtRNA info about tmRNA, and tmRNA genes',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['Functional, regulatory and non-coding RNA', 'Genomics']},\n",
       " {'id': 'aarnet_filesender',\n",
       "  'description': 'FileSender is an open source web application for sending files of any size, quickly and securely. FileSender is available to all institutions affiliated with the Australian Access Federation (AAF). You can use your institution login details to access and use FileSender.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'arriba',\n",
       "  'description': 'Arriba is a command-line tool to detect gene fusions from RNA-Seq data based on the STAR aligner. In addition to fusions, it can detect exon duplications/inversions and truncations of genes (i.e., breakpoints in introns and intergenic regions). Arriba is the winner of the DREAM SMC-RNA Challenge.',\n",
       "  'edam_inputs': [{'formats': ['BAM', 'SAM', 'CRAM', 'FASTQ-illumina'],\n",
       "    'term': 'FASTQ-illumina'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Structural variation detection', 'Split read mapping'],\n",
       "  'edam_topics': ['Gene transcripts',\n",
       "   'RNA splicing',\n",
       "   'DNA structural variation']},\n",
       " {'id': 'artic',\n",
       "  'description': 'A bioinformatics pipeline for working with virus sequencing data sequenced with nanopore',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-like format (text)'],\n",
       "    'term': 'FASTQ-like format (text)'},\n",
       "   {'formats': ['BED'], 'term': 'BED'}],\n",
       "  'edam_outputs': [{'formats': ['BAM'], 'term': 'BAM'},\n",
       "   {'formats': ['VCF'], 'term': 'VCF'}],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'augustus',\n",
       "  'description': 'AUGUSTUS is a eukaryotic gene prediction tool. It can integrate evidence, e.g. from RNA-Seq, ESTs, proteomics, but can also predict genes ab initio. The PPX extension to AUGUSTUS can take a protein sequence multiple sequence alignment as input to find new members of the family in a genome. It can be run through a web interface (see https://bio.tools/webaugustus), or downloaded and run locally.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['Gene transcripts', 'Gene and protein families']},\n",
       " {'id': 'autodock_vina',\n",
       "  'description': 'AutoDock Vina is a new open-source program for drug discovery, molecular docking and virtual screening, offering multi-core capability, high performance and enhanced accuracy and ease of use.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Molecular docking'],\n",
       "  'edam_topics': ['Drug discovery',\n",
       "   'Molecular modelling',\n",
       "   'Molecular docking']},\n",
       " {'id': 'b2btools',\n",
       "  'description': \"The bio2byte tools server (b2btools) offers the following single protein sequence based predictions:\\n- Backbone and sidechain dynamics (DynaMine)\\n- Helix, sheet, coil and polyproline-II propensity\\n- Early folding propensity (EFoldMine)\\n- Disorder (DisoMine)\\n- Beta-sheet aggregation (Agmata)\\n\\nIn addition, multiple sequence alignments (MSAs) can be uploaded to scan the 'biophysical limits' of a protein family as defined in the MSA\",\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Protein disorder prediction',\n",
       "   'Protein secondary structure prediction',\n",
       "   'Protein feature detection'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'bakta',\n",
       "  'description': 'Rapid & standardized annotation of bacterial genomes, MAGs & plasmids',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['GFF3'], 'term': 'GFF3'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['HMMER3'], 'term': 'HMMER3'}],\n",
       "  'edam_outputs': [{'formats': ['GenBank format',\n",
       "     'EMBL format',\n",
       "     'GFF3',\n",
       "     'JSON',\n",
       "     'TSV'],\n",
       "    'term': 'TSV'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['PNG', 'SVG'], 'term': 'SVG'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['Genomics', 'Sequence analysis', 'Bioinformatics']},\n",
       " {'id': 'bamtools',\n",
       "  'description': 'BamTools provides a fast, flexible C++ API & toolkit for reading, writing, and managing BAM files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Sequence alignment analysis'],\n",
       "  'edam_topics': ['Sequencing', 'Data management', 'Sequence analysis']},\n",
       " {'id': 'bamutil_diff',\n",
       "  'description': 'Bamutil provides a serie of programs to work on SAM/BAM files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Data management']},\n",
       " {'id': 'bandage',\n",
       "  'description': 'GUI program that allows users to interact with the assembly graphs made by de novo assemblers such as Velvet, SPAdes, MEGAHIT and others. It visualises assembly graphs, with connections, using graph layout algorithms.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence assembly'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly visualisation'],\n",
       "  'edam_topics': ['Genomics', 'Sequence assembly']},\n",
       " {'id': 'baredsc',\n",
       "  'description': 'The baredSC (Bayesian Approach to Retreive Expression Distribution of Single Cell) is a tool that uses a Monte-Carlo Markov Chain to estimate a confidence interval on the probability density function (PDF) of expression of one or two genes from single-cell RNA-seq data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data retrieval',\n",
       "   'Expression correlation analysis',\n",
       "   'Differential gene expression profiling'],\n",
       "  'edam_topics': ['RNA-Seq',\n",
       "   'Cytometry',\n",
       "   'Transcriptomics',\n",
       "   'Gene transcripts',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'barrnap',\n",
       "  'description': 'Predict the location of ribosomal RNA genes in genomes. It supports bacteria (5S,23S,16S), archaea (5S,5.8S,23S,16S), mitochondria (12S,16S) and eukaryotes (5S,5.8S,28S,18S).',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['GFF3'], 'term': 'GFF3'}],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['Genomics', 'Model organisms', 'Model organisms']},\n",
       " {'id': 'basespace',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bbmap',\n",
       "  'description': 'BBMap is a fast splice-aware aligner for RNA and DNA.  It is faster than almost all short-read aligners, yet retains unrivaled sensitivity and specificity, particularly for reads with many errors and indels.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['RNA-Seq analysis',\n",
       "   'Sequence trimming',\n",
       "   'Read mapping',\n",
       "   'Sequence contamination filtering',\n",
       "   'Read binning',\n",
       "   'Sequence alignment'],\n",
       "  'edam_topics': ['Sequencing',\n",
       "   'RNA splicing',\n",
       "   'Whole genome sequencing',\n",
       "   'Phylogenetics',\n",
       "   'Metagenomics',\n",
       "   'RNA-Seq']},\n",
       " {'id': 'bbtools',\n",
       "  'description': 'BBTools is a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data. BBTools can handle common sequencing file formats such as fastq, fasta, sam, scarf, fasta+qual, compressed or raw, with autodetection of quality encoding and interleaving. It is written in Java and works on any platform supporting Java, including Linux, MacOS, and Microsoft Windows and Linux; there are no dependencies other than Java (version 7 or higher). Program descriptions and options are shown when running the shell scripts with no parameters.\\n\\nBBTools is open source and free for unlimited use, and is used regularly by DOE JGI and other institutions around the world.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'bcbiogff',\n",
       "  'description': 'A tool for filling the gap created by genomic data processing/analysis by rebasing some analysis results against the parent features which were originally analysed.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Identifier (by type of data)'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Formatting'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'bcftools',\n",
       "  'description': 'BCFtools is a set of utilities that manipulate variant calls in the Variant Call Format (VCF) and its binary counterpart BCF. All commands work transparently with both VCFs and BCFs, both uncompressed and BGZF-compressed.',\n",
       "  'edam_inputs': [{'formats': ['BCF', 'VCF'], 'term': 'VCF'}],\n",
       "  'edam_outputs': [{'formats': ['BCF', 'VCF'], 'term': 'VCF'}],\n",
       "  'edam_operations': ['Variant calling', 'Data handling'],\n",
       "  'edam_topics': ['Genetic variation',\n",
       "   'DNA polymorphism',\n",
       "   'GWAS study',\n",
       "   'Genotyping experiment']},\n",
       " {'id': 'bcl2fastq2',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'beacon2',\n",
       "  'description': 'A global search engine for genetic mutations.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Service discovery',\n",
       "   'Database search',\n",
       "   'Genetic variation analysis'],\n",
       "  'edam_topics': ['Genetic variation',\n",
       "   'Population genetics',\n",
       "   'Data security',\n",
       "   'Rare diseases']},\n",
       " {'id': 'beagle',\n",
       "  'description': 'Beagle is a software package that performs genotype calling, genotype phasing, imputation of ungenotyped markers, and identity-by-descent segment detection.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phasing', 'Imputation'],\n",
       "  'edam_topics': ['Population genetics']},\n",
       " {'id': 'beagle-lib',\n",
       "  'description': 'BEAGLE is a high-performance library that can perform the core calculations at the heart of most Bayesian and Maximum Likelihood phylogenetics packages.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic inference',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Phylogenetics',\n",
       "   'Statistics and probability',\n",
       "   'Computer science']},\n",
       " {'id': 'beast',\n",
       "  'description': 'The Bayesian Evolutionary Analysis Sampling Trees is a cross-platform program for Bayesian analysis of molecular sequences using MCMC (Markov chain Monte Carlo). It is entirely orientated towards rooted, time-measured phylogenies inferred using strict or relaxed molecular clock models. It can be used as a method of reconstructing phylogenies but is also a framework for testing evolutionary hypotheses without conditioning on a single tree topology.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Multiple sequence alignment',\n",
       "   'Phylogenetic tree generation (from molecular sequences)',\n",
       "   'Phylogenetic tree generation (maximum likelihood and Bayesian methods)'],\n",
       "  'edam_topics': ['Phylogenetics', 'Sequence analysis']},\n",
       " {'id': 'beast2',\n",
       "  'description': 'Bayesian phylogenetic analysis of molecular sequences. It estimates rooted, time-measured phylogenies using strict or relaxed molecular clock models. It can be used as a method of reconstructing phylogenies but is also a framework for testing evolutionary hypotheses without conditioning on a single tree topology. It uses Markov chain Monte Carlo (MCMC) to average over tree space, so that each tree is weighted proportional to its posterior probability. It includes a graphical user-interface for setting up standard analyses and a suit of programs for analysing the results.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree reconstruction',\n",
       "   'Phylogenetic tree analysis (shape)'],\n",
       "  'edam_topics': ['Phylogenetics', 'Phylogenomics']},\n",
       " {'id': 'bed_to_protein_map',\n",
       "  'description': 'Convert a BED format file of the proteins from a proteomics search database into a tabular format for the Multiomics Visualization Platform (MVP).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Conversion'],\n",
       "  'edam_topics': ['Literature and language']},\n",
       " {'id': 'bedtools',\n",
       "  'description': 'BEDTools is an extensive suite of utilities for comparing genomic features in BED format.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Mapping'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'bellerophon',\n",
       "  'description': 'The Bellerophon pipeline, improving de novo transcriptomes and removing chimeras.\\n\\nBellerophon is a pipeline created to remove falsely assembled chimeric transcripts in de novo transcriptome assemblies.\\n\\nThe pipeline can be downloaded as a vragrant virtual machine (https://app.vagrantup.com/bellerophon/boxes/bellerophon). This is recommended, as it avoids backwards compatibility problems with TransRate',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly',\n",
       "   'Validation',\n",
       "   'de Novo sequencing',\n",
       "   'Transcriptome assembly',\n",
       "   'Chimera detection'],\n",
       "  'edam_topics': ['Transcriptomics',\n",
       "   'Sequence assembly',\n",
       "   'RNA-Seq',\n",
       "   'Workflows']},\n",
       " {'id': 'berokka',\n",
       "  'description': 'Trim, circularise, orient and filter long read bacterial genome assemblies',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Filtering', 'Sequence trimming'],\n",
       "  'edam_topics': ['Computer science']},\n",
       " {'id': 'bftools',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'binchicken',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bindcraft',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'binette',\n",
       "  'description': 'Binette is a fast and accurate binning refinement tool to constructs high quality MAGs from the output of multiple binning tools.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read binning'],\n",
       "  'edam_topics': ['Metagenomics']},\n",
       " {'id': 'bio-searchio-hmmer',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bio3d',\n",
       "  'description': 'Bio3D is an R package containing utilities for the analysis of protein structure, sequence and trajectory data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Essential dynamics',\n",
       "   'Molecular dynamics',\n",
       "   'Protein structure analysis',\n",
       "   'Principal component visualisation',\n",
       "   'Sequence trimming'],\n",
       "  'edam_topics': ['Structure analysis',\n",
       "   'Gene and protein families',\n",
       "   'Bioinformatics']},\n",
       " {'id': 'biobakery_workflows',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bio-db-hts',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bioformats2raw',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'biom-format',\n",
       "  'description': 'This package includes basic tools for reading biom-format files, accessing and subsetting data tables from a biom object, as well as limited support for writing a biom-object back to a biom-format file. The design of this API is intended to match the python API and other tools included with the biom-format project, but with a decidedly \"R flavor\" that should be familiar to R users. This includes S4 classes and methods, as well as extensions of common core functions/methods.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Formatting'],\n",
       "  'edam_topics': ['Laboratory information management', 'Sequence analysis']},\n",
       " {'id': 'bionano_solve',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bioperl',\n",
       "  'description': 'A collection of Perl modules that facilitate the development of Perl scripts for bioinformatics applications.  It provides software modules for many of the typical tasks of bioinformatics programming.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Service invocation'],\n",
       "  'edam_topics': ['Genomics', 'Software engineering', 'Data management']},\n",
       " {'id': 'biopython',\n",
       "  'description': 'Biopython is a set of freely available tools for biological computation written in Python by an international team of developers.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Deposition',\n",
       "   'Service invocation',\n",
       "   'Query and retrieval',\n",
       "   'Aggregation'],\n",
       "  'edam_topics': ['Bioinformatics',\n",
       "   'Data management',\n",
       "   'Software engineering',\n",
       "   'Computational biology']},\n",
       " {'id': 'biotransformer',\n",
       "  'description': 'BioTransformer is a freely available web server that supports accurate, rapid and comprehensive in silico metabolism prediction.',\n",
       "  'edam_inputs': [{'formats': ['SMILES'], 'term': 'SMILES'}],\n",
       "  'edam_outputs': [{'formats': ['SMILES'], 'term': 'SMILES'}],\n",
       "  'edam_operations': ['Metabolic pathway prediction',\n",
       "   'PTM site prediction',\n",
       "   'Natural product identification'],\n",
       "  'edam_topics': ['Small molecules',\n",
       "   'Endocrinology and metabolism',\n",
       "   'Metabolomics',\n",
       "   'Carbohydrates',\n",
       "   'NMR']},\n",
       " {'id': 'bismark',\n",
       "  'description': 'Bismark is a tool to map bisulfite treated sequencing reads and perform methylation calling in a quick and easy-to-use fashion.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Methylation analysis',\n",
       "   'Read mapping',\n",
       "   'Bisulfite mapping'],\n",
       "  'edam_topics': ['Epigenomics', 'Genomics', 'Epigenetics']},\n",
       " {'id': 'blast',\n",
       "  'description': 'A tool that finds regions of similarity between biological sequences. The program compares nucleotide or protein sequences to sequence databases and calculates the statistical significance.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['BLAST XML results format',\n",
       "     'BLAST XML v2 results format',\n",
       "     'TSV',\n",
       "     'CSV',\n",
       "     'ASN.1 sequence format',\n",
       "     'JSON',\n",
       "     'Binary format',\n",
       "     'Textual format'],\n",
       "    'term': 'Textual format'}],\n",
       "  'edam_operations': ['Sequence similarity search', 'Sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis', 'Bioinformatics']},\n",
       " {'id': 'blast+',\n",
       "  'description': 'A tool that finds regions of similarity between biological sequences. The program compares nucleotide or protein sequences to sequence databases and calculates the statistical significance.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['BLAST XML results format',\n",
       "     'BLAST XML v2 results format',\n",
       "     'TSV',\n",
       "     'CSV',\n",
       "     'ASN.1 sequence format',\n",
       "     'JSON',\n",
       "     'Binary format',\n",
       "     'Textual format'],\n",
       "    'term': 'Textual format'}],\n",
       "  'edam_operations': ['Sequence similarity search', 'Sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis', 'Bioinformatics']},\n",
       " {'id': 'blat',\n",
       "  'description': 'Fast, accurate spliced alignment of DNA sequences.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'Blockbuster',\n",
       "  'description': 'detect blocks of overlapping reads using a gaussian-distribution approach',\n",
       "  'edam_inputs': [{'formats': ['BED'], 'term': 'BED'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['RNA-Seq analysis'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'BlockClust',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'blue-crab',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bolt-lmm',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'boltz',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'boost',\n",
       "  'description': 'Boost is a set of libraries for the C++ programming language that provides support for tasks and structures such as linear algebra, pseudorandom number generation, multithreading, image processing, regular expressions, and unit testing.',\n",
       "  'edam_inputs': [{'formats': ['Workflow format'], 'term': 'Workflow format'},\n",
       "   {'formats': [], 'term': 'Image'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis'],\n",
       "  'edam_topics': ['Pure mathematics', 'Imaging', 'Software engineering']},\n",
       " {'id': 'bowtie',\n",
       "  'description': 'Bowtie is an ultrafast, memory-efficient short read aligner.',\n",
       "  'edam_inputs': [{'formats': ['HTML', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['FASTQ-solexa', 'FASTQ-sanger', 'FASTQ-illumina'],\n",
       "    'term': 'FASTQ-illumina'}],\n",
       "  'edam_outputs': [{'formats': ['SAM'], 'term': 'SAM'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['Mapping', 'Genomics', 'Sequencing', 'Sequence assembly']},\n",
       " {'id': 'bowtie2',\n",
       "  'description': 'Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s or 1,000s of characters, and particularly good at aligning to relatively long (e.g. mammalian) genomes. Bowtie 2 indexes the genome with an FM Index to keep its memory footprint small: for the human genome, its memory footprint is typically around 3.2 GB. Bowtie 2 supports gapped, local, and paired-end alignment modes.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['BAM', 'SAM'], 'term': 'SAM'}],\n",
       "  'edam_operations': ['Read mapping'],\n",
       "  'edam_topics': ['Mapping', 'Genomics', 'Mapping']},\n",
       " {'id': 'bracken',\n",
       "  'description': 'Bracken is a companion program to Kraken 1, KrakenUniq, or Kraken 2 While Kraken classifies reads to multiple levels in the taxonomic tree, Bracken allows estimation of abundance at a single level using those classifications (e.g. Bracken can estimate abundance of species within a sample).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Statistical calculation'],\n",
       "  'edam_topics': ['Metagenomics', 'Microbial ecology']},\n",
       " {'id': 'braker',\n",
       "  'description': 'Pipeline for unsupervised RNA-Seq-based genome annotation with GeneMark-ET and AUGUSTUS.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['RNA-Seq', 'RNA', 'Workflows']},\n",
       " {'id': 'breseq',\n",
       "  'description': 'Runs Breseq software on a set of fastq files.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['SAM'], 'term': 'SAM'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Polymorphism detection'],\n",
       "  'edam_topics': ['Sequencing', 'Sequence analysis', 'DNA mutation']},\n",
       " {'id': 'busco',\n",
       "  'description': 'Provides measures for quantitative assessment of genome assembly, gene set, and transcriptome completeness based on evolutionarily informed expectations of gene content from near-universal single-copy orthologs.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence report'}],\n",
       "  'edam_operations': ['Sequence assembly validation'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Genomics',\n",
       "   'Transcriptomics',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'buttery-eel',\n",
       "  'description': 'Accelerated nanopore basecalling with SLOW5 data format.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Base-calling', 'Formatting', 'Splitting'],\n",
       "  'edam_topics': ['Computer science',\n",
       "   'Protein expression',\n",
       "   'Workflows',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'bwa',\n",
       "  'description': 'Fast, accurate, memory-efficient aligner for short and long sequencing reads',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Genome index'}],\n",
       "  'edam_operations': ['Genome indexing'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'bwa-mem2',\n",
       "  'description': 'Bwa-mem2 is the next version of the bwa-mem algorithm in bwa. It produces alignment identical to bwa and is ~1.3-3.1x faster depending on the use-case,  dataset and the running machine.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'BWA-meth',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bwakit',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bx-python',\n",
       "  'description': 'Tools for manipulating biological data, particularly multiple sequence alignments.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'c3s',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'cactus',\n",
       "  'description': 'Cactus is a reference-free whole-genome multiple alignment program.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Multiple sequence alignment', 'Genome alignment'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'Sequence analysis',\n",
       "   'Phylogeny',\n",
       "   'Sequence assembly',\n",
       "   'Mapping',\n",
       "   'Phylogenetics']},\n",
       " {'id': 'camera',\n",
       "  'description': 'Annotation of peaklists generated by xcms, rule based annotation of isotopes and adducts, isotope validation, EIC correlation based tagging of unknown adducts and fragments.',\n",
       "  'edam_inputs': [{'formats': ['Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_outputs': [{'formats': ['DSV'], 'term': 'DSV'}],\n",
       "  'edam_operations': ['Annotation', 'Spectral analysis'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'cami-amber',\n",
       "  'description': 'AMBER is an evaluation package for the comparative assessment of genome reconstructions and taxonomic assignments from metagenome benchmark datasets. It provides performance metrics, results rankings, and comparative visualizations for assessing multiple programs or parameter effects. The provided metrics were used in the first community benchmarking challenge of the initiative for the Critical Assessment of Metagenomic Interpretation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read binning'],\n",
       "  'edam_topics': ['Metagenomics']},\n",
       " {'id': 'canu',\n",
       "  'description': 'De-novo assembly tool for long read chemistry like Nanopore data and PacBio data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'cap3',\n",
       "  'description': 'Web-based contig assembly.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'Mapping assembly',\n",
       "   'Sequence assembly',\n",
       "   'Scaffolding'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Sequencing',\n",
       "   'Sequence analysis',\n",
       "   'Genetic variation']},\n",
       " {'id': 'cardinal',\n",
       "  'description': 'Implements statistical and computational tools for analyzing mass spectrometry imaging datasets, including methods for efficient pre-processing, spatial segmentation, and classification.',\n",
       "  'edam_inputs': [{'formats': ['imzML metadata file', 'ibd'], 'term': 'ibd'}],\n",
       "  'edam_outputs': [{'formats': ['imzML metadata file', 'ibd'], 'term': 'ibd'}],\n",
       "  'edam_operations': ['Spectral analysis',\n",
       "   'Standardisation and normalisation',\n",
       "   'Mass spectra calibration'],\n",
       "  'edam_topics': ['Proteomics', 'Data visualisation']},\n",
       " {'id': 'cat_bins',\n",
       "  'description': 'Contig Annotation Tool (CAT) and Bin Annotation Tool (BAT) are pipelines for the taxonomic classification of long DNA sequences and metagenome assembled genomes (MAGs/bins) of both known and (highly) unknown microorganisms, as generated by contemporary metagenomics studies. The core algorithm of both programs involves gene calling, mapping of predicted ORFs against the nr protein database, and voting-based classification of the entire contig / MAG based on classification of the individual ORFs.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Taxonomic classification',\n",
       "   'Sequence assembly',\n",
       "   'Coding region prediction'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Metagenomic sequencing',\n",
       "   'Taxonomy',\n",
       "   'Sequence assembly']},\n",
       " {'id': 'cd-hit',\n",
       "  'description': 'Cluster a nucleotide dataset into representative sequences.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Sequence clustering'],\n",
       "  'edam_topics': ['Sequencing']},\n",
       " {'id': 'cebraem',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'celseq2',\n",
       "  'description': 'celseq2 is a Python framework for generating the UMI count matrix from CEL-Seq2 sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Demultiplexing', 'RNA-Seq analysis'],\n",
       "  'edam_topics': ['RNA-Seq', 'Transcriptomics']},\n",
       " {'id': 'cellbender',\n",
       "  'description': 'a deep generative model for unsupervised removal of background noise from scRNA-seq datasets.\\n\\nCellBender is a software package for eliminating technical artifacts from high-throughput single-cell RNA sequencing (scRNA-seq) data.\\n\\nWelcome to CellBenders documentation!  CellBender documentation.\\n\\nFree document hosting provided by Read the Docs.\\n\\nStephen J Fleming, John C Marioni, and Mehrtash Babadi. CellBender remove-background: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets. bioRxiv 791699; doi: https://doi.org/10.1101/791699.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Quantification',\n",
       "   'Data retrieval',\n",
       "   'Expression analysis'],\n",
       "  'edam_topics': ['Gene expression', 'RNA', 'RNA-Seq']},\n",
       " {'id': 'cellpose',\n",
       "  'description': 'Cellpose is a generalist algorithm for cellular segmentation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Neurite measurement',\n",
       "   'Microscope image visualisation',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Imaging',\n",
       "   'Cell biology',\n",
       "   'Machine learning',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'CellProfiler',\n",
       "  'description': 'Tool for quantifying data from biological images, particularly in high-throughput experiments.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Quantification', 'Image analysis', 'Parsing'],\n",
       "  'edam_topics': ['Imaging',\n",
       "   'Microarray experiment',\n",
       "   'Genotype and phenotype']},\n",
       " {'id': 'cellranger',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'cellsnp-lite',\n",
       "  'description': 'cellsnp-lite is an efficient tool for genotyping single cells. cellsnp-lite was initially designed to pileup the expressed alleles in single-cell or bulk RNA-seq data, which can be directly used for donor deconvolution in multiplexed single-cell RNA-seq data, particularly with vireo, which assigns cells to donors and detects doublets, even without genotyping reference. Now besides RNA-seq data, cellsnp-lite could also be applied on DNA-seq and ATAC-seq data, either in bulk or single-cell.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping', 'SNP detection'],\n",
       "  'edam_topics': ['Genotype and phenotype', 'DNA polymorphism']},\n",
       " {'id': 'cellxgene',\n",
       "  'description': 'cellxgene (pronounced \"cell-by-gene\") is an interactive data explorer for single-cell transcriptomics datasets, such as those coming from the Human Cell Atlas.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Visualisation', 'Essential dynamics'],\n",
       "  'edam_topics': ['Transcriptomics', 'Cell biology']},\n",
       " {'id': 'cemitool',\n",
       "  'description': 'It unifies the discovery and the analysis of coexpression gene modules in a fully automatic manner, while providing a user-friendly html report with high quality graphs. Our tool evaluates if modules contain genes that are over-represented by specific pathways or that are altered in a specific sample group. Additionally, CEMiTool is able to integrate transcriptomic data with interactome information, identifying the potential hubs on each network.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Enrichment analysis'],\n",
       "  'edam_topics': ['Gene expression',\n",
       "   'Transcriptomics',\n",
       "   'Microarray experiment']},\n",
       " {'id': 'chai-1',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'checkm',\n",
       "  'description': 'CheckM provides a set of tools for assessing the quality of genomes recovered from isolates, single cells, or metagenomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly validation',\n",
       "   'Validation',\n",
       "   'Sequence composition calculation',\n",
       "   'Sequencing quality control',\n",
       "   'Statistical calculation'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'Phylogenomics',\n",
       "   'Phylogenetics',\n",
       "   'Taxonomy',\n",
       "   'Metagenomics',\n",
       "   'Data quality management']},\n",
       " {'id': 'checkm-database',\n",
       "  'description': 'CheckM provides a set of tools for assessing the quality of genomes recovered from isolates, single cells, or metagenomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly validation',\n",
       "   'Validation',\n",
       "   'Sequence composition calculation',\n",
       "   'Sequencing quality control',\n",
       "   'Statistical calculation'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'Phylogenomics',\n",
       "   'Phylogenetics',\n",
       "   'Taxonomy',\n",
       "   'Metagenomics',\n",
       "   'Data quality management']},\n",
       " {'id': 'chembl',\n",
       "  'description': 'Database of bioactive compounds, their quantitative properties and bioactivities (binding constants, pharmacology and ADMET, etc). The data is abstracted and curated from the primary scientific literature.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Compound ID (ChEMBL)'}],\n",
       "  'edam_outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Data retrieval'],\n",
       "  'edam_topics': ['Small molecules', 'Chemical biology', 'Cheminformatics']},\n",
       " {'id': 'chemfp',\n",
       "  'description': 'Fast cheminformatics fingerprint search, at your fingertips.\\n\\nChemfp is a set of command-line tools and a Python library for fingerprint generation and high-performance similarity search.\\n\\nThere are two ways to try out chemfp. From the download page page you can request an evaluation copy of the most recent version of chemfp, or you can download an earlier version for no cost under the MIT license',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Database search',\n",
       "   'Standardisation and normalisation',\n",
       "   'Formatting'],\n",
       "  'edam_topics': ['Cheminformatics',\n",
       "   'Chemistry',\n",
       "   'Transcription factors and regulatory sites',\n",
       "   'Literature and language']},\n",
       " {'id': 'chemicaltoolbox',\n",
       "  'description': 'ChemicalToolbox is a publicly available web server for performing cheminformatics analysis. The ChemicalToolbox provides an intuitive, graphical interface for common tools for downloading, filtering, visualizing and simulating small molecules and proteins. The ChemicalToolbox is based on Galaxy, an open-source web-based platform which enables accessible and reproducible data analysis. There is already an active Galaxy cheminformatics community using and developing tools. Based on their work, we provide four example workflows which illustrate the capabilities of the ChemicalToolbox, covering assembly of a compound library, hole filling, protein-ligand docking, and construction of a quantitative structure-activity relationship (QSAR) model.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Protein-ligand docking',\n",
       "   'Molecular dynamics',\n",
       "   'Sequence assembly'],\n",
       "  'edam_topics': ['Cheminformatics',\n",
       "   'Molecular modelling',\n",
       "   'Compound libraries and screening',\n",
       "   'Workflows',\n",
       "   'Small molecules']},\n",
       " {'id': 'chipseeker',\n",
       "  'description': 'This package implements functions to retrieve the nearest genes around the peak, annotate the genomic region of the peak, statistical methods for estimating the significance of overlap among ChIP peak data sets, and incorporate GEO database to compare the own dataset with those deposited in database. The comparison can be used to infer cooperative regulation and thus can be used to generate hypotheses.',\n",
       "  'edam_inputs': [{'formats': ['BED', 'Sequence range format'],\n",
       "    'term': 'Sequence range format'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Sequence annotation',\n",
       "   'Visualisation',\n",
       "   'Gene-set enrichment analysis',\n",
       "   'Sequence feature comparison'],\n",
       "  'edam_topics': ['ChIP-seq', 'Data visualisation', 'Data mining']},\n",
       " {'id': 'chira',\n",
       "  'description': 'ChiRA is a tool suite to analyze RNA-RNA interactome experimental data such as CLASH, CLEAR-CLIP, PARIS, SPLASH, etc.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['RNA',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'Functional, regulatory and non-coding RNA']},\n",
       " {'id': 'chrombpnet',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'chromeister',\n",
       "  'description': 'An ultra fast, heuristic approach to detect conserved signals in extremely large pairwise genome comparisons (dotplot).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Dot plot plotting'],\n",
       "  'edam_topics': ['Comparative genomics', 'Sequence analysis']},\n",
       " {'id': 'circexplorer',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'circlator',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'circos',\n",
       "  'description': 'Circos is tool for visualizing data in a circular format. It was developed for genomic data but can work for many other kinds of data as well.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Image format'], 'term': 'Image format'}],\n",
       "  'edam_operations': ['Visualisation'],\n",
       "  'edam_topics': ['Comparative genomics', 'Data visualisation']},\n",
       " {'id': 'CITE-seq-Count',\n",
       "  'description': 'Tool for counting antibody TAGS from a CITE-seq and/or cell hashing experiment.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_outputs': [{'formats': ['Matrix format'], 'term': 'Matrix format'},\n",
       "   {'formats': ['YAML'], 'term': 'YAML'}],\n",
       "  'edam_operations': ['RNA-Seq quantification'],\n",
       "  'edam_topics': ['Transcriptomics', 'Immunoproteins and antigens']},\n",
       " {'id': 'clair3',\n",
       "  'description': 'Clair3 is a germline small variant caller for long-reads. Clair3 makes the best of two major method categories: pileup calling handles most variant candidates with speed, and full-alignment tackles complicated candidates to maximize precision and recall. Clair3 runs fast and has superior performance, especially at lower coverage. Clair3 is simple and modular for easy deployment and integration.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling'],\n",
       "  'edam_topics': ['Molecular genetics']},\n",
       " {'id': 'clifinder',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'climate_stripes',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'clinker',\n",
       "  'description': 'Automatic generation of gene cluster comparison figures.\\n\\nGene cluster comparison figure generator.\\n\\nA d3 chart for generating gene cluster comparison figures.\\n\\nclinker is a pipeline for easily generating publication-quality gene cluster comparison figures.\\n\\nGiven a set of GenBank files, clinker will automatically extract protein translations, perform global alignments between sequences in each cluster, determine the optimal display order based on cluster similarity, and generate an interactive visualisation (using clustermap.js) that can be extensively tweaked before being exported as an SVG file.\\n\\nclustermap.js is an interactive, reusable d3 chart designed to visualise homology between multiple gene clusters.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence clustering',\n",
       "   'Sequence cluster visualisation',\n",
       "   'Dendrogram visualisation'],\n",
       "  'edam_topics': ['Molecular interactions, pathways and networks',\n",
       "   'Sequence analysis',\n",
       "   'Workflows',\n",
       "   'Imaging']},\n",
       " {'id': 'clipkit',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'clustalo',\n",
       "  'description': 'Multiple sequence alignment software. The name is occassionally spelled as ClustalOmega, Clustal , Clustal, Clustal O, ClustalO.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Multiple sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'clustalw',\n",
       "  'description': 'Multiple sequence alignment software. Old deprecated versions. Even older versions were CLUSTAL and CLUSTAL V (ClustalV).',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Multiple sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'cnvkit',\n",
       "  'description': 'CNVkit is a software toolkit to infer and visualize copy number from targeted DNA sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling'],\n",
       "  'edam_topics': ['DNA structural variation']},\n",
       " {'id': 'cnv-vcf2json',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'cobrapy',\n",
       "  'description': 'COnstraints-Based Reconstruction and Analysis for Python.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Metabolic network modelling'],\n",
       "  'edam_topics': ['Gene expression',\n",
       "   'Genomics',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'colabfold',\n",
       "  'description': 'ColabFold databases are MMseqs2 expandable profile databases to generate diverse multiple sequence alignments to predict protein structures.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Database search',\n",
       "   'Protein structure prediction',\n",
       "   'Fold recognition'],\n",
       "  'edam_topics': ['Protein folds and structural domains',\n",
       "   'Protein folding, stability and design',\n",
       "   'Structure prediction',\n",
       "   'Sequence sites, features and motifs',\n",
       "   'Metagenomics']},\n",
       " {'id': 'colabfold_batch',\n",
       "  'description': 'ColabFold databases are MMseqs2 expandable profile databases to generate diverse multiple sequence alignments to predict protein structures.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Database search',\n",
       "   'Protein structure prediction',\n",
       "   'Fold recognition'],\n",
       "  'edam_topics': ['Protein folds and structural domains',\n",
       "   'Protein folding, stability and design',\n",
       "   'Structure prediction',\n",
       "   'Sequence sites, features and motifs',\n",
       "   'Metagenomics']},\n",
       " {'id': 'compleasm',\n",
       "  'description': 'Compleasm: a faster and more accurate reimplementation of BUSCO.\\nIt provides measures for quantitative assessment of genome assembly, gene set, and transcriptome completeness based on evolutionarily informed expectations of gene content from near-universal single-copy orthologs.',\n",
       "  'edam_inputs': [{'formats': ['FASTA-like'], 'term': 'FASTA-like'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence report'}],\n",
       "  'edam_operations': ['Sequence assembly validation'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Genomics',\n",
       "   'Transcriptomics',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'compose_text_param',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'compucell3d',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'concoct',\n",
       "  'description': 'A program for unsupervised binning of metagenomic contigs by using nucleotide composition, coverage data in multiple samples and linkage data from paired end reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence clustering', 'Read binning'],\n",
       "  'edam_topics': ['Metagenomics']},\n",
       " {'id': 'constava',\n",
       "  'description': 'Constava calculates conformational states probability and conformational state variability from protein structure ensembles.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Protein geometry data'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Amino acid property'}],\n",
       "  'edam_operations': ['Protein geometry calculation'],\n",
       "  'edam_topics': ['Protein folding, stability and design']},\n",
       " {'id': 'cookiecutter',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'Count1',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'coverm',\n",
       "  'description': 'Read coverage calculator for metagenomics',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['FASTA-like (text)'], 'term': 'FASTA-like (text)'},\n",
       "   {'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Statistical estimate score'}],\n",
       "  'edam_operations': ['Local alignment'],\n",
       "  'edam_topics': ['Metagenomics', 'Sequence analysis']},\n",
       " {'id': 'cpat',\n",
       "  'description': 'CPAT (Coding-Potential Assessment Tool) is a logistic regression model-based classifier that can accurately and quickly distinguish protein-coding and noncoding RNAs using pure linguistic features calculated from the RNA sequences. CPAT takes as input the nucleotides sequences or genomic coordinates of RNAs and outputs the probabilities p (0  p  1), which measure the likelihood of protein coding.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Regression analysis', 'Query and retrieval'],\n",
       "  'edam_topics': ['Functional, regulatory and non-coding RNA',\n",
       "   'Gene transcripts',\n",
       "   'RNA-Seq',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'crisflash',\n",
       "  'description': 'Software to generate CRISPR guide RNAs against genomes annotated with individual variation.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Genetic variation']},\n",
       " {'id': 'cryocare',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'cryodrgn',\n",
       "  'description': 'CryoDRGN is a neural network based algorithm for heterogeneous cryo-EM reconstruction.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Single particle analysis',\n",
       "   'Trajectory visualization',\n",
       "   'Simulation analysis',\n",
       "   'Principal component visualisation',\n",
       "   'Clustering',\n",
       "   'Classification'],\n",
       "  'edam_topics': ['Machine learning',\n",
       "   'Imaging',\n",
       "   'Molecular dynamics',\n",
       "   'Electron microscopy']},\n",
       " {'id': 'csvtk',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ctsm_fates',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'cufflinks',\n",
       "  'description': 'Cufflinks assembles transcripts and estimates their abundances in RNA-Seq samples. It accepts aligned RNA-Seq reads and assembles the alignments into a parsimonious set of transcripts. Cufflinks then estimates the relative abundances of these transcripts based on how many reads support each one.',\n",
       "  'edam_inputs': [{'formats': ['BAM', 'SAM'], 'term': 'SAM'}],\n",
       "  'edam_outputs': [{'formats': ['GTF'], 'term': 'GTF'},\n",
       "   {'formats': [], 'term': 'Gene transcript report'},\n",
       "   {'formats': [], 'term': 'Gene expression data'}],\n",
       "  'edam_operations': ['RNA-seq time series data analysis',\n",
       "   'Transcriptome assembly'],\n",
       "  'edam_topics': ['RNA splicing', 'Transcriptomics', 'RNA-Seq']},\n",
       " {'id': 'cummeRbund',\n",
       "  'description': 'Allows for persistent storage, access, exploration, and manipulation of Cufflinks high-throughput sequencing data. In addition, provides numerous plotting functions for commonly used visualizations.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis', 'Sequence visualisation'],\n",
       "  'edam_topics': ['Sequencing']},\n",
       " {'id': 'customprodb',\n",
       "  'description': 'Generate customized protein sequence database from RNA-Seq data for proteomics search.',\n",
       "  'edam_inputs': [{'formats': ['BAM', 'Textual format', 'VCF', 'FASTA'],\n",
       "    'term': 'FASTA'},\n",
       "   {'formats': ['BAM', 'Textual format', 'VCF', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['BAM', 'Textual format', 'VCF', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['BAM', 'Textual format', 'VCF', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['BAM', 'Textual format', 'VCF', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Generation', 'Sequence annotation'],\n",
       "  'edam_topics': ['Data integration and warehousing',\n",
       "   'RNA-Seq',\n",
       "   'Sequencing']},\n",
       " {'id': 'cutadapt',\n",
       "  'description': 'Find and remove adapter sequences, primers, poly-A tails and other types of unwanted sequence from your high-throughput sequencing reads.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_operations': ['Sequence trimming',\n",
       "   'Primer removal',\n",
       "   'Read pre-processing'],\n",
       "  'edam_topics': ['Genomics', 'Probes and primers', 'Sequencing']},\n",
       " {'id': 'cutesv',\n",
       "  'description': 'Long Read based Human Genomic Structural Variation Detection with cuteSV | Long-read sequencing technologies enable to comprehensively discover structural variations (SVs). However, it is still non-trivial for state-of-the-art approaches to detect SVs with high sensitivity or high performance or both. Herein, we propose cuteSV, a sensitive, fast and lightweight SV detection approach. cuteSV uses tailored methods to comprehensively collect various types of SV signatures, and a clustering-and-refinement method to implement a stepwise SV detection, which enables to achieve high sensitivity without loss of accuracy. Benchmark results demonstrate that cuteSV has better yields on real datasets. Further, its speed and scalability are outstanding and promising to large-scale data analysis',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Split read mapping',\n",
       "   'Genotyping',\n",
       "   'Structural variation detection'],\n",
       "  'edam_topics': ['DNA structural variation',\n",
       "   'Sequencing',\n",
       "   'Computer science']},\n",
       " {'id': 'dada2',\n",
       "  'description': 'This package infers exact sequence variants (SVs) from amplicon data, replacing the commonly used and coarser OTU clustering approach. This pipeline inputs demultiplexed fastq files, and outputs the sequence variants and their sample-wise abundances after removing substitution and chimera errors. Taxonomic classification is available via a native implementation of the RDP naive Bayesian classifier.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling', 'DNA barcoding'],\n",
       "  'edam_topics': ['Sequencing',\n",
       "   'Genetic variation',\n",
       "   'Microbial ecology',\n",
       "   'Metagenomics']},\n",
       " {'id': 'dadi',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'dastool',\n",
       "  'description': 'DAS Tool is an automated method that integrates the results of a flexible number of binning algorithms to calculate an optimized, non-redundant set of bins from a single assembly.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read binning'],\n",
       "  'edam_topics': ['Metagenomics']},\n",
       " {'id': 'datamash',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'dbbuilder',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'dcm2niix',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'decoupler',\n",
       "  'description': 'Ensemble of computational methods to infer biological activities from omics data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data retrieval', 'Visualisation'],\n",
       "  'edam_topics': ['Transcriptomics',\n",
       "   'Laboratory techniques',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'deepconsensus-gpu',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'deepconsensus-cpu',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'deeplabcut',\n",
       "  'description': 'DeepLabCut is a toolbox for state-of-the-art markerless pose estimation of animals performing various behaviors.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Machine learning', 'Neurobiology']},\n",
       " {'id': 'deeptools',\n",
       "  'description': 'User-friendly tools for the normalization and visualization of deep-sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Standardisation and normalisation', 'Visualisation'],\n",
       "  'edam_topics': ['Genomics', 'ChIP-seq']},\n",
       " {'id': 'dendropy',\n",
       "  'description': 'DendroPy is a Python library for phylogenetic computing. It provides classes and functions for the simulation, processing, and manipulation of phylogenetic trees and character matrices, and supports the reading and writing of phylogenetic data in a range of formats.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic inference',\n",
       "   'Phylogenetic tree distances calculation',\n",
       "   'Phylogenetic tree editing',\n",
       "   'Phylogenetic tree topology analysis',\n",
       "   'Phylogenetic tree bootstrapping'],\n",
       "  'edam_topics': ['Phylogenetics', 'Phylogenomics']},\n",
       " {'id': 'deseq2',\n",
       "  'description': 'R/Bioconductor package for differential gene expression analysis based on the negative binomial distribution. Estimate variance-mean dependence in count data from high-throughput sequencing assays and test for differential expression based on a model using the negative binomial distribution.',\n",
       "  'edam_inputs': [{'formats': ['DSV'], 'term': 'DSV'}],\n",
       "  'edam_outputs': [{'formats': ['DSV'], 'term': 'DSV'}],\n",
       "  'edam_operations': ['Differential gene expression analysis',\n",
       "   'RNA-Seq analysis'],\n",
       "  'edam_topics': ['RNA-Seq']},\n",
       " {'id': 'DEXSeq',\n",
       "  'description': 'The package is focused on finding differential exon usage using RNA-seq exon counts between samples with different experimental designs. It provides functions that allows the user to make the necessary statistical tests based on a model that uses the negative binomial distribution to estimate the variance between biological replicates and generalized linear models for testing. The package also provides functions for the visualization and exploration of the results.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Enrichment analysis',\n",
       "   'Exonic splicing enhancer prediction'],\n",
       "  'edam_topics': ['RNA-Seq']},\n",
       " {'id': 'dfam',\n",
       "  'description': 'The Dfam database is a open collection of Transposable Element DNA sequence alignments, hidden Markov Models (HMMs), consensus sequences, and genome annotations.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation',\n",
       "   'Sequence visualisation',\n",
       "   'Pairwise sequence alignment'],\n",
       "  'edam_topics': ['DNA',\n",
       "   'Phylogeny',\n",
       "   'Mobile genetic elements',\n",
       "   'Sequence assembly',\n",
       "   'Plant biology']},\n",
       " {'id': 'diann',\n",
       "  'description': 'Neural networks and interference correction enable deep proteome coverage in high throughput.\\n\\nDIA-NN - a fast and easy to use tool for processing data independent acquisition (DIA) proteomics data.\\n\\nNone required (for .raw, .mzML and .dia processing). Two executables are provided: DiaNN.exe (a command-line tool) and DIA-NN.exe (a GUI implemented as a wrapper for DiaNN.exe)',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Quantification',\n",
       "   'Spectral library search',\n",
       "   'Protein identification'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Proteomics experiment',\n",
       "   'Sequence analysis',\n",
       "   'Machine learning',\n",
       "   'Small molecules']},\n",
       " {'id': 'diaumpire',\n",
       "  'description': 'DIA-Umpire is an open source Java program for computational analysis of data independent acquisition (DIA) mass spectrometry-based proteomics data. It enables untargeted peptide and protein identification and quantitation using DIA data, and also incorporates targeted extraction to reduce the number of cases of missing quantitation.',\n",
       "  'edam_inputs': [{'formats': ['mzXML'], 'term': 'mzXML'}],\n",
       "  'edam_outputs': [{'formats': ['MGF'], 'term': 'MGF'}],\n",
       "  'edam_operations': ['Peak detection',\n",
       "   'Spectrum calculation',\n",
       "   'Retention time prediction',\n",
       "   'Mass spectra calibration'],\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment']},\n",
       " {'id': 'diamond',\n",
       "  'description': 'Sequence aligner for protein and translated DNA searches and functions as a drop-in replacement for the NCBI BLAST software tools. It is suitable for protein-protein search as well as DNA-protein search on short reads and longer sequences including contigs and assemblies, providing a speedup of BLAST ranging up to x20,000.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment analysis'],\n",
       "  'edam_topics': ['Sequence analysis', 'Proteins']},\n",
       " {'id': 'diapysef',\n",
       "  'description': 'diaPASEF is an appproch for parallel accumulation-serial fragmentation combined with data-independent acquisition.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Deisotoping', 'Ion counting', 'Protein identification'],\n",
       "  'edam_topics': ['Proteomics experiment',\n",
       "   'Proteomics',\n",
       "   'Sequence analysis',\n",
       "   'Sequencing',\n",
       "   'Small molecules']},\n",
       " {'id': 'diffbind',\n",
       "  'description': 'Compute differentially bound sites from multiple ChIP-seq experiments using affinity (quantitative) data. Also enables occupancy (overlap) analysis and plotting functions.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Differential binding analysis'],\n",
       "  'edam_topics': ['ChIP-seq']},\n",
       " {'id': 'diffdock',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'dorado',\n",
       "  'description': 'Dorado is a high-performance, easy-to-use, open source basecaller for Oxford Nanopore reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Base-calling', 'Sequence generation'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'dram',\n",
       "  'description': 'Distilled and Refined Annotation of Metabolism: A tool for the annotation and curation of function for microbial and viral genomes',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene functional annotation'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Biological databases',\n",
       "   'Molecular genetics']},\n",
       " {'id': 'drep',\n",
       "  'description': 'Fast and accurate genomic comparisons that enables improved genome recovery from metagenomes through de-replication.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome comparison'],\n",
       "  'edam_topics': ['Metagenomics', 'Genomics', 'Sequence analysis']},\n",
       " {'id': 'dropletutils',\n",
       "  'description': 'Provides a number of utility functions for handling single-cell (RNA-seq) data from droplet technologies such as 10X Genomics. This includes data loading, identification of cells from empty droplets, removal of barcode-swapped pseudo-cells, and downsampling of the count matrix.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Loading'],\n",
       "  'edam_topics': ['Gene expression',\n",
       "   'RNA-seq',\n",
       "   'Sequencing',\n",
       "   'Transcriptomics']},\n",
       " {'id': 'dwt',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'eagleimp',\n",
       "  'description': 'Fast and Accurate Genome-wide Phasing and Imputation in a Single Tool.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Imputation',\n",
       "   'Phasing',\n",
       "   'Genotyping',\n",
       "   'Genome indexing'],\n",
       "  'edam_topics': ['Genotype and phenotype',\n",
       "   'Sequencing',\n",
       "   'GWAS study',\n",
       "   'DNA',\n",
       "   'Computer science']},\n",
       " {'id': 'easybuild',\n",
       "  'description': 'EasyBuild is a software build and installation framework that allows you to manage (scientific) software on High Performance Computing (HPC) systems in an efficient way.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Service management'],\n",
       "  'edam_topics': ['Computer science']},\n",
       " {'id': 'ectyper',\n",
       "  'description': 'Ectyper is a standalone serotyping module for Escherichia coli. It supports fasta and fastq file formats. (Galaxy Version 1.0.0)',\n",
       "  'edam_inputs': [{'formats': ['FASTA', 'FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly',\n",
       "   'Read pre-processing',\n",
       "   'Variant calling'],\n",
       "  'edam_topics': ['Whole genome sequencing',\n",
       "   'Public health and epidemiology',\n",
       "   'Sequence assembly',\n",
       "   'Genotype and phenotype',\n",
       "   'Workflows']},\n",
       " {'id': 'edger',\n",
       "  'description': 'Differential expression analysis of RNA-seq expression profiles with biological replication. Implements a range of statistical methodology based on the negative binomial distributions, including empirical Bayes estimation, exact tests, generalized linear models and quasi-likelihood tests. As well as RNA-seq, it be applied to differential signal analysis of other types of genomic data that produce counts, including ChIP-seq, SAGE and CAGE.',\n",
       "  'edam_inputs': [{'formats': ['Sequence record format'],\n",
       "    'term': 'Sequence record format'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format', 'Image format'],\n",
       "    'term': 'Image format'},\n",
       "   {'formats': ['Textual format', 'Image format'], 'term': 'Image format'}],\n",
       "  'edam_operations': ['Differential gene expression analysis'],\n",
       "  'edam_topics': ['Genetics', 'RNA-Seq', 'ChIP-seq']},\n",
       " {'id': 'edger-repenrich',\n",
       "  'description': 'Differential expression analysis of RNA-seq expression profiles with biological replication. Implements a range of statistical methodology based on the negative binomial distributions, including empirical Bayes estimation, exact tests, generalized linear models and quasi-likelihood tests. As well as RNA-seq, it be applied to differential signal analysis of other types of genomic data that produce counts, including ChIP-seq, SAGE and CAGE.',\n",
       "  'edam_inputs': [{'formats': ['Sequence record format'],\n",
       "    'term': 'Sequence record format'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format', 'Image format'],\n",
       "    'term': 'Image format'},\n",
       "   {'formats': ['Textual format', 'Image format'], 'term': 'Image format'}],\n",
       "  'edam_operations': ['Differential gene expression analysis'],\n",
       "  'edam_topics': ['Genetics', 'RNA-Seq', 'ChIP-seq']},\n",
       " {'id': 'edirect',\n",
       "  'description': 'Entrez Direct (EDirect) is a command-line tool for Entrez databases. EDirect connects to Entrez through the Entrez Programming Utilities interface. It supports searching by indexed terms, looking up precomputed neighbors or links, filtering results by date or category, and downloading record summaries or reports.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Information retrieval'],\n",
       "  'edam_topics': ['Data integration and warehousing']},\n",
       " {'id': 'edta',\n",
       "  'description': 'The EDTA package was designed to filter out false discoveries in raw TE candidates and generate a high-quality non-redundant TE library for whole-genome TE annotations. Selection of initial search programs were based on benckmarkings on the annotation performance using a manually curated TE library in the rice genome.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly', 'Deisotoping', 'Genome annotation'],\n",
       "  'edam_topics': ['Workflows',\n",
       "   'Mobile genetic elements',\n",
       "   'Plant biology',\n",
       "   'Transcription factors and regulatory sites',\n",
       "   'Model organisms']},\n",
       " {'id': 'eggnog-mapper',\n",
       "  'description': 'For fast functional annotation of novel sequences. It uses precomputed orthologous groups and phylogenies from the eggNOG database to transfer functional information from fine-grained orthologs only. Its common uses include the annotation of novel genomes, transcriptomes or even metagenomic gene catalogs.\\n\\nThe use of orthology predictions for functional annotation permits a higher precision than traditional homology searches, as it avoids transferring annotations from close paralogs.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Database search'],\n",
       "  'edam_topics': ['Metagenomics', 'Genomics']},\n",
       " {'id': 'egsea',\n",
       "  'description': 'This package implements the Ensemble of Gene Set Enrichment Analyses method for gene set testing.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Gene set testing'],\n",
       "  'edam_topics': ['Systems biology']},\n",
       " {'id': 'elastix',\n",
       "  'description': 'Evaluation of an Open Source Registration Package for Automatic Contour Propagation in Online Adaptive Intensity-Modulated Proton Therapy of Prostate Cancer.\\n\\nHome : About : FAQ : wiki : Download : News : Legal stuff : Documentation.\\n\\nWelcome to elastix : a toolbox for rigid and nonrigid registration of images.\\n\\nelastix is open source software, based on the well-known Insight Segmentation and Registration Toolkit (ITK). The software consists of a collection of algorithms that are commonly used to solve (medical) image registration problems. The modular design of elastix allows the user to quickly configure, test, and compare different registration methods for a specific application. A command-line interface enables automated processing of large numbers of data sets, by means of scripting. Nowadays elastix is accompanied by SimpleElastix , making it available in languages like C++, Python, Java, R, Ruby, C# and Lua.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis'],\n",
       "  'edam_topics': ['Medical imaging', 'Oncology', 'Model organisms']},\n",
       " {'id': 'emboss',\n",
       "  'description': 'Diverse suite of tools for sequence analysis; many programs analagous to GCG; context-sensitive help for each tool.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis',\n",
       "   'Local alignment',\n",
       "   'Sequence alignment analysis',\n",
       "   'Global alignment',\n",
       "   'Sequence alignment'],\n",
       "  'edam_topics': ['Molecular biology', 'Sequence analysis', 'Biology']},\n",
       " {'id': 'emmtyper',\n",
       "  'description': 'emmtyper is a command line tool for emm-typing of Streptococcus pyogenes using a de novo or complete assembly.\\n\\nBy default, we use the U.S. Centers for Disease Control and Prevention trimmed emm subtype database, which can be found here (https://www2a.cdc.gov/ncidod/biotech/strepblast.asp). The database is curated by Dr. Velusamy Srinivasan. \\n\\nInner workings\\nThe difficulty in performing M-typing is that there is a single gene of interest (emm), but two other homologue genes (enn and mrp), often referred to as emm-like. The homologue genes may or may not occur in the isolate of interest. When performing emm-typing from an assembly, we can distinguish betweeen one or more clusters of matches on the contigs. The best match for each of the clusters identified is then parsed from the BLAST results. Where possible, we try to distinguish between matches to the emm gene, and matches to one of the emm-like genes.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ena_upload',\n",
       "  'description': \"A globally comprehensive data resource for nucleotide sequence, spanning raw data, alignments and assemblies, functional and taxonomic annotation and rich contextual data relating to sequenced samples and experimental design. Serving both as the database of record for the output of the world's sequencing activity and as a platform for the management, sharing and publication of sequence data.\",\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence similarity search',\n",
       "   'Deposition',\n",
       "   'Sequence assembly',\n",
       "   'Data retrieval'],\n",
       "  'edam_topics': ['Data submission, annotation and curation',\n",
       "   'Transcription factors and regulatory sites',\n",
       "   'Gene transcripts',\n",
       "   'Nucleic acid sites, features and motifs']},\n",
       " {'id': 'encyclopedia',\n",
       "  'description': 'EncyclopeDIA is library search engine comprised of several algorithms for DIA data analysis and can search for peptides using either DDA-based spectrum libraries or DIA-based chromatogram libraries.',\n",
       "  'edam_inputs': [{'formats': ['TraML'], 'term': 'TraML'}],\n",
       "  'edam_outputs': [{'formats': ['SQLite format'], 'term': 'SQLite format'}],\n",
       "  'edam_operations': ['Formatting'],\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment']},\n",
       " {'id': 'enrichm',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'epiScanpy',\n",
       "  'description': 'Epigenomics Single Cell Analysis in Python.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Enrichment analysis', 'Imputation'],\n",
       "  'edam_topics': ['Epigenomics', 'Cell biology', 'DNA']},\n",
       " {'id': 'escher',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ete3',\n",
       "  'description': 'The Environment for Tree Exploration (ETE) is a computational framework that simplifies the reconstruction, analysis, and visualization of phylogenetic trees and multiple sequence alignments. Here, we present ETE v3, featuring numerous improvements in the underlying library of methods, and providing a novel set of standalone tools to perform common tasks in comparative genomics and phylogenetics. The new features include (i) building gene-based and supermatrix-based phylogenies using a single command, (ii) testing and visualizing evolutionary models, (iii) calculating distances between trees of different size or including duplications, and (iv) providing seamless integration with the NCBI taxonomy database. ETE is freely available at http://etetoolkit.org',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic analysis'],\n",
       "  'edam_topics': ['Phylogenetics']},\n",
       " {'id': 'ethercalc',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'eupathdb',\n",
       "  'description': 'Integrated database covering the eukaryotic pathogens of the genera Cryptosporidium, Giardia, Leishmania, Neospora, Plasmodium, Toxoplasma, Trichomonas and Trypanosoma. The database portal offers an entry point to all these resources, and the opportunity to leverage orthology for searches across genera.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Database search',\n",
       "   'Deposition',\n",
       "   'Generation',\n",
       "   'Query and retrieval',\n",
       "   'Data retrieval'],\n",
       "  'edam_topics': ['Model organisms',\n",
       "   'Model organisms',\n",
       "   'Zoology',\n",
       "   'Gene transcripts']},\n",
       " {'id': 'evobind',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'exabayes',\n",
       "  'description': 'ExaBayes is a software package for Bayesian phylogenetic tree inference. It is particularly suitable for large-scale analyses on computer clusters.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic inference',\n",
       "   'Phylogenetic tree editing',\n",
       "   'Phylogenetic tree topology analysis'],\n",
       "  'edam_topics': ['Phylogenetics']},\n",
       " {'id': 'examl',\n",
       "  'description': 'Tool for phylogenomic analyses on supercomputers.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree generation (maximum likelihood and Bayesian methods)'],\n",
       "  'edam_topics': ['Phylogeny', 'Phylogenomics', 'Literature and language']},\n",
       " {'id': 'exonerate',\n",
       "  'description': 'A tool for pairwise sequence alignment. It enables alignment for DNA-DNA and DNA-protein pairs and also gapped and ungapped alignment.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Pairwise sequence alignment',\n",
       "   'Protein threading',\n",
       "   'Genome alignment'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Sequence sites, features and motifs',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'export_remote',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'export2graphlan',\n",
       "  'description': 'export2graphlan is a conversion software tool for producing both annotation and tree file for GraPhlAn. In particular, the annotation file tries to highlight specific sub-trees deriving automatically from input file what nodes are important.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Conversion'],\n",
       "  'edam_topics': ['Taxonomy', 'Metabolomics', 'Biomarkers']},\n",
       " {'id': 'eXpress',\n",
       "  'description': 'Streaming tool for quantifying the abundances of a set of target sequences from sampled subsequences. Example applications include transcript-level RNA-Seq quantification, allele-specific/haplotype expression analysis (from RNA-Seq), transcription factor binding quantification in ChIP-Seq, and analysis of metagenomic data. It can be used to resolve ambiguous mappings in other high-throughput sequencing based applications.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['BAM', 'SAM'], 'term': 'SAM'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Report'},\n",
       "   {'formats': [], 'term': 'Data'},\n",
       "   {'formats': [], 'term': 'Report'}],\n",
       "  'edam_operations': ['Imputation'],\n",
       "  'edam_topics': ['Gene expression', 'Metagenomics', 'RNA-Seq', 'ChIP-seq']},\n",
       " {'id': 'f5c',\n",
       "  'description': 'GPU Accelerated Adaptive Banded Event Alignment for Rapid Comparative Nanopore Signal Analysis | Re-engineered and optimised Nanopolish call-methylation module (supports CUDA acceleration) | An optimised re-implementation of the call-methylation module in Nanopolish. Given a set of basecalled Nanopore reads and the raw signals, f5c detects the methylated cytosine bases. f5c can optionally utilise NVIDIA graphics cards for acceleration',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Local alignment', 'Genotyping', 'Base-calling'],\n",
       "  'edam_topics': ['Computer science',\n",
       "   'Epigenetics',\n",
       "   'Whole genome sequencing']},\n",
       " {'id': 'falco',\n",
       "  'description': 'A high-speed FastQC emulation for quality control of sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequencing quality control',\n",
       "   'Visualisation',\n",
       "   'Read mapping'],\n",
       "  'edam_topics': ['Workflows', 'Mapping', 'Imaging']},\n",
       " {'id': 'pb-assembly',\n",
       "  'description': 'Experimental PacBio diploid assembler.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'fasta_compute_length',\n",
       "  'description': 'Add length of sequence to fasta header.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Nucleic acids', 'Sequence analysis', 'Data management']},\n",
       " {'id': 'fastahack',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'fastani',\n",
       "  'description': 'FastANI is developed for fast alignment-free computation of whole-genome Average Nucleotide Identity (ANI). ANI is defined as mean nucleotide identity of orthologous gene pairs shared between two microbial genomes. FastANI supports pairwise comparison of both complete and draft genome assemblies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome alignment', 'Sequence similarity search'],\n",
       "  'edam_topics': ['Microbiology', 'Genetic variation']},\n",
       " {'id': 'fastool',\n",
       "  'description': 'Read huge FastQ and FastA files (both normal and gzipped) an demanipulate them.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Formatting'],\n",
       "  'edam_topics': ['Data management']},\n",
       " {'id': 'fastp',\n",
       "  'description': 'A tool designed to provide fast all-in-one preprocessing for FastQ files. This tool is developed in C++ with multithreading supported to afford high performance.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequencing quality control',\n",
       "   'Sequence contamination filtering'],\n",
       "  'edam_topics': ['Sequence analysis', 'Probes and primers']},\n",
       " {'id': 'fastplong',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'fastqc',\n",
       "  'description': 'This tool aims to provide a QC report which can spot problems or biases which originate either in the sequencer or in the starting library material. It can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-like format (text)',\n",
       "     'SAM',\n",
       "     'FASTQ',\n",
       "     'BAM'],\n",
       "    'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Sequence composition calculation',\n",
       "   'Sequencing quality control',\n",
       "   'Statistical calculation'],\n",
       "  'edam_topics': ['Sequencing',\n",
       "   'Data quality management',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'fastqe',\n",
       "  'description': 'Compute quality stats for FASTQ files and print those stats as emoji... for some reason.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequencing quality control'],\n",
       "  'edam_topics': ['Sequence analysis', 'Sequencing']},\n",
       " {'id': 'fasttree',\n",
       "  'description': 'Infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree generation (from molecular sequences)',\n",
       "   'Phylogenetic tree generation (maximum likelihood and Bayesian methods)'],\n",
       "  'edam_topics': ['Phylogenetics', 'Sequence analysis']},\n",
       " {'id': 'fastx',\n",
       "  'description': 'Collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read pre-processing', 'Sequence editing'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'fastx_toolkit',\n",
       "  'description': 'Collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read pre-processing', 'Sequence editing'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'feelnc',\n",
       "  'description': 'A tool to annotate long non-coding RNAs from RNA-seq assembled transcripts.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Annotation'],\n",
       "  'edam_topics': ['RNA-seq', 'Functional, regulatory and non-coding RNA']},\n",
       " {'id': 'fermi-lite',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'featurecounts',\n",
       "  'description': 'featureCounts is a very efficient read quantifier. It can be used to summarize RNA-seq reads and gDNA-seq reads to a variety of genomic features such as genes, exons, promoters, gene bodies and genomic bins. It is included in the Bioconductor Rsubread package and also in the SourceForge Subread package.',\n",
       "  'edam_inputs': [{'formats': ['GFF'], 'term': 'GFF'},\n",
       "   {'formats': ['GTF'], 'term': 'GTF'},\n",
       "   {'formats': ['BAM'], 'term': 'BAM'},\n",
       "   {'formats': ['SAM'], 'term': 'SAM'}],\n",
       "  'edam_outputs': [{'formats': ['DSV'], 'term': 'DSV'}],\n",
       "  'edam_operations': ['Read summarisation', 'RNA-Seq quantification'],\n",
       "  'edam_topics': ['RNA-Seq']},\n",
       " {'id': 'fgenesh',\n",
       "  'description': 'HMM-based gene structure prediction (multiple genes, both chains); Program for predicting multiple genes in genomic DNA sequences.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['Gene structure']},\n",
       " {'id': 'fgsea',\n",
       "  'description': 'The package implements an algorithm for fast gene set enrichment analysis. Using the fast algorithm allows to make more permutations and get more fine grained p-values, which allows to use accurate stantard approaches to multiple hypothesis correction.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene-set enrichment analysis'],\n",
       "  'edam_topics': ['Genetics']},\n",
       " {'id': 'filter_transcripts_via_tracking',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'filtlong',\n",
       "  'description': 'Filtlong is a tool for filtering long reads by quality. It can take a set of long reads and produce a smaller, better subset. It uses both read length (longer is better) and read identity (higher is better) when choosing which reads pass the filter.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Filtering', 'Sequencing quality control'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'flashlfq',\n",
       "  'description': 'FlashLFQ is an ultrafast label-free quantification algorithm for mass-spectrometry proteomics.',\n",
       "  'edam_inputs': [{'formats': ['Thermo RAW', 'Waters RAW'],\n",
       "    'term': 'Waters RAW'},\n",
       "   {'formats': ['TSV', 'Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Label-free quantification'],\n",
       "  'edam_topics': ['Proteomics experiment', 'Proteomics']},\n",
       " {'id': 'flex',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'flye',\n",
       "  'description': 'Flye is a de novo assembler for single molecule sequencing reads, such as those produced by PacBio and Oxford Nanopore Technologies. It is designed for a wide range of datasets, from small bacterial projects to large mammalian-scale assemblies. The package represents a complete pipeline: it takes raw PB / ONT reads as input and outputs polished contigs.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'De-novo assembly',\n",
       "   'Mapping assembly',\n",
       "   'Cross-assembly'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Metagenomics',\n",
       "   'Whole genome sequencing',\n",
       "   'Genomics']},\n",
       " {'id': 'flymine',\n",
       "  'description': 'An integrated database for Drosophila and Anopheles genomics.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence features'},\n",
       "   {'formats': [], 'term': 'Ontology data'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Over-representation data'},\n",
       "   {'formats': [], 'term': 'Query script'},\n",
       "   {'formats': [], 'term': 'Sequence features'}],\n",
       "  'edam_operations': ['Data retrieval',\n",
       "   'Visualisation',\n",
       "   'Gene-set enrichment analysis',\n",
       "   'Query and retrieval'],\n",
       "  'edam_topics': ['Data management', 'Genomics', 'Proteomics']},\n",
       " {'id': 'foldseek',\n",
       "  'description': 'Foldseek enables fast and sensitive comparisons of large structure sets. It reaches sensitivities similar to state-of-the-art structural aligners while being at least 20,000 times faster.',\n",
       "  'edam_inputs': [{'formats': ['PDB', 'mmCIF'], 'term': 'mmCIF'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Protein structure alignment'}],\n",
       "  'edam_operations': ['Structure alignment'],\n",
       "  'edam_topics': ['Proteins']},\n",
       " {'id': 'fpocket',\n",
       "  'description': 'Web server which detects small molecule pockets by relying on the geometric alpha sphere theory. It also tracks pockets during molecular dynamics so to provide insight on pocket dynamics (mdpocket) and transposes mdpocket to the combined analysis of homologous structures (hpocket).',\n",
       "  'edam_inputs': [{'formats': ['PDB'], 'term': 'PDB'}],\n",
       "  'edam_outputs': [{'formats': ['PDB'], 'term': 'PDB'}],\n",
       "  'edam_operations': ['Molecular surface analysis'],\n",
       "  'edam_topics': ['Protein structural motifs and surfaces']},\n",
       " {'id': 'fraggenescan',\n",
       "  'description': 'Application for finding (fragmented) genes in short reads',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['GFF'], 'term': 'GFF'}],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['Genetics', 'Sequence analysis']},\n",
       " {'id': 'fragpipe',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'freebayes',\n",
       "  'description': 'Bayesian genetic variant detector designed to find small polymorphisms, specifically SNPs, indels, multi-nucleotide polymorphisms, and complex events (composite insertion and substitution events) smaller than the length of a short-read sequencing alignment.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling', 'Statistical calculation'],\n",
       "  'edam_topics': ['Genomics', 'Genetic variation', 'Rare diseases']},\n",
       " {'id': 'freec',\n",
       "  'description': 'A tool for control-free copy number alteration (CNA) and allelic imbalances (LOH) detection using deep-sequencing data, particularly useful for cancer studies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Copy number estimation',\n",
       "   'Variant calling',\n",
       "   'Genome alignment'],\n",
       "  'edam_topics': ['DNA structural variation',\n",
       "   'Oncology',\n",
       "   'Human genetics',\n",
       "   'Data mining']},\n",
       " {'id': 'freesurfer',\n",
       "  'description': \"Wrapper functions that interface with 'Freesurfer', a powerful and commonly-used 'neuroimaging' software, using system commands. The goal is to be able to interface with 'Freesurfer' completely in R, where you pass R objects of class 'nifti', implemented by package 'oro.nifti', and the function executes an 'Freesurfer' command and returns an R object of class 'nifti' or necessary output.\",\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Service invocation'],\n",
       "  'edam_topics': ['MRI']},\n",
       " {'id': 'fsom',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'funannotate',\n",
       "  'description': 'funannotate is a pipeline for genome annotation (built specifically for fungi, but will also work with higher eukaryotes).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'gaeval',\n",
       "  'description': 'Gene Annotation EVAluation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence annotation'],\n",
       "  'edam_topics': ['Sequence analysis', 'Gene structure']},\n",
       " {'id': 'galaxy_collection_operations',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'Galaxy CONVERTER',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_data_sources',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_filter_and_sort',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_graph_display',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_fetch_alignments_sequences',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_image_analysis',\n",
       "  'description': 'Galaxy Image Analysis is mainly developed by the Biomedical Computer Vision (BMCV) Group at Heidelberg University. Its dedication is to provide tools for image analysis and image processing for the Galaxy platform.',\n",
       "  'edam_inputs': [{'formats': ['OME-TIFF', 'TIFF'], 'term': 'TIFF'}],\n",
       "  'edam_outputs': [{'formats': ['Binary format'], 'term': 'Binary format'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Image analysis'],\n",
       "  'edam_topics': ['Imaging', 'Bioinformatics']},\n",
       " {'id': 'galaxy_genomic_intervals',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_join_subtract_and_group',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_text_manipulation',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_sequence_utils',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'galaxy_statistics',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'gatk',\n",
       "  'description': 'The Genome Analysis Toolkit (GATK) is a set of bioinformatic tools for analyzing high-throughput sequencing (HTS) and variant call format (VCF) data. The toolkit is well established for germline short variant discovery from whole genome and exome sequencing data. GATK4 expands functionality into copy number and somatic analyses and offers pipeline scripts for workflows.  \\n\\nVersion 4 (GATK4) is open-source at https://github.com/broadinstitute/gatk.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Polymorphism detection',\n",
       "   'Sequence analysis',\n",
       "   'Genotyping',\n",
       "   'Statistical calculation'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Genetic variation',\n",
       "   'Sequencing',\n",
       "   'Workflows']},\n",
       " {'id': 'gblock',\n",
       "  'description': 'Cleaning aligned sequences.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Sequence alignment analysis'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'gc-meox-tms',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'gcta',\n",
       "  'description': 'Genome-wide Complex Trait Analysis. Estimate the proportion of phenotypic variance explained by genome- or chromosome-wide SNPs for complex traits (the GREML method), and has subsequently extended for many other analyses to better understand the genetic architecture of complex traits.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genetic variation analysis'],\n",
       "  'edam_topics': ['GWAS study']},\n",
       " {'id': 'gecko',\n",
       "  'description': 'Software aimed at pairwise sequence comparison generating high quality results (equivalent to MUMmer) with controlled memory consumption and comparable or faster execution times particularly with long sequences.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'gemini',\n",
       "  'description': 'GEMINI (GEnome MINIng) is a flexible framework for exploring genetic variation in the context of the wealth of genome annotations available for the human genome. By placing genetic variants, sample phenotypes and genotypes, as well as genome annotations into an integrated database framework, GEMINI provides a simple, flexible, and powerful system for exploring genetic variation for disease and population genetics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis', 'Genetic variation analysis'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'gemoma',\n",
       "  'description': 'Gene Model Mapper is a homology-based gene prediction program. GeMoMa uses the annotation of protein-coding genes in a reference genome to infer the annotation of protein-coding genes in a target genome. Thereby, it utilizes amino acid sequence and intron position conservation. In addition, it allows to incorporate RNA-seq evidence for splice site prediction.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['Genomics', 'Homology modeling']},\n",
       " {'id': 'geneiobio',\n",
       "  'description': 'An interactive web tool for versatile, clinically-driven variant interrogation and prioritization.\\n\\nIOBIO is a suite of web apps for visually driven real-time analysis of genomic data.\\n\\nVisually driven real-time analysis of genomic data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant prioritisation',\n",
       "   'Variant calling',\n",
       "   'Variant classification'],\n",
       "  'edam_topics': ['Exome sequencing',\n",
       "   'Genomics',\n",
       "   'Medicine',\n",
       "   'Genetic variation',\n",
       "   'Genotype and phenotype']},\n",
       " {'id': 'generate_count_matrix',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'generate_pc_lda_matrix',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'generode',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'genomescope',\n",
       "  'description': 'Reference-free profiling of polyploid genomes | We have developed GenomeScope 2.0, which applies classical insights from combinatorial theory to establish a detailed mathematical model of how k-mer frequencies will be distributed in heterozygous and polyploid genomes | Average k-mer coverage for polyploid genome | Upload results from running Jellyfish or KMC',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping', 'k-mer counting'],\n",
       "  'edam_topics': ['Sequence assembly', 'Mathematics', 'Mapping']},\n",
       " {'id': 'genometools',\n",
       "  'description': 'Free collection of bioinformatics tools for genome informatics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Nucleic acid sites, features and motifs',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'genrich',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'getorganelle',\n",
       "  'description': 'A fast and versatile toolkit for accurate de novo assembly of organelle genomes.\\n\\nThis toolkit assemblies organelle genome from genomic skimming data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly',\n",
       "   'Genome assembly',\n",
       "   'Mapping assembly',\n",
       "   'Mapping',\n",
       "   'Sequence trimming'],\n",
       "  'edam_topics': ['Cell biology',\n",
       "   'Sequence assembly',\n",
       "   'Whole genome sequencing',\n",
       "   'Plant biology',\n",
       "   'Model organisms']},\n",
       " {'id': 'gfastats',\n",
       "  'description': 'gfastats is a single fast and exhaustive tool for summary statistics and simultaneous genome assembly file manipulation. gfastats also allows seamless fasta/fastq/gfa conversion.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Nucleic acid sequence'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Computational biology']},\n",
       " {'id': 'gffcompare',\n",
       "  'description': 'Program for comparing, annotating, merging and tracking transcripts in GFF files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence annotation'],\n",
       "  'edam_topics': ['Nucleic acids', 'Sequence analysis']},\n",
       " {'id': 'gffread',\n",
       "  'description': 'program for filtering, converting and manipulating GFF files',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence annotation'],\n",
       "  'edam_topics': ['Nucleic acids', 'Sequence analysis']},\n",
       " {'id': 'gff3sort',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ggplot2',\n",
       "  'description': 'Plotting system for R, based on the grammar of graphics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [{'formats': ['Image format'], 'term': 'Image format'}],\n",
       "  'edam_operations': ['Visualisation'],\n",
       "  'edam_topics': ['Data visualisation']},\n",
       " {'id': 'ggsashimi',\n",
       "  'description': 'Command-line tool for the visualization of splicing events across multiple samples',\n",
       "  'edam_inputs': [{'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['PDF', 'SVG', 'TIFF', 'PNG', 'JPG'],\n",
       "    'term': 'JPG'}],\n",
       "  'edam_operations': ['Visualisation', 'Aggregation'],\n",
       "  'edam_topics': ['RNA splicing', 'RNA-Seq', 'Data visualisation']},\n",
       " {'id': 'ghostscript',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'GMAJ',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'gmap',\n",
       "  'description': 'Genomic Mapping and Alignment Program for mRNA and EST Sequences.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-illumina', 'FASTQ-sanger', 'FASTA'],\n",
       "    'term': 'FASTA'},\n",
       "   {'formats': ['FASTQ-illumina', 'FASTQ-sanger', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Genetic mapping', 'Sequence alignment'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'gmap-gsnap',\n",
       "  'description': 'Genomic Mapping and Alignment Program for mRNA and EST Sequences.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-illumina', 'FASTQ-sanger', 'FASTA'],\n",
       "    'term': 'FASTA'},\n",
       "   {'formats': ['FASTQ-illumina', 'FASTQ-sanger', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Genetic mapping', 'Sequence alignment'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'parallel',\n",
       "  'description': 'GNU parallel is a shell tool for executing jobs in parallel using one or more computers. A job can be a single command or a small script that has to be run for each of the lines in the input. The typical input is a list of files, a list of hosts, a list of users, a list of URLs, or a list of tables. A job can also be a command that reads from a pipe. GNU parallel can then split the input and pipe it into commands in parallel.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'URL'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Splitting'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'gsl',\n",
       "  'description': 'The GNU Scientific Library (GSL) is a numerical library for C and C++ programmers.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'goenrichment',\n",
       "  'description': 'GOEnrichment is a tool for performing GO enrichment analysis of gene sets, such as those obtained from RNA-seq or Microarray experiments, to help characterize them at the functional level. It is available in Galaxy Europe and as a stand-alone tool.\\nGOEnrichment is flexible in that it allows the user to use any version of the Gene Ontology and any GO annotation file they desire. To enable the use of GO slims, it is accompanied by a sister tool GOSlimmer, which can convert annotation files from full GO to any specified GO slim.\\nThe tool features an optional graph clustering algorithm to reduce the redundancy in the set of enriched GO terms and simplify its output.\\nIt was developed by the BioData.pt / ELIXIR-PT team at the Instituto Gulbenkian de Cincia.',\n",
       "  'edam_inputs': [{'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['OBO format', 'OWL format'], 'term': 'OWL format'}],\n",
       "  'edam_outputs': [{'formats': ['png', 'Textual format'],\n",
       "    'term': 'Textual format'}],\n",
       "  'edam_operations': ['Gene-set enrichment analysis'],\n",
       "  'edam_topics': ['Transcriptomics']},\n",
       " {'id': 'goseq',\n",
       "  'description': 'Detect Gene Ontology and/or other user defined categories which are over/under represented in RNA-seq data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene functional annotation'],\n",
       "  'edam_topics': ['RNA-Seq']},\n",
       " {'id': 'gramenemart',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'graphlan',\n",
       "  'description': 'GraPhlAn is a software tool for producing high-quality circular representations of taxonomic and phylogenetic trees. GraPhlAn focuses on concise, integrative, informative, and publication-ready representations of phylogenetically- and taxonomically-driven investigation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic inference',\n",
       "   'Phylogenetic tree visualisation',\n",
       "   'Phylogenetic tree editing',\n",
       "   'Taxonomic classification'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Phylogenetics',\n",
       "   'Phylogenomics',\n",
       "   'Cladistics']},\n",
       " {'id': 'gromacs',\n",
       "  'description': 'Versatile package to perform molecular dynamics, i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles. It is primarily designed for biochemical molecules like proteins, lipids and nucleic acids that have a lot of complicated bonded interactions, but since it is extremely fast at calculating the nonbonded interactions (that usually dominate simulations) many groups are also using it for research on non-biological systems, e.g. polymers.',\n",
       "  'edam_inputs': [{'formats': ['PDB'], 'term': 'PDB'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Molecular dynamics'],\n",
       "  'edam_topics': ['Molecular dynamics']},\n",
       " {'id': 'gtdb_to_taxdump',\n",
       "  'description': 'Tool with multiple functions. Main functions are to create a DIAMOND database from the GTDB taxonomy data or create a NCBI taxdump format out of this data. This tool can also create a mapping between the taxonomy classification between GTDB and NCBI.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Mapping', 'Generation'],\n",
       "  'edam_topics': ['Computational biology']},\n",
       " {'id': 'gtdb-tk',\n",
       "  'description': 'a toolkit to classify genomes with the Genome Taxonomy Database.\\n\\nGTDB-Tk: a toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes.\\n\\nGTDB-Tk is a software toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes based on the Genome Database Taxonomy GTDB. It is designed to work with recent advances that allow hundreds or thousands of metagenome-assembled genomes (MAGs) to be obtained directly from environmental samples. It can also be applied to isolate and single-cell genomes. The GTDB-Tk is open source and released under the GNU General Public License (Version 3).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome alignment',\n",
       "   'Taxonomic classification',\n",
       "   'Sequence assembly',\n",
       "   'Query and retrieval'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Taxonomy',\n",
       "   'Phylogenetics',\n",
       "   'Database management',\n",
       "   'Proteins']},\n",
       " {'id': 'gubbins',\n",
       "  'description': 'Gubbins is a tool for rapid phylogenetic analysis of large samples of recombinant bacterial whole genome sequences.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping',\n",
       "   'Phylogenetic inference',\n",
       "   'Ancestral reconstruction'],\n",
       "  'edam_topics': ['Phylogeny',\n",
       "   'Genotype and phenotype',\n",
       "   'Whole genome sequencing']},\n",
       " {'id': 'gzip',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'hapcut2',\n",
       "  'description': 'HapCUT2 is a maximum-likelihood-based tool for assembling haplotypes from DNA sequence reads, designed to \"just work\" with excellent speed and accuracy across a range of long- and short-read sequencing technologies.\\nThe output is in Haplotype block format described here: https://github.com/vibansal/HapCUT2/blob/master/outputformat.md',\n",
       "  'edam_inputs': [{'formats': ['BAM'], 'term': 'BAM'},\n",
       "   {'formats': ['VCF'], 'term': 'VCF'}],\n",
       "  'edam_outputs': [{'formats': ['plain text format (unformatted)'],\n",
       "    'term': 'plain text format (unformatted)'},\n",
       "   {'formats': ['VCF'], 'term': 'VCF'}],\n",
       "  'edam_operations': ['Haplotype mapping', 'Variant classification'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'hapflk',\n",
       "  'description': 'Software for the detection of selection signatures based on multiple population genotyping data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping'],\n",
       "  'edam_topics': ['Genotyping experiment',\n",
       "   'Genetics',\n",
       "   'Genotype and phenotype',\n",
       "   'Population genetics']},\n",
       " {'id': 'hapog',\n",
       "  'description': 'Hapo-G is a tool that aims to improve the quality of genome assemblies by polishing the consensus with accurate reads. It capable of incorporating phasing information from high-quality reads (short or long-reads) to polish genome assemblies and in particular assemblies of diploid and heterozygous genomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'Optimisation and refinement'],\n",
       "  'edam_topics': ['Sequence assembly', 'Genomics']},\n",
       " {'id': 'hatchling',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'hbvar',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'heinz',\n",
       "  'description': 'Tool for single-species active module discovery.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Pathway or network analysis'],\n",
       "  'edam_topics': ['Genetics',\n",
       "   'Gene expression',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'helixer',\n",
       "  'description': 'Deep Learning to predict gene annotations',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction', 'Genome annotation'],\n",
       "  'edam_topics': ['Sequence analysis', 'Gene transcripts']},\n",
       " {'id': 'hgv_david',\n",
       "  'description': 'This tool provides functional annotation for a list of genes by connecting with DAVID database.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Database search',\n",
       "   'Service invocation',\n",
       "   'Gene functional annotation'],\n",
       "  'edam_topics': ['Ontology and terminology', 'Gene expression']},\n",
       " {'id': 'hgv_ldtools',\n",
       "  'description': 'This tool can be used to analyze the patterns of linkage disequilibrium (LD) between polymorphic sites in a locus.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Data handling',\n",
       "   'Linkage disequilibrium calculation',\n",
       "   'Variant pattern analysis'],\n",
       "  'edam_topics': ['Mapping',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'DNA polymorphism',\n",
       "   'Data mining']},\n",
       " {'id': 'hgv_linkToGProfile',\n",
       "  'description': 'This tool creates a link to the g:GOSt tool (Gene Group Functional Profiling), which provides functional profiling of gene lists.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Data handling',\n",
       "   'Service invocation',\n",
       "   'Gene functional annotation'],\n",
       "  'edam_topics': ['Ontology and terminology',\n",
       "   'Function analysis',\n",
       "   'Data mining']},\n",
       " {'id': 'hicexplorer',\n",
       "  'description': 'A web server for reproducible Hi-C, capture Hi-C and single-cell Hi-C data analysis, quality control and visualization.\\n\\nHiCExplorer  HiCExplorer 3.6 documentation.\\n\\nscHiCExplorer  scHiCExplorer 7 documentation.\\n\\nFree document hosting provided by Read the Docs.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Validation',\n",
       "   'Visualisation',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Workflows',\n",
       "   'DNA',\n",
       "   'Chromosome conformation capture',\n",
       "   'ChIP-seq',\n",
       "   'Epigenetics']},\n",
       " {'id': 'hifiadapterfilt',\n",
       "  'description': 'Remove CCS reads with remnant PacBio adapter sequences and convert outputs to a compressed .fastq (.fastq.gz).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Primer removal',\n",
       "   'De-novo assembly',\n",
       "   'Genome assembly',\n",
       "   'Sequence trimming'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Agricultural science',\n",
       "   'Workflows',\n",
       "   'Whole genome sequencing']},\n",
       " {'id': 'hifiasm',\n",
       "  'description': 'Hifiasm: a haplotype-resolved assembler for accurate Hifi reads',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'De-novo assembly'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'hifiasm_meta',\n",
       "  'description': 'Hifiasm_meta - de novo metagenome assembler, based on hifiasm, a haplotype-resolved de novo assembler for PacBio Hifi reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly'],\n",
       "  'edam_topics': ['Sequence assembly', 'Metagenomics']},\n",
       " {'id': 'hisat2',\n",
       "  'description': 'Alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes (as well as to a single reference genome).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['RNA-seq']},\n",
       " {'id': 'hmmcleaner',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'hmmer',\n",
       "  'description': 'This tool is used for searching sequence databases for homologs of protein sequences, and for making protein sequence alignments. It implements methods using probabilistic models called profile hidden Markov models. The new HMMER3 project, HMMER is now as fast as BLAST for protein search.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence alignment'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence alignment'}],\n",
       "  'edam_operations': ['Formatting'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Sequence sites, features and motifs',\n",
       "   'Gene and protein families']},\n",
       " {'id': 'htseq',\n",
       "  'description': 'Python framework to process and analyse high-throughput sequencing (HTS) data',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Nucleic acid sequence analysis'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'htslib',\n",
       "  'description': 'The main purpose of HTSlib is to provide access to genomic information files, both alignment data (SAM, BAM, and CRAM formats) and variant data (VCF and BCF formats). The library also provides interfaces to access and index genome reference data in FASTA format and tab-delimited files with genomic coordinates. It is utilized and incorporated into both SAMtools and BCFtools.',\n",
       "  'edam_inputs': [{'formats': ['SAM', 'CRAM', 'FASTQ', 'BAM', 'FASTA'],\n",
       "    'term': 'FASTA'},\n",
       "   {'formats': ['VCF', 'BCF'], 'term': 'BCF'}],\n",
       "  'edam_outputs': [{'formats': ['SAM', 'CRAM', 'FASTQ', 'BAM', 'FASTA'],\n",
       "    'term': 'FASTA'},\n",
       "   {'formats': ['VCF', 'BCF'], 'term': 'BCF'}],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Data management']},\n",
       " {'id': 'humann',\n",
       "  'description': 'HUMAnN is a pipeline for efficiently and accurately profiling the presence/absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data (typically millions of short DNA/RNA reads). This process, referred to as functional profiling, aims to describe the metabolic potential of a microbial community and its members. More generally, functional profiling answers the question What are the microbes in my community-of-interest doing (or are capable of doing)?',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Species frequency estimation',\n",
       "   'Taxonomic classification',\n",
       "   'Phylogenetic analysis'],\n",
       "  'edam_topics': ['Metagenomics', 'Phylogenomics']},\n",
       " {'id': 'humann2',\n",
       "  'description': 'HUMAnN 2.0 is a pipeline for efficiently and accurately profiling the presence/absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data (typically millions of short DNA/RNA reads). This process, referred to as functional profiling, aims to describe the metabolic potential of a microbial community and its members. More generally, functional profiling answers the question What are the microbes in my community-of-interest doing (or capable of doing)?',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Metabolic pathway prediction'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'huygens',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'hybpiper',\n",
       "  'description': 'Paralogs and off-target sequences improve phylogenetic resolution in a densely-sampled study of the breadfruit genus (Artocarpus, Moraceae).\\n\\nRecovering genes from targeted sequence capture data.\\n\\nCurrent version: 1.3.1 (August 2018).\\n\\n-- Read our article in Applications in Plant Sciences (Open Access).\\n\\nHybPiper was designed for targeted sequence capture, in which DNA sequencing libraries are enriched for gene regions of interest, especially for phylogenetics. HybPiper is a suite of Python scripts that wrap and connect bioinformatics tools in order to extract target sequences from high-throughput DNA sequencing reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence trimming',\n",
       "   'Sequence assembly',\n",
       "   'Read mapping'],\n",
       "  'edam_topics': ['Phylogenetics',\n",
       "   'Plant biology',\n",
       "   'Gene transcripts',\n",
       "   'Sequence assembly',\n",
       "   'Phylogenomics']},\n",
       " {'id': 'HyPhy',\n",
       "  'description': 'Software package for the analysis of genetic sequences using techniques in phylogenetics, molecular evolution, and machine learning.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Statistical calculation'],\n",
       "  'edam_topics': ['Phylogeny',\n",
       "   'Small molecules',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'ilastik',\n",
       "  'description': 'Tool for interactive bioimage classification, segmentation and analysis.',\n",
       "  'edam_inputs': [{'formats': ['ppm',\n",
       "     'GIF',\n",
       "     'HDF5',\n",
       "     'TIFF',\n",
       "     'pgm',\n",
       "     'JPG',\n",
       "     'PNG',\n",
       "     'pbm',\n",
       "     'BMP'],\n",
       "    'term': 'BMP'}],\n",
       "  'edam_outputs': [{'formats': ['HDF5', 'TIFF'], 'term': 'TIFF'},\n",
       "   {'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_operations': ['Quantification', 'Image analysis', 'Classification'],\n",
       "  'edam_topics': ['Machine learning', 'Biomedical science']},\n",
       " {'id': 'imagemagick',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pbipa',\n",
       "  'description': 'Improved Phased Assembler (IPA) is the official PacBio software for HiFi genome assembly. IPA was designed to utilize the accuracy of PacBio HiFi reads to produce high-quality phased genome assemblies',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'Phasing'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'infernal',\n",
       "  'description': 'Infernal (\"INFERence of RNA ALignment\") is for searching DNA sequence databases for RNA structure and sequence similarities. It is an implementation of a special case of profile stochastic context-free grammars called covariance models (CMs). A CM is like a sequence profile, but it scores a combination of sequence consensus and RNA secondary structure consensus, so in many cases, it is more capable of identifying RNA homologs that conserve their secondary structure more than their primary sequence.',\n",
       "  'edam_inputs': [{'formats': ['Sequence profile format'],\n",
       "    'term': 'Sequence profile format'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Database search results'}],\n",
       "  'edam_operations': ['Nucleic acid feature detection'],\n",
       "  'edam_topics': ['Sequence sites, features and motifs',\n",
       "   'Structural genomics']},\n",
       " {'id': 'integron_finder',\n",
       "  'description': 'A tool to detect Integron in DNA sequences.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['GenBank format'], 'term': 'GenBank format'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Nucleic acid feature detection',\n",
       "   'Sequence motif recognition',\n",
       "   'Protein feature detection',\n",
       "   'Genome annotation'],\n",
       "  'edam_topics': ['Functional genomics',\n",
       "   'Mobile genetic elements',\n",
       "   'Molecular biology',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'intermine',\n",
       "  'description': 'Open source data warehouse built specifically for the integration and analysis of complex biological data. It enables the creation of biological databases accessed by sophisticated web query tools. Parsers are provided for integrating data from many common biological data sources and formats, and there is a framework for adding your own data.',\n",
       "  'edam_inputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Database search'],\n",
       "  'edam_topics': ['Data integration and warehousing', 'Rare diseases']},\n",
       " {'id': 'interproscan',\n",
       "  'description': 'Scan sequences against the InterPro protein signature databases.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Protein sequence'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence signature matches'},\n",
       "   {'formats': [], 'term': 'Protein features'}],\n",
       "  'edam_operations': ['Sequence motif recognition',\n",
       "   'Protein feature detection'],\n",
       "  'edam_topics': ['Gene and protein families', 'Sequence analysis']},\n",
       " {'id': 'ipyrad',\n",
       "  'description': 'Interactive assembly and analysis of RADseq datasets.\\n\\nipyrad: interactive assembly and analysis of RAD-seq data sets.\\n\\nWelcome to ipyrad, an interactive toolkit for assembly and analysis of restriction-site associated genomic data sets (e.g., RAD, ddRAD, GBS) for population genetic and phylogenetic studies.\\n\\nWelcome to ipyrad  ipyrad documentation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly',\n",
       "   'de Novo sequencing',\n",
       "   'Haplotype mapping'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Phylogenetics',\n",
       "   'Population genetics']},\n",
       " {'id': 'iq-tree',\n",
       "  'description': 'Very efficient phylogenetic software for reconstructing maximum-likelihood trees and assessing branch supports with the ultrafast bootstrap approximation. It is based on the IQPNNI algorithm with 10-fold speedup together with substantially additional features.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree generation (maximum likelihood and Bayesian methods)'],\n",
       "  'edam_topics': ['Phylogenetics',\n",
       "   'Statistics and probability',\n",
       "   'Phylogenomics']},\n",
       " {'id': 'irfinder',\n",
       "  'description': 'A comprehensive suite to discover and explore intron retention.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Quantification',\n",
       "   'Alternative splicing prediction',\n",
       "   'Visualisation'],\n",
       "  'edam_topics': ['Gene transcripts',\n",
       "   'PCR experiment',\n",
       "   'RNA-Seq',\n",
       "   'Machine learning',\n",
       "   'RNA splicing']},\n",
       " {'id': 'isee',\n",
       "  'description': 'Provides functions for creating an interactive Shiny-based graphical user interface for exploring data stored in SummarizedExperiment objects, including row- and column-level metadata. Particular attention is given to single-cell data in a SingleCellExperiment object with visualization of dimensionality reduction results.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Clustering'],\n",
       "  'edam_topics': ['Gene expression', 'Transcriptomics']},\n",
       " {'id': 'isescan',\n",
       "  'description': 'Automated identification of insertion sequence elements in prokaryotic genomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Structural variation detection'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'DNA structural variation',\n",
       "   'Sequence analysis',\n",
       "   'Genetic variation']},\n",
       " {'id': 'isoformswitchanalyzer',\n",
       "  'description': 'Enables identification of isoform switches with predicted functional consequences from RNA-seq data. Consequences can be chosen from a long list but includes protein domains gain/loss changes in NMD sensitivity etc. It directly supports import of data from Cufflinks/Cuffdiff, Kallisto, Salmon and RSEM but other transcript qunatification tools are easy to import as well.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence comparison', 'Sequence analysis'],\n",
       "  'edam_topics': ['Computational biology', 'Gene transcripts']},\n",
       " {'id': 'isolib',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'isoseq3',\n",
       "  'description': 'IsoSeq v3 contains the newest tools to identify transcripts in PacBio single-molecule sequencing data. Starting in SMRT Link v6.0.0, those tools power the IsoSeq GUI-based analysis application. A composable workflow of existing tools and algorithms, combined with a new clustering technique.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Demultiplexing',\n",
       "   'Sequence trimming',\n",
       "   'Clustering',\n",
       "   'Primer removal',\n",
       "   'DNA barcoding'],\n",
       "  'edam_topics': ['Gene transcripts']},\n",
       " {'id': 'itk-snap',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'itsx',\n",
       "  'description': 'TSx is an open source software utility to extract the highly variable ITS1 and ITS2 subregions from ITS sequences, which is commonly used as a molecular barcode for e.g. fungi. As the inclusion of parts of the neighbouring, very conserved, ribosomal genes (SSU, 5S and LSU rRNA sequences) in the sequence identification process can lead to severely misleading results, ITSx identifies and extracts only the ITS regions themselves.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence feature detection'],\n",
       "  'edam_topics': ['Functional, regulatory and non-coding RNA',\n",
       "   'Microbiology']},\n",
       " {'id': 'ivar',\n",
       "  'description': 'Interpretation-oriented tool to manage the update and revision of variant annotation and classification.\\n\\niVar - DataBase of Genomics Variants.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant classification',\n",
       "   'Text annotation',\n",
       "   'Variant calling'],\n",
       "  'edam_topics': ['Biological databases',\n",
       "   'Evolutionary biology',\n",
       "   'Sequencing']},\n",
       " {'id': 'iwtomics',\n",
       "  'description': 'Implementation of the Interval-Wise Testing (IWT) for omics data. This inferential procedure tests for differences in \"Omics\" data between two groups of genomic regions (or between a group of genomic regions and a reference center of symmetry), and does not require fixing location and scale at the outset.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Differential gene expression analysis'],\n",
       "  'edam_topics': ['Statistics and probability']},\n",
       " {'id': 'jags',\n",
       "  'description': 'JAGS is Just Another Gibbs Sampler. It is a program for analysis of Bayesian hierarchical models using Markov Chain Monte Carlo (MCMC) simulation.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Statistics and probability']},\n",
       " {'id': 'jasminesv',\n",
       "  'description': 'JASMINE (Jointly Accurate Sv Merging with Intersample Network Edges) is an automated pipeline for alignment and SV calling in long-read datasets. The tool is used to merge structural variants (SVs) across samples. Each sample has a number of SV calls, consisting of position information (chromosome, start, end, length), type and strand information, and a number of other values. Jasmine represents the set of all SVs across samples as a network, and uses a modified minimum spanning forest algorithm to determine the best way of merging the variants such that each merged variants represents a set of analogous variants occurring in different samples.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping',\n",
       "   'Variant calling',\n",
       "   'Genome indexing',\n",
       "   'Visualisation'],\n",
       "  'edam_topics': ['Structural variation',\n",
       "   'DNA polymorphism',\n",
       "   'Workflows',\n",
       "   'Mapping',\n",
       "   'RNA-Seq']},\n",
       " {'id': 'jbigkit',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'jbrowse',\n",
       "  'description': 'Slick, speedy genome browser with a responsive and dynamic AJAX interface for visualization of genome data. Being developed by the GMOD project as a successor to GBrowse.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome visualisation'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'jbrowse2',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'jcvi',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'jellyfish',\n",
       "  'description': 'A command-line algorithm for counting k-mers in DNA sequence.',\n",
       "  'edam_inputs': [{'formats': ['FASTA', 'FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['K-mer countgraph'],\n",
       "    'term': 'K-mer countgraph'}],\n",
       "  'edam_operations': ['k-mer counting'],\n",
       "  'edam_topics': ['Sequence analysis', 'Genomics']},\n",
       " {'id': 'jq',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'juicer',\n",
       "  'description': 'Juicer is a platform for analyzing kilobase resolution Hi-C data. In this distribution, we include the pipeline for generating Hi-C maps from fastq raw data files and command line tools for feature annotation on the Hi-C maps.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome indexing', 'Visualisation'],\n",
       "  'edam_topics': ['Sequencing', 'Mapping', 'Genomics', 'Epigenetics']},\n",
       " {'id': 'jupyterlab',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'jupyter_notebook',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'jupyter-server',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'kallisto',\n",
       "  'description': 'A program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene expression profiling'],\n",
       "  'edam_topics': ['Transcriptomics', 'RNA-Seq', 'Gene expression']},\n",
       " {'id': 'kat',\n",
       "  'description': 'Suite of tools that generate, analyse and compare k-mer spectra produced from sequence files',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence composition calculation',\n",
       "   'Sequence feature detection'],\n",
       "  'edam_topics': ['Sequence analysis', 'Genomics']},\n",
       " {'id': 'kentutils',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'khmer',\n",
       "  'description': 'khmer is a set of command-line tools for working with DNA shotgun sequencing data from genomes, transcriptomes, metagenomes, and single cells. khmer can make de novo assemblies faster, and sometimes better. khmer can also identify (and fix) problems with shotgun data.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-like format (text)', 'FASTA'],\n",
       "    'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTQ-like format (text)', 'FASTA'],\n",
       "    'term': 'FASTA'}],\n",
       "  'edam_operations': ['Standardisation and normalisation'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'kmc',\n",
       "  'description': 'KMC is a utility designed for counting k-mers (sequences of consecutive k symbols) in a set of reads from genome sequencing projects.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['k-mer counting'],\n",
       "  'edam_topics': ['Whole genome sequencing', 'Genomics', 'Sequence assembly']},\n",
       " {'id': 'kofamscan',\n",
       "  'description': 'KofamScan is a gene function annotation tool based on KEGG Orthology and hidden Markov model. You need KOfam database to use this tool.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis', 'Gene functional annotation'],\n",
       "  'edam_topics': ['Genomics', 'Structure analysis']},\n",
       " {'id': 'kraken',\n",
       "  'description': 'System for assigning taxonomic labels to short DNA sequences, usually obtained through metagenomic studies. Previous attempts by other bioinformatics software to accomplish this task have often used sequence alignment or machine learning techniques that were quite slow, leading to the development of less sensitive but much faster abundance estimation programs. It aims to achieve high sensitivity and high speed by utilizing exact alignments of k-mers and a novel classification algorithm.',\n",
       "  'edam_inputs': [{'formats': ['FASTA-like'], 'term': 'FASTA-like'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Taxonomic classification'],\n",
       "  'edam_topics': ['Taxonomy', 'Metagenomics']},\n",
       " {'id': 'kraken2',\n",
       "  'description': 'Kraken 2 is the newest version of Kraken, a taxonomic classification system using exact k-mer matches to achieve high accuracy and fast classification speeds. This classifier matches each k-mer within a query sequence to the lowest common ancestor (LCA) of all genomes containing the given k-mer. The k-mer assignments inform the classification algorithm.\\nAny assumption that Krakens raw read assignments can be directly translated into species or strain-level abundance estimates is flawed. Bracken (Bayesian Reestimation of Abundance after Classification with KrakEN), estimates species abundances in metagenomics samples by probabilistically re-distributing reads in the taxonomic tree. (Lu, Jennifer et al. Bracken: estimating species abundance in metagenomics data.)',\n",
       "  'edam_inputs': [{'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Taxonomic classification'],\n",
       "  'edam_topics': ['Taxonomy',\n",
       "   'Metagenomics',\n",
       "   'Microbial ecology',\n",
       "   'Microbiology']},\n",
       " {'id': 'krakentools',\n",
       "  'description': 'KrakenTools provides individual scripts to analyze Kraken/Kraken2/Bracken/KrakenUniq output files',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Visualisation'],\n",
       "  'edam_topics': ['Taxonomy', 'Metagenomics']},\n",
       " {'id': 'krona',\n",
       "  'description': 'Krona creates interactive HTML5 charts of hierarchical data (such as taxonomic abundance in a metagenome).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Visualisation'],\n",
       "  'edam_topics': ['Metagenomics']},\n",
       " {'id': 'kronatools',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'lama',\n",
       "  'description': 'Automated image analysis for developmental phenotyping of mouse embryos.\\n\\nLAMA (Lightweight Analysis of Morphological Abnormalities).\\n\\nWelcome to LAMA, an open source pipeline to automatically identify embryo dysmorphology from 3D volumetric images.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis',\n",
       "   'Image annotation',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Tomography', 'Genotype and phenotype', 'MRI', 'Workflows']},\n",
       " {'id': 'lastz',\n",
       "  'description': 'A tool for (1) aligning two DNA sequences, and (2) inferring appropriate scoring parameters automatically.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment', 'Read mapping'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'length_and_gc_content',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'LFQ-Analyst',\n",
       "  'description': 'An Easy-To-Use Interactive Web Platform To Analyze and Visualize Label-Free Proteomics Data Preprocessed with MaxQuant.\\n\\nA tool for analysing label-free quantitative proteomics dataset https://bioinformatics.erc.monash.edu/apps/LFQ-Analyst/.\\n\\nLFQ-Analyst: An easy-to-use interactive web-platform to analyze and visualize proteomics data preprocessed with MaxQuant.\\n\\nLFQ-Analyst is an easy-to-use, interactive web application developed to perform differential expression analysis with one click and to visualize label-free quantitative proteomic datasets preprocessed with MaxQuant. LFQ-Analyst provides a wealth of user-analytic features and offers numerous publication-quality result output graphics and tables to facilitate statistical and exploratory analysis of label-free quantitative datasets',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Essential dynamics',\n",
       "   'Principal component visualisation',\n",
       "   'Label-free quantification',\n",
       "   'Imputation',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Sequence analysis',\n",
       "   'Small molecules',\n",
       "   'Metabolomics',\n",
       "   'Proteomics experiment']},\n",
       " {'id': 'liftOver1',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'limma',\n",
       "  'description': 'Data analysis, linear models and differential expression for microarray data.',\n",
       "  'edam_inputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_outputs': [{'formats': ['Image format'], 'term': 'Image format'}],\n",
       "  'edam_operations': ['RNA-Seq analysis'],\n",
       "  'edam_topics': ['Molecular biology', 'Genetics']},\n",
       " {'id': 'links',\n",
       "  'description': 'LINKS (Long Interval Nucleotide K-mer Scaffolder) is a genomics application for scaffolding genome assemblies with long reads, such as those produced by Oxford Nanopore Technologies Ltd. It can be used to scaffold high-quality draft genome assemblies with any long sequences (eg. ONT reads, PacBio reads, other draft genomes, etc). It is also used to scaffold contig pairs linked by ARCS/ARKS.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Scaffolding',\n",
       "   'Genome assembly',\n",
       "   'Read mapping',\n",
       "   'Read pre-processing',\n",
       "   'Sequence trimming'],\n",
       "  'edam_topics': ['Sequence assembly', 'Mapping', 'Sequencing']},\n",
       " {'id': 'lofreq',\n",
       "  'description': 'LoFreq* (i.e. LoFreq version 2) is a fast and sensitive variant-caller for inferring SNVs and indels from next-generation sequencing data. It makes full use of base-call qualities and other sources of errors inherent in sequencing (e.g. mapping or base/indel alignment uncertainty), which are usually ignored by other methods or only used for filtering.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling'],\n",
       "  'edam_topics': ['Genomics', 'Genetic variation']},\n",
       " {'id': 'longshot',\n",
       "  'description': 'Longshot is a variant calling tool for diploid genomes using long error prone reads such as Pacific Biosciences (PacBio) SMRT and Oxford Nanopore Technologies (ONT). It takes as input an aligned BAM file and outputs a phased VCF file with variants and haplotype information. It can also genotype and phase input VCF files. It can output haplotype-separated BAM files that can be used for downstream analysis. Currently, it only calls single nucleotide variants (SNVs), but it can genotype indels if they are given in an input VCF.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling'],\n",
       "  'edam_topics': ['Genetic variation']},\n",
       " {'id': 'longtr',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'lotus2',\n",
       "  'description': 'LotuS2 is a lightweight and user-friendly pipeline that is fast, precise, and streamlined, using extensive pre- and post-ASV/OTU clustering steps to further increase data quality. High data usage rates and reliability enable high-throughput microbiome analysis in minutes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence feature detection', 'DNA barcoding'],\n",
       "  'edam_topics': ['Metagenomics', 'Taxonomy', 'Microbial ecology']},\n",
       " {'id': 'lpsolve',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ltr_retriever',\n",
       "  'description': 'LTR_retriever is a highly accurate and sensitive program for identification of LTR retrotransposons; The LTR Assembly Index (LAI) is also included in this package.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'lua',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'macs2',\n",
       "  'description': 'Model-based Analysis of ChIP-seq data.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['BAM', 'BED', 'SAM', 'TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['BED', 'WIG'], 'term': 'WIG'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Peak calling',\n",
       "   'Enrichment analysis',\n",
       "   'Gene regulatory network analysis'],\n",
       "  'edam_topics': ['ChIP-seq',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'macs3',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'maeparser',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'mafft',\n",
       "  'description': 'MAFFT (Multiple Alignment using Fast Fourier Transform) is a high speed multiple sequence alignment program.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Multiple sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'MAGeCK',\n",
       "  'description': 'Computational tool to identify important genes from the recent genome-scale CRISPR-Cas9 knockout screens technology.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genetic variation analysis'],\n",
       "  'edam_topics': ['Genetics', 'Genetic variation', 'Genomics']},\n",
       " {'id': 'maker',\n",
       "  'description': 'Portable and easily configurable genome annotation pipeline. Its purpose is to allow smaller eukaryotic and prokaryotic genome projects to independently annotate their genomes and to create genome databases.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['Genomics', 'DNA', 'Sequence analysis']},\n",
       " {'id': 'maldi_quant',\n",
       "  'description': 'MALDIquant is a complete analysis pipeline for matrix-assisted laser desorption/ionization-time-of-flight (MALDI-TOF) and other two-dimensional mass spectrometry data. In addition to commonly used plotting and processing methods it includes distinctive features, namely baseline subtraction methods such as morphological filters (TopHat) or the statistics-sensitive non-linear iterative peak-clipping algorithm (SNIP), peak alignment using warping functions, handling of replicated measurements as well as allowing spectra with different resolutions.',\n",
       "  'edam_inputs': [{'formats': ['imzML metadata file'],\n",
       "    'term': 'imzML metadata file'},\n",
       "   {'formats': ['ibd'], 'term': 'ibd'}],\n",
       "  'edam_outputs': [{'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_operations': ['Quantification',\n",
       "   'Spectrum calculation',\n",
       "   'Peak detection'],\n",
       "  'edam_topics': ['Proteomics experiment', 'Proteomics']},\n",
       " {'id': 'mash',\n",
       "  'description': 'Fast genome and metagenome distance estimation using MinHash.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence distance matrix generation'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'Metagenomics',\n",
       "   'Statistics and probability',\n",
       "   'Sequence analysis',\n",
       "   'DNA mutation']},\n",
       " {'id': 'master2pgSnp',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'masurca',\n",
       "  'description': 'Whole genome assembly software. It combines the efficiency of the de\\xa0Bruijn graph and Overlap-Layout-Consensus (OLC) approaches. MaSuRCA can assemble data sets containing only short reads from Illumina sequencing or a mixture of short reads and long reads (Sanger, 454).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Whole genome sequencing',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'matchms',\n",
       "  'description': 'Tool to import, process, clean, and compare mass spectrometry data.',\n",
       "  'edam_inputs': [{'formats': ['MGF', 'JSON'], 'term': 'JSON'}],\n",
       "  'edam_outputs': [{'formats': ['JSON'], 'term': 'JSON'},\n",
       "   {'formats': ['JSON'], 'term': 'JSON'},\n",
       "   {'formats': ['Mass spectrometry data format'],\n",
       "    'term': 'Mass spectrometry data format'}],\n",
       "  'edam_operations': ['Spectral library search',\n",
       "   'Format validation',\n",
       "   'Filtering'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'matplotlib',\n",
       "  'description': 'Matplotlib is a comprehensive library for creating static, animated, and interactive visualizations in Python. Matplotlib makes easy things easy and hard things possible.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Visualisation'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'maxbin',\n",
       "  'description': 'Software for binning assembled metagenomic sequences based on an Expectation-Maximization algorithm.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly'],\n",
       "  'edam_topics': ['Metagenomics', 'Sequence assembly', 'Microbiology']},\n",
       " {'id': 'maxquant',\n",
       "  'description': 'Quantitative proteomics software package designed for analyzing large mass-spectrometric data sets. It is specifically aimed at high-resolution MS data.',\n",
       "  'edam_inputs': [{'formats': ['CSV', 'TSV', 'SEQUEST .out file', 'xlsx'],\n",
       "    'term': 'xlsx'}],\n",
       "  'edam_outputs': [{'formats': ['CSV'], 'term': 'CSV'},\n",
       "   {'formats': ['JPG', 'PNG'], 'term': 'PNG'}],\n",
       "  'edam_operations': ['Imputation',\n",
       "   'Visualisation',\n",
       "   'Protein quantification',\n",
       "   'Statistical calculation',\n",
       "   'Standardisation and normalisation',\n",
       "   'Heat map generation',\n",
       "   'Clustering',\n",
       "   'Principal component plotting'],\n",
       "  'edam_topics': ['Proteomics experiment',\n",
       "   'Proteomics',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'mcl',\n",
       "  'description': 'MCL is a clustering algorithm widely used in bioinformatics and gaining traction in other fields.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Pathway or network'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Clustering',\n",
       "   'Network analysis',\n",
       "   'Gene regulatory network analysis'],\n",
       "  'edam_topics': ['Molecular interactions, pathways and networks']},\n",
       " {'id': 'mcquant',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'mdanalysis',\n",
       "  'description': 'MDAnalysis is an object-oriented python toolkit to analyze molecular dynamics trajectories generated by CHARMM, Gromacs, NAMD, LAMMPS, Amber or DL_POLY; it also reads other formats (e.g. PDB files and XYZ format trajectories; see the supported coordinate formats for the full list). It can write most of these formats, too, together with atom selections for use in Gromacs, CHARMM, VMD and PyMol',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Molecular dynamics'],\n",
       "  'edam_topics': ['Molecular dynamics']},\n",
       " {'id': 'MDTraj',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'medaka',\n",
       "  'description': 'medaka is a tool to create consensus sequences and variant calls from nanopore sequencing data. This task is performed using neural networks applied a pileup of individual sequencing reads against a draft assembly.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Base-calling', 'Variant calling', 'Sequence assembly'],\n",
       "  'edam_topics': ['Sequence assembly', 'Machine learning']},\n",
       " {'id': 'megahit',\n",
       "  'description': 'Single node assembler for large and complex metagenomics NGS reads, such as soil. It makes use of succinct de Bruijn graph to achieve low memory usage, whereas its goal is not to make memory usage as low as possible.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Sequencing',\n",
       "   'Ecology',\n",
       "   'Sequence assembly']},\n",
       " {'id': 'megan',\n",
       "  'description': 'Metagenome Analysis Software - MEGAN (MEtaGenome ANalyzer\\x9d) is a new computer program that allows laptop analysis of large metagenomic datasets. In a preprocessing step, the set of DNA reads (or contigs) is compared against databases of known sequences using BLAST or another comparison tool. MEGAN can then be used to compute and interactively explore the taxonomical content of the dataset, employing the NCBI taxonomy to summarize and order the results.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis', 'Taxonomic classification'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'memote',\n",
       "  'description': 'Testing of metabolic models',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Metabolic network modelling',\n",
       "   'Metabolic pathway prediction'],\n",
       "  'edam_topics': ['Endocrinology and metabolism',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'merqury',\n",
       "  'description': 'Reference-free quality, completeness, and phasing assessment for genome assemblies.\\n\\nEvaluate genome assemblies with k-mers and more.\\nOften, genome assembly projects have illumina whole genome sequencing reads available for the assembled individual.\\n\\nMerqury provides a set of tools for this purpose.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'k-mer counting',\n",
       "   'Scaffolding',\n",
       "   'Phasing',\n",
       "   'De-novo assembly'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Whole genome sequencing',\n",
       "   'Plant biology']},\n",
       " {'id': 'meryl',\n",
       "  'description': 'Meryl is a tool for counting and working with sets of k-mers that was originally developed for use in the Celera Assembler and has since been migrated and maintained as part of Canu.',\n",
       "  'edam_inputs': [{'formats': ['FASTA', 'FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['K-mer countgraph'],\n",
       "    'term': 'K-mer countgraph'}],\n",
       "  'edam_operations': ['k-mer counting'],\n",
       "  'edam_topics': ['Whole genome sequencing',\n",
       "   'Genomics',\n",
       "   'Sequence analysis',\n",
       "   'Sequencing']},\n",
       " {'id': 'metabat',\n",
       "  'description': 'an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies | MetaBAT2 clusters metagenomic contigs into different \"bins\", each of which should correspond to a putative genome | MetaBAT2 uses nucleotide composition information and source strain abundance (measured by depth-of-coverage by aligning the reads to the contigs) to perform binning',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read binning',\n",
       "   'Sequence assembly',\n",
       "   'Genome annotation'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Sequence assembly',\n",
       "   'Metagenomic sequencing']},\n",
       " {'id': 'metabolic',\n",
       "  'description': 'A scalable high-throughput metabolic and biogeochemical functional trait profiler based on microbial genomes | A scalable high-throughput metabolic and biogeochemical functional trait profiler | This software enables the prediction of metabolic and biogeochemical functional trait profiles to any given genome datasets. These genome datasets can either be metagenome-assembled genomes (MAGs), single-cell amplified genomes (SAGs) or pure culture sequenced genomes. It can also calculate the genome coverage. The information is parsed and diagrams for elemental/biogeochemical cycling pathways (currently Nitrogen, Carbon, Sulfur and \"other\") are produced',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Metabolic network modelling',\n",
       "   'Metabolic labeling',\n",
       "   'Parsing'],\n",
       "  'edam_topics': ['Endocrinology and metabolism',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'Microbial ecology']},\n",
       " {'id': 'MetaDEGalaxy',\n",
       "  'description': 'Galaxy workflow for differential abundance analysis of 16s metagenomic data.\\n\\nYou are over your disk quota. Tool execution is on hold until your disk usage drops below your allocated quota.\\n\\nThis history is empty. You can load your own data or get data from an external source',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read pre-processing',\n",
       "   'Standardisation and normalisation',\n",
       "   'Sequence merging'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Metagenomic sequencing',\n",
       "   'Workflows',\n",
       "   'Microbial ecology',\n",
       "   'RNA-Seq']},\n",
       " {'id': 'metaeuk',\n",
       "  'description': 'MetaEuk - sensitive, high-throughput gene discovery and annotation for large-scale eukaryotic metagenomics',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Homology-based gene prediction'],\n",
       "  'edam_topics': ['Metagenomics', 'Gene and protein families']},\n",
       " {'id': 'metanovo',\n",
       "  'description': 'An open-source pipeline for probabilistic peptide discovery in complex metaproteomic datasets.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Target-Decoy',\n",
       "   'de Novo sequencing',\n",
       "   'Tag-based peptide identification',\n",
       "   'Protein identification',\n",
       "   'Expression analysis'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Microbial ecology',\n",
       "   'Metagenomics',\n",
       "   'Proteomics experiment',\n",
       "   'Small molecules']},\n",
       " {'id': 'metaphlan',\n",
       "  'description': 'Computational tool for profiling the composition of microbial communities from metagenomic shotgun sequencing data.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ', 'FASTA', 'SAM'], 'term': 'SAM'}],\n",
       "  'edam_outputs': [{'formats': ['DSV'], 'term': 'DSV'}],\n",
       "  'edam_operations': ['Taxonomic classification'],\n",
       "  'edam_topics': ['Metagenomics', 'Phylogenomics']},\n",
       " {'id': 'metaQuantome',\n",
       "  'description': 'metaQuantome software suite analyzes the state of a microbiome by leveraging complex taxonomic and functional hierarchies to summarize peptide-level quantitative information. metaQuantome offers differential abundance analysis, principal components analysis, and clustered heat map visualizations, as well as exploratory analysis for a single sample or experimental condition.',\n",
       "  'edam_inputs': [{'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_outputs': [{'formats': ['Image format'], 'term': 'Image format'},\n",
       "   {'formats': ['Image format'], 'term': 'Image format'},\n",
       "   {'formats': ['Image format'], 'term': 'Image format'}],\n",
       "  'edam_operations': ['Principal component visualisation',\n",
       "   'Visualisation',\n",
       "   'Functional clustering',\n",
       "   'Query and retrieval',\n",
       "   'Differential protein expression analysis',\n",
       "   'Heat map generation',\n",
       "   'Quantification',\n",
       "   'Indexing',\n",
       "   'Filtering',\n",
       "   'Statistical inference'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Metatranscriptomics',\n",
       "   'Microbial ecology',\n",
       "   'Proteomics experiment',\n",
       "   'Metagenomics']},\n",
       " {'id': 'metaspades',\n",
       "  'description': 'Genome assembler for metagenomics datasets.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Genome alignment'],\n",
       "  'edam_topics': ['Metagenomics', 'Sequence analysis']},\n",
       " {'id': 'metawrap',\n",
       "  'description': 'MetaWRAP aims to be an easy-to-use metagenomic wrapper suite that accomplishes the core tasks of metagenomic analysis from start to finish: read quality control, assembly, visualization, taxonomic profiling, extracting draft genomes (binning), and functional annotation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read binning',\n",
       "   'Sequence assembly',\n",
       "   'Genome annotation',\n",
       "   'Sequence trimming',\n",
       "   'Demultiplexing'],\n",
       "  'edam_topics': ['Whole genome sequencing',\n",
       "   'Metagenomic sequencing',\n",
       "   'Metagenomics']},\n",
       " {'id': 'MethylDackel',\n",
       "  'description': 'A (mostly) universal methylation extractor for BS-seq experiments.',\n",
       "  'edam_inputs': [{'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['bedgraph'], 'term': 'bedgraph'}],\n",
       "  'edam_operations': ['Sequence alignment conversion', 'Methylation calling'],\n",
       "  'edam_topics': ['Epigenomics']},\n",
       " {'id': 'metilene',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'MIGRATE',\n",
       "  'description': 'Estimates effective population sizes,past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes, and population divergences or admixture.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Haplotype mapping',\n",
       "   'Phylogenetic tree generation (maximum likelihood and Bayesian methods)',\n",
       "   'Cell migration analysis'],\n",
       "  'edam_topics': ['Population genetics',\n",
       "   'Nucleic acid structure analysis',\n",
       "   'Zoology']},\n",
       " {'id': 'mikado',\n",
       "  'description': 'A lightweight Python3 pipeline whose purpose is to facilitate the identification of expressed loci from RNA-Seq data and to select the best models in each locus.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Splice transcript prediction',\n",
       "   'Gene prediction',\n",
       "   'RNA-Seq analysis'],\n",
       "  'edam_topics': ['RNA-seq', 'Gene transcripts']},\n",
       " {'id': 'mimodd',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'minia',\n",
       "  'description': 'Short-read assembler based on a de Bruijn graph, capable of assembling a human genome on a desktop computer in a day.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-like format (text)', 'FASTA'],\n",
       "    'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'miniasm',\n",
       "  'description': 'Miniasm is a very fast OLC-based de novo assembler for noisy long reads. It takes all-vs-all read self-mappings (typically by minimap) as input and outputs an assembly graph in the GFA format.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly'],\n",
       "  'edam_topics': ['Genomics', 'Sequence assembly']},\n",
       " {'id': 'miniconda',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'minigraph',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'minimap2',\n",
       "  'description': 'Pairwise aligner for genomic and spliced nucleotide sequences',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Pairwise sequence alignment'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'miniprot',\n",
       "  'description': 'Miniprot aligns a protein sequence against a genome with affine gap penalty, splicing and frameshift. It is primarily intended for annotating protein-coding genes in a new species using known genes from other species.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment', 'Protein sequence analysis'],\n",
       "  'edam_topics': ['Sequence sites, features and motifs', 'Sequence analysis']},\n",
       " {'id': 'mira',\n",
       "  'description': 'MIRA 3 - Whole Genome Shotgun and EST Sequence Assembler',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Local alignment',\n",
       "   'k-mer counting',\n",
       "   'Sequence assembly',\n",
       "   'Read mapping'],\n",
       "  'edam_topics': ['Sequence assembly', 'RNA-Seq']},\n",
       " {'id': 'mircounts',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'mirdeep2',\n",
       "  'description': 'miRDeep2 discovers active known or novel miRNAs from deep sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['miRNA target prediction', 'miRNA expression analysis'],\n",
       "  'edam_topics': ['Functional, regulatory and non-coding RNA',\n",
       "   'RNA immunoprecipitation']},\n",
       " {'id': 'mitobim',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'mitohifi',\n",
       "  'description': 'Find, circularise and annotate mitogenome from PacBio assemblies',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'De-novo assembly',\n",
       "   'Genome annotation',\n",
       "   'Sequence classification'],\n",
       "  'edam_topics': ['Sequence assembly', 'Sequence analysis', 'Genomics']},\n",
       " {'id': 'mitos2',\n",
       "  'description': 'De novo metazoan mitochondrial genome annotation.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['BED', 'GFF', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['Zoology', 'Whole genome sequencing']},\n",
       " {'id': 'mlst',\n",
       "  'description': 'Multi Locus Sequence Typing from an assembled genome or from a set of reads.',\n",
       "  'edam_inputs': [{'formats': ['FASTA-like', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_operations': ['Multilocus sequence typing'],\n",
       "  'edam_topics': ['Immunoproteins and antigens']},\n",
       " {'id': 'mmseqs2',\n",
       "  'description': 'MMseqs2 (Many-against-Many sequence searching) is a software suite to search and cluster huge protein and nucleotide sequence sets. MMseqs2 is open source software implemented in C++ for Linux, MacOS, and (as beta version, via cygwin) Windows. The software is designed to run on multiple cores and servers and exhibits very good scalability. MMseqs2 can run 10000 times faster than BLAST. At 100 times its speed it achieves almost the same sensitivity. It can perform profile searches with the same sensitivity as PSI-BLAST at over 400 times its speed.\\n\\nMMseqs2 includes Linclust, the first clustering algorithm whose runtime scales linearly With Linclust we clustered 1.6 billion metagenomic sequence fragments in 10\\u2009h on a single server to 50% sequence identity.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-like format (text)',\n",
       "     'FASTA-like (text)'],\n",
       "    'term': 'FASTA-like (text)'}],\n",
       "  'edam_outputs': [{'formats': ['BLAST results'], 'term': 'BLAST results'}],\n",
       "  'edam_operations': ['Sequence similarity search', 'Sequence alignment'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Sequence analysis',\n",
       "   'Proteins',\n",
       "   'Nucleic acids',\n",
       "   'Gene and protein families',\n",
       "   'Taxonomy']},\n",
       " {'id': 'mob-suite',\n",
       "  'description': 'Universal whole-sequence-based plasmid typing and its utility to prediction of host range and epidemiological surveillance.\\n\\nMOB-suite: Software tools for clustering, reconstruction and typing of plasmids from draft assemblies.\\n\\nPlasmids are mobile genetic elements (MGEs), which allow for rapid evolution and adaption of bacteria to new niches through horizontal transmission of novel traits to different genetic backgrounds. The MOB-suite is designed to be a modular set of tools for the typing and reconstruction of plasmid sequences from WGS assemblies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Clustering',\n",
       "   'Plasmid map drawing',\n",
       "   'Sequence assembly',\n",
       "   'Deposition'],\n",
       "  'edam_topics': ['Mobile genetic elements',\n",
       "   'Whole genome sequencing',\n",
       "   'Public health and epidemiology',\n",
       "   'Sequence assembly',\n",
       "   'Biomarkers']},\n",
       " {'id': 'moff',\n",
       "  'description': 'A modest Feature Finder to extract features in MS1 Data.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Mass spectrometry data'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence motif (protein)'}],\n",
       "  'edam_operations': ['Protein feature detection'],\n",
       "  'edam_topics': ['Proteomics']},\n",
       " {'id': 'monailabel',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'morpheus',\n",
       "  'description': 'A proteomics search algorithm specifically designed for high-resolution tandem mass spectra.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Peptide database search'],\n",
       "  'edam_topics': ['Proteomics']},\n",
       " {'id': 'mosdepth',\n",
       "  'description': 'Fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Nucleic acid sequence analysis', 'Read depth analysis'],\n",
       "  'edam_topics': ['Statistics and probability', 'Data quality management']},\n",
       " {'id': 'mothur',\n",
       "  'description': 'Open-source, platform-independent, community-supported software for describing and comparing microbial communities',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['DNA barcoding',\n",
       "   'Sequencing quality control',\n",
       "   'Sequence clustering',\n",
       "   'Taxonomic classification',\n",
       "   'Visualisation',\n",
       "   'Sequence read processing',\n",
       "   'Phylogenetic analysis'],\n",
       "  'edam_topics': ['Microbial ecology',\n",
       "   'Taxonomy',\n",
       "   'Sequence analysis',\n",
       "   'Phylogeny']},\n",
       " {'id': 'mousemine',\n",
       "  'description': 'Data warehouse for accessing mouse data from Mouse Genome Informatics (MGI). Supports powerful query, reporting, and analysis capabilities, the ability to save and combine results from different queries, easy integration into larger workflows, and a comprehensive Web Services layer.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Query and retrieval'],\n",
       "  'edam_topics': ['Model organisms', 'Genotype and phenotype']},\n",
       " {'id': 'mrbayes',\n",
       "  'description': 'Program for Bayesian inference and model choice across a wide range of phylogenetic and evolutionary models. It uses Markov chain Monte Carlo (MCMC) methods to estimate the posterior distribution of model parameters.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree generation (maximum likelihood and Bayesian methods)'],\n",
       "  'edam_topics': ['Phylogenetics']},\n",
       " {'id': 'mrtrix',\n",
       "  'description': 'A fast, flexible and open software framework for medical image processing and visualisation | MRtrix3 is an open-source, cross-platform software package for medical image processing, analysis and visualisation, with a particular emphasis on the investigation of the brain using diffusion MRI. It is implemented using a fast, modular and flexible general-purpose code framework for image data access and manipulation, enabling efficient development of new applications, whilst retaining high computational performance and a consistent command-line interface between applications. In this article, we provide a high-level overview of the features of the MRtrix3 framework and general-purpose image processing applications provided with the software',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis', 'Visualisation'],\n",
       "  'edam_topics': ['MRI', 'Medical imaging', 'Data visualisation']},\n",
       " {'id': 'msconvert',\n",
       "  'description': 'msConvert is a command-line utility for converting between various mass spectrometry data formats, including from raw data from several commercial companies (with vendor libraries, Windows-only). For Windows users, there is also a GUI, msConvertGUI.',\n",
       "  'edam_inputs': [{'formats': ['Textual format',\n",
       "     'Thermo RAW',\n",
       "     'mzXML',\n",
       "     'mzML',\n",
       "     'MGF',\n",
       "     'mzIdentML'],\n",
       "    'term': 'mzIdentML'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format',\n",
       "     'Thermo RAW',\n",
       "     'mzXML',\n",
       "     'mzML',\n",
       "     'MGF',\n",
       "     'mzIdentML'],\n",
       "    'term': 'mzIdentML'}],\n",
       "  'edam_operations': ['Filtering', 'Formatting'],\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment']},\n",
       " {'id': 'msfinder',\n",
       "  'description': 'This is a modified copy of MS-FINDER with source code modifications to make the tool accessible in Galaxy.\\n\\nMS-FINDER - software for structure elucidation of unknown spectra with hydrogen rearrangement (HR) rules\\n\\nThe program supports molecular formula prediction, metabolie class prediction, and structure elucidation for EI-MS and MS/MS spectra, and the assembly is licensed under the CC-BY 4.0.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Mass spectrometry data'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Expression data'}],\n",
       "  'edam_operations': ['Annotation'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'msmetaenhancer',\n",
       "  'description': 'Tool for mass spectra metadata annotation.',\n",
       "  'edam_inputs': [{'formats': ['MGF', 'JSON', 'CSV', 'TSV', 'xlsx'],\n",
       "    'term': 'xlsx'},\n",
       "   {'formats': ['SMILES', 'InChI', 'InChIKey', 'Textual format'],\n",
       "    'term': 'Textual format'}],\n",
       "  'edam_outputs': [{'formats': ['MGF', 'JSON', 'CSV', 'TSV', 'xlsx'],\n",
       "    'term': 'xlsx'},\n",
       "   {'formats': [], 'term': 'Sample annotation'},\n",
       "   {'formats': ['InChI', 'SMILES'], 'term': 'SMILES'},\n",
       "   {'formats': [], 'term': 'Database ID'},\n",
       "   {'formats': [], 'term': 'PubChem ID'},\n",
       "   {'formats': [], 'term': 'Chemical name (IUPAC)'},\n",
       "   {'formats': [], 'term': 'CAS number'},\n",
       "   {'formats': [], 'term': 'ChEBI ID'}],\n",
       "  'edam_operations': ['Annotation', 'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Metabolomics',\n",
       "   'Compound libraries and screening',\n",
       "   'Data submission, annotation and curation']},\n",
       " {'id': 'msstats',\n",
       "  'description': 'Statistical tool for quantitative mass spectrometry-based proteomics.',\n",
       "  'edam_inputs': [{'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_outputs': [{'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_operations': ['Statistical modelling',\n",
       "   'Data handling',\n",
       "   'Differential protein expression analysis',\n",
       "   'Statistical inference',\n",
       "   'Quantification',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Proteomics experiment',\n",
       "   'Statistics and probability',\n",
       "   'Workflows']},\n",
       " {'id': 'bioconductor-msstatstmt',\n",
       "  'description': 'Tools for detecting differentially abundant peptides and proteins in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Spectrum calculation',\n",
       "   'Tag-based peptide identification',\n",
       "   'Differential protein expression profiling'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Proteomics experiment',\n",
       "   'Protein expression']},\n",
       " {'id': 'mtag',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'multigsea',\n",
       "  'description': 'A GSEA-based pathway enrichment analysis for multi-omics data.\\n\\nmultiGSEA: a GSEA-based pathway enrichment analysis for multi-omics data, BMC Bioinformatics 21, 561 (2020).\\n\\nCombining GSEA-based pathway enrichment with multi omics data integration.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene-set enrichment analysis',\n",
       "   'Aggregation',\n",
       "   'Pathway analysis'],\n",
       "  'edam_topics': ['Metabolomics',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'Proteomics',\n",
       "   'Transcriptomics',\n",
       "   'Small molecules']},\n",
       " {'id': 'mti',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'multiqc',\n",
       "  'description': \"MultiQC aggregates results from multiple bioinformatics analyses across many samples into a single report. It searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.\",\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [{'formats': ['CSV', 'JSON', 'HTML', 'YAML', 'TSV'],\n",
       "    'term': 'TSV'},\n",
       "   {'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Validation', 'Sequencing quality control'],\n",
       "  'edam_topics': ['Sequencing',\n",
       "   'Bioinformatics',\n",
       "   'Sequence analysis',\n",
       "   'Genomics']},\n",
       " {'id': 'mummer',\n",
       "  'description': 'MUMmer is a modular system for the rapid whole genome alignment of finished or draft sequence. Basically it is a ultra-fast alignment of large-scale DNA and protein sequences',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment', 'Read mapping'],\n",
       "  'edam_topics': ['Genomics', 'Mapping', 'Sequencing']},\n",
       " {'id': 'music_compare',\n",
       "  'description': 'MuSiC is a suite of programs that evaluate the biophysical effects of amino acid mutations in proteins. They request the experimental or modeled 3-dimensional protein structure as input, and predict the impact of specific single-site mutations requested by the user or of all possible single-site mutations.\\n\\nPoPMuSiC and HoTMuSiC predict the changes in thermodynamic and thermal stability, respectively, upon mutation.  They are helpful for the rational design of modified proteins with controlled stability properties. \\n\\nSNPMuSiC predicts whether protein variants are deleterious or benign due to stability issues, thus providing a molecular-level interpretation of disease phenotype.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant effect prediction',\n",
       "   'Design',\n",
       "   'Protein property calculation'],\n",
       "  'edam_topics': ['Protein folding, stability and design',\n",
       "   'Genetic variation',\n",
       "   'Computational biology',\n",
       "   'Bioengineering',\n",
       "   'Protein properties']},\n",
       " {'id': 'music_deconvolution',\n",
       "  'description': 'MuSiC utilizes cell-type specific gene expression from single-cell RNA sequencing (RNA-seq) data to characterize cell type compositions from bulk RNA-seq data in complex tissues. By appropriate weighting of genes showing cross-subject and cross-cell consistency, MuSiC enables the transfer of cell type-specific gene expression information from one dataset to another.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'mztosqlite',\n",
       "  'description': 'Convert proteomics data files into a SQLite database\\n\\nExtract data from proteomics mzIdentML and mass spec scan files: mzML, MGF, etc and store in a SQLite database.\\nThe intended purpose is to provide a Galaxy dataset that can support an interactive Galaxy visualization plugin.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Conversion', 'Peptide database search'],\n",
       "  'edam_topics': ['Proteomics', 'Biological databases']},\n",
       " {'id': 'nanocompore',\n",
       "  'description': 'RNA modifications detection by comparative Nanopore direct RNA sequencing.\\n\\nRNA modifications detection from Nanopore dRNA-Seq data.\\n\\nNanocompore identifies differences in ONT nanopore sequencing raw signal corresponding to RNA modifications by comparing 2 samples.\\n\\nAnalyses performed for the nanocompore paper.\\n\\nNanocompore compares 2 ONT nanopore direct RNA sequencing datasets from different experimental conditions expected to have a significant impact on RNA modifications. It is recommended to have at least 2 replicates per condition. For example one can use a control condition with a significantly reduced number of modifications such as a cell line for which a modification writing enzyme was knocked-down or knocked-out. Alternatively, on a smaller scale transcripts of interests could be synthesized in-vitro',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Post-translation modification site prediction',\n",
       "   'PolyA signal detection',\n",
       "   'Genotyping',\n",
       "   'k-mer counting'],\n",
       "  'edam_topics': ['Functional, regulatory and non-coding RNA',\n",
       "   'RNA-Seq',\n",
       "   'Gene transcripts',\n",
       "   'Transcriptomics',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'nanofilt',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'nanoplot',\n",
       "  'description': 'NanoPlot is a tool with various visualizations of sequencing data in bam, cram, fastq, fasta or platform-specific TSV summaries, mainly intended for long-read sequencing from Oxford Nanopore Technologies and Pacific Biosciences',\n",
       "  'edam_inputs': [{'formats': ['FASTA-like', 'nucleotide'],\n",
       "    'term': 'nucleotide'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Scatter plot plotting', 'Box-Whisker plot plotting'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'nanopolish',\n",
       "  'description': 'A package for detecting cytosine methylations and genetic variations from nanopore MinION sequencing data.',\n",
       "  'edam_inputs': [{'formats': ['FASTA-like (text)'],\n",
       "    'term': 'FASTA-like (text)'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence alignment report'}],\n",
       "  'edam_operations': ['Statistical calculation',\n",
       "   'Genetic variation analysis',\n",
       "   'Sequence assembly',\n",
       "   'Methylation analysis',\n",
       "   'Mapping'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Sequencing',\n",
       "   'Genetic variation',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'nanosv',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'natural_product_likeness_scorer',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ncbi-datasets-cli',\n",
       "  'description': 'NCBI Datasets is a new resource that lets you easily gather data from across NCBI databases. Find and download sequence, annotation, and metadata for genes and genomes using our command-line tools or web interface.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling',\n",
       "   'Sequence database search',\n",
       "   'Data retrieval'],\n",
       "  'edam_topics': ['Biological databases']},\n",
       " {'id': 'ncbi_acc_download',\n",
       "  'description': 'The National Center for Biotechnology Information (NCBI) provides analysis and retrieval resources for the data in GenBank and other biological data made available through the NCBI web site.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Deposition',\n",
       "   'Service discovery',\n",
       "   'Query and retrieval',\n",
       "   'PCR primer design'],\n",
       "  'edam_topics': ['Data integration and warehousing',\n",
       "   'Database management',\n",
       "   'Data mining',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'Rare diseases']},\n",
       " {'id': 'ncbi_fcs',\n",
       "  'description': 'The NCBI Foreign Contamination Screen (FCS) is a tool suite for identifying and removing contaminant sequences in genome assemblies. Contaminants are defined as sequences in a dataset that do not originate from the biological source organism and can arise from a variety of environmental and laboratory sources. FCS will help you remove contaminants from genomes before submission to GenBank.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly validation',\n",
       "   'Sequence trimming',\n",
       "   'Sequence contamination filtering'],\n",
       "  'edam_topics': ['Sequence analysis', 'Sequence assembly']},\n",
       " {'id': 'ncbi-vdb',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'netcdf',\n",
       "  'description': 'NetCDF (Network Common Data Form) is a set of software libraries and machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'newick_utils',\n",
       "  'description': 'The Newick Utilities are a set of command-line tools for processing phylogenetic trees. They can process arbitrarily large amounts of data and do not require user interaction, which makes them suitable for automating phylogeny processing tasks.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree generation',\n",
       "   'Phylogenetic tree analysis',\n",
       "   'Phylogenetic tree reconstruction'],\n",
       "  'edam_topics': ['Phylogeny', 'Genomics', 'Computer science']},\n",
       " {'id': 'nextclade',\n",
       "  'description': 'Nextclade is an open-source project for viral genome alignment, mutation calling, clade assignment, quality checks and phylogenetic placement.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Methylation analysis', 'Variant calling'],\n",
       "  'edam_topics': ['Genomics', 'Sequence analysis', 'Cladistics']},\n",
       " {'id': 'nextflow',\n",
       "  'description': 'Nextflow enables scalable and reproducible scientific workflows using software containers. It allows the adaptation of pipelines written in the most common scripting languages.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Service composition', 'Service invocation'],\n",
       "  'edam_topics': ['Software engineering',\n",
       "   'Workflows',\n",
       "   'Bioinformatics',\n",
       "   'Genomics']},\n",
       " {'id': 'nextpolish',\n",
       "  'description': 'A fast and efficient genome polishing tool for long read assembly.\\n\\nFast and accurately polish the genome generated by noisy long reads.\\n\\nNextPolish is used to fix base errors (SNV/Indel) in the genome generated by noisy long reads, it can be used with short read data only or long read data only or a combination of both. It contains two core modules, and use a stepwise fashion to correct the error bases in reference genome. To correct the raw third-generation sequencing (TGS) long reads with approximately 15-10% sequencing errors, please use NextDenovo',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'Scaffolding', 'k-mer counting'],\n",
       "  'edam_topics': ['Sequence assembly', 'Plant biology', 'Mapping']},\n",
       " {'id': 'nextpolish2',\n",
       "  'description': 'A fast and efficient genome polishing tool for long read assembly.\\n\\nFast and accurately polish the genome generated by noisy long reads.\\n\\nNextPolish is used to fix base errors (SNV/Indel) in the genome generated by noisy long reads, it can be used with short read data only or long read data only or a combination of both. It contains two core modules, and use a stepwise fashion to correct the error bases in reference genome. To correct the raw third-generation sequencing (TGS) long reads with approximately 15-10% sequencing errors, please use NextDenovo',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'Scaffolding', 'k-mer counting'],\n",
       "  'edam_topics': ['Sequence assembly', 'Plant biology', 'Mapping']},\n",
       " {'id': 'ngs',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ngsutils',\n",
       "  'description': 'NGSUtils is a suite of software tools for working with next-generation sequencing datasets',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read pre-processing',\n",
       "   'Sequencing quality control',\n",
       "   'Variant calling',\n",
       "   'Formatting',\n",
       "   'Sequence contamination filtering'],\n",
       "  'edam_topics': ['Genomics', 'Transcriptomics']},\n",
       " {'id': 'ninja',\n",
       "  'description': 'Nearly Infinite Neighbor Joining Application',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Phylogeny']},\n",
       " {'id': 'nnu-net',\n",
       "  'description': 'nnU-Net is the first segmentation method that is designed to deal with the dataset diversity found in the domain. It condenses and automates the keys decisions for designing a successful segmentation pipeline for any given dataset.\\n\\nIn 3D biomedical image segmentation, dataset properties like imaging modality, image sizes, voxel spacings, class ratios etc vary drastically.\\n\\nIn current research practice, segmentation pipelines are designed manually and with one specific dataset in mind.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis', 'Visualisation', 'Quantification'],\n",
       "  'edam_topics': ['Imaging',\n",
       "   'Workflows',\n",
       "   'Protein expression',\n",
       "   'Machine learning']},\n",
       " {'id': 'NOVOPlasty',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'numpy',\n",
       "  'description': 'The fundamental package for scientific computing with Python',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Calculation'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'nvc',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'bg_find_subsequences',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'obitools',\n",
       "  'description': 'Set of python programs developed to simplify the manipulation of sequence files. They were mainly designed to help us for analyzing Next Generation Sequencer outputs (454 or Illumina) in the context of DNA Metabarcoding.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Sequence analysis', 'DNA', 'Sequencing']},\n",
       " {'id': 'omark',\n",
       "  'description': 'OMArk is a software for proteome (protein-coding gene repertoire) quality assessment. It provides measures of proteome completeness, characterizes the consistency of all protein coding genes with regard to their homologs, and identifies the presence of contamination from other species. OMArk relies on the OMA orthology database, from which it exploits orthology relationships, and on the OMAmer software for fast placement of all proteins into gene families.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly validation',\n",
       "   'Differential protein expression profiling'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Sequence analysis',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'ont-fast5-api',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'openbabel',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'openfold',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'openms',\n",
       "  'description': 'Open source library and a collection of tools and interfaces for the analysis of mass spectrometry data. Includes over 200 standalone (TOPP) tools that can be combined to a workflow with the integrated workflow editor TOPPAS. Raw and intermediate mass spectrometry data can be visualised with the included viewer TOPPView.',\n",
       "  'edam_inputs': [{'formats': ['pepXML',\n",
       "     'mzQuantML',\n",
       "     'dta',\n",
       "     'FASTA',\n",
       "     'protXML',\n",
       "     'idXML',\n",
       "     'OBO',\n",
       "     'XML',\n",
       "     'TraML',\n",
       "     'mzML',\n",
       "     'mzIdentML',\n",
       "     'mzXML',\n",
       "     'MGF',\n",
       "     'qcML'],\n",
       "    'term': 'qcML'},\n",
       "   {'formats': ['pepXML',\n",
       "     'mzQuantML',\n",
       "     'dta',\n",
       "     'FASTA',\n",
       "     'protXML',\n",
       "     'idXML',\n",
       "     'OBO',\n",
       "     'XML',\n",
       "     'TraML',\n",
       "     'mzML',\n",
       "     'mzIdentML',\n",
       "     'mzXML',\n",
       "     'MGF',\n",
       "     'qcML'],\n",
       "    'term': 'qcML'},\n",
       "   {'formats': ['pepXML',\n",
       "     'mzQuantML',\n",
       "     'dta',\n",
       "     'FASTA',\n",
       "     'protXML',\n",
       "     'idXML',\n",
       "     'OBO',\n",
       "     'XML',\n",
       "     'TraML',\n",
       "     'mzML',\n",
       "     'mzIdentML',\n",
       "     'mzXML',\n",
       "     'MGF',\n",
       "     'qcML'],\n",
       "    'term': 'qcML'}],\n",
       "  'edam_outputs': [{'formats': ['pepXML',\n",
       "     'mzQuantML',\n",
       "     'dta',\n",
       "     'FASTA',\n",
       "     'idXML',\n",
       "     'mzXML',\n",
       "     'TraML',\n",
       "     'mzML',\n",
       "     'mzIdentML',\n",
       "     'MGF',\n",
       "     'qcML'],\n",
       "    'term': 'qcML'},\n",
       "   {'formats': ['pepXML',\n",
       "     'mzQuantML',\n",
       "     'dta',\n",
       "     'FASTA',\n",
       "     'idXML',\n",
       "     'mzXML',\n",
       "     'TraML',\n",
       "     'mzML',\n",
       "     'mzIdentML',\n",
       "     'MGF',\n",
       "     'qcML'],\n",
       "    'term': 'qcML'}],\n",
       "  'edam_operations': ['Protein identification',\n",
       "   'Parsing',\n",
       "   'Data handling',\n",
       "   'Chromatographic alignment',\n",
       "   'Mass spectra calibration',\n",
       "   'Formatting',\n",
       "   'Protein quantification',\n",
       "   'Retention time prediction',\n",
       "   'Annotation',\n",
       "   'Protein feature detection',\n",
       "   'Isotope-coded protein label',\n",
       "   'Label-free quantification',\n",
       "   'iTRAQ',\n",
       "   'Peptide identification',\n",
       "   'Peak detection',\n",
       "   'Mass spectrum visualisation',\n",
       "   'Labeled quantification',\n",
       "   'Splitting',\n",
       "   'Spectral analysis',\n",
       "   'Deisotoping',\n",
       "   'Target-Decoy',\n",
       "   'Prediction and recognition (protein)',\n",
       "   'Comparison',\n",
       "   'Standardisation and normalisation'],\n",
       "  'edam_topics': ['Proteomics experiment', 'Proteomics', 'Metabolomics']},\n",
       " {'id': 'optitype',\n",
       "  'description': 'OptiType is a novel HLA genotyping algorithm based on integer linear programming, capable of producing accurate 4-digit HLA genotyping predictions from NGS data by simultaneously selecting all major and minor HLA Class I alleles.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping'],\n",
       "  'edam_topics': ['Sequencing', 'Genotype and phenotype']},\n",
       " {'id': 'orthofinder',\n",
       "  'description': 'OrthoFinder is a fast, accurate and comprehensive platform for comparative genomics. It finds orthogroups and orthologs, infers rooted gene trees for all orthogroups and identifies all of the gene duplcation events in those gene trees. It also infers a rooted species tree for the species being analysed and maps the gene duplication events from the gene trees to branches in the species tree. OrthoFinder also provides comprehensive statistics for comparative genomic analyses.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome comparison',\n",
       "   'Phylogenetic tree generation (from molecular sequences)',\n",
       "   'Phylogenetic tree analysis',\n",
       "   'Genome alignment'],\n",
       "  'edam_topics': ['Phylogenetics',\n",
       "   'Phylogenomics',\n",
       "   'Bioinformatics',\n",
       "   'Comparative genomics',\n",
       "   'Sequence analysis']},\n",
       " {'id': 'pbtk',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pampa',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'panaroo',\n",
       "  'description': 'Producing Polished Prokaryotic Pangenomes with the Panaroo Pipeline.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly', 'Genome assembly', 'Clustering'],\n",
       "  'edam_topics': ['Workflows',\n",
       "   'Sequence assembly',\n",
       "   'Gene and protein families',\n",
       "   'Plant biology',\n",
       "   'Sequencing']},\n",
       " {'id': 'pandoc',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pangolin',\n",
       "  'description': \"Pangolin is a deep-learning based method for predicting splice site strengths (for details, see Zeng and Li, Genome Biology 2022). It is available as a command-line tool that can be run on a VCF or CSV file containing variants of interest; Pangolin will predict changes in splice site strength due to each variant, and return a file of the same format. Pangolin's models can also be used with custom sequences.\",\n",
       "  'edam_inputs': [{'formats': ['VCF', 'CSV'], 'term': 'CSV'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Splice site prediction',\n",
       "   'Sequence merging',\n",
       "   'Alternative splicing prediction',\n",
       "   'Genotyping',\n",
       "   'Variant classification'],\n",
       "  'edam_topics': ['RNA splicing',\n",
       "   'Genetic variation',\n",
       "   'Gene transcripts',\n",
       "   'RNA-Seq',\n",
       "   'Exome sequencing']},\n",
       " {'id': 'param_value_from_file',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'parse_mito_blast',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'parsnp',\n",
       "  'description': 'Parsnp is a command-line-tool for efficient microbial core genome alignment and SNP detection. Parsnp was designed to work in tandem with Gingr, a flexible platform for visualizing genome alignments and phylogenetic trees.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['SNP detection', 'Genome alignment'],\n",
       "  'edam_topics': ['DNA polymorphism', 'Phylogenetics', 'Structure analysis']},\n",
       " {'id': 'pathview',\n",
       "  'description': 'Tool set for pathway based data integration and visualization that maps and renders a wide variety of biological data on relevant pathway graphs. It downloads the pathway graph data, parses the data file, maps user data to the pathway, and render pathway graph with the mapped data. In addition, it integrates with pathway and gene set (enrichment) analysis tools for large-scale and fully automated analysis.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Pathway or network analysis',\n",
       "   'Pathway or network visualisation'],\n",
       "  'edam_topics': ['Molecular interactions, pathways and networks',\n",
       "   'Systems biology',\n",
       "   'Data visualisation']},\n",
       " {'id': 'pavian',\n",
       "  'description': 'Web application for exploring metagenomics classification results, with a special focus on infectious disease diagnosis. Pinpointing pathogens in metagenomics classification results is often complicated by host and laboratory contaminants as well as many non-pathogenic microbiota. Researchers can analyze, display and transform results from the Kraken and Centrifuge classifiers using interactive tables, heatmaps and flow diagrams.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Plot'}],\n",
       "  'edam_operations': ['Annotation', 'Taxonomic classification'],\n",
       "  'edam_topics': ['Metagenomics', 'Model organisms']},\n",
       " {'id': 'pblat',\n",
       "  'description': 'Multithread blat algorithm speeding up aligning sequences to genomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Protein threading', 'Read mapping'],\n",
       "  'edam_topics': ['RNA-seq', 'Mapping', 'Gene transcripts']},\n",
       " {'id': 'pdb',\n",
       "  'description': 'Global repository for all bona fide protein structure data. Single worldwide repository of information about the 3D structures of large biological molecules, including proteins and nucleic acids. It also has RESTful interface to search and retrieve data from the Protein Data Bank (PDB).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Protein structure analysis'],\n",
       "  'edam_topics': ['Structural biology',\n",
       "   'Electron microscopy',\n",
       "   'NMR',\n",
       "   'Protein structure analysis']},\n",
       " {'id': 'pe_histogram',\n",
       "  'description': 'Productive visualization of high-throughput sequencing data using the SeqCode open portable platform.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Visualisation',\n",
       "   'Peak calling',\n",
       "   'Essential dynamics',\n",
       "   'Phylogenetic footprinting'],\n",
       "  'edam_topics': ['ChIP-seq',\n",
       "   'RNA-Seq',\n",
       "   'Epigenetics',\n",
       "   'Methylated DNA immunoprecipitation',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'pear',\n",
       "  'description': 'Paired-end read merger. PEAR evaluates all possible paired-end read overlaps without requiring the target fragment size as input. In addition, it implements a statistical test for minimizing false-positive results.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence merging'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'PepPointer',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pepquery2',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'peptideshaker',\n",
       "  'description': 'PeptideShaker is a search engine independent platform for interpretation of proteomics identification results from multiple search engines, currently supporting X!Tandem, MS-GF+, MS Amanda, OMSSA, MyriMatch, Comet, Tide, Mascot, Andromeda and mzIdentML. By combining the results from multiple search engines, while re-calculating PTM localization scores and redoing the protein inference, PeptideShaker attempts to give you the best possible understanding of your proteomics data',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['mzIdentML'], 'term': 'mzIdentML'},\n",
       "   {'formats': ['MGF'], 'term': 'MGF'}],\n",
       "  'edam_outputs': [{'formats': ['JPG', 'SVG'], 'term': 'SVG'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'},\n",
       "   {'formats': ['Image format'], 'term': 'Image format'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Mass spectrum visualisation',\n",
       "   'Peptide identification',\n",
       "   'Validation',\n",
       "   'PTM localisation',\n",
       "   'Enrichment analysis',\n",
       "   'Structure visualisation',\n",
       "   'Protein identification'],\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment', 'Proteome']},\n",
       " {'id': 'peptide_genomic_coordinate',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'percolator',\n",
       "  'description': 'Semi-supervised learning for peptide identification from MS/MS data.',\n",
       "  'edam_inputs': [{'formats': ['XML'], 'term': 'XML'}],\n",
       "  'edam_outputs': [{'formats': ['XML'], 'term': 'XML'}],\n",
       "  'edam_operations': ['Protein fragment weight comparison',\n",
       "   'Protein identification',\n",
       "   'Blind peptide database search',\n",
       "   'Peptide identification',\n",
       "   'Target-Decoy',\n",
       "   'Validation of peptide-spectrum matches'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Proteomics experiment',\n",
       "   'Statistics and probability',\n",
       "   'Machine learning']},\n",
       " {'id': 'pfamscan',\n",
       "  'description': 'This tool is used to search a FASTA sequence against a library of Pfam HMM.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Protein sequence analysis'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'pharokka',\n",
       "  'description': 'Pharokka is a rapid standardised annotation tool for bacteriophage genomes and metagenomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation',\n",
       "   'Antimicrobial resistance prediction',\n",
       "   'tRNA gene prediction',\n",
       "   'Formatting',\n",
       "   'Sequence assembly'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Sequence sites, features and motifs',\n",
       "   'Workflows',\n",
       "   'Functional, regulatory and non-coding RNA']},\n",
       " {'id': 'phinch',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'phyloseq',\n",
       "  'description': 'Provides a set of classes and tools to facilitate the import, storage, analysis, and graphical display of microbiome census data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Deposition', 'Analysis', 'Visualisation'],\n",
       "  'edam_topics': ['Microbiology', 'Sequence analysis', 'Metagenomics']},\n",
       " {'id': 'phyml',\n",
       "  'description': 'Phylogenetic estimation software using Maximum Likelihood',\n",
       "  'edam_inputs': [{'formats': ['Phylip format variant'],\n",
       "    'term': 'Phylip format variant'},\n",
       "   {'formats': ['Phylip format variant'], 'term': 'Phylip format variant'}],\n",
       "  'edam_outputs': [{'formats': ['newick'], 'term': 'newick'}],\n",
       "  'edam_operations': ['Phylogenetic tree generation (maximum likelihood and Bayesian methods)'],\n",
       "  'edam_topics': ['Phylogenetics', 'Bioinformatics', 'Phylogenetics']},\n",
       " {'id': 'picard',\n",
       "  'description': 'A set of command line tools for manipulating high-throughput sequencing (HTS) data in formats such as SAM/BAM/CRAM and VCF. Available as a standalone program or within the GATK4 program.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genetic variation analysis'],\n",
       "  'edam_topics': ['Sequencing', 'Data management']},\n",
       " {'id': 'picrust',\n",
       "  'description': 'PICRUSt (Phylogenetic Investigation of Communities by Reconstruction of Unobserved States) is a bioinformatics software package designed to predict metagenome functional content from marker gene (e.g., 16S rRNA) surveys and full genomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic reconstruction',\n",
       "   'Expression analysis',\n",
       "   'Genome annotation',\n",
       "   'DNA barcoding'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Microbial ecology',\n",
       "   'Functional, regulatory and non-coding RNA',\n",
       "   'Metagenomic sequencing']},\n",
       " {'id': 'picrust2',\n",
       "  'description': 'PICRUSt2 (Phylogenetic Investigation of Communities by Reconstruction of Unobserved States) is a software for predicting functional abundances based only on marker gene sequences.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic reconstruction',\n",
       "   'Expression analysis',\n",
       "   'Rarefaction',\n",
       "   'Pathway analysis'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Microbiology',\n",
       "   'Phylogenetics',\n",
       "   'Metagenomic sequencing']},\n",
       " {'id': 'pileup_interval',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pileup_parser',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pilon',\n",
       "  'description': 'Read alignment analysis to diagnose, report, and automatically improve de novo genome assemblies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly'],\n",
       "  'edam_topics': ['Assembly']},\n",
       " {'id': 'pipe_t',\n",
       "  'description': 'PIPE-T is a Galaxy Workflow for processing and analyzing miR expression profiles by RTqPCR.\\n\\nIt is a tool that offers several state-of-the-art options for parsing, filtering, normalizing, imputing and analyzing RT-qPCR expression data. Integration of PIPE-T into Galaxy allows experimentalists with strong bioinformatic background, as well as those without any programming or development expertise, to perform complex analysis in a simple to use, transparent, accessible, reproducible, and user-friendly environment',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Essential dynamics',\n",
       "   'Quantification',\n",
       "   'Standardisation and normalisation',\n",
       "   'Imputation',\n",
       "   'Parsing'],\n",
       "  'edam_topics': ['PCR experiment',\n",
       "   'Transcriptomics',\n",
       "   'Biomarkers',\n",
       "   'Microarray experiment']},\n",
       " {'id': 'PlasFlow',\n",
       "  'description': 'PlasFlow is a set of scripts used for prediction of plasmid sequences in metagenomic contigs.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Sequence assembly',\n",
       "   'Mobile genetic elements',\n",
       "   'Machine learning']},\n",
       " {'id': 'plasmidfinder',\n",
       "  'description': 'PlasmidFinder is a tool for the identification and typing of Plasmid Replicons in Whole-Genome Sequencing (WGS).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'Scaffolding',\n",
       "   'Multilocus sequence typing'],\n",
       "  'edam_topics': ['Whole genome sequencing',\n",
       "   'Sequence assembly',\n",
       "   'Mapping',\n",
       "   'Probes and primers']},\n",
       " {'id': 'plink',\n",
       "  'description': 'Free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genetic variation analysis'],\n",
       "  'edam_topics': ['GWAS study']},\n",
       " {'id': 'poisson2test',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'polypolish',\n",
       "  'description': 'Polypolish is a tool for polishing genome assemblies with short reads. Unlike other tools in this category, Polypolish uses SAM files where each read has been aligned to all possible locations (not just a single best location). This allows it to repair errors in repeat regions that other alignment-based polishers cannot fix.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'Read mapping',\n",
       "   'Mapping assembly',\n",
       "   'Sequencing error detection'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Sequence composition, complexity and repeats',\n",
       "   'Mapping']},\n",
       " {'id': 'porechop',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'porechop_abi',\n",
       "  'description': 'Porechop_ABI (ab initio) is an extension of Porechop whose purpose is to process adapter sequences in ONT reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence trimming',\n",
       "   'Base-calling',\n",
       "   'k-mer counting',\n",
       "   'Demultiplexing',\n",
       "   'Primer removal'],\n",
       "  'edam_topics': ['Genomics', 'Bioinformatics']},\n",
       " {'id': 'poretools',\n",
       "  'description': 'Flexible toolkit for exploring datasets generated by nanopore sequencing devices from MinION for the purposes of quality control and downstream analysis.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Nucleic acid sequence analysis'],\n",
       "  'edam_topics': ['DNA', 'Sequencing']},\n",
       " {'id': 'pplacer',\n",
       "  'description': 'Tools for performing taxonomic assignment based on phylogeny using pplacer and clst.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Taxonomic classification'],\n",
       "  'edam_topics': ['Taxonomy']},\n",
       " {'id': 'presto',\n",
       "  'description': 'Integrated collection of platform-independent Python modules for processing raw reads from high-throughput (next-generation) sequencing of lymphocyte repertoires.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Nucleic acid sequence analysis'],\n",
       "  'edam_topics': ['Sequencing', 'DNA', 'Immunology']},\n",
       " {'id': 'pretext_map',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pretextview',\n",
       "  'description': 'Pretext is an OpenGL-powered pretext contact map viewer.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Scaffolding'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'prodigal',\n",
       "  'description': 'The pipeline runs PRODIGAL gene predictions on all genomes, runs pan-reciprocal BLAST, and identifies ortholog sets. For a set of orthologous genes, if the positions of the PRODIGAL selected starts coincide in a multiple sequence alignment, they are accepted. If they do not coincide, a consistent start position is sought where a majority of the highest-scoring PRODIGAL selected sites coincide. If such a position is found, it is accepted, and the predictions are changed for the outlying genes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['DNA',\n",
       "   'Sequence analysis',\n",
       "   'Nucleic acid sites, features and motifs']},\n",
       " {'id': 'profia',\n",
       "  'description': 'Flow Injection Analysis coupled to High-Resolution Mass Spectrometry is a promising approach for high-throughput metabolomics. FIA-HRMS data, however, cannot be pre-processed with current software tools which rely on liquid chromatography separation, or handle low resolution data only. Here we present the package that implements a new methodology to pre-process FIA-HRMS raw data (netCDF, mzData, mzXML, and mzML) and generates the peak table.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Peak calling'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'progressbar2',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'prokka',\n",
       "  'description': 'Software tool to annotate bacterial, archaeal and viral genomes quickly and produce standards-compliant output files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction',\n",
       "   'Coding region prediction',\n",
       "   'Genome annotation'],\n",
       "  'edam_topics': ['Genomics', 'Model organisms', 'Virology']},\n",
       " {'id': 'proteinmpnn',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pslcdnafilter',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'purge_dups',\n",
       "  'description': 'Identifying and removing haplotypic duplication in primary genome assemblies | haplotypic duplication identification tool | scripts/pd_config.py: script to generate a configuration file used by run_purge_dups.py | purge haplotigs and overlaps in an assembly based on read depth | Given a primary assembly pri_asm and an alternative assembly hap_asm (optional, if you have one), follow the steps shown below to build your own purge_dups pipeline, steps with same number can be run simultaneously. Among all the steps, although step 4 is optional, we highly recommend our users to do so, because assemblers may produce overrepresented seqeuences. In such a case, The final step 4 can be applied to remove those seqeuences',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly', 'Read binning', 'Scaffolding'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'pybigwig',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pycoqc',\n",
       "  'description': 'PycoQC computes metrics and generates interactive QC plots for Oxford Nanopore technologies sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequencing quality control', 'Statistical calculation'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Data quality management',\n",
       "   'Sequencing']},\n",
       " {'id': 'pygenometracks',\n",
       "  'description': 'reproducible plots for multivariate genomic data sets.\\n\\nStandalone program and library to plot beautiful genome browser tracks.\\n\\npyGenomeTracks aims to produce high-quality genome browser tracks that are highly customizable. Currently, it is possible to plot:.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Visualisation', 'Formatting'],\n",
       "  'edam_topics': ['Model organisms', 'Imaging', 'Workflows']},\n",
       " {'id': 'pyprophet',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'pysam',\n",
       "  'description': 'A Python module for reading and manipulating SAM/BAM/VCF/BCF files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Sequence file editing'],\n",
       "  'edam_topics': ['Sequencing', 'Mapping']},\n",
       " {'id': 'pytorch',\n",
       "  'description': 'PyTorch is an optimized tensor library for deep learning using GPUs and CPUs.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Machine learning', 'Computer science']},\n",
       " {'id': 'qapa',\n",
       "  'description': 'RNA-seq Quantification of Alternative Polyadenylation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['PolyA signal detection'],\n",
       "  'edam_topics': ['RNA-Seq',\n",
       "   'Genetic variation',\n",
       "   'Molecular genetics',\n",
       "   'Sequence sites, features and motifs']},\n",
       " {'id': 'qcxms',\n",
       "  'description': 'This Galaxy tool is based on the quantum chemical code QCxMS for calculating mass spectra using Born-Oppenheimer Molecular Dynamics. This version supports only Electron Ionization and uses the GFN2-xTB and GFN1-xTB quantum chemistry methods for the simulations. The mass spectrum is generated in three steps: neutral run, production run, and result extraction.',\n",
       "  'edam_inputs': [{'formats': ['XYZ'], 'term': 'XYZ'}],\n",
       "  'edam_outputs': [{'formats': ['Format'], 'term': 'Format'}],\n",
       "  'edam_operations': ['Spectrum calculation'],\n",
       "  'edam_topics': ['Computational chemistry', 'Molecular dynamics']},\n",
       " {'id': 'qiime2',\n",
       "  'description': 'QIIME 2 is an AI-ready microbiome multi-omics data science platform that is trusted, free, open source, extensible, and community developed and supported bioinformatics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Demultiplexing',\n",
       "   'Visualisation',\n",
       "   'Taxonomic classification'],\n",
       "  'edam_topics': ['Microbial ecology', 'Metatranscriptomics', 'Metagenomics']},\n",
       " {'id': 'qiime2-amplicon',\n",
       "  'description': 'QIIME 2 is an AI-ready microbiome multi-omics data science platform that is trusted, free, open source, extensible, and community developed and supported bioinformatics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Demultiplexing',\n",
       "   'Visualisation',\n",
       "   'Taxonomic classification'],\n",
       "  'edam_topics': ['Microbial ecology', 'Metatranscriptomics', 'Metagenomics']},\n",
       " {'id': 'qiime2-shotgun',\n",
       "  'description': 'QIIME 2 is an AI-ready microbiome multi-omics data science platform that is trusted, free, open source, extensible, and community developed and supported bioinformatics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Demultiplexing',\n",
       "   'Visualisation',\n",
       "   'Taxonomic classification'],\n",
       "  'edam_topics': ['Microbial ecology', 'Metatranscriptomics', 'Metagenomics']},\n",
       " {'id': 'qualimap',\n",
       "  'description': 'Platform-independent application written in Java and R that provides both a Graphical User Inteface (GUI) and a command-line interface to facilitate the quality control of alignment sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequencing quality control'],\n",
       "  'edam_topics': ['Data quality management']},\n",
       " {'id': 'quasitools',\n",
       "  'description': 'A Collection of Tools for Viral Quasispecies Analysis | Abstract Summary quasitools is a collection of newly-developed, open-source tools for analyzing viral quasispcies data. The application suite includes tools with the ability to create consensus sequences, call nucleotide, codon, and amino acid variants, calculate the complexity of a quasispecies, and measure the genetic distance between two similar quasispecies. These tools may be run independently or in user-created workflows. Availability The quasitools suite is a freely available application licensed under the Apache License, Version 2.0. The source code, documentation, and file specifications are available at: https: phac-nml.github.io quasitools Contact gary.vandomselaar@canada.ca',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence trimming', 'Genotyping', 'Base-calling'],\n",
       "  'edam_topics': ['Virology', 'Small molecules', 'Sequencing']},\n",
       " {'id': 'quast',\n",
       "  'description': 'QUAST stands for QUality ASsessment Tool.  \\nIt evaluates a quality of genome assemblies by computing various metrics and providing nice reports.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Visualisation', 'Sequence assembly validation'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'query_tabular',\n",
       "  'description': 'Query Tabular is a Galaxy-based tool which manipulates tabular files.  Query Tabular automatically creates a SQLite database directly from a tabular file within a Galaxy workflow.  The SQLite database can be saved to the Galaxy history, and further process to generate tabular outputs containing desired information and formatting.',\n",
       "  'edam_inputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Data handling',\n",
       "   'Parsing',\n",
       "   'Query and retrieval',\n",
       "   'Filtering'],\n",
       "  'edam_topics': ['Bioinformatics', 'Workflows']},\n",
       " {'id': 'qupath',\n",
       "  'description': 'Aims to help improve the speed, objectivity and reproducibility of digital pathology analysis and biomarker interpretation by offering an open, powerful, flexible, extensible software platform for whole slide image analysis.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis'],\n",
       "  'edam_topics': ['Light microscopy',\n",
       "   'Data visualisation',\n",
       "   'Pathology',\n",
       "   'Medical imaging']},\n",
       " {'id': 'r',\n",
       "  'description': 'Free software environment for statistical computing and graphics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Analysis', 'Data handling', 'Visualisation'],\n",
       "  'edam_topics': ['Mathematics',\n",
       "   'Data architecture, analysis and design',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'r-raceid',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'racon',\n",
       "  'description': 'Consensus module for raw de novo DNA assembly of long uncorrected reads\\n\\nRacon is intended as a standalone consensus module to correct raw contigs generated by rapid assembly methods which do not include a consensus step. The goal of Racon is to generate genomic consensus which is of similar or better quality compared to the output generated by assembly methods which employ both error correction and consensus steps, while providing a speedup of several times compared to those methods. It supports data produced by both Pacific Biosciences and Oxford Nanopore Technologies.',\n",
       "  'edam_inputs': [{'formats': ['BAM', 'SAM'], 'term': 'SAM'},\n",
       "   {'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly', 'Mapping assembly'],\n",
       "  'edam_topics': ['Whole genome sequencing', 'Sequence assembly']},\n",
       " {'id': 'ragtag',\n",
       "  'description': 'RagTag is a collection of software tools for scaffolding and improving modern genome assemblies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'ramclustr',\n",
       "  'description': 'A feature clustering algorithm for non-targeted mass spectrometric metabolomics data.',\n",
       "  'edam_inputs': [{'formats': ['Binary format', 'CSV'], 'term': 'CSV'},\n",
       "   {'formats': ['Binary format', 'CSV'], 'term': 'CSV'},\n",
       "   {'formats': ['Binary format', 'CSV'], 'term': 'CSV'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Mass spectrum'},\n",
       "   {'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_operations': ['Imputation',\n",
       "   'Standardisation and normalisation',\n",
       "   'Clustering',\n",
       "   'Correlation'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'rapidnj',\n",
       "  'description': 'A tool for fast canonical neighbor-joining tree construction.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree generation'],\n",
       "  'edam_topics': ['Phylogeny']},\n",
       " {'id': 'ratatosk',\n",
       "  'description': 'Ratatosk  Hybrid error correction of long reads enables accurate variant calling and assembly.\\n\\nPhased hybrid error correction of long reads using colored de Bruijn graphs.\\n\\nRatatosk is a phased error correction tool for erroneous long reads based on compacted and colored de Bruijn graphs built from accurate short reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly',\n",
       "   'Sequencing error detection',\n",
       "   'Variant calling',\n",
       "   'Phasing',\n",
       "   'Read binning'],\n",
       "  'edam_topics': ['DNA polymorphism',\n",
       "   'Sequence assembly',\n",
       "   'Whole genome sequencing']},\n",
       " {'id': 'raven',\n",
       "  'description': 'a de novo genome assembler for long reads.\\n\\nRaven is a de novo genome assembler for long uncorrected reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly',\n",
       "   'Genome assembly',\n",
       "   'Read pre-processing'],\n",
       "  'edam_topics': ['Sequence assembly', 'Whole genome sequencing', 'Genomics']},\n",
       " {'id': 'rawtools',\n",
       "  'description': 'A standalone tool for extracting data directly from raw files generated by Thermo Orbitrap family instruments.',\n",
       "  'edam_inputs': [{'formats': ['Thermo RAW'], 'term': 'Thermo RAW'}],\n",
       "  'edam_outputs': [{'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Parsing'],\n",
       "  'edam_topics': ['Proteomics experiment', 'Proteomics']},\n",
       " {'id': 'raxml',\n",
       "  'description': 'A tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis', 'Phylogenetic tree analysis'],\n",
       "  'edam_topics': ['Phylogenetics', 'Sequence analysis']},\n",
       " {'id': 'rdkit',\n",
       "  'description': 'RDKit is an Open-Source Cheminformatics Software. Fast, Efficient Fragment-Based Coordinate Generation for Open Babel.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Nucleic acid structure prediction',\n",
       "   'Molecular dynamics',\n",
       "   'Protein geometry validation'],\n",
       "  'edam_topics': ['Cheminformatics',\n",
       "   'Medicinal chemistry',\n",
       "   'Molecular biology']},\n",
       " {'id': 'rdock',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'recon',\n",
       "  'description': 'Tool for calculating the probability of nucleosome formation along a DNA sequence input by the user.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Nucleosome formation or exclusion sequence prediction',\n",
       "   'Formatting'],\n",
       "  'edam_topics': ['DNA',\n",
       "   'DNA binding sites',\n",
       "   'Statistics and probability',\n",
       "   'DNA packaging',\n",
       "   'Transcription factors and regulatory sites']},\n",
       " {'id': 'Recentrifuge',\n",
       "  'description': 'Robust comparative analysis and contamination removal for metagenomics.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Taxonomic classification',\n",
       "   'Expression analysis',\n",
       "   'Cross-assembly'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Microbial ecology',\n",
       "   'Metagenomic sequencing']},\n",
       " {'id': 'recetox-aplcms',\n",
       "  'description': 'recetox-aplcms is a tool for peak detection in mass spectrometry data. The tool performs (1) noise removal, (2) peak detection, (3) retention time drift correction, (4) peak alignment and (5) weaker signal recovery as well as (6) suspect screening.',\n",
       "  'edam_inputs': [{'formats': ['mzML'], 'term': 'mzML'}],\n",
       "  'edam_outputs': [{'formats': ['Binary format'], 'term': 'Binary format'},\n",
       "   {'formats': ['Binary format'], 'term': 'Binary format'},\n",
       "   {'formats': ['Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_operations': ['Alignment',\n",
       "   'Quantification',\n",
       "   'Chromatographic alignment',\n",
       "   'Peak detection',\n",
       "   'Feature extraction'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'recetox_galaxytools',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'red',\n",
       "  'description': 'This is a program to detect and visualize RNA editing events at genomic scale using next-generation sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['RNA-Seq analysis'],\n",
       "  'edam_topics': ['RNA', 'Sequencing', 'Data visualisation']},\n",
       " {'id': 'regenie',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'repeatmasker',\n",
       "  'description': 'A program that screens DNA sequences for interspersed repeats and low complexity DNA sequences. The output of the program is a detailed annotation of the repeats that are present in the query sequence as well as a modified version of the query sequence in which all the annotated repeats have been masked (default: replaced by Ns).',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome annotation'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Sequence composition, complexity and repeats']},\n",
       " {'id': 'repeatmodeler',\n",
       "  'description': 'RepeatModeler is a de novo transposable element (TE) family identification and modeling package. At the heart of RepeatModeler are three de-novo repeat finding programs ( RECON, RepeatScout and LtrHarvest/Ltr_retriever ) which employ complementary computational methods for identifying repeat element boundaries and family relationships from sequence data.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'de Novo sequencing',\n",
       "   'Genome annotation'],\n",
       "  'edam_topics': ['Model organisms',\n",
       "   'Sequence composition, complexity and repeats',\n",
       "   'Whole genome sequencing',\n",
       "   'Sequence assembly',\n",
       "   'Mobile genetic elements']},\n",
       " {'id': 'repeatscout',\n",
       "  'description': 'RepeatScout is a tool to discover repetitive substrings in DNA.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['k-mer counting'],\n",
       "  'edam_topics': ['DNA']},\n",
       " {'id': 'repenrich',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'rfantibody',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'rfdiffusion',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'riassigner',\n",
       "  'description': 'RIAssigner is a python tool for retention index (RI) computation for GC-MS data.',\n",
       "  'edam_inputs': [{'formats': ['CSV', 'TSV'], 'term': 'TSV'}],\n",
       "  'edam_outputs': [{'formats': ['CSV', 'TSV'], 'term': 'TSV'},\n",
       "   {'formats': [], 'term': 'Data index'}],\n",
       "  'edam_operations': ['Standardisation and normalisation'],\n",
       "  'edam_topics': ['Metabolomics',\n",
       "   'Compound libraries and screening',\n",
       "   'Data submission, annotation and curation']},\n",
       " {'id': 'rjags',\n",
       "  'description': 'The rjags package provides an interface from R to the JAGS library for Bayesian data analysis. JAGS uses Markov Chain Monte Carlo (MCMC) to generate a sequence of dependent samples\\nfrom the posterior distribution of the parameters.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Statistics and probability']},\n",
       " {'id': 'rmassbank',\n",
       "  'description': 'Workflow to process tandem MS files and build MassBank records. Functions include automated extraction of tandem MS spectra, formula assignment to tandem MS fragments, recalibration of tandem MS spectra with assigned fragments, spectrum cleanup, automated retrieval of compound information from Internet databases, and export to MassBank records.',\n",
       "  'edam_inputs': [{'formats': ['XML'], 'term': 'XML'}],\n",
       "  'edam_outputs': [{'formats': ['Mass spectrometry data format'],\n",
       "    'term': 'Mass spectrometry data format'}],\n",
       "  'edam_operations': ['Clustering'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'rmats-turbo',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'rmblast',\n",
       "  'description': 'RMBlast is a RepeatMasker compatible version of the standard NCBI blastn program. The primary difference between this distribution and the NCBI distribution is the addition of a new program \"rmblastn\" for use with RepeatMasker and RepeatModeler.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'rnachipintegrator',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'rnaquast',\n",
       "  'description': 'Quality assessment tool for de novo transcriptome assemblies.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['De-novo assembly',\n",
       "   'Transcriptome assembly',\n",
       "   'Sequence assembly validation'],\n",
       "  'edam_topics': ['Sequence assembly', 'Transcriptomics', 'RNA-Seq']},\n",
       " {'id': 'roary',\n",
       "  'description': 'A high speed stand alone pan genome pipeline, which takes annotated assemblies in GFF3 format (produced by Prokka (Seemann, 2014)) and calculates the pan genome.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['DNA', 'Genomics', 'Mapping']},\n",
       " {'id': 'rrmscorer',\n",
       "  'description': 'RRMScorer provides quick predictions for any RNA recognition motif (RRM) and any RNA target purely based on their sequences.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Protein sequence record'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Score'}],\n",
       "  'edam_operations': ['Multiple sequence alignment',\n",
       "   'Calculation',\n",
       "   'Visualisation'],\n",
       "  'edam_topics': ['Computational biology']},\n",
       " {'id': 'rsem',\n",
       "  'description': 'We present a generative statistical model and associated inference methods that handle read mapping uncertainty in a principled manner. Through simulations parameterized by real RNASeq data, we show that our method is more accurate than previous methods. Our improved accuracy is the result of handling read mapping uncertainty with a statistical model and the estimation of gene expression levels as the sum of isoform expression levels.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['RNA-Seq quantification'],\n",
       "  'edam_topics': ['RNA-Seq']},\n",
       " {'id': 'rseqc',\n",
       "  'description': 'Provides a number of useful modules that can comprehensively evaluate high throughput sequence data especially RNA-seq data. Some basic modules quickly inspect sequence quality, nucleotide composition bias, PCR bias and GC bias, while RNA-seq specific modules evaluate sequencing saturation, mapped reads distribution, coverage uniformity, strand specificity, transcript level RNA integrity etc.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Sequencing']},\n",
       " {'id': 'rstudio',\n",
       "  'description': 'Integrated development environment (IDE) for the R programming language.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Visualisation'],\n",
       "  'edam_topics': ['Mathematics',\n",
       "   'Data architecture, analysis and design',\n",
       "   'Statistics and probability']},\n",
       " {'id': 'rtg-tools',\n",
       "  'description': 'RTG Core: Software for alignment and analysis of next-gen sequencing data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment', 'Sequence alignment analysis'],\n",
       "  'edam_topics': ['Genomics']},\n",
       " {'id': 'rtracklayer',\n",
       "  'description': 'Extensible framework for interacting with multiple genome browsers (currently UCSC built-in) and manipulating annotation tracks in various formats (currently GFF, BED, bedGraph, BED15, WIG, BigWig and 2bit built-in). The user may export/import tracks to/from the supported browsers, as well as query and modify the browser state, such as the current viewport.',\n",
       "  'edam_inputs': [{'formats': ['BED', 'GFF', 'WIG'], 'term': 'WIG'}],\n",
       "  'edam_outputs': [{'formats': ['BED', 'GFF', 'WIG'], 'term': 'WIG'}],\n",
       "  'edam_operations': ['Genome visualisation', 'Genome annotation'],\n",
       "  'edam_topics': ['Genetics', 'Molecular biology']},\n",
       " {'id': 'rxdock',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 's3segmenter',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sailfish',\n",
       "  'description': 'A software tool that implements a novel, is an alignment-free algorithm for the estimation of isoform abundances directly from a set of reference sequences and RNA-seq reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['RNA-Seq analysis'],\n",
       "  'edam_topics': ['RNA-Seq', 'RNA', 'Sequencing', 'Transcriptomics']},\n",
       " {'id': 'salmon',\n",
       "  'description': 'A tool for transcript expression quantification from RNA-seq data',\n",
       "  'edam_inputs': [{'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'}],\n",
       "  'edam_operations': ['Sequence composition calculation',\n",
       "   'RNA-Seq quantification',\n",
       "   'Gene expression analysis'],\n",
       "  'edam_topics': ['RNA-Seq', 'Gene expression', 'Transcriptomics']},\n",
       " {'id': 'salmon_kallisto_mtx_to_10x',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'salsa',\n",
       "  'description': \"> VERY_LOW CONFIDENCE! | > CORRECT NAME OF TOOL COULD ALSO BE 'chromosome-scale', 'reference-quality', 'Hi-C', 'scaffolder' | Integrating Hi-C links with assembly graphs for chromosome-scale assembly | SALSA: A tool to scaffold long read assemblies with Hi-C data | SALSA: A tool to scaffold long read assemblies with Hi-C | This code is used to scaffold your assemblies using Hi-C data. This version implements some improvements in the original SALSA algorithm. If you want to use the old version, it can be found in the old_salsa branch\",\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'De-novo assembly', 'Scaffolding'],\n",
       "  'edam_topics': ['Sequence assembly', 'DNA binding sites', 'Mapping']},\n",
       " {'id': 'sam_bitwise_flag_filter',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sam_pileup',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sam2interval',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sambamba',\n",
       "  'description': 'This tool is a high performance modern robust and fast tool (and library), written in the D programming language, for working with SAM, BAM and CRAM formats.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis', 'Sequence alignment'],\n",
       "  'edam_topics': ['DNA', 'Sequence analysis', 'RNA-Seq']},\n",
       " {'id': 'samtools',\n",
       "  'description': 'SAMtools and BCFtools are widely used programs for processing and analysing high-throughput sequencing data. They include tools for file format conversion and manipulation, sorting, querying, statistics, variant calling, and effect analysis amongst other methods.',\n",
       "  'edam_inputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       "  'edam_operations': ['Data filtering',\n",
       "   'Visualisation',\n",
       "   'Data formatting',\n",
       "   'Indexing',\n",
       "   'Data parsing',\n",
       "   'Data sorting',\n",
       "   'Data editing'],\n",
       "  'edam_topics': ['Mapping',\n",
       "   'Sequence analysis',\n",
       "   'Sequencing',\n",
       "   'Rare diseases']},\n",
       " {'id': 'sanntis',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sccaf',\n",
       "  'description': 'Single Cell Clustering Assessment Framework (SCCAF) is a novel method for automated identification of putative cell types from single cell RNA-seq (scRNA-seq) data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Annotation', 'Expression analysis', 'RNA-Seq analysis'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'scanpy',\n",
       "  'description': 'Scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.',\n",
       "  'edam_inputs': [{'formats': ['HDF5'], 'term': 'HDF5'}],\n",
       "  'edam_outputs': [{'formats': ['HDF5'], 'term': 'HDF5'}],\n",
       "  'edam_operations': ['Differential gene expression analysis',\n",
       "   'Principal component visualisation',\n",
       "   'Expression profile clustering'],\n",
       "  'edam_topics': ['Gene expression',\n",
       "   'Cell biology',\n",
       "   'Genetics',\n",
       "   'Single-cell sequencing']},\n",
       " {'id': 'scater',\n",
       "  'description': 'Pre-processing, quality control, normalization and visualization of single-cell RNA-seq data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read pre-processing',\n",
       "   'Sequencing quality control',\n",
       "   'Sequence visualisation'],\n",
       "  'edam_topics': ['RNA-Seq', 'Quality affairs', 'Molecular genetics']},\n",
       " {'id': 'sceasy',\n",
       "  'description': 'sceasy is a package that helps easy conversion of different single-cell data formats to each other.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'scenicplus',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'schrodinger',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'scikit-build',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'scikit-image',\n",
       "  'description': 'Scikit-image contains image processing algorithms for SciPy, including IO, morphology, filtering, warping, color manipulation, object detection, etc.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis',\n",
       "   'Image annotation',\n",
       "   'Visualisation',\n",
       "   'Data handling'],\n",
       "  'edam_topics': ['Imaging',\n",
       "   'Software engineering',\n",
       "   'Literature and language']},\n",
       " {'id': 'scikit',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'scipio',\n",
       "  'description': 'Scipio is a tool to determine the precise exon-intron gene structure given a protein sequence and a genome. It identifies splice sites and is able to cope with sequencing errors and genes spanning several contigs. The output contains information about discrepancies that may result from sequencing errors. Scipio has also successfully been used to find homologous genes in related species. WebScipio, allows to search for mutually exclusive spliced exons and tandemly arrayed gene duplicates.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction',\n",
       "   'Query and retrieval',\n",
       "   'Protein feature detection',\n",
       "   'Protein structure prediction'],\n",
       "  'edam_topics': ['Gene transcripts',\n",
       "   'Gene structure',\n",
       "   'Gene and protein families',\n",
       "   'Model organisms',\n",
       "   'Model organisms']},\n",
       " {'id': 'scran',\n",
       "  'description': 'Implements a variety of low-level analyses of single-cell RNA-seq data. Methods are provided for normalization of cell-specific biases, assignment of cell cycle phase, and detection of highly variable and significantly correlated genes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['RNA-Seq analysis'],\n",
       "  'edam_topics': ['Cell biology', 'RNA-seq']},\n",
       " {'id': 'screen_assembly',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'scvelo',\n",
       "  'description': 'Generalizing RNA velocity to transient cell states through dynamical modeling.\\nsingle-cell RNA velocity generalized to transient cell states',\n",
       "  'edam_inputs': [{'formats': ['Loom', 'HDF5'], 'term': 'HDF5'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Molecular dynamics'],\n",
       "  'edam_topics': ['Gene transcripts',\n",
       "   'Cell biology',\n",
       "   'Molecular interactions, pathways and networks',\n",
       "   'Gene expression',\n",
       "   'RNA-Seq']},\n",
       " {'id': 'scxa-plots',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'semibin',\n",
       "  'description': 'Command tool for metagenomic binning with semi-supervised deep learning using information from reference genomes.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence assembly', 'Read binning'],\n",
       "  'edam_topics': ['Metagenomics',\n",
       "   'Machine learning',\n",
       "   'Microbial ecology',\n",
       "   'Sequence assembly']},\n",
       " {'id': 'sepp',\n",
       "  'description': 'SEPP stands for SAT-Enabled Phylogenetic Placement and addresses the problem of phylogenetic placement for meta-genomic short reads',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Phylogenetic tree'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Phylogenetic data'}],\n",
       "  'edam_operations': ['Phylogenetic analysis'],\n",
       "  'edam_topics': ['Phylogenetics']},\n",
       " {'id': 'seq2hla',\n",
       "  'description': 'seq2HLA is a computational tool to determine Human Leukocyte Antigen (HLA) directly from existing and future short RNA-Seq reads. \\nIt takes standard RNA-Seq sequence reads in fastq format as input, uses a bowtie index comprising known HLA alleles and outputs the most likely HLA class I and class II types, a p-value for each call, and the expression of each class.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read mapping', 'Genetic variation analysis'],\n",
       "  'edam_topics': ['Transcriptomics', 'Mapping']},\n",
       " {'id': 'seq_select_by_id',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'seqkit',\n",
       "  'description': 'FASTA and FASTQ are basic and ubiquitous formats for storing nucleotide and protein sequences. Common manipulations of FASTA/Q file include converting, searching, filtering, deduplication, splitting, shuffling, and sampling. Existing tools only implement some of these manipulations, and not particularly efficiently, and some are only available for certain operating systems. Furthermore, the complicated installation process of required packages and running environments can render these programs less user friendly. SeqKit demonstrates competitive performance in execution time and memory usage compared to similar tools. The efficiency and usability of SeqKit enable researchers to rapidly accomplish common FASTA/Q file manipulations.',\n",
       "  'edam_inputs': [{'formats': ['FASTA', 'FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA', 'FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_operations': ['DNA transcription',\n",
       "   'Sequence trimming',\n",
       "   'DNA translation',\n",
       "   'Sequence conversion'],\n",
       "  'edam_topics': ['Database management', 'Sequence analysis']},\n",
       " {'id': 'seqlib',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'seqtk',\n",
       "  'description': 'A tool for processing sequences in the FASTA or FASTQ format. It parses both FASTA and FASTQ files which can also be optionally compressed by gzip.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Sequence file editing'],\n",
       "  'edam_topics': ['Data management']},\n",
       " {'id': 'seurat',\n",
       "  'description': 'Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['RNA-Seq', 'Transcriptomics']},\n",
       " {'id': 'shasta',\n",
       "  'description': 'De novo assembly from Oxford Nanopore reads.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Genomics', 'Sequence assembly']},\n",
       " {'id': 'shovill',\n",
       "  'description': 'Shovill is a pipeline for assembly of bacterial isolate genomes from Illumina paired-end reads.  Shovill uses SPAdes at its core, but alters the steps before and after the primary assembly step to get similar results in less time. Shovill also supports other assemblers like SKESA, Velvet and Megahit, so you can take advantage of the pre- and post-processing the Shovill provides with those too.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Genomics', 'Microbiology', 'Sequence assembly']},\n",
       " {'id': 'sicer',\n",
       "  'description': 'A clustering approach for identification of enriched domains from histone modification ChIP-seq data.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence contamination filtering'],\n",
       "  'edam_topics': ['ChIP-seq', 'Epigenomics']},\n",
       " {'id': 'simtext',\n",
       "  'description': 'A text mining framework for interactive analysis and visualization of similarities among biomedical entities.\\n\\nFor each search query, PMIDs or abstracts from PubMed are saved.\\n\\n$ git clone https://github.com/dlal-group/simtext.\\n\\nFor all PMIDs in each row of a table the according abstracts are saved in additional columns.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Text mining',\n",
       "   'Database search',\n",
       "   'Text annotation',\n",
       "   'Clustering'],\n",
       "  'edam_topics': ['Natural language processing',\n",
       "   'Workflows',\n",
       "   'Machine learning',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'singlem',\n",
       "  'description': 'Novelty-inclusive microbial community profiling of shotgun metagenomes',\n",
       "  'edam_inputs': [{'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['CSV'], 'term': 'CSV'}],\n",
       "  'edam_operations': ['Taxonomic classification'],\n",
       "  'edam_topics': ['Metagenomics']},\n",
       " {'id': 'sinto',\n",
       "  'description': 'Sinto is a toolkit for processing aligned single-cell data.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Bioinformatics', 'Cell biology']},\n",
       " {'id': 'sip',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sistr',\n",
       "  'description': 'The Salmonella In Silico Typing Resource (SISTR) is an open-source and freely available web application for rapid in silico typing and serovar prediction from Salmonella genome assemblies using cgMLST and O and H antigen gene searching.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'Visualisation',\n",
       "   'Multilocus sequence typing'],\n",
       "  'edam_topics': ['Whole genome sequencing',\n",
       "   'Immunoproteins and antigens',\n",
       "   'Sequence assembly',\n",
       "   'Zoology',\n",
       "   'Public health and epidemiology']},\n",
       " {'id': 'slow5-dorado',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'slow5-guppy',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'slow5tools',\n",
       "  'description': 'Slow5tools is a simple toolkit for converting (FAST5 <-> SLOW5), compressing, viewing, indexing and manipulating data in SLOW5 format.\\n\\nAbout SLOW5 format:\\nSLOW5 is a new file format for storing signal data from Oxford Nanopore Technologies (ONT) devices. SLOW5 was developed to overcome inherent limitations in the standard FAST5 signal data format that prevent efficient, scalable analysis and cause many headaches for developers. SLOW5 can be encoded in human-readable ASCII format, or a more compact and efficient binary format (BLOW5) - this is analogous to the seminal SAM/BAM format for storing DNA sequence alignments. The BLOW5 binary format supports zlib (DEFLATE) compression, or other compression methods, thereby minimising the data storage footprint while still permitting efficient parallel access. Detailed benchmarking experiments have shown that SLOW5 format is an order of magnitude faster and significantly smaller than FAST5.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'smithwaterman',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'Smudgeplots',\n",
       "  'description': 'Reference-free profiling of polyploid genomes | Inference of ploidy and heterozygosity structure using whole genome sequencing data | Smudgeplots are computed from raw or even better from trimmed reads and show the haplotype structure using heterozygous kmer pairs. For example: | This tool extracts heterozygous kmer pairs from kmer dump files and performs gymnastics with them. We are able to disentangle genome structure by comparing the sum of kmer pair coverages (CovA + CovB) to their relative coverage (CovA / (CovA + CovB)). Such an approach also allows us to analyze obscure genomes with duplications, various ploidy levels, etc | GenomeScope 2.0 and Smudgeplots: Reference-free profiling of polyploid genomes Timothy Rhyker Ranallo-Benavidez, Kamil S. Jaron, Michael C. Schatz bioRxiv 747568; doi: https://doi.org/10.1101/747568',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence trimming', 'Genotyping', 'k-mer counting'],\n",
       "  'edam_topics': ['Sequence assembly', 'Genetic variation', 'Mathematics']},\n",
       " {'id': 'snakemake',\n",
       "  'description': 'Workflow engine and language. It aims to reduce the complexity of creating workflows by providing a fast and comfortable execution environment, together with a clean and modern domain specific specification language (DSL) in python style.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Service composition'],\n",
       "  'edam_topics': ['Bioinformatics', 'Workflows', 'Software engineering']},\n",
       " {'id': 'snap',\n",
       "  'description': 'The Semi-HMM-based Nucleic Acid Parser is a gene prediction tool.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction'],\n",
       "  'edam_topics': ['DNA', 'DNA polymorphism', 'Genetics']},\n",
       " {'id': 'snapatac',\n",
       "  'description': 'SnapATAC (Single Nucleus Analysis Pipeline for ATAC-seq) is a fast, accurate and comprehensive method for analyzing single cell ATAC-seq datasets.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Essential dynamics',\n",
       "   'Dimensionality reduction',\n",
       "   'Gene expression profiling'],\n",
       "  'edam_topics': ['Epigenomics',\n",
       "   'Cell biology',\n",
       "   'Chromosome conformation capture',\n",
       "   'Mapping',\n",
       "   'Sequencing']},\n",
       " {'id': 'snappy',\n",
       "  'description': 'a snakemake pipeline for scalable HIV-1 subtyping by phylogenetic pairing | SNAPPy is a Snakemake pipeline for HIV-1 subtyping by phylogenetic pairing | This is the repository for SNAPPy, a Snakemake pipeline for HIV-1 subtyping by phylogenetic pairing. SNAPPy allows high-throughput HIV-1 subtyping locally while being resource efficient and scalable. This pipeline was constructed using Snakemake , and it uses MAFFT and for multiple sequence alignment, BLAST for similarirty querys, IQ-TREE for phylogenetic inference, and several Biopython modules for data parsing an analysis. For in-depth information on how the tool works please visit the documentation page. SNAPPy was design for Linux based operative systems | Welcome to snappys documentation!  SNAPPy-HIV1-Subtyping 1.0.0 documentation | Free document hosting provided by Read the Docs',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence trimming', 'Parsing', 'Genotyping'],\n",
       "  'edam_topics': ['Phylogenetics',\n",
       "   'Workflows',\n",
       "   'Public health and epidemiology']},\n",
       " {'id': 'sniffles',\n",
       "  'description': 'An algorithm for structural variation detection from third generation sequencing alignment.',\n",
       "  'edam_inputs': [{'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['VCF'], 'term': 'VCF'}],\n",
       "  'edam_operations': ['Sequence analysis', 'Structural variation detection'],\n",
       "  'edam_topics': ['DNA structural variation', 'Sequencing']},\n",
       " {'id': 'snippy',\n",
       "  'description': 'Rapid haploid variant calling and core SNP phylogeny generation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree visualisation',\n",
       "   'Phylogenetic tree generation',\n",
       "   'Variant calling'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'Model organisms',\n",
       "   'DNA polymorphism',\n",
       "   'Phylogenetics']},\n",
       " {'id': 'snp_sites',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'snp-dists',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'snpeff',\n",
       "  'description': 'Variant annotation and effect prediction tool. It annotates and predicts the effects of variants on genes and proteins (such as amino acid changes).',\n",
       "  'edam_inputs': [{'formats': ['TSV', 'Textual format'],\n",
       "    'term': 'Textual format'},\n",
       "   {'formats': ['VCF'], 'term': 'VCF'},\n",
       "   {'formats': ['BED'], 'term': 'BED'},\n",
       "   {'formats': ['GFF', 'GTF'], 'term': 'GTF'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['SNP detection'],\n",
       "  'edam_topics': ['DNA polymorphism',\n",
       "   'Genetic variation',\n",
       "   'Nucleic acid sites, features and motifs']},\n",
       " {'id': 'snpfreq',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'snpfreqplot',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'snpsift',\n",
       "  'description': 'Toolbox that allows you to filter and manipulate annotated vcf files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant filtering'],\n",
       "  'edam_topics': ['Data management',\n",
       "   'Data architecture, analysis and design']},\n",
       " {'id': 'sortmerna',\n",
       "  'description': 'Sequence analysis tool for filtering, mapping and OTU-picking NGS reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence similarity search',\n",
       "   'Sequence comparison',\n",
       "   'Sequence alignment analysis'],\n",
       "  'edam_topics': ['Metatranscriptomics', 'Metagenomics']},\n",
       " {'id': 'spaceranger',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'spades',\n",
       "  'description': 'St. Petersburg genome assembler  is intended for both standard isolates and single-cell MDA bacteria assemblies. SPAdes 3.9 works with Illumina or IonTorrent reads and is capable of providing hybrid assemblies using PacBio, Oxford Nanopore and Sanger reads. Additional contigs can be provided and can be used as long reads.',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Genome assembly'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'spaln',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'spec2vec',\n",
       "  'description': 'Improved mass spectral similarity scoring through learning of structural relationships.\\n\\nSpec2vec is a novel spectral similarity score inspired by a natural language processing algorithm -- Word2Vec. Where Word2Vec learns relationships between words in sentences, spec2vec does so for mass fragments and neutral losses in MS/MS spectra. The spectral similarity score is based on spectral embeddings learnt from the fragmental relationships within a large set of spectral data.\\n\\nAnalysis and benchmarking of mass spectra similarity measures using gnps data set.',\n",
       "  'edam_inputs': [{'formats': ['MGF', 'MSP', 'JSON'], 'term': 'JSON'}],\n",
       "  'edam_outputs': [{'formats': ['DSV', 'JSON'], 'term': 'JSON'}],\n",
       "  'edam_operations': ['Database search',\n",
       "   'Comparison',\n",
       "   'Spectral library search',\n",
       "   'Natural product identification'],\n",
       "  'edam_topics': ['Proteomics experiment',\n",
       "   'Metabolomics',\n",
       "   'Natural language processing',\n",
       "   'Proteomics']},\n",
       " {'id': 'spectra',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'srf',\n",
       "  'description': 'Spectral Repeat Finder (SRF) is a program to find repeats through an analysis of the power spectrum of a given DNA sequence.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Analysis'],\n",
       "  'edam_topics': ['Computational biology']},\n",
       " {'id': 'sqlite',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'sqanti3',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'squidpy',\n",
       "  'description': 'Squidpy - Spatial Single Cell Analysis in Python.\\n\\nSquidpy is a tool for the analysis and visualization of spatial molecular data. It builds on top of scanpy and anndata, from which it inherits modularity and scalability. It provides analysis tools that leverages the spatial coordinates of the data, as well as tissue images if available.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Image analysis',\n",
       "   'Visualisation',\n",
       "   'Gene expression profiling'],\n",
       "  'edam_topics': ['Imaging',\n",
       "   'Metabolomics',\n",
       "   'Transcriptomics',\n",
       "   'Proteomics',\n",
       "   'Cytometry']},\n",
       " {'id': 'squirrel',\n",
       "  'description': 'Some QUIck Reconstruction to Resolve Evolutionary Links\\n\\nSquirrel provides a rapid way of producing reliable alignments for MPXV and also enable maximum-likelihood phylogenetics pipeline tree estimation.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic analysis'],\n",
       "  'edam_topics': ['Virology']},\n",
       " {'id': 'sra-tools',\n",
       "  'description': 'The SRA Toolkit and SDK from NCBI is a collection of tools and libraries for using data in the INSDC Sequence Read Archives.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['DNA', 'Genomics', 'Sequencing']},\n",
       " {'id': 'srst2',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ssw',\n",
       "  'description': 'A fast implementation of the Smith-Waterman algorithm whose API that can be flexibly used by programs written in C, C++ and other languages.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling', 'Sequence alignment'],\n",
       "  'edam_topics': ['Sample collections',\n",
       "   'Genomics',\n",
       "   'Data governance',\n",
       "   'Nucleic acid structure analysis']},\n",
       " {'id': 'stacks',\n",
       "  'description': 'Developed to work with restriction enzyme based sequence data, such as RADseq, for building genetic maps and conducting population genomics and phylogeography analysis.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Mapping', 'Population genetics']},\n",
       " {'id': 'star',\n",
       "  'description': 'Ultrafast universal RNA-seq data aligner',\n",
       "  'edam_inputs': [{'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_outputs': [{'formats': ['BAM', 'SAM'], 'term': 'SAM'},\n",
       "   {'formats': [], 'term': 'Gene expression matrix'}],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['RNA-Seq', 'Transcriptomics']},\n",
       " {'id': 'staramr',\n",
       "  'description': 'staramr (*AMR) scans bacterial genome contigs against the ResFinder, PointFinder, and PlasmidFinder databases (used by the ResFinder webservice and other webservices offered by the Center for Genomic Epidemiology) and compiles a summary report of detected antimicrobial resistance genes. The star|* in staramr indicates that it can handle all of the ResFinder, PointFinder, and PlasmidFinder databases.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Antimicrobial resistance prediction'],\n",
       "  'edam_topics': ['Microbiology',\n",
       "   'Public health and epidemiology',\n",
       "   'Infectious disease']},\n",
       " {'id': 'star_fusion',\n",
       "  'description': 'STAR-Fusion, a method that is both fast and accurate in identifying fusion transcripts from RNA-Seq data',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence analysis'],\n",
       "  'edam_topics': ['Gene structure', 'RNA-Seq']},\n",
       " {'id': 'strainberry',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'stringtie',\n",
       "  'description': 'Fast and highly efficient assembler of RNA-Seq alignments into potential transcripts. It uses a novel network flow algorithm as well as an optional de novo assembly step to assemble and quantitate full-length transcripts representing multiple splice variants for each gene locus.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Transcriptome assembly', 'RNA-Seq analysis'],\n",
       "  'edam_topics': ['Transcriptomics', 'RNA-Seq']},\n",
       " {'id': 'subread',\n",
       "  'description': 'Subread is a general-purpose read aligner which can be used to map both genomic DNA-seq reads and RNA-seq reads. It uses a new mapping paradigm called \"seed-and-vote\" to achieve fast, accurate and scalable read mapping. It automatically determines if a read should be globally or locally aligned, therefore particularly powerful in mapping RNA-seq reads. It supports indel detection and can map reads with both fixed and variable lengths.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Read mapping'],\n",
       "  'edam_topics': ['Sequencing',\n",
       "   'Mapping',\n",
       "   'Whole genome sequencing',\n",
       "   'RNA-Seq']},\n",
       " {'id': 'subtom',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'superfocus',\n",
       "  'description': 'An agile homology-based approach using a reduced SEED database to report the subsystems present in metagenomic samples and profile their abundances.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence feature detection'],\n",
       "  'edam_topics': ['DNA', 'Metagenomics', 'Sequence analysis']},\n",
       " {'id': 'suppa',\n",
       "  'description': 'This tool generates Alternative Splicing (AS) events from an annotation and calculates the PSI (\"Percentage Spliced In\") value for each event exploiting fast quantification of transcript abundances from multiple samples.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Splicing model analysis'],\n",
       "  'edam_topics': ['Transcriptomics', 'RNA splicing']},\n",
       " {'id': 'svim-asm',\n",
       "  'description': 'Structural variant detection from haploid and diploid genome assemblies.\\n\\nSVIM-asm - Structural variant identification method (Assembly edition).\\n\\nSVIM-asm (pronounced SWIM-assem) is a structural variant caller for haploid or diploid genome-genome alignments. It analyzes a given sorted BAM file (preferably from minimap2) and detects five different variant classes between the query assembly and the reference: deletions, insertions, tandem and interspersed duplications and inversions.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly', 'Variant calling', 'Genotyping'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'Genotype and phenotype',\n",
       "   'Sequencing',\n",
       "   'DNA polymorphism']},\n",
       " {'id': 'syri',\n",
       "  'description': 'SyRI is tool for finding genomic rearrangements and local sequence differences from whole-genome assemblies.\\n\\nGenomic differences range from single nucleotide differences to complex structural variations. Current methods typically annotate sequence differences ranging from SNPs to large indels accurately but do not unravel the full complexity of structural rearrangements, including inversions, translocations, and duplications, where highly similar sequence changes in location, orientation, or copy number. Here, we present SyRI, a pairwise whole-genome comparison tool for chromosome-level assemblies. SyRI starts by finding rearranged regions and then searches for differences in the sequences, which are distinguished for residing in syntenic or rearranged regions. This distinction is important as rearranged regions are inherited differently compared to syntenic regions.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Haplotype mapping',\n",
       "   'Variant calling',\n",
       "   'Genotyping',\n",
       "   'Sequence assembly',\n",
       "   'Read mapping'],\n",
       "  'edam_topics': ['DNA polymorphism',\n",
       "   'Sequence assembly',\n",
       "   'DNA structural variation',\n",
       "   'Mapping',\n",
       "   'Sequencing']},\n",
       " {'id': 'targetfinder',\n",
       "  'description': 'Targetfinder.org provides a web based resource that allows users to find genes that have a similar expression to a query gene signature.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene prediction',\n",
       "   'Gene expression profiling',\n",
       "   'Gene expression analysis'],\n",
       "  'edam_topics': ['Gene expression',\n",
       "   'Transcription factors and regulatory sites',\n",
       "   'Gene transcripts',\n",
       "   'Small molecules',\n",
       "   'Gene regulation']},\n",
       " {'id': 'taxonkit',\n",
       "  'description': 'TaxonKit is a practical and efficient NCBI taxonomy toolkit.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Formatting', 'Data retrieval'],\n",
       "  'edam_topics': ['Taxonomy', 'Biotechnology', 'Ecology']},\n",
       " {'id': 'tb_variant_filter',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'tbl2asn',\n",
       "  'description': 'Tbl2asn is a command-line program that automates the creation of sequence records for submission to GenBank. It uses many of the same functions as Genome Workbench but is driven generally by data files. Tbl2asn generates .sqn files for submission to GenBank.',\n",
       "  'edam_inputs': [{'formats': ['FASTA'], 'term': 'FASTA'},\n",
       "   {'formats': ['Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_outputs': [{'formats': ['Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_operations': ['Deposition'],\n",
       "  'edam_topics': ['Data submission, annotation and curation']},\n",
       " {'id': 'tbprofiler',\n",
       "  'description': 'A tool for drug resistance prediction from _M. tuberculosis_ genomic data (sequencing reads, alignments or variants).',\n",
       "  'edam_inputs': [{'formats': ['FASTQ-like format (text)'],\n",
       "    'term': 'FASTQ-like format (text)'}],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Antimicrobial resistance prediction'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'tbvcfreport',\n",
       "  'description': 'The COMBAT-TB Workbench is an IRIDA based, module workbench for M. tuberculosis bioinformatics. It is designed to be easily deployed on a single server.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Whole genome sequencing',\n",
       "   'Microbiology',\n",
       "   'Workflows',\n",
       "   'Phylogeny']},\n",
       " {'id': 'tesorter',\n",
       "  'description': 'lineage-level classification of transposable elements using conserved protein domains.\\n\\nNote: do not move or hard link TEsorter.py alone to anywhere else, as it rely on database/ and bin/. You can add the directory to PATH or soft link TEsorter.py to PATH',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Phylogenetic tree reconstruction'],\n",
       "  'edam_topics': ['Protein folds and structural domains',\n",
       "   'Phylogenetics',\n",
       "   'Transcription factors and regulatory sites',\n",
       "   'Plant biology',\n",
       "   'Model organisms']},\n",
       " {'id': 'TEtranscripts',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'text_processing',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'ThermoRawFileParser',\n",
       "  'description': 'Open-source, crossplatform tool that converts Thermo RAW files into open file formats such as MGF and to the HUPO-PSI standard file format mzML',\n",
       "  'edam_inputs': [{'formats': ['Thermo RAW'], 'term': 'Thermo RAW'}],\n",
       "  'edam_outputs': [{'formats': ['MGF', 'mzML'], 'term': 'mzML'}],\n",
       "  'edam_operations': ['Conversion', 'Filtering', 'Peak detection'],\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment', 'Metabolomics']},\n",
       " {'id': 'tmt-analyst',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'tophat',\n",
       "  'description': 'Program that aligns RNA-Seq reads to a genome in order to identify exon-exon splice junctions. It is built on the ultrafast short read mapping program Bowtie. A stable SAMtools version is now packaged with the program.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Genome identifier'},\n",
       "   {'formats': ['FASTQ', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['SAM'], 'term': 'SAM'},\n",
       "   {'formats': ['BED'], 'term': 'BED'},\n",
       "   {'formats': ['BED'], 'term': 'BED'}],\n",
       "  'edam_operations': ['Splice site prediction', 'Read mapping'],\n",
       "  'edam_topics': ['RNA-Seq', 'Mapping', 'Rare diseases']},\n",
       " {'id': 'transdecoder',\n",
       "  'description': 'TransDecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Coding region prediction',\n",
       "   'de Novo sequencing',\n",
       "   'De-novo assembly'],\n",
       "  'edam_topics': ['Genomics',\n",
       "   'Gene transcripts',\n",
       "   'RNA-Seq',\n",
       "   'Gene expression',\n",
       "   'Sequence assembly',\n",
       "   'Whole genome sequencing']},\n",
       " {'id': 'transit',\n",
       "  'description': 'A tool for the analysis of Tn-Seq data. It provides an easy to use graphical interface and access to three different analysis methods that allow the user to determine essentiality in a single condition as well as between conditions.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Transposon prediction'],\n",
       "  'edam_topics': ['DNA', 'Sequencing', 'Mobile genetic elements']},\n",
       " {'id': 'tpp',\n",
       "  'description': 'Institute for Systems Biology \"Trans-Proteomic Pipeline\"',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment']},\n",
       " {'id': 'transvar',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'treeshrink',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'trf',\n",
       "  'description': 'Tandem Repeats Finder. Find tandem repeats in DNA sequences without the need to specify either the pattern or pattern size. It uses the method of k-tuple matching to avoid the need for full scale alignment matrix computations. It requires no a priori knowledge of the pattern, pattern size or number of copies. There are no restrictions on the size of the repeats that can be detected. It determines a consensus pattern for the smallest repetitive unit in the tandem repeat.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Repeat sequence detection'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Sequence composition, complexity and repeats']},\n",
       " {'id': 'trf-mod',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'trimal',\n",
       "  'description': 'Tool for the automated removal of spurious sequences or poorly aligned regions from a multiple sequence alignment.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence alignment'}],\n",
       "  'edam_outputs': [{'formats': ['phylipnon',\n",
       "     'FASTA',\n",
       "     'ClustalW format',\n",
       "     'PHYLIP format'],\n",
       "    'term': 'PHYLIP format'},\n",
       "   {'formats': ['Textual format'], 'term': 'Textual format'}],\n",
       "  'edam_operations': ['Multiple sequence alignment'],\n",
       "  'edam_topics': ['Sequence analysis',\n",
       "   'Sequencing',\n",
       "   'Sequence sites, features and motifs']},\n",
       " {'id': 'trim_galore',\n",
       "  'description': 'A wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files, with some extra functionality for MspI-digested RRBS-type (Reduced Representation Bisufite-Seq) libraries.',\n",
       "  'edam_inputs': [{'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_outputs': [{'formats': ['BAM'], 'term': 'BAM'}],\n",
       "  'edam_operations': ['Sequence trimming',\n",
       "   'Primer removal',\n",
       "   'Read pre-processing'],\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'trimmomatic',\n",
       "  'description': 'A flexible read trimming tool for Illumina NGS data',\n",
       "  'edam_inputs': [{'formats': ['Binary format',\n",
       "     'FASTQ-solexa',\n",
       "     'FASTQ-illumina',\n",
       "     'FASTQ-sanger',\n",
       "     'FASTQ'],\n",
       "    'term': 'FASTQ'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['TSV'], 'term': 'TSV'},\n",
       "   {'formats': ['FASTQ'], 'term': 'FASTQ'}],\n",
       "  'edam_operations': ['Sequence trimming'],\n",
       "  'edam_topics': ['Sequencing',\n",
       "   'Sequence analysis',\n",
       "   'Data quality management']},\n",
       " {'id': 'trinity',\n",
       "  'description': 'Trinity is a transcriptome assembler which relies on three different tools, inchworm an assembler, chrysalis which pools contigs and butterfly which amongst others compacts a graph resulting from butterfly with reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Transcriptome assembly'],\n",
       "  'edam_topics': ['Transcriptomics', 'Gene expression', 'Gene transcripts']},\n",
       " {'id': 'trinotate',\n",
       "  'description': 'Comprehensive annotation suite designed for automatic functional annotation of transcriptomes, particularly de novo assembled transcriptomes, from model or non-model organisms.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Gene functional annotation'],\n",
       "  'edam_topics': ['Gene expression', 'Transcriptomics']},\n",
       " {'id': 'trnascan-se',\n",
       "  'description': 'A program for improved detection of transfer RNA genes in genomic sequence.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['tRNA gene prediction', 'Sequence feature detection'],\n",
       "  'edam_topics': ['RNA', 'Genomics', 'Sequence analysis']},\n",
       " {'id': 'trycycler',\n",
       "  'description': 'Trycycler: consensus long-read assemblies for bacterial genomes',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genome assembly',\n",
       "   'Sequence trimming',\n",
       "   'Scaffold gap completion',\n",
       "   'Sequence assembly validation'],\n",
       "  'edam_topics': ['Sequence assembly',\n",
       "   'DNA',\n",
       "   'Human biology',\n",
       "   'Whole genome sequencing',\n",
       "   'Genomics']},\n",
       " {'id': 'ucsc',\n",
       "  'description': 'Utilities for handling sequences and assemblies from the UCSC Genome Browser project.',\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': ['Sequence analysis']},\n",
       " {'id': 'umi_tools',\n",
       "  'description': 'Tools for handling Unique Molecular Identifiers in NGS data sets.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequencing quality control'],\n",
       "  'edam_topics': ['NGS',\n",
       "   'Sequence sites, features and motifs',\n",
       "   'Quality affairs']},\n",
       " {'id': 'unicycler',\n",
       "  'description': 'A tool for assembling bacterial genomes from a combination of short (2nd generation) and long (3rd generation) sequencing reads.',\n",
       "  'edam_inputs': [{'formats': ['Raw sequence format'],\n",
       "    'term': 'Raw sequence format'}],\n",
       "  'edam_outputs': [{'formats': ['Graph format', 'FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_operations': ['Genome assembly', 'Aggregation'],\n",
       "  'edam_topics': ['Microbiology',\n",
       "   'Genomics',\n",
       "   'Sequencing',\n",
       "   'Sequence assembly']},\n",
       " {'id': 'unipept',\n",
       "  'description': 'Metaproteomics data analysis with a focus on interactive data visualizations.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Protein sequence'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence set'},\n",
       "   {'formats': [], 'term': 'Phylogenetic tree'}],\n",
       "  'edam_operations': ['Prediction and recognition', 'Visualisation'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Proteogenomics',\n",
       "   'Biodiversity',\n",
       "   'Workflows']},\n",
       " {'id': 'UniProt_Downloader',\n",
       "  'description': 'The universal protein knowledgebase in 2021.\\n\\nYou are using a version of browser that may not display all the features of this website.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Text annotation',\n",
       "   'Data retrieval',\n",
       "   'Database search',\n",
       "   'Deposition'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Sequence analysis',\n",
       "   'Literature and language',\n",
       "   'Model organisms',\n",
       "   'Proteomics experiment']},\n",
       " {'id': 'uniprot_rest_interface',\n",
       "  'description': 'The universal protein knowledgebase in 2021.\\n\\nYou are using a version of browser that may not display all the features of this website.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Text annotation',\n",
       "   'Data retrieval',\n",
       "   'Database search',\n",
       "   'Deposition'],\n",
       "  'edam_topics': ['Proteomics',\n",
       "   'Sequence analysis',\n",
       "   'Literature and language',\n",
       "   'Model organisms',\n",
       "   'Proteomics experiment']},\n",
       " {'id': 'unzip',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'vep',\n",
       "  'description': 'Tool for predicting effects of variants for any genome in Ensembl or with genome annotation (via GFF). This includes vertebrates and also plants, fungi, protists, metazoa and bacteria.  There is a web and a REST API version but the most powerful is the Perl script version. See McLaren et al., 2016, Genome Biology',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['SNP annotation', 'Genetic variation analysis'],\n",
       "  'edam_topics': ['Genetic variation', 'Genomics']},\n",
       " {'id': 'validate_fasta_database',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'vapor',\n",
       "  'description': 'VAPOR is a tool for classification of Influenza samples from raw short read sequence data for downstream bioinformatics analysis. VAPOR is provided with a fasta file of full-length sequences (> 20,000) for a given segment, a set of reads, and attempts to retrieve a reference that is closest to the sample strain.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data retrieval', 'De-novo assembly', 'Read mapping'],\n",
       "  'edam_topics': ['Whole genome sequencing', 'Mapping', 'Sequence assembly']},\n",
       " {'id': 'varscan',\n",
       "  'description': 'VarScan, an open source tool for variant detection that is compatible with several short read align-ers.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Variant calling'],\n",
       "  'edam_topics': ['Sequencing', 'DNA polymorphism']},\n",
       " {'id': 'vcflib',\n",
       "  'description': 'API and command line utilities for the manipulation of VCF files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling'],\n",
       "  'edam_topics': ['Genomics', 'Data management']},\n",
       " {'id': 'vcftools',\n",
       "  'description': 'Provide easily accessible methods for working with complex genetic variation data in the form of VCF files.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Data handling',\n",
       "   'Variant filtering',\n",
       "   'Genetic variation analysis'],\n",
       "  'edam_topics': ['Genetic variation']},\n",
       " {'id': 'velocyto',\n",
       "  'description': 'Estimating RNA velocity in single cell RNA sequencing datasets',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['RNA-Seq analysis', 'Splicing analysis'],\n",
       "  'edam_topics': None},\n",
       " {'id': 'velvet',\n",
       "  'description': 'A de novo genomic assembler specially designed for short read sequencing technologies, such as Solexa or 454 or SOLiD.',\n",
       "  'edam_inputs': [{'formats': ['BAM', 'SAM', 'FASTQ', 'FASTA'],\n",
       "    'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence'},\n",
       "   {'formats': [], 'term': 'Resource metadata'}],\n",
       "  'edam_operations': ['Formatting'],\n",
       "  'edam_topics': ['Sequence assembly']},\n",
       " {'id': 'verkko',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'vitessce',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'vsearch',\n",
       "  'description': 'High-throughput search and clustering sequence analysis tool. It supports de novo and reference based chimera detection, clustering, full-length and prefix dereplication, reverse complementation, masking, all-vs-all pairwise global alignment, exact and global alignment searching, shuffling, subsampling and sorting. It also supports FASTQ file analysis, filtering and conversion.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Nucleic acid sequence'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Sequence similarity score'},\n",
       "   {'formats': [], 'term': 'Sequence alignment'}],\n",
       "  'edam_operations': ['DNA mapping', 'Chimera detection'],\n",
       "  'edam_topics': ['Metagenomics', 'Sequence analysis']},\n",
       " {'id': 'waveica',\n",
       "  'description': 'Removal of batch effects for large-scale untargeted metabolomics data based on wavelet transform.',\n",
       "  'edam_inputs': [{'formats': ['CSV', 'TSV', 'Binary format'],\n",
       "    'term': 'Binary format'},\n",
       "   {'formats': ['CSV', 'TSV', 'Binary format'], 'term': 'Binary format'}],\n",
       "  'edam_outputs': [{'formats': ['CSV', 'DSV', 'Binary format'],\n",
       "    'term': 'Binary format'}],\n",
       "  'edam_operations': ['Standardisation and normalisation'],\n",
       "  'edam_topics': ['Metabolomics']},\n",
       " {'id': 'weblogo3',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'whatshap',\n",
       "  'description': 'Software for phasing genomic variants using DNA sequencing reads, also called haplotype assembly. It is especially suitable for long reads, but works also well with short reads.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Genotyping'],\n",
       "  'edam_topics': ['DNA polymorphism', 'Sequence assembly']},\n",
       " {'id': 'windowmasker',\n",
       "  'description': 'windowmasker identifies and masks highly repetitive DNA sequences in a genome, using only the sequence of the genome itself.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Repeat sequence detection', 'Repeat sequence analysis'],\n",
       "  'edam_topics': ['Sequencing']},\n",
       " {'id': 'winnowmap',\n",
       "  'description': 'Winnowmap is a long-read mapping algorithm optimized for mapping ONT and PacBio reads to repetitive reference sequences. Winnowmap development began on top of minimap2 codebase, and since then we have incorporated the following two ideas to improve mapping accuracy within repeats',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Sequence alignment'],\n",
       "  'edam_topics': ['Mapping']},\n",
       " {'id': 'workflow4metabolomics',\n",
       "  'description': 'First fully open-source and collaborative online platform for computational metabolomics. It includes preprocessing, normalization, quality control, statistical analysis of LC/MS, FIA-MS, GC/MS and NMR data.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'NMR spectrum'},\n",
       "   {'formats': ['mzData', 'mzML', 'NetCDF', 'mzXML'], 'term': 'mzXML'}],\n",
       "  'edam_outputs': [{'formats': ['Chemical formula format'],\n",
       "    'term': 'Chemical formula format'},\n",
       "   {'formats': [], 'term': 'Distance matrix'}],\n",
       "  'edam_operations': ['Annotation',\n",
       "   'Spectral analysis',\n",
       "   'Enrichment analysis',\n",
       "   'Statistical calculation',\n",
       "   'Visualisation'],\n",
       "  'edam_topics': ['Systems biology', 'Metabolomics', 'Computational biology']},\n",
       " {'id': 'wormbase',\n",
       "  'description': 'Caenorhabditis elegans genome database. International consortium of biologists and computer scientists dedicated to providing the research community with accurate, current, accessible information concerning the genetics, genomics and biology of C. elegans and related nematodes. Founded in 2000, the Consortium is led by Paul Sternberg of CalTech, Paul Kersey of the EBI, Matt Berriman of the Wellcome Trust Sanger Institute, and Lincoln Stein of the Ontario Institute for Cancer Research.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Gene symbol'}],\n",
       "  'edam_outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam_operations': ['Query and retrieval'],\n",
       "  'edam_topics': ['Model organisms']},\n",
       " {'id': 'xarray',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'xcms',\n",
       "  'description': 'Framework for processing and visualization of chromatographically separated and single-spectra mass spectral data.  The packages enables imports from AIA/ANDI NetCDF, mzXML, mzData and mzML files and preprocesses data for high-throughput, untargeted analyte profiling.',\n",
       "  'edam_inputs': [{'formats': ['mzML', 'NetCDF', 'mzXML'], 'term': 'mzXML'},\n",
       "   {'formats': ['mzML', 'NetCDF', 'mzXML'], 'term': 'mzXML'}],\n",
       "  'edam_outputs': [{'formats': ['mzQuantML', 'NetCDF'], 'term': 'NetCDF'}],\n",
       "  'edam_operations': ['Mass spectrum visualisation',\n",
       "   'Chromatographic alignment',\n",
       "   'Label-free quantification',\n",
       "   'Validation',\n",
       "   'Data filtering',\n",
       "   'Peak detection'],\n",
       "  'edam_topics': ['Data visualisation', 'Metabolomics']},\n",
       " {'id': 'xml4ena',\n",
       "  'description': None,\n",
       "  'edam_inputs': None,\n",
       "  'edam_outputs': None,\n",
       "  'edam_operations': None,\n",
       "  'edam_topics': None},\n",
       " {'id': 'xpore',\n",
       "  'description': 'Detection of differential RNA modifications from direct RNA sequencing of human cell lines.\\n\\nPython package for detection of differential RNA modifications from direct RNA sequencing.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Quantification',\n",
       "   'Expression analysis',\n",
       "   'DMR identification'],\n",
       "  'edam_topics': ['RNA-Seq',\n",
       "   'RNA',\n",
       "   'Gene expression',\n",
       "   'Genotype and phenotype',\n",
       "   'Molecular interactions, pathways and networks']},\n",
       " {'id': 'xtandem',\n",
       "  'description': 'Matches tandem mass spectra with peptide sequences.',\n",
       "  'edam_inputs': [{'formats': ['dta',\n",
       "     'pkl',\n",
       "     'mzXML',\n",
       "     'mzData',\n",
       "     'MGF',\n",
       "     'X!Tandem XML',\n",
       "     'mzML'],\n",
       "    'term': 'mzML'},\n",
       "   {'formats': ['FASTA'], 'term': 'FASTA'}],\n",
       "  'edam_outputs': [{'formats': ['X!Tandem XML', 'mzIdentML'],\n",
       "    'term': 'mzIdentML'}],\n",
       "  'edam_operations': ['Blind peptide database search',\n",
       "   'PTM identification',\n",
       "   'Peptide identification',\n",
       "   'Peptide database search',\n",
       "   'Protein identification'],\n",
       "  'edam_topics': ['Proteomics', 'Proteomics experiment']},\n",
       " {'id': 'xtb',\n",
       "  'description': 'The xTB molecular optimization tool, based on the Semiempirical Tight Binding method (GFNn-xTB), is implemented in the xtb (extended tight binding) program package. This tool handles molecular .XYZ input format and provides various levels of optimization to suit different computational needs. The xTB method offers an efficient approach for molecular structure optimization, making it valuable for large-scale quantum chemical simulations.',\n",
       "  'edam_inputs': [{'formats': ['XYZ'], 'term': 'XYZ'}],\n",
       "  'edam_outputs': [{'formats': ['XYZ'], 'term': 'XYZ'}],\n",
       "  'edam_operations': ['3D profile generation'],\n",
       "  'edam_topics': ['Computational chemistry']},\n",
       " {'id': 'yahs',\n",
       "  'description': 'YaHS is scaffolding tool using Hi-C data. It relies on a new algorithm for contig joining detection which considers the topological distribution of Hi-C signals aiming to distinguish real interaction signals from mapping noises.',\n",
       "  'edam_inputs': [],\n",
       "  'edam_outputs': [],\n",
       "  'edam_operations': ['Scaffolding', 'Genome assembly', 'De-novo assembly'],\n",
       "  'edam_topics': ['Sequence assembly', 'DNA binding sites']},\n",
       " {'id': 'yeastmine',\n",
       "  'description': 'Search and retrieve S. cerevisiae data, populated by SGD and powered by InterMine',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence features'},\n",
       "   {'formats': [], 'term': 'Ontology data'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Over-representation data'},\n",
       "   {'formats': [], 'term': 'Query script'},\n",
       "   {'formats': [], 'term': 'Sequence features'}],\n",
       "  'edam_operations': ['Data retrieval',\n",
       "   'Visualisation',\n",
       "   'Gene-set enrichment analysis',\n",
       "   'Query and retrieval'],\n",
       "  'edam_topics': ['Data management', 'Genomics', 'Proteomics']},\n",
       " {'id': 'zebrafishmine',\n",
       "  'description': 'ZebrafishMine is powered by the InterMIne data warehouse system, and integrates biological data sets from multiple sources.\\r\\nIt currently includes updates of data from ZFIN, the zebrafish model organism database. There is also data from the Panther database.',\n",
       "  'edam_inputs': [{'formats': [], 'term': 'Sequence features'},\n",
       "   {'formats': [], 'term': 'Ontology data'}],\n",
       "  'edam_outputs': [{'formats': [], 'term': 'Over-representation data'},\n",
       "   {'formats': [], 'term': 'Query script'},\n",
       "   {'formats': [], 'term': 'Sequence features'}],\n",
       "  'edam_operations': ['Data retrieval',\n",
       "   'Visualisation',\n",
       "   'Gene-set enrichment analysis',\n",
       "   'Query and retrieval'],\n",
       "  'edam_topics': ['Data management', 'Genomics', 'Proteomics']}]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collected_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d00e4a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate set of unique edam operations\n",
    "x = { op\n",
    "        for tool in collected_data\n",
    "        for op in (tool.get(\"edam_operations\") or [])\n",
    "}\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65fd7edd",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pronto import Ontology\n",
    "from bio_mcp.cache.embeddings import load_edam_obo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06d83a6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "obo = load_edam_obo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b04fbb0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "obo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1df1f35c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Test concept extract\n",
    "from bio_mcp.cache.embeddings import extract_concepts_from_biotools_yaml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6b95bfd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "concepts = extract_concepts_from_biotools_yaml(yaml_path=Path(\"/home/ubuntu/cache/johan.yaml\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a6b174eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'3d de novo assembly 3d dna is a pipeline for de novo assembly using hic': {'3d-dna'},\n",
       " 'mass screening of contigs for antimicrobial resistance or virulence genes': {'abricate'},\n",
       " 'an amr gene detection pipeline that runs amrfinderplus on a single or list of given isolates and collates the results into a table separating genes identified into functionally relevant groups': {'abritamr'},\n",
       " 'de novo genome sequence assembler using short reads': {'abyss'},\n",
       " 'another gff analysis toolkit agat suite of tools to handle gene annotations in any gtf gff format': {'agat'},\n",
       " 'ai guided 3d structure prediction of novel proteins alphafold uses neural networks to predict the tertiary 3d structure of proteins alphafold accepts an amino acid sequence as an input which it will then will fold into a 3d model': {'alphafold_2'},\n",
       " 'the assisted model building with energy refinement tool refers to two things a set of molecular mechanical force fields for the simulation of biomolecules which are in the public domain and are used in a variety of simulation programs and a package of molecular simulation programs which includes source code and demos': {'amber'},\n",
       " 'consists of several independently developed packages that work well by themselves and with amber assisted model building with energy refinement itself the suite can also be used to carry out complete non periodic molecular dynamics simulations using nab with generalized born solvent models': {'ambertools'},\n",
       " 'software package specially developed for the study of genes primary structure it uses gene sequences downloaded from public databases as fasta and genbank and it applies a set of statistical and visualization methods in different ways to reveal information about codon context codon usage nucleotide repeats within open reading frames orfeome and others': {'anaconda'},\n",
       " 'from https anndata readthedocs io en latest python package for handling annotated data matrices in memory and on disk positioned between pandas and xarray': {'anndata'},\n",
       " 'this tool can get annotation for a generic set of ids using the bioconductor annotation data packages supported organisms are human mouse rat fruit fly and zebrafish the org db packages that are used here are primarily based on mapping using entrez gene identifiers more information on the annotation packages can be found at the bioconductor website for example information on the human annotation package org hs eg db can be found here': {'annotatemyids'},\n",
       " 'rapid genome wide identification annotation and analysis of secondary metabolite biosynthesis gene clusters in bacterial and fungal genomes it integrates and cross links with a large number of in silico secondary metabolite analysis tools that have been published earlier': {'antismash'},\n",
       " 'convert various sequence formats to fasta': {'any2fasta'},\n",
       " 'apollo is a genome annotation viewer and editor apollo allows researchers to explore genomic annotations at many levels of detail and to perform expert annotation curation all in a graphical environment': {'apollo'},\n",
       " 'aragorn detects trna mtrna info about tmrna and tmrna genes': {'aragorn'},\n",
       " 'filesender is an open source web application for sending files of any size quickly and securely filesender is available to all institutions affiliated with the australian access federation aaf you can use your institution login details to access and use filesender': {'aarnet_filesender'},\n",
       " 'arriba is a command line tool to detect gene fusions from rna seq data based on the star aligner in addition to fusions it can detect exon duplications inversions and truncations of genes i e breakpoints in introns and intergenic regions arriba is the winner of the dream smc rna challenge': {'arriba'},\n",
       " 'a bioinformatics pipeline for working with virus sequencing data sequenced with nanopore': {'artic'},\n",
       " 'augustus is a eukaryotic gene prediction tool it can integrate evidence e g from rna seq ests proteomics but can also predict genes ab initio the ppx extension to augustus can take a protein sequence multiple sequence alignment as input to find new members of the family in a genome it can be run through a web interface see https bio tools webaugustus or downloaded and run locally': {'augustus'},\n",
       " 'autodock vina is a new open source program for drug discovery molecular docking and virtual screening offering multi core capability high performance and enhanced accuracy and ease of use': {'autodock_vina'},\n",
       " 'the bio2byte tools server b2btools offers the following single protein sequence based predictions backbone and sidechain dynamics dynamine helix sheet coil and polyproline ii propensity early folding propensity efoldmine disorder disomine beta sheet aggregation agmata in addition multiple sequence alignments msas can be uploaded to scan the biophysical limits of a protein family as defined in the msa': {'b2btools'},\n",
       " 'rapid standardized annotation of bacterial genomes mags plasmids': {'bakta'},\n",
       " 'bamtools provides a fast flexible c api toolkit for reading writing and managing bam files': {'bamtools'},\n",
       " 'bamutil provides a serie of programs to work on sam bam files': {'bamutil_diff'},\n",
       " 'gui program that allows users to interact with the assembly graphs made by de novo assemblers such as velvet spades megahit and others it visualises assembly graphs with connections using graph layout algorithms': {'bandage'},\n",
       " 'the baredsc bayesian approach to retreive expression distribution of single cell is a tool that uses a monte carlo markov chain to estimate a confidence interval on the probability density function pdf of expression of one or two genes from single cell rna seq data': {'baredsc'},\n",
       " 'predict the location of ribosomal rna genes in genomes it supports bacteria 5s 23s 16s archaea 5s 5 8s 23s 16s mitochondria 12s 16s and eukaryotes 5s 5 8s 28s 18s': {'barrnap'},\n",
       " 'bbmap is a fast splice aware aligner for rna and dna it is faster than almost all short read aligners yet retains unrivaled sensitivity and specificity particularly for reads with many errors and indels': {'bbmap'},\n",
       " 'bbtools is a suite of fast multithreaded bioinformatics tools designed for analysis of dna and rna sequence data bbtools can handle common sequencing file formats such as fastq fasta sam scarf fasta qual compressed or raw with autodetection of quality encoding and interleaving it is written in java and works on any platform supporting java including linux macos and microsoft windows and linux there are no dependencies other than java version 7 or higher program descriptions and options are shown when running the shell scripts with no parameters bbtools is open source and free for unlimited use and is used regularly by doe jgi and other institutions around the world': {'bbtools'},\n",
       " 'a tool for filling the gap created by genomic data processing analysis by rebasing some analysis results against the parent features which were originally analysed': {'bcbiogff'},\n",
       " 'bcftools is a set of utilities that manipulate variant calls in the variant call format vcf and its binary counterpart bcf all commands work transparently with both vcfs and bcfs both uncompressed and bgzf compressed': {'bcftools'},\n",
       " 'a global search engine for genetic mutations': {'beacon2'},\n",
       " 'beagle is a software package that performs genotype calling genotype phasing imputation of ungenotyped markers and identity by descent segment detection': {'beagle'},\n",
       " 'beagle is a high performance library that can perform the core calculations at the heart of most bayesian and maximum likelihood phylogenetics packages': {'beagle-lib'},\n",
       " 'the bayesian evolutionary analysis sampling trees is a cross platform program for bayesian analysis of molecular sequences using mcmc markov chain monte carlo it is entirely orientated towards rooted time measured phylogenies inferred using strict or relaxed molecular clock models it can be used as a method of reconstructing phylogenies but is also a framework for testing evolutionary hypotheses without conditioning on a single tree topology': {'beast'},\n",
       " 'bayesian phylogenetic analysis of molecular sequences it estimates rooted time measured phylogenies using strict or relaxed molecular clock models it can be used as a method of reconstructing phylogenies but is also a framework for testing evolutionary hypotheses without conditioning on a single tree topology it uses markov chain monte carlo mcmc to average over tree space so that each tree is weighted proportional to its posterior probability it includes a graphical user interface for setting up standard analyses and a suit of programs for analysing the results': {'beast2'},\n",
       " 'convert a bed format file of the proteins from a proteomics search database into a tabular format for the multiomics visualization platform mvp': {'bed_to_protein_map'},\n",
       " 'bedtools is an extensive suite of utilities for comparing genomic features in bed format': {'bedtools'},\n",
       " 'the bellerophon pipeline improving de novo transcriptomes and removing chimeras bellerophon is a pipeline created to remove falsely assembled chimeric transcripts in de novo transcriptome assemblies the pipeline can be downloaded as a vragrant virtual machine https app vagrantup com bellerophon boxes bellerophon this is recommended as it avoids backwards compatibility problems with transrate': {'bellerophon'},\n",
       " 'trim circularise orient and filter long read bacterial genome assemblies': {'berokka'},\n",
       " 'binette is a fast and accurate binning refinement tool to constructs high quality mags from the output of multiple binning tools': {'binette'},\n",
       " 'bio3d is an r package containing utilities for the analysis of protein structure sequence and trajectory data': {'bio3d'},\n",
       " 'this package includes basic tools for reading biom format files accessing and subsetting data tables from a biom object as well as limited support for writing a biom object back to a biom format file the design of this api is intended to match the python api and other tools included with the biom format project but with a decidedly r flavor that should be familiar to r users this includes s4 classes and methods as well as extensions of common core functions methods': {'biom-format'},\n",
       " 'a collection of perl modules that facilitate the development of perl scripts for bioinformatics applications it provides software modules for many of the typical tasks of bioinformatics programming': {'bioperl'},\n",
       " 'biopython is a set of freely available tools for biological computation written in python by an international team of developers': {'biopython'},\n",
       " 'biotransformer is a freely available web server that supports accurate rapid and comprehensive in silico metabolism prediction': {'biotransformer'},\n",
       " 'bismark is a tool to map bisulfite treated sequencing reads and perform methylation calling in a quick and easy to use fashion': {'bismark'},\n",
       " 'a tool that finds regions of similarity between biological sequences the program compares nucleotide or protein sequences to sequence databases and calculates the statistical significance': {'blast',\n",
       "  'blast+'},\n",
       " 'fast accurate spliced alignment of dna sequences': {'blat'},\n",
       " 'detect blocks of overlapping reads using a gaussian distribution approach': {'Blockbuster'},\n",
       " 'boost is a set of libraries for the c programming language that provides support for tasks and structures such as linear algebra pseudorandom number generation multithreading image processing regular expressions and unit testing': {'boost'},\n",
       " 'bowtie is an ultrafast memory efficient short read aligner': {'bowtie'},\n",
       " 'bowtie 2 is an ultrafast and memory efficient tool for aligning sequencing reads to long reference sequences it is particularly good at aligning reads of about 50 up to 100s or 1 000s of characters and particularly good at aligning to relatively long e g mammalian genomes bowtie 2 indexes the genome with an fm index to keep its memory footprint small for the human genome its memory footprint is typically around 3 2 gb bowtie 2 supports gapped local and paired end alignment modes': {'bowtie2'},\n",
       " 'bracken is a companion program to kraken 1 krakenuniq or kraken 2 while kraken classifies reads to multiple levels in the taxonomic tree bracken allows estimation of abundance at a single level using those classifications e g bracken can estimate abundance of species within a sample': {'bracken'},\n",
       " 'pipeline for unsupervised rna seq based genome annotation with genemark et and augustus': {'braker'},\n",
       " 'runs breseq software on a set of fastq files': {'breseq'},\n",
       " 'provides measures for quantitative assessment of genome assembly gene set and transcriptome completeness based on evolutionarily informed expectations of gene content from near universal single copy orthologs': {'busco'},\n",
       " 'accelerated nanopore basecalling with slow5 data format': {'buttery-eel'},\n",
       " 'fast accurate memory efficient aligner for short and long sequencing reads': {'bwa'},\n",
       " 'bwa mem2 is the next version of the bwa mem algorithm in bwa it produces alignment identical to bwa and is 1 3 3 1x faster depending on the use case dataset and the running machine': {'bwa-mem2'},\n",
       " 'tools for manipulating biological data particularly multiple sequence alignments': {'bx-python'},\n",
       " 'cactus is a reference free whole genome multiple alignment program': {'cactus'},\n",
       " 'annotation of peaklists generated by xcms rule based annotation of isotopes and adducts isotope validation eic correlation based tagging of unknown adducts and fragments': {'camera'},\n",
       " 'amber is an evaluation package for the comparative assessment of genome reconstructions and taxonomic assignments from metagenome benchmark datasets it provides performance metrics results rankings and comparative visualizations for assessing multiple programs or parameter effects the provided metrics were used in the first community benchmarking challenge of the initiative for the critical assessment of metagenomic interpretation': {'cami-amber'},\n",
       " 'de novo assembly tool for long read chemistry like nanopore data and pacbio data': {'canu'},\n",
       " 'web based contig assembly': {'cap3'},\n",
       " 'implements statistical and computational tools for analyzing mass spectrometry imaging datasets including methods for efficient pre processing spatial segmentation and classification': {'cardinal'},\n",
       " 'contig annotation tool cat and bin annotation tool bat are pipelines for the taxonomic classification of long dna sequences and metagenome assembled genomes mags bins of both known and highly unknown microorganisms as generated by contemporary metagenomics studies the core algorithm of both programs involves gene calling mapping of predicted orfs against the nr protein database and voting based classification of the entire contig mag based on classification of the individual orfs': {'cat_bins'},\n",
       " 'cluster a nucleotide dataset into representative sequences': {'cd-hit'},\n",
       " 'celseq2 is a python framework for generating the umi count matrix from cel seq2 sequencing data': {'celseq2'},\n",
       " 'a deep generative model for unsupervised removal of background noise from scrna seq datasets cellbender is a software package for eliminating technical artifacts from high throughput single cell rna sequencing scrna seq data welcome to cellbender s documentation cellbender documentation free document hosting provided by read the docs stephen j fleming john c marioni and mehrtash babadi cellbender remove background a deep generative model for unsupervised removal of background noise from scrna seq datasets biorxiv 791699 doi https doi org 10 1101 791699': {'cellbender'},\n",
       " 'cellpose is a generalist algorithm for cellular segmentation': {'cellpose'},\n",
       " 'tool for quantifying data from biological images particularly in high throughput experiments': {'CellProfiler'},\n",
       " 'cellsnp lite is an efficient tool for genotyping single cells cellsnp lite was initially designed to pileup the expressed alleles in single cell or bulk rna seq data which can be directly used for donor deconvolution in multiplexed single cell rna seq data particularly with vireo which assigns cells to donors and detects doublets even without genotyping reference now besides rna seq data cellsnp lite could also be applied on dna seq and atac seq data either in bulk or single cell': {'cellsnp-lite'},\n",
       " 'cellxgene pronounced cell by gene is an interactive data explorer for single cell transcriptomics datasets such as those coming from the human cell atlas': {'cellxgene'},\n",
       " 'it unifies the discovery and the analysis of coexpression gene modules in a fully automatic manner while providing a user friendly html report with high quality graphs our tool evaluates if modules contain genes that are over represented by specific pathways or that are altered in a specific sample group additionally cemitool is able to integrate transcriptomic data with interactome information identifying the potential hubs on each network': {'cemitool'},\n",
       " 'checkm provides a set of tools for assessing the quality of genomes recovered from isolates single cells or metagenomes': {'checkm',\n",
       "  'checkm-database'},\n",
       " 'database of bioactive compounds their quantitative properties and bioactivities binding constants pharmacology and admet etc the data is abstracted and curated from the primary scientific literature': {'chembl'},\n",
       " 'fast cheminformatics fingerprint search at your fingertips chemfp is a set of command line tools and a python library for fingerprint generation and high performance similarity search there are two ways to try out chemfp from the download page page you can request an evaluation copy of the most recent version of chemfp or you can download an earlier version for no cost under the mit license': {'chemfp'},\n",
       " 'chemicaltoolbox is a publicly available web server for performing cheminformatics analysis the chemicaltoolbox provides an intuitive graphical interface for common tools for downloading filtering visualizing and simulating small molecules and proteins the chemicaltoolbox is based on galaxy an open source web based platform which enables accessible and reproducible data analysis there is already an active galaxy cheminformatics community using and developing tools based on their work we provide four example workflows which illustrate the capabilities of the chemicaltoolbox covering assembly of a compound library hole filling protein ligand docking and construction of a quantitative structure activity relationship qsar model': {'chemicaltoolbox'},\n",
       " 'this package implements functions to retrieve the nearest genes around the peak annotate the genomic region of the peak statistical methods for estimating the significance of overlap among chip peak data sets and incorporate geo database to compare the own dataset with those deposited in database the comparison can be used to infer cooperative regulation and thus can be used to generate hypotheses': {'chipseeker'},\n",
       " 'chira is a tool suite to analyze rna rna interactome experimental data such as clash clear clip paris splash etc': {'chira'},\n",
       " 'an ultra fast heuristic approach to detect conserved signals in extremely large pairwise genome comparisons dotplot': {'chromeister'},\n",
       " 'circos is tool for visualizing data in a circular format it was developed for genomic data but can work for many other kinds of data as well': {'circos'},\n",
       " 'tool for counting antibody tags from a cite seq and or cell hashing experiment': {'CITE-seq-Count'},\n",
       " 'clair3 is a germline small variant caller for long reads clair3 makes the best of two major method categories pileup calling handles most variant candidates with speed and full alignment tackles complicated candidates to maximize precision and recall clair3 runs fast and has superior performance especially at lower coverage clair3 is simple and modular for easy deployment and integration': {'clair3'},\n",
       " 'automatic generation of gene cluster comparison figures gene cluster comparison figure generator a d3 chart for generating gene cluster comparison figures clinker is a pipeline for easily generating publication quality gene cluster comparison figures given a set of genbank files clinker will automatically extract protein translations perform global alignments between sequences in each cluster determine the optimal display order based on cluster similarity and generate an interactive visualisation using clustermap js that can be extensively tweaked before being exported as an svg file clustermap js is an interactive reusable d3 chart designed to visualise homology between multiple gene clusters': {'clinker'},\n",
       " 'multiple sequence alignment software the name is occassionally spelled as clustalomega clustal clustal clustal o clustalo': {'clustalo'},\n",
       " 'multiple sequence alignment software old deprecated versions even older versions were clustal and clustal v clustalv': {'clustalw'},\n",
       " 'cnvkit is a software toolkit to infer and visualize copy number from targeted dna sequencing data': {'cnvkit'},\n",
       " 'constraints based reconstruction and analysis for python': {'cobrapy'},\n",
       " 'colabfold databases are mmseqs2 expandable profile databases to generate diverse multiple sequence alignments to predict protein structures': {'colabfold',\n",
       "  'colabfold_batch'},\n",
       " 'compleasm a faster and more accurate reimplementation of busco it provides measures for quantitative assessment of genome assembly gene set and transcriptome completeness based on evolutionarily informed expectations of gene content from near universal single copy orthologs': {'compleasm'},\n",
       " 'a program for unsupervised binning of metagenomic contigs by using nucleotide composition coverage data in multiple samples and linkage data from paired end reads': {'concoct'},\n",
       " 'constava calculates conformational states probability and conformational state variability from protein structure ensembles': {'constava'},\n",
       " 'read coverage calculator for metagenomics': {'coverm'},\n",
       " 'cpat coding potential assessment tool is a logistic regression model based classifier that can accurately and quickly distinguish protein coding and noncoding rnas using pure linguistic features calculated from the rna sequences cpat takes as input the nucleotides sequences or genomic coordinates of rnas and outputs the probabilities p 0 p 1 which measure the likelihood of protein coding': {'cpat'},\n",
       " 'software to generate crispr guide rnas against genomes annotated with individual variation': {'crisflash'},\n",
       " 'cryodrgn is a neural network based algorithm for heterogeneous cryo em reconstruction': {'cryodrgn'},\n",
       " 'cufflinks assembles transcripts and estimates their abundances in rna seq samples it accepts aligned rna seq reads and assembles the alignments into a parsimonious set of transcripts cufflinks then estimates the relative abundances of these transcripts based on how many reads support each one': {'cufflinks'},\n",
       " 'allows for persistent storage access exploration and manipulation of cufflinks high throughput sequencing data in addition provides numerous plotting functions for commonly used visualizations': {'cummeRbund'},\n",
       " 'generate customized protein sequence database from rna seq data for proteomics search': {'customprodb'},\n",
       " 'find and remove adapter sequences primers poly a tails and other types of unwanted sequence from your high throughput sequencing reads': {'cutadapt'},\n",
       " 'long read based human genomic structural variation detection with cutesv long read sequencing technologies enable to comprehensively discover structural variations svs however it is still non trivial for state of the art approaches to detect svs with high sensitivity or high performance or both herein we propose cutesv a sensitive fast and lightweight sv detection approach cutesv uses tailored methods to comprehensively collect various types of sv signatures and a clustering and refinement method to implement a stepwise sv detection which enables to achieve high sensitivity without loss of accuracy benchmark results demonstrate that cutesv has better yields on real datasets further its speed and scalability are outstanding and promising to large scale data analysis': {'cutesv'},\n",
       " 'this package infers exact sequence variants svs from amplicon data replacing the commonly used and coarser otu clustering approach this pipeline inputs demultiplexed fastq files and outputs the sequence variants and their sample wise abundances after removing substitution and chimera errors taxonomic classification is available via a native implementation of the rdp naive bayesian classifier': {'dada2'},\n",
       " 'das tool is an automated method that integrates the results of a flexible number of binning algorithms to calculate an optimized non redundant set of bins from a single assembly': {'dastool'},\n",
       " 'ensemble of computational methods to infer biological activities from omics data': {'decoupler'},\n",
       " 'deeplabcut is a toolbox for state of the art markerless pose estimation of animals performing various behaviors': {'deeplabcut'},\n",
       " 'user friendly tools for the normalization and visualization of deep sequencing data': {'deeptools'},\n",
       " 'dendropy is a python library for phylogenetic computing it provides classes and functions for the simulation processing and manipulation of phylogenetic trees and character matrices and supports the reading and writing of phylogenetic data in a range of formats': {'dendropy'},\n",
       " 'r bioconductor package for differential gene expression analysis based on the negative binomial distribution estimate variance mean dependence in count data from high throughput sequencing assays and test for differential expression based on a model using the negative binomial distribution': {'deseq2'},\n",
       " 'the package is focused on finding differential exon usage using rna seq exon counts between samples with different experimental designs it provides functions that allows the user to make the necessary statistical tests based on a model that uses the negative binomial distribution to estimate the variance between biological replicates and generalized linear models for testing the package also provides functions for the visualization and exploration of the results': {'DEXSeq'},\n",
       " 'the dfam database is a open collection of transposable element dna sequence alignments hidden markov models hmms consensus sequences and genome annotations': {'dfam'},\n",
       " 'neural networks and interference correction enable deep proteome coverage in high throughput dia nn a fast and easy to use tool for processing data independent acquisition dia proteomics data none required for raw mzml and dia processing two executables are provided diann exe a command line tool and dia nn exe a gui implemented as a wrapper for diann exe': {'diann'},\n",
       " 'dia umpire is an open source java program for computational analysis of data independent acquisition dia mass spectrometry based proteomics data it enables untargeted peptide and protein identification and quantitation using dia data and also incorporates targeted extraction to reduce the number of cases of missing quantitation': {'diaumpire'},\n",
       " 'sequence aligner for protein and translated dna searches and functions as a drop in replacement for the ncbi blast software tools it is suitable for protein protein search as well as dna protein search on short reads and longer sequences including contigs and assemblies providing a speedup of blast ranging up to x20 000': {'diamond'},\n",
       " 'diapasef is an appproch for parallel accumulation serial fragmentation combined with data independent acquisition': {'diapysef'},\n",
       " 'compute differentially bound sites from multiple chip seq experiments using affinity quantitative data also enables occupancy overlap analysis and plotting functions': {'diffbind'},\n",
       " 'dorado is a high performance easy to use open source basecaller for oxford nanopore reads': {'dorado'},\n",
       " 'distilled and refined annotation of metabolism a tool for the annotation and curation of function for microbial and viral genomes': {'dram'},\n",
       " 'fast and accurate genomic comparisons that enables improved genome recovery from metagenomes through de replication': {'drep'},\n",
       " 'provides a number of utility functions for handling single cell rna seq data from droplet technologies such as 10x genomics this includes data loading identification of cells from empty droplets removal of barcode swapped pseudo cells and downsampling of the count matrix': {'dropletutils'},\n",
       " 'fast and accurate genome wide phasing and imputation in a single tool': {'eagleimp'},\n",
       " 'easybuild is a software build and installation framework that allows you to manage scientific software on high performance computing hpc systems in an efficient way': {'easybuild'},\n",
       " 'ectyper is a standalone serotyping module for escherichia coli it supports fasta and fastq file formats galaxy version 1 0 0': {'ectyper'},\n",
       " 'differential expression analysis of rna seq expression profiles with biological replication implements a range of statistical methodology based on the negative binomial distributions including empirical bayes estimation exact tests generalized linear models and quasi likelihood tests as well as rna seq it be applied to differential signal analysis of other types of genomic data that produce counts including chip seq sage and cage': {'edger',\n",
       "  'edger-repenrich'},\n",
       " 'entrez direct edirect is a command line tool for entrez databases edirect connects to entrez through the entrez programming utilities interface it supports searching by indexed terms looking up precomputed neighbors or links filtering results by date or category and downloading record summaries or reports': {'edirect'},\n",
       " 'the edta package was designed to filter out false discoveries in raw te candidates and generate a high quality non redundant te library for whole genome te annotations selection of initial search programs were based on benckmarkings on the annotation performance using a manually curated te library in the rice genome': {'edta'},\n",
       " 'for fast functional annotation of novel sequences it uses precomputed orthologous groups and phylogenies from the eggnog database to transfer functional information from fine grained orthologs only its common uses include the annotation of novel genomes transcriptomes or even metagenomic gene catalogs the use of orthology predictions for functional annotation permits a higher precision than traditional homology searches as it avoids transferring annotations from close paralogs': {'eggnog-mapper'},\n",
       " 'this package implements the ensemble of gene set enrichment analyses method for gene set testing': {'egsea'},\n",
       " 'evaluation of an open source registration package for automatic contour propagation in online adaptive intensity modulated proton therapy of prostate cancer home about faq wiki download news legal stuff documentation welcome to elastix a toolbox for rigid and nonrigid registration of images elastix is open source software based on the well known insight segmentation and registration toolkit itk the software consists of a collection of algorithms that are commonly used to solve medical image registration problems the modular design of elastix allows the user to quickly configure test and compare different registration methods for a specific application a command line interface enables automated processing of large numbers of data sets by means of scripting nowadays elastix is accompanied by simpleelastix making it available in languages like c python java r ruby c and lua': {'elastix'},\n",
       " 'diverse suite of tools for sequence analysis many programs analagous to gcg context sensitive help for each tool': {'emboss'},\n",
       " 'emmtyper is a command line tool for emm typing of streptococcus pyogenes using a de novo or complete assembly by default we use the u s centers for disease control and prevention trimmed emm subtype database which can be found here https www2a cdc gov ncidod biotech strepblast asp the database is curated by dr velusamy srinivasan inner workings the difficulty in performing m typing is that there is a single gene of interest emm but two other homologue genes enn and mrp often referred to as emm like the homologue genes may or may not occur in the isolate of interest when performing emm typing from an assembly we can distinguish betweeen one or more clusters of matches on the contigs the best match for each of the clusters identified is then parsed from the blast results where possible we try to distinguish between matches to the emm gene and matches to one of the emm like genes': {'emmtyper'},\n",
       " 'a globally comprehensive data resource for nucleotide sequence spanning raw data alignments and assemblies functional and taxonomic annotation and rich contextual data relating to sequenced samples and experimental design serving both as the database of record for the output of the world s sequencing activity and as a platform for the management sharing and publication of sequence data': {'ena_upload'},\n",
       " 'encyclopedia is library search engine comprised of several algorithms for dia data analysis and can search for peptides using either dda based spectrum libraries or dia based chromatogram libraries': {'encyclopedia'},\n",
       " 'epigenomics single cell analysis in python': {'epiScanpy'},\n",
       " 'the environment for tree exploration ete is a computational framework that simplifies the reconstruction analysis and visualization of phylogenetic trees and multiple sequence alignments here we present ete v3 featuring numerous improvements in the underlying library of methods and providing a novel set of standalone tools to perform common tasks in comparative genomics and phylogenetics the new features include i building gene based and supermatrix based phylogenies using a single command ii testing and visualizing evolutionary models iii calculating distances between trees of different size or including duplications and iv providing seamless integration with the ncbi taxonomy database ete is freely available at http etetoolkit org': {'ete3'},\n",
       " 'integrated database covering the eukaryotic pathogens of the genera cryptosporidium giardia leishmania neospora plasmodium toxoplasma trichomonas and trypanosoma the database portal offers an entry point to all these resources and the opportunity to leverage orthology for searches across genera': {'eupathdb'},\n",
       " 'exabayes is a software package for bayesian phylogenetic tree inference it is particularly suitable for large scale analyses on computer clusters': {'exabayes'},\n",
       " 'tool for phylogenomic analyses on supercomputers': {'examl'},\n",
       " 'a tool for pairwise sequence alignment it enables alignment for dna dna and dna protein pairs and also gapped and ungapped alignment': {'exonerate'},\n",
       " 'export2graphlan is a conversion software tool for producing both annotation and tree file for graphlan in particular the annotation file tries to highlight specific sub trees deriving automatically from input file what nodes are important': {'export2graphlan'},\n",
       " 'streaming tool for quantifying the abundances of a set of target sequences from sampled subsequences example applications include transcript level rna seq quantification allele specific haplotype expression analysis from rna seq transcription factor binding quantification in chip seq and analysis of metagenomic data it can be used to resolve ambiguous mappings in other high throughput sequencing based applications': {'eXpress'},\n",
       " 'gpu accelerated adaptive banded event alignment for rapid comparative nanopore signal analysis re engineered and optimised nanopolish call methylation module supports cuda acceleration an optimised re implementation of the call methylation module in nanopolish given a set of basecalled nanopore reads and the raw signals f5c detects the methylated cytosine bases f5c can optionally utilise nvidia graphics cards for acceleration': {'f5c'},\n",
       " 'a high speed fastqc emulation for quality control of sequencing data': {'falco'},\n",
       " 'experimental pacbio diploid assembler': {'pb-assembly'},\n",
       " 'add length of sequence to fasta header': {'fasta_compute_length'},\n",
       " 'fastani is developed for fast alignment free computation of whole genome average nucleotide identity ani ani is defined as mean nucleotide identity of orthologous gene pairs shared between two microbial genomes fastani supports pairwise comparison of both complete and draft genome assemblies': {'fastani'},\n",
       " 'read huge fastq and fasta files both normal and gzipped an demanipulate them': {'fastool'},\n",
       " 'a tool designed to provide fast all in one preprocessing for fastq files this tool is developed in c with multithreading supported to afford high performance': {'fastp'},\n",
       " 'this tool aims to provide a qc report which can spot problems or biases which originate either in the sequencer or in the starting library material it can be run in one of two modes it can either run as a stand alone interactive application for the immediate analysis of small numbers of fastq files or it can be run in a non interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files': {'fastqc'},\n",
       " 'compute quality stats for fastq files and print those stats as emoji for some reason': {'fastqe'},\n",
       " 'infers approximately maximum likelihood phylogenetic trees from alignments of nucleotide or protein sequences': {'fasttree'},\n",
       " 'collection of command line tools for short reads fasta fastq files preprocessing': {'fastx',\n",
       "  'fastx_toolkit'},\n",
       " 'a tool to annotate long non coding rnas from rna seq assembled transcripts': {'feelnc'},\n",
       " 'featurecounts is a very efficient read quantifier it can be used to summarize rna seq reads and gdna seq reads to a variety of genomic features such as genes exons promoters gene bodies and genomic bins it is included in the bioconductor rsubread package and also in the sourceforge subread package': {'featurecounts'},\n",
       " 'hmm based gene structure prediction multiple genes both chains program for predicting multiple genes in genomic dna sequences': {'fgenesh'},\n",
       " 'the package implements an algorithm for fast gene set enrichment analysis using the fast algorithm allows to make more permutations and get more fine grained p values which allows to use accurate stantard approaches to multiple hypothesis correction': {'fgsea'},\n",
       " 'filtlong is a tool for filtering long reads by quality it can take a set of long reads and produce a smaller better subset it uses both read length longer is better and read identity higher is better when choosing which reads pass the filter': {'filtlong'},\n",
       " 'flashlfq is an ultrafast label free quantification algorithm for mass spectrometry proteomics': {'flashlfq'},\n",
       " 'flye is a de novo assembler for single molecule sequencing reads such as those produced by pacbio and oxford nanopore technologies it is designed for a wide range of datasets from small bacterial projects to large mammalian scale assemblies the package represents a complete pipeline it takes raw pb ont reads as input and outputs polished contigs': {'flye'},\n",
       " 'an integrated database for drosophila and anopheles genomics': {'flymine'},\n",
       " 'foldseek enables fast and sensitive comparisons of large structure sets it reaches sensitivities similar to state of the art structural aligners while being at least 20 000 times faster': {'foldseek'},\n",
       " 'web server which detects small molecule pockets by relying on the geometric alpha sphere theory it also tracks pockets during molecular dynamics so to provide insight on pocket dynamics mdpocket and transposes mdpocket to the combined analysis of homologous structures hpocket': {'fpocket'},\n",
       " 'application for finding fragmented genes in short reads': {'fraggenescan'},\n",
       " 'bayesian genetic variant detector designed to find small polymorphisms specifically snps indels multi nucleotide polymorphisms and complex events composite insertion and substitution events smaller than the length of a short read sequencing alignment': {'freebayes'},\n",
       " 'a tool for control free copy number alteration cna and allelic imbalances loh detection using deep sequencing data particularly useful for cancer studies': {'freec'},\n",
       " 'wrapper functions that interface with freesurfer a powerful and commonly used neuroimaging software using system commands the goal is to be able to interface with freesurfer completely in r where you pass r objects of class nifti implemented by package oro nifti and the function executes an freesurfer command and returns an r object of class nifti or necessary output': {'freesurfer'},\n",
       " 'funannotate is a pipeline for genome annotation built specifically for fungi but will also work with higher eukaryotes': {'funannotate'},\n",
       " 'gene annotation evaluation': {'gaeval'},\n",
       " 'galaxy image analysis is mainly developed by the biomedical computer vision bmcv group at heidelberg university its dedication is to provide tools for image analysis and image processing for the galaxy platform': {'galaxy_image_analysis'},\n",
       " 'the genome analysis toolkit gatk is a set of bioinformatic tools for analyzing high throughput sequencing hts and variant call format vcf data the toolkit is well established for germline short variant discovery from whole genome and exome sequencing data gatk4 expands functionality into copy number and somatic analyses and offers pipeline scripts for workflows version 4 gatk4 is open source at https github com broadinstitute gatk': {'gatk'},\n",
       " 'cleaning aligned sequences': {'gblock'},\n",
       " 'genome wide complex trait analysis estimate the proportion of phenotypic variance explained by genome or chromosome wide snps for complex traits the greml method and has subsequently extended for many other analyses to better understand the genetic architecture of complex traits': {'gcta'},\n",
       " 'software aimed at pairwise sequence comparison generating high quality results equivalent to mummer with controlled memory consumption and comparable or faster execution times particularly with long sequences': {'gecko'},\n",
       " 'gemini genome mining is a flexible framework for exploring genetic variation in the context of the wealth of genome annotations available for the human genome by placing genetic variants sample phenotypes and genotypes as well as genome annotations into an integrated database framework gemini provides a simple flexible and powerful system for exploring genetic variation for disease and population genetics': {'gemini'},\n",
       " 'gene model mapper is a homology based gene prediction program gemoma uses the annotation of protein coding genes in a reference genome to infer the annotation of protein coding genes in a target genome thereby it utilizes amino acid sequence and intron position conservation in addition it allows to incorporate rna seq evidence for splice site prediction': {'gemoma'},\n",
       " 'an interactive web tool for versatile clinically driven variant interrogation and prioritization iobio is a suite of web apps for visually driven real time analysis of genomic data visually driven real time analysis of genomic data': {'geneiobio'},\n",
       " 'reference free profiling of polyploid genomes we have developed genomescope 2 0 which applies classical insights from combinatorial theory to establish a detailed mathematical model of how k mer frequencies will be distributed in heterozygous and polyploid genomes average k mer coverage for polyploid genome upload results from running jellyfish or kmc': {'genomescope'},\n",
       " 'free collection of bioinformatics tools for genome informatics': {'genometools'},\n",
       " 'a fast and versatile toolkit for accurate de novo assembly of organelle genomes this toolkit assemblies organelle genome from genomic skimming data': {'getorganelle'},\n",
       " 'gfastats is a single fast and exhaustive tool for summary statistics and simultaneous genome assembly file manipulation gfastats also allows seamless fasta fastq gfa conversion': {'gfastats'},\n",
       " 'program for comparing annotating merging and tracking transcripts in gff files': {'gffcompare'},\n",
       " 'program for filtering converting and manipulating gff files': {'gffread'},\n",
       " 'plotting system for r based on the grammar of graphics': {'ggplot2'},\n",
       " 'command line tool for the visualization of splicing events across multiple samples': {'ggsashimi'},\n",
       " 'genomic mapping and alignment program for mrna and est sequences': {'gmap',\n",
       "  'gmap-gsnap'},\n",
       " 'gnu parallel is a shell tool for executing jobs in parallel using one or more computers a job can be a single command or a small script that has to be run for each of the lines in the input the typical input is a list of files a list of hosts a list of users a list of urls or a list of tables a job can also be a command that reads from a pipe gnu parallel can then split the input and pipe it into commands in parallel': {'parallel'},\n",
       " 'the gnu scientific library gsl is a numerical library for c and c programmers': {'gsl'},\n",
       " 'goenrichment is a tool for performing go enrichment analysis of gene sets such as those obtained from rna seq or microarray experiments to help characterize them at the functional level it is available in galaxy europe and as a stand alone tool goenrichment is flexible in that it allows the user to use any version of the gene ontology and any go annotation file they desire to enable the use of go slims it is accompanied by a sister tool goslimmer which can convert annotation files from full go to any specified go slim the tool features an optional graph clustering algorithm to reduce the redundancy in the set of enriched go terms and simplify its output it was developed by the biodata pt elixir pt team at the instituto gulbenkian de ci ncia': {'goenrichment'},\n",
       " 'detect gene ontology and or other user defined categories which are over under represented in rna seq data': {'goseq'},\n",
       " 'graphlan is a software tool for producing high quality circular representations of taxonomic and phylogenetic trees graphlan focuses on concise integrative informative and publication ready representations of phylogenetically and taxonomically driven investigation': {'graphlan'},\n",
       " 'versatile package to perform molecular dynamics i e simulate the newtonian equations of motion for systems with hundreds to millions of particles it is primarily designed for biochemical molecules like proteins lipids and nucleic acids that have a lot of complicated bonded interactions but since it is extremely fast at calculating the nonbonded interactions that usually dominate simulations many groups are also using it for research on non biological systems e g polymers': {'gromacs'},\n",
       " 'tool with multiple functions main functions are to create a diamond database from the gtdb taxonomy data or create a ncbi taxdump format out of this data this tool can also create a mapping between the taxonomy classification between gtdb and ncbi': {'gtdb_to_taxdump'},\n",
       " 'a toolkit to classify genomes with the genome taxonomy database gtdb tk a toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes gtdb tk is a software toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes based on the genome database taxonomy gtdb it is designed to work with recent advances that allow hundreds or thousands of metagenome assembled genomes mags to be obtained directly from environmental samples it can also be applied to isolate and single cell genomes the gtdb tk is open source and released under the gnu general public license version 3': {'gtdb-tk'},\n",
       " 'gubbins is a tool for rapid phylogenetic analysis of large samples of recombinant bacterial whole genome sequences': {'gubbins'},\n",
       " 'hapcut2 is a maximum likelihood based tool for assembling haplotypes from dna sequence reads designed to just work with excellent speed and accuracy across a range of long and short read sequencing technologies the output is in haplotype block format described here https github com vibansal hapcut2 blob master outputformat md': {'hapcut2'},\n",
       " 'software for the detection of selection signatures based on multiple population genotyping data': {'hapflk'},\n",
       " 'hapo g is a tool that aims to improve the quality of genome assemblies by polishing the consensus with accurate reads it capable of incorporating phasing information from high quality reads short or long reads to polish genome assemblies and in particular assemblies of diploid and heterozygous genomes': {'hapog'},\n",
       " 'tool for single species active module discovery': {'heinz'},\n",
       " 'deep learning to predict gene annotations': {'helixer'},\n",
       " 'this tool provides functional annotation for a list of genes by connecting with david database': {'hgv_david'},\n",
       " 'this tool can be used to analyze the patterns of linkage disequilibrium ld between polymorphic sites in a locus': {'hgv_ldtools'},\n",
       " 'this tool creates a link to the g gost tool gene group functional profiling which provides functional profiling of gene lists': {'hgv_linkToGProfile'},\n",
       " 'a web server for reproducible hi c capture hi c and single cell hi c data analysis quality control and visualization hicexplorer hicexplorer 3 6 documentation schicexplorer schicexplorer 7 documentation free document hosting provided by read the docs': {'hicexplorer'},\n",
       " 'remove ccs reads with remnant pacbio adapter sequences and convert outputs to a compressed fastq fastq gz': {'hifiadapterfilt'},\n",
       " 'hifiasm a haplotype resolved assembler for accurate hifi reads': {'hifiasm'},\n",
       " 'hifiasm meta de novo metagenome assembler based on hifiasm a haplotype resolved de novo assembler for pacbio hifi reads': {'hifiasm_meta'},\n",
       " 'alignment program for mapping next generation sequencing reads both dna and rna to a population of human genomes as well as to a single reference genome': {'hisat2'},\n",
       " 'this tool is used for searching sequence databases for homologs of protein sequences and for making protein sequence alignments it implements methods using probabilistic models called profile hidden markov models the new hmmer3 project hmmer is now as fast as blast for protein search': {'hmmer'},\n",
       " 'python framework to process and analyse high throughput sequencing hts data': {'htseq'},\n",
       " 'the main purpose of htslib is to provide access to genomic information files both alignment data sam bam and cram formats and variant data vcf and bcf formats the library also provides interfaces to access and index genome reference data in fasta format and tab delimited files with genomic coordinates it is utilized and incorporated into both samtools and bcftools': {'htslib'},\n",
       " 'humann is a pipeline for efficiently and accurately profiling the presence absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data typically millions of short dna rna reads this process referred to as functional profiling aims to describe the metabolic potential of a microbial community and its members more generally functional profiling answers the question what are the microbes in my community of interest doing or are capable of doing': {'humann'},\n",
       " 'humann 2 0 is a pipeline for efficiently and accurately profiling the presence absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data typically millions of short dna rna reads this process referred to as functional profiling aims to describe the metabolic potential of a microbial community and its members more generally functional profiling answers the question what are the microbes in my community of interest doing or capable of doing': {'humann2'},\n",
       " 'paralogs and off target sequences improve phylogenetic resolution in a densely sampled study of the breadfruit genus artocarpus moraceae recovering genes from targeted sequence capture data current version 1 3 1 august 2018 read our article in applications in plant sciences open access hybpiper was designed for targeted sequence capture in which dna sequencing libraries are enriched for gene regions of interest especially for phylogenetics hybpiper is a suite of python scripts that wrap and connect bioinformatics tools in order to extract target sequences from high throughput dna sequencing reads': {'hybpiper'},\n",
       " 'software package for the analysis of genetic sequences using techniques in phylogenetics molecular evolution and machine learning': {'HyPhy'},\n",
       " 'tool for interactive bioimage classification segmentation and analysis': {'ilastik'},\n",
       " 'improved phased assembler ipa is the official pacbio software for hifi genome assembly ipa was designed to utilize the accuracy of pacbio hifi reads to produce high quality phased genome assemblies': {'pbipa'},\n",
       " 'infernal inference of rna alignment is for searching dna sequence databases for rna structure and sequence similarities it is an implementation of a special case of profile stochastic context free grammars called covariance models cms a cm is like a sequence profile but it scores a combination of sequence consensus and rna secondary structure consensus so in many cases it is more capable of identifying rna homologs that conserve their secondary structure more than their primary sequence': {'infernal'},\n",
       " 'a tool to detect integron in dna sequences': {'integron_finder'},\n",
       " 'open source data warehouse built specifically for the integration and analysis of complex biological data it enables the creation of biological databases accessed by sophisticated web query tools parsers are provided for integrating data from many common biological data sources and formats and there is a framework for adding your own data': {'intermine'},\n",
       " 'scan sequences against the interpro protein signature databases': {'interproscan'},\n",
       " 'interactive assembly and analysis of radseq datasets ipyrad interactive assembly and analysis of rad seq data sets welcome to ipyrad an interactive toolkit for assembly and analysis of restriction site associated genomic data sets e g rad ddrad gbs for population genetic and phylogenetic studies welcome to ipyrad ipyrad documentation': {'ipyrad'},\n",
       " 'very efficient phylogenetic software for reconstructing maximum likelihood trees and assessing branch supports with the ultrafast bootstrap approximation it is based on the iqpnni algorithm with 10 fold speedup together with substantially additional features': {'iq-tree'},\n",
       " 'a comprehensive suite to discover and explore intron retention': {'irfinder'},\n",
       " 'provides functions for creating an interactive shiny based graphical user interface for exploring data stored in summarizedexperiment objects including row and column level metadata particular attention is given to single cell data in a singlecellexperiment object with visualization of dimensionality reduction results': {'isee'},\n",
       " 'automated identification of insertion sequence elements in prokaryotic genomes': {'isescan'},\n",
       " 'enables identification of isoform switches with predicted functional consequences from rna seq data consequences can be chosen from a long list but includes protein domains gain loss changes in nmd sensitivity etc it directly supports import of data from cufflinks cuffdiff kallisto salmon and rsem but other transcript qunatification tools are easy to import as well': {'isoformswitchanalyzer'},\n",
       " 'isoseq v3 contains the newest tools to identify transcripts in pacbio single molecule sequencing data starting in smrt link v6 0 0 those tools power the isoseq gui based analysis application a composable workflow of existing tools and algorithms combined with a new clustering technique': {'isoseq3'},\n",
       " 'tsx is an open source software utility to extract the highly variable its1 and its2 subregions from its sequences which is commonly used as a molecular barcode for e g fungi as the inclusion of parts of the neighbouring very conserved ribosomal genes ssu 5s and lsu rrna sequences in the sequence identification process can lead to severely misleading results itsx identifies and extracts only the its regions themselves': {'itsx'},\n",
       " 'interpretation oriented tool to manage the update and revision of variant annotation and classification ivar database of genomics variants': {'ivar'},\n",
       " 'implementation of the interval wise testing iwt for omics data this inferential procedure tests for differences in omics data between two groups of genomic regions or between a group of genomic regions and a reference center of symmetry and does not require fixing location and scale at the outset': {'iwtomics'},\n",
       " 'jags is just another gibbs sampler it is a program for analysis of bayesian hierarchical models using markov chain monte carlo mcmc simulation': {'jags'},\n",
       " 'jasmine jointly accurate sv merging with intersample network edges is an automated pipeline for alignment and sv calling in long read datasets the tool is used to merge structural variants svs across samples each sample has a number of sv calls consisting of position information chromosome start end length type and strand information and a number of other values jasmine represents the set of all svs across samples as a network and uses a modified minimum spanning forest algorithm to determine the best way of merging the variants such that each merged variants represents a set of analogous variants occurring in different samples': {'jasminesv'},\n",
       " 'slick speedy genome browser with a responsive and dynamic ajax interface for visualization of genome data being developed by the gmod project as a successor to gbrowse': {'jbrowse'},\n",
       " 'a command line algorithm for counting k mers in dna sequence': {'jellyfish'},\n",
       " 'juicer is a platform for analyzing kilobase resolution hi c data in this distribution we include the pipeline for generating hi c maps from fastq raw data files and command line tools for feature annotation on the hi c maps': {'juicer'},\n",
       " 'a program for quantifying abundances of transcripts from rna seq data or more generally of target sequences using high throughput sequencing reads it is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets without the need for alignment': {'kallisto'},\n",
       " 'suite of tools that generate analyse and compare k mer spectra produced from sequence files': {'kat'},\n",
       " 'khmer is a set of command line tools for working with dna shotgun sequencing data from genomes transcriptomes metagenomes and single cells khmer can make de novo assemblies faster and sometimes better khmer can also identify and fix problems with shotgun data': {'khmer'},\n",
       " 'kmc is a utility designed for counting k mers sequences of consecutive k symbols in a set of reads from genome sequencing projects': {'kmc'},\n",
       " 'kofamscan is a gene function annotation tool based on kegg orthology and hidden markov model you need kofam database to use this tool': {'kofamscan'},\n",
       " 'system for assigning taxonomic labels to short dna sequences usually obtained through metagenomic studies previous attempts by other bioinformatics software to accomplish this task have often used sequence alignment or machine learning techniques that were quite slow leading to the development of less sensitive but much faster abundance estimation programs it aims to achieve high sensitivity and high speed by utilizing exact alignments of k mers and a novel classification algorithm': {'kraken'},\n",
       " 'kraken 2 is the newest version of kraken a taxonomic classification system using exact k mer matches to achieve high accuracy and fast classification speeds this classifier matches each k mer within a query sequence to the lowest common ancestor lca of all genomes containing the given k mer the k mer assignments inform the classification algorithm any assumption that kraken s raw read assignments can be directly translated into species or strain level abundance estimates is flawed bracken bayesian reestimation of abundance after classification with kraken estimates species abundances in metagenomics samples by probabilistically re distributing reads in the taxonomic tree lu jennifer et al bracken estimating species abundance in metagenomics data': {'kraken2'},\n",
       " 'krakentools provides individual scripts to analyze kraken kraken2 bracken krakenuniq output files': {'krakentools'},\n",
       " 'krona creates interactive html5 charts of hierarchical data such as taxonomic abundance in a metagenome': {'krona'},\n",
       " 'automated image analysis for developmental phenotyping of mouse embryos lama lightweight analysis of morphological abnormalities welcome to lama an open source pipeline to automatically identify embryo dysmorphology from 3d volumetric images': {'lama'},\n",
       " 'a tool for 1 aligning two dna sequences and 2 inferring appropriate scoring parameters automatically': {'lastz'},\n",
       " 'an easy to use interactive web platform to analyze and visualize label free proteomics data preprocessed with maxquant a tool for analysing label free quantitative proteomics dataset https bioinformatics erc monash edu apps lfq analyst lfq analyst an easy to use interactive web platform to analyze and visualize proteomics data preprocessed with maxquant lfq analyst is an easy to use interactive web application developed to perform differential expression analysis with one click and to visualize label free quantitative proteomic datasets preprocessed with maxquant lfq analyst provides a wealth of user analytic features and offers numerous publication quality result output graphics and tables to facilitate statistical and exploratory analysis of label free quantitative datasets': {'LFQ-Analyst'},\n",
       " 'data analysis linear models and differential expression for microarray data': {'limma'},\n",
       " 'links long interval nucleotide k mer scaffolder is a genomics application for scaffolding genome assemblies with long reads such as those produced by oxford nanopore technologies ltd it can be used to scaffold high quality draft genome assemblies with any long sequences eg ont reads pacbio reads other draft genomes etc it is also used to scaffold contig pairs linked by arcs arks': {'links'},\n",
       " 'lofreq i e lofreq version 2 is a fast and sensitive variant caller for inferring snvs and indels from next generation sequencing data it makes full use of base call qualities and other sources of errors inherent in sequencing e g mapping or base indel alignment uncertainty which are usually ignored by other methods or only used for filtering': {'lofreq'},\n",
       " 'longshot is a variant calling tool for diploid genomes using long error prone reads such as pacific biosciences pacbio smrt and oxford nanopore technologies ont it takes as input an aligned bam file and outputs a phased vcf file with variants and haplotype information it can also genotype and phase input vcf files it can output haplotype separated bam files that can be used for downstream analysis currently it only calls single nucleotide variants snvs but it can genotype indels if they are given in an input vcf': {'longshot'},\n",
       " 'lotus2 is a lightweight and user friendly pipeline that is fast precise and streamlined using extensive pre and post asv otu clustering steps to further increase data quality high data usage rates and reliability enable high throughput microbiome analysis in minutes': {'lotus2'},\n",
       " 'ltr retriever is a highly accurate and sensitive program for identification of ltr retrotransposons the ltr assembly index lai is also included in this package': {'ltr_retriever'},\n",
       " 'model based analysis of chip seq data': {'macs2'},\n",
       " 'mafft multiple alignment using fast fourier transform is a high speed multiple sequence alignment program': {'mafft'},\n",
       " 'computational tool to identify important genes from the recent genome scale crispr cas9 knockout screens technology': {'MAGeCK'},\n",
       " 'portable and easily configurable genome annotation pipeline it s purpose is to allow smaller eukaryotic and prokaryotic genome projects to independently annotate their genomes and to create genome databases': {'maker'},\n",
       " 'maldiquant is a complete analysis pipeline for matrix assisted laser desorption ionization time of flight maldi tof and other two dimensional mass spectrometry data in addition to commonly used plotting and processing methods it includes distinctive features namely baseline subtraction methods such as morphological filters tophat or the statistics sensitive non linear iterative peak clipping algorithm snip peak alignment using warping functions handling of replicated measurements as well as allowing spectra with different resolutions': {'maldi_quant'},\n",
       " 'fast genome and metagenome distance estimation using minhash': {'mash'},\n",
       " 'whole genome assembly software it combines the efficiency of the de bruijn graph and overlap layout consensus olc approaches masurca can assemble data sets containing only short reads from illumina sequencing or a mixture of short reads and long reads sanger 454': {'masurca'},\n",
       " 'tool to import process clean and compare mass spectrometry data': {'matchms'},\n",
       " 'matplotlib is a comprehensive library for creating static animated and interactive visualizations in python matplotlib makes easy things easy and hard things possible': {'matplotlib'},\n",
       " 'software for binning assembled metagenomic sequences based on an expectation maximization algorithm': {'maxbin'},\n",
       " 'quantitative proteomics software package designed for analyzing large mass spectrometric data sets it is specifically aimed at high resolution ms data': {'maxquant'},\n",
       " 'mcl is a clustering algorithm widely used in bioinformatics and gaining traction in other fields': {'mcl'},\n",
       " 'mdanalysis is an object oriented python toolkit to analyze molecular dynamics trajectories generated by charmm gromacs namd lammps amber or dl poly it also reads other formats e g pdb files and xyz format trajectories see the supported coordinate formats for the full list it can write most of these formats too together with atom selections for use in gromacs charmm vmd and pymol': {'mdanalysis'},\n",
       " 'medaka is a tool to create consensus sequences and variant calls from nanopore sequencing data this task is performed using neural networks applied a pileup of individual sequencing reads against a draft assembly': {'medaka'},\n",
       " 'single node assembler for large and complex metagenomics ngs reads such as soil it makes use of succinct de bruijn graph to achieve low memory usage whereas its goal is not to make memory usage as low as possible': {'megahit'},\n",
       " 'metagenome analysis software megan metagenome analyzer is a new computer program that allows laptop analysis of large metagenomic datasets in a preprocessing step the set of dna reads or contigs is compared against databases of known sequences using blast or another comparison tool megan can then be used to compute and interactively explore the taxonomical content of the dataset employing the ncbi taxonomy to summarize and order the results': {'megan'},\n",
       " 'testing of metabolic models': {'memote'},\n",
       " 'reference free quality completeness and phasing assessment for genome assemblies evaluate genome assemblies with k mers and more often genome assembly projects have illumina whole genome sequencing reads available for the assembled individual merqury provides a set of tools for this purpose': {'merqury'},\n",
       " 'meryl is a tool for counting and working with sets of k mers that was originally developed for use in the celera assembler and has since been migrated and maintained as part of canu': {'meryl'},\n",
       " 'an adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies metabat2 clusters metagenomic contigs into different bins each of which should correspond to a putative genome metabat2 uses nucleotide composition information and source strain abundance measured by depth of coverage by aligning the reads to the contigs to perform binning': {'metabat'},\n",
       " 'a scalable high throughput metabolic and biogeochemical functional trait profiler based on microbial genomes a scalable high throughput metabolic and biogeochemical functional trait profiler this software enables the prediction of metabolic and biogeochemical functional trait profiles to any given genome datasets these genome datasets can either be metagenome assembled genomes mags single cell amplified genomes sags or pure culture sequenced genomes it can also calculate the genome coverage the information is parsed and diagrams for elemental biogeochemical cycling pathways currently nitrogen carbon sulfur and other are produced': {'metabolic'},\n",
       " 'galaxy workflow for differential abundance analysis of 16s metagenomic data you are over your disk quota tool execution is on hold until your disk usage drops below your allocated quota this history is empty you can load your own data or get data from an external source': {'MetaDEGalaxy'},\n",
       " 'metaeuk sensitive high throughput gene discovery and annotation for large scale eukaryotic metagenomics': {'metaeuk'},\n",
       " 'an open source pipeline for probabilistic peptide discovery in complex metaproteomic datasets': {'metanovo'},\n",
       " 'computational tool for profiling the composition of microbial communities from metagenomic shotgun sequencing data': {'metaphlan'},\n",
       " 'metaquantome software suite analyzes the state of a microbiome by leveraging complex taxonomic and functional hierarchies to summarize peptide level quantitative information metaquantome offers differential abundance analysis principal components analysis and clustered heat map visualizations as well as exploratory analysis for a single sample or experimental condition': {'metaQuantome'},\n",
       " 'genome assembler for metagenomics datasets': {'metaspades'},\n",
       " 'metawrap aims to be an easy to use metagenomic wrapper suite that accomplishes the core tasks of metagenomic analysis from start to finish read quality control assembly visualization taxonomic profiling extracting draft genomes binning and functional annotation': {'metawrap'},\n",
       " 'a mostly universal methylation extractor for bs seq experiments': {'MethylDackel'},\n",
       " 'estimates effective population sizes past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes and population divergences or admixture': {'MIGRATE'},\n",
       " 'a lightweight python3 pipeline whose purpose is to facilitate the identification of expressed loci from rna seq data and to select the best models in each locus': {'mikado'},\n",
       " 'short read assembler based on a de bruijn graph capable of assembling a human genome on a desktop computer in a day': {'minia'},\n",
       " 'miniasm is a very fast olc based de novo assembler for noisy long reads it takes all vs all read self mappings typically by minimap as input and outputs an assembly graph in the gfa format': {'miniasm'},\n",
       " 'pairwise aligner for genomic and spliced nucleotide sequences': {'minimap2'},\n",
       " 'miniprot aligns a protein sequence against a genome with affine gap penalty splicing and frameshift it is primarily intended for annotating protein coding genes in a new species using known genes from other species': {'miniprot'},\n",
       " 'mira 3 whole genome shotgun and est sequence assembler': {'mira'},\n",
       " 'mirdeep2 discovers active known or novel mirnas from deep sequencing data': {'mirdeep2'},\n",
       " 'find circularise and annotate mitogenome from pacbio assemblies': {'mitohifi'},\n",
       " 'de novo metazoan mitochondrial genome annotation': {'mitos2'},\n",
       " 'multi locus sequence typing from an assembled genome or from a set of reads': {'mlst'},\n",
       " 'mmseqs2 many against many sequence searching is a software suite to search and cluster huge protein and nucleotide sequence sets mmseqs2 is open source software implemented in c for linux macos and as beta version via cygwin windows the software is designed to run on multiple cores and servers and exhibits very good scalability mmseqs2 can run 10000 times faster than blast at 100 times its speed it achieves almost the same sensitivity it can perform profile searches with the same sensitivity as psi blast at over 400 times its speed mmseqs2 includes linclust the first clustering algorithm whose runtime scales linearly with linclust we clustered 1 6 billion metagenomic sequence fragments in 10 h on a single server to 50 sequence identity': {'mmseqs2'},\n",
       " 'universal whole sequence based plasmid typing and its utility to prediction of host range and epidemiological surveillance mob suite software tools for clustering reconstruction and typing of plasmids from draft assemblies plasmids are mobile genetic elements mges which allow for rapid evolution and adaption of bacteria to new niches through horizontal transmission of novel traits to different genetic backgrounds the mob suite is designed to be a modular set of tools for the typing and reconstruction of plasmid sequences from wgs assemblies': {'mob-suite'},\n",
       " 'a modest feature finder to extract features in ms1 data': {'moff'},\n",
       " 'a proteomics search algorithm specifically designed for high resolution tandem mass spectra': {'morpheus'},\n",
       " 'fast bam cram depth calculation for wgs exome or targeted sequencing': {'mosdepth'},\n",
       " 'open source platform independent community supported software for describing and comparing microbial communities': {'mothur'},\n",
       " 'data warehouse for accessing mouse data from mouse genome informatics mgi supports powerful query reporting and analysis capabilities the ability to save and combine results from different queries easy integration into larger workflows and a comprehensive web services layer': {'mousemine'},\n",
       " 'program for bayesian inference and model choice across a wide range of phylogenetic and evolutionary models it uses markov chain monte carlo mcmc methods to estimate the posterior distribution of model parameters': {'mrbayes'},\n",
       " 'a fast flexible and open software framework for medical image processing and visualisation mrtrix3 is an open source cross platform software package for medical image processing analysis and visualisation with a particular emphasis on the investigation of the brain using diffusion mri it is implemented using a fast modular and flexible general purpose code framework for image data access and manipulation enabling efficient development of new applications whilst retaining high computational performance and a consistent command line interface between applications in this article we provide a high level overview of the features of the mrtrix3 framework and general purpose image processing applications provided with the software': {'mrtrix'},\n",
       " 'msconvert is a command line utility for converting between various mass spectrometry data formats including from raw data from several commercial companies with vendor libraries windows only for windows users there is also a gui msconvertgui': {'msconvert'},\n",
       " 'this is a modified copy of ms finder with source code modifications to make the tool accessible in galaxy ms finder software for structure elucidation of unknown spectra with hydrogen rearrangement hr rules the program supports molecular formula prediction metabolie class prediction and structure elucidation for ei ms and ms ms spectra and the assembly is licensed under the cc by 4 0': {'msfinder'},\n",
       " 'tool for mass spectra metadata annotation': {'msmetaenhancer'},\n",
       " 'statistical tool for quantitative mass spectrometry based proteomics': {'msstats'},\n",
       " 'tools for detecting differentially abundant peptides and proteins in shotgun mass spectrometry based proteomic experiments with tandem mass tag tmt labeling': {'bioconductor-msstatstmt'},\n",
       " 'a gsea based pathway enrichment analysis for multi omics data multigsea a gsea based pathway enrichment analysis for multi omics data bmc bioinformatics 21 561 2020 combining gsea based pathway enrichment with multi omics data integration': {'multigsea'},\n",
       " 'multiqc aggregates results from multiple bioinformatics analyses across many samples into a single report it searches a given directory for analysis logs and compiles a html report it s a general use tool perfect for summarising the output from numerous bioinformatics tools': {'multiqc'},\n",
       " 'mummer is a modular system for the rapid whole genome alignment of finished or draft sequence basically it is a ultra fast alignment of large scale dna and protein sequences': {'mummer'},\n",
       " 'music is a suite of programs that evaluate the biophysical effects of amino acid mutations in proteins they request the experimental or modeled 3 dimensional protein structure as input and predict the impact of specific single site mutations requested by the user or of all possible single site mutations popmusic and hotmusic predict the changes in thermodynamic and thermal stability respectively upon mutation they are helpful for the rational design of modified proteins with controlled stability properties snpmusic predicts whether protein variants are deleterious or benign due to stability issues thus providing a molecular level interpretation of disease phenotype': {'music_compare'},\n",
       " 'music utilizes cell type specific gene expression from single cell rna sequencing rna seq data to characterize cell type compositions from bulk rna seq data in complex tissues by appropriate weighting of genes showing cross subject and cross cell consistency music enables the transfer of cell type specific gene expression information from one dataset to another': {'music_deconvolution'},\n",
       " 'convert proteomics data files into a sqlite database extract data from proteomics mzidentml and mass spec scan files mzml mgf etc and store in a sqlite database the intended purpose is to provide a galaxy dataset that can support an interactive galaxy visualization plugin': {'mztosqlite'},\n",
       " 'rna modifications detection by comparative nanopore direct rna sequencing rna modifications detection from nanopore drna seq data nanocompore identifies differences in ont nanopore sequencing raw signal corresponding to rna modifications by comparing 2 samples analyses performed for the nanocompore paper nanocompore compares 2 ont nanopore direct rna sequencing datasets from different experimental conditions expected to have a significant impact on rna modifications it is recommended to have at least 2 replicates per condition for example one can use a control condition with a significantly reduced number of modifications such as a cell line for which a modification writing enzyme was knocked down or knocked out alternatively on a smaller scale transcripts of interests could be synthesized in vitro': {'nanocompore'},\n",
       " 'nanoplot is a tool with various visualizations of sequencing data in bam cram fastq fasta or platform specific tsv summaries mainly intended for long read sequencing from oxford nanopore technologies and pacific biosciences': {'nanoplot'},\n",
       " 'a package for detecting cytosine methylations and genetic variations from nanopore minion sequencing data': {'nanopolish'},\n",
       " 'ncbi datasets is a new resource that lets you easily gather data from across ncbi databases find and download sequence annotation and metadata for genes and genomes using our command line tools or web interface': {'ncbi-datasets-cli'},\n",
       " 'the national center for biotechnology information ncbi provides analysis and retrieval resources for the data in genbank and other biological data made available through the ncbi web site': {'ncbi_acc_download'},\n",
       " 'the ncbi foreign contamination screen fcs is a tool suite for identifying and removing contaminant sequences in genome assemblies contaminants are defined as sequences in a dataset that do not originate from the biological source organism and can arise from a variety of environmental and laboratory sources fcs will help you remove contaminants from genomes before submission to genbank': {'ncbi_fcs'},\n",
       " 'netcdf network common data form is a set of software libraries and machine independent data formats that support the creation access and sharing of array oriented scientific data': {'netcdf'},\n",
       " 'the newick utilities are a set of command line tools for processing phylogenetic trees they can process arbitrarily large amounts of data and do not require user interaction which makes them suitable for automating phylogeny processing tasks': {'newick_utils'},\n",
       " 'nextclade is an open source project for viral genome alignment mutation calling clade assignment quality checks and phylogenetic placement': {'nextclade'},\n",
       " 'nextflow enables scalable and reproducible scientific workflows using software containers it allows the adaptation of pipelines written in the most common scripting languages': {'nextflow'},\n",
       " 'a fast and efficient genome polishing tool for long read assembly fast and accurately polish the genome generated by noisy long reads nextpolish is used to fix base errors snv indel in the genome generated by noisy long reads it can be used with short read data only or long read data only or a combination of both it contains two core modules and use a stepwise fashion to correct the error bases in reference genome to correct the raw third generation sequencing tgs long reads with approximately 15 10 sequencing errors please use nextdenovo': {'nextpolish',\n",
       "  'nextpolish2'},\n",
       " 'ngsutils is a suite of software tools for working with next generation sequencing datasets': {'ngsutils'},\n",
       " 'nearly infinite neighbor joining application': {'ninja'},\n",
       " 'nnu net is the first segmentation method that is designed to deal with the dataset diversity found in the domain it condenses and automates the keys decisions for designing a successful segmentation pipeline for any given dataset in 3d biomedical image segmentation dataset properties like imaging modality image sizes voxel spacings class ratios etc vary drastically in current research practice segmentation pipelines are designed manually and with one specific dataset in mind': {'nnu-net'},\n",
       " 'the fundamental package for scientific computing with python': {'numpy'},\n",
       " 'set of python programs developed to simplify the manipulation of sequence files they were mainly designed to help us for analyzing next generation sequencer outputs 454 or illumina in the context of dna metabarcoding': {'obitools'},\n",
       " 'omark is a software for proteome protein coding gene repertoire quality assessment it provides measures of proteome completeness characterizes the consistency of all protein coding genes with regard to their homologs and identifies the presence of contamination from other species omark relies on the oma orthology database from which it exploits orthology relationships and on the omamer software for fast placement of all proteins into gene families': {'omark'},\n",
       " 'open source library and a collection of tools and interfaces for the analysis of mass spectrometry data includes over 200 standalone topp tools that can be combined to a workflow with the integrated workflow editor toppas raw and intermediate mass spectrometry data can be visualised with the included viewer toppview': {'openms'},\n",
       " 'optitype is a novel hla genotyping algorithm based on integer linear programming capable of producing accurate 4 digit hla genotyping predictions from ngs data by simultaneously selecting all major and minor hla class i alleles': {'optitype'},\n",
       " 'orthofinder is a fast accurate and comprehensive platform for comparative genomics it finds orthogroups and orthologs infers rooted gene trees for all orthogroups and identifies all of the gene duplcation events in those gene trees it also infers a rooted species tree for the species being analysed and maps the gene duplication events from the gene trees to branches in the species tree orthofinder also provides comprehensive statistics for comparative genomic analyses': {'orthofinder'},\n",
       " 'producing polished prokaryotic pangenomes with the panaroo pipeline': {'panaroo'},\n",
       " 'pangolin is a deep learning based method for predicting splice site strengths for details see zeng and li genome biology 2022 it is available as a command line tool that can be run on a vcf or csv file containing variants of interest pangolin will predict changes in splice site strength due to each variant and return a file of the same format pangolin s models can also be used with custom sequences': {'pangolin'},\n",
       " 'parsnp is a command line tool for efficient microbial core genome alignment and snp detection parsnp was designed to work in tandem with gingr a flexible platform for visualizing genome alignments and phylogenetic trees': {'parsnp'},\n",
       " 'tool set for pathway based data integration and visualization that maps and renders a wide variety of biological data on relevant pathway graphs it downloads the pathway graph data parses the data file maps user data to the pathway and render pathway graph with the mapped data in addition it integrates with pathway and gene set enrichment analysis tools for large scale and fully automated analysis': {'pathview'},\n",
       " 'web application for exploring metagenomics classification results with a special focus on infectious disease diagnosis pinpointing pathogens in metagenomics classification results is often complicated by host and laboratory contaminants as well as many non pathogenic microbiota researchers can analyze display and transform results from the kraken and centrifuge classifiers using interactive tables heatmaps and flow diagrams': {'pavian'},\n",
       " 'multithread blat algorithm speeding up aligning sequences to genomes': {'pblat'},\n",
       " 'global repository for all bona fide protein structure data single worldwide repository of information about the 3d structures of large biological molecules including proteins and nucleic acids it also has restful interface to search and retrieve data from the protein data bank pdb': {'pdb'},\n",
       " 'productive visualization of high throughput sequencing data using the seqcode open portable platform': {'pe_histogram'},\n",
       " 'paired end read merger pear evaluates all possible paired end read overlaps without requiring the target fragment size as input in addition it implements a statistical test for minimizing false positive results': {'pear'},\n",
       " 'peptideshaker is a search engine independent platform for interpretation of proteomics identification results from multiple search engines currently supporting x tandem ms gf ms amanda omssa myrimatch comet tide mascot andromeda and mzidentml by combining the results from multiple search engines while re calculating ptm localization scores and redoing the protein inference peptideshaker attempts to give you the best possible understanding of your proteomics data': {'peptideshaker'},\n",
       " 'semi supervised learning for peptide identification from ms ms data': {'percolator'},\n",
       " 'this tool is used to search a fasta sequence against a library of pfam hmm': {'pfamscan'},\n",
       " 'pharokka is a rapid standardised annotation tool for bacteriophage genomes and metagenomes': {'pharokka'},\n",
       " 'provides a set of classes and tools to facilitate the import storage analysis and graphical display of microbiome census data': {'phyloseq'},\n",
       " 'phylogenetic estimation software using maximum likelihood': {'phyml'},\n",
       " 'a set of command line tools for manipulating high throughput sequencing hts data in formats such as sam bam cram and vcf available as a standalone program or within the gatk4 program': {'picard'},\n",
       " 'picrust phylogenetic investigation of communities by reconstruction of unobserved states is a bioinformatics software package designed to predict metagenome functional content from marker gene e g 16s rrna surveys and full genomes': {'picrust'},\n",
       " 'picrust2 phylogenetic investigation of communities by reconstruction of unobserved states is a software for predicting functional abundances based only on marker gene sequences': {'picrust2'},\n",
       " 'read alignment analysis to diagnose report and automatically improve de novo genome assemblies': {'pilon'},\n",
       " 'pipe t is a galaxy workflow for processing and analyzing mir expression profiles by rtqpcr it is a tool that offers several state of the art options for parsing filtering normalizing imputing and analyzing rt qpcr expression data integration of pipe t into galaxy allows experimentalists with strong bioinformatic background as well as those without any programming or development expertise to perform complex analysis in a simple to use transparent accessible reproducible and user friendly environment': {'pipe_t'},\n",
       " 'plasflow is a set of scripts used for prediction of plasmid sequences in metagenomic contigs': {'PlasFlow'},\n",
       " 'plasmidfinder is a tool for the identification and typing of plasmid replicons in whole genome sequencing wgs': {'plasmidfinder'},\n",
       " 'free open source whole genome association analysis toolset designed to perform a range of basic large scale analyses in a computationally efficient manner': {'plink'},\n",
       " 'polypolish is a tool for polishing genome assemblies with short reads unlike other tools in this category polypolish uses sam files where each read has been aligned to all possible locations not just a single best location this allows it to repair errors in repeat regions that other alignment based polishers cannot fix': {'polypolish'},\n",
       " 'porechop abi ab initio is an extension of porechop whose purpose is to process adapter sequences in ont reads': {'porechop_abi'},\n",
       " 'flexible toolkit for exploring datasets generated by nanopore sequencing devices from minion for the purposes of quality control and downstream analysis': {'poretools'},\n",
       " 'tools for performing taxonomic assignment based on phylogeny using pplacer and clst': {'pplacer'},\n",
       " 'integrated collection of platform independent python modules for processing raw reads from high throughput next generation sequencing of lymphocyte repertoires': {'presto'},\n",
       " 'pretext is an opengl powered pretext contact map viewer': {'pretextview'},\n",
       " 'the pipeline runs prodigal gene predictions on all genomes runs pan reciprocal blast and identifies ortholog sets for a set of orthologous genes if the positions of the prodigal selected starts coincide in a multiple sequence alignment they are accepted if they do not coincide a consistent start position is sought where a majority of the highest scoring prodigal selected sites coincide if such a position is found it is accepted and the predictions are changed for the outlying genes': {'prodigal'},\n",
       " 'flow injection analysis coupled to high resolution mass spectrometry is a promising approach for high throughput metabolomics fia hrms data however cannot be pre processed with current software tools which rely on liquid chromatography separation or handle low resolution data only here we present the package that implements a new methodology to pre process fia hrms raw data netcdf mzdata mzxml and mzml and generates the peak table': {'profia'},\n",
       " 'software tool to annotate bacterial archaeal and viral genomes quickly and produce standards compliant output files': {'prokka'},\n",
       " 'identifying and removing haplotypic duplication in primary genome assemblies haplotypic duplication identification tool scripts pd config py script to generate a configuration file used by run purge dups py purge haplotigs and overlaps in an assembly based on read depth given a primary assembly pri asm and an alternative assembly hap asm optional if you have one follow the steps shown below to build your own purge dups pipeline steps with same number can be run simultaneously among all the steps although step 4 is optional we highly recommend our users to do so because assemblers may produce overrepresented seqeuences in such a case the final step 4 can be applied to remove those seqeuences': {'purge_dups'},\n",
       " 'pycoqc computes metrics and generates interactive qc plots for oxford nanopore technologies sequencing data': {'pycoqc'},\n",
       " 'reproducible plots for multivariate genomic data sets standalone program and library to plot beautiful genome browser tracks pygenometracks aims to produce high quality genome browser tracks that are highly customizable currently it is possible to plot': {'pygenometracks'},\n",
       " 'a python module for reading and manipulating sam bam vcf bcf files': {'pysam'},\n",
       " 'pytorch is an optimized tensor library for deep learning using gpus and cpus': {'pytorch'},\n",
       " 'rna seq quantification of alternative polyadenylation': {'qapa'},\n",
       " 'this galaxy tool is based on the quantum chemical code qcxms for calculating mass spectra using born oppenheimer molecular dynamics this version supports only electron ionization and uses the gfn2 xtb and gfn1 xtb quantum chemistry methods for the simulations the mass spectrum is generated in three steps neutral run production run and result extraction': {'qcxms'},\n",
       " 'qiime 2 is an ai ready microbiome multi omics data science platform that is trusted free open source extensible and community developed and supported bioinformatics': {'qiime2',\n",
       "  'qiime2-amplicon',\n",
       "  'qiime2-shotgun'},\n",
       " 'platform independent application written in java and r that provides both a graphical user inteface gui and a command line interface to facilitate the quality control of alignment sequencing data': {'qualimap'},\n",
       " 'a collection of tools for viral quasispecies analysis abstract summary quasitools is a collection of newly developed open source tools for analyzing viral quasispcies data the application suite includes tools with the ability to create consensus sequences call nucleotide codon and amino acid variants calculate the complexity of a quasispecies and measure the genetic distance between two similar quasispecies these tools may be run independently or in user created workflows availability the quasitools suite is a freely available application licensed under the apache license version 2 0 the source code documentation and file specifications are available at https phac nml github io quasitools contact gary vandomselaar canada ca': {'quasitools'},\n",
       " 'quast stands for quality assessment tool it evaluates a quality of genome assemblies by computing various metrics and providing nice reports': {'quast'},\n",
       " 'query tabular is a galaxy based tool which manipulates tabular files query tabular automatically creates a sqlite database directly from a tabular file within a galaxy workflow the sqlite database can be saved to the galaxy history and further process to generate tabular outputs containing desired information and formatting': {'query_tabular'},\n",
       " 'aims to help improve the speed objectivity and reproducibility of digital pathology analysis and biomarker interpretation by offering an open powerful flexible extensible software platform for whole slide image analysis': {'qupath'},\n",
       " 'free software environment for statistical computing and graphics': {'r'},\n",
       " 'consensus module for raw de novo dna assembly of long uncorrected reads racon is intended as a standalone consensus module to correct raw contigs generated by rapid assembly methods which do not include a consensus step the goal of racon is to generate genomic consensus which is of similar or better quality compared to the output generated by assembly methods which employ both error correction and consensus steps while providing a speedup of several times compared to those methods it supports data produced by both pacific biosciences and oxford nanopore technologies': {'racon'},\n",
       " 'ragtag is a collection of software tools for scaffolding and improving modern genome assemblies': {'ragtag'},\n",
       " 'a feature clustering algorithm for non targeted mass spectrometric metabolomics data': {'ramclustr'},\n",
       " 'a tool for fast canonical neighbor joining tree construction': {'rapidnj'},\n",
       " 'ratatosk hybrid error correction of long reads enables accurate variant calling and assembly phased hybrid error correction of long reads using colored de bruijn graphs ratatosk is a phased error correction tool for erroneous long reads based on compacted and colored de bruijn graphs built from accurate short reads': {'ratatosk'},\n",
       " 'a de novo genome assembler for long reads raven is a de novo genome assembler for long uncorrected reads': {'raven'},\n",
       " 'a standalone tool for extracting data directly from raw files generated by thermo orbitrap family instruments': {'rawtools'},\n",
       " 'a tool for phylogenetic analysis and post analysis of large phylogenies': {'raxml'},\n",
       " 'rdkit is an open source cheminformatics software fast efficient fragment based coordinate generation for open babel': {'rdkit'},\n",
       " 'tool for calculating the probability of nucleosome formation along a dna sequence input by the user': {'recon'},\n",
       " 'robust comparative analysis and contamination removal for metagenomics': {'Recentrifuge'},\n",
       " 'recetox aplcms is a tool for peak detection in mass spectrometry data the tool performs 1 noise removal 2 peak detection 3 retention time drift correction 4 peak alignment and 5 weaker signal recovery as well as 6 suspect screening': {'recetox-aplcms'},\n",
       " 'this is a program to detect and visualize rna editing events at genomic scale using next generation sequencing data': {'red'},\n",
       " 'a program that screens dna sequences for interspersed repeats and low complexity dna sequences the output of the program is a detailed annotation of the repeats that are present in the query sequence as well as a modified version of the query sequence in which all the annotated repeats have been masked default replaced by ns': {'repeatmasker'},\n",
       " 'repeatmodeler is a de novo transposable element te family identification and modeling package at the heart of repeatmodeler are three de novo repeat finding programs recon repeatscout and ltrharvest ltr retriever which employ complementary computational methods for identifying repeat element boundaries and family relationships from sequence data': {'repeatmodeler'},\n",
       " 'repeatscout is a tool to discover repetitive substrings in dna': {'repeatscout'},\n",
       " 'riassigner is a python tool for retention index ri computation for gc ms data': {'riassigner'},\n",
       " 'the rjags package provides an interface from r to the jags library for bayesian data analysis jags uses markov chain monte carlo mcmc to generate a sequence of dependent samples from the posterior distribution of the parameters': {'rjags'},\n",
       " 'workflow to process tandem ms files and build massbank records functions include automated extraction of tandem ms spectra formula assignment to tandem ms fragments recalibration of tandem ms spectra with assigned fragments spectrum cleanup automated retrieval of compound information from internet databases and export to massbank records': {'rmassbank'},\n",
       " 'rmblast is a repeatmasker compatible version of the standard ncbi blastn program the primary difference between this distribution and the ncbi distribution is the addition of a new program rmblastn for use with repeatmasker and repeatmodeler': {'rmblast'},\n",
       " 'quality assessment tool for de novo transcriptome assemblies': {'rnaquast'},\n",
       " 'a high speed stand alone pan genome pipeline which takes annotated assemblies in gff3 format produced by prokka seemann 2014 and calculates the pan genome': {'roary'},\n",
       " 'rrmscorer provides quick predictions for any rna recognition motif rrm and any rna target purely based on their sequences': {'rrmscorer'},\n",
       " 'we present a generative statistical model and associated inference methods that handle read mapping uncertainty in a principled manner through simulations parameterized by real rnaseq data we show that our method is more accurate than previous methods our improved accuracy is the result of handling read mapping uncertainty with a statistical model and the estimation of gene expression levels as the sum of isoform expression levels': {'rsem'},\n",
       " 'provides a number of useful modules that can comprehensively evaluate high throughput sequence data especially rna seq data some basic modules quickly inspect sequence quality nucleotide composition bias pcr bias and gc bias while rna seq specific modules evaluate sequencing saturation mapped reads distribution coverage uniformity strand specificity transcript level rna integrity etc': {'rseqc'},\n",
       " 'integrated development environment ide for the r programming language': {'rstudio'},\n",
       " 'rtg core software for alignment and analysis of next gen sequencing data': {'rtg-tools'},\n",
       " 'extensible framework for interacting with multiple genome browsers currently ucsc built in and manipulating annotation tracks in various formats currently gff bed bedgraph bed15 wig bigwig and 2bit built in the user may export import tracks to from the supported browsers as well as query and modify the browser state such as the current viewport': {'rtracklayer'},\n",
       " 'a software tool that implements a novel is an alignment free algorithm for the estimation of isoform abundances directly from a set of reference sequences and rna seq reads': {'sailfish'},\n",
       " 'a tool for transcript expression quantification from rna seq data': {'salmon'},\n",
       " 'very low confidence correct name of tool could also be chromosome scale reference quality hi c scaffolder integrating hi c links with assembly graphs for chromosome scale assembly salsa a tool to scaffold long read assemblies with hi c data salsa a tool to scaffold long read assemblies with hi c this code is used to scaffold your assemblies using hi c data this version implements some improvements in the original salsa algorithm if you want to use the old version it can be found in the old salsa branch': {'salsa'},\n",
       " 'this tool is a high performance modern robust and fast tool and library written in the d programming language for working with sam bam and cram formats': {'sambamba'},\n",
       " 'samtools and bcftools are widely used programs for processing and analysing high throughput sequencing data they include tools for file format conversion and manipulation sorting querying statistics variant calling and effect analysis amongst other methods': {'samtools'},\n",
       " 'single cell clustering assessment framework sccaf is a novel method for automated identification of putative cell types from single cell rna seq scrna seq data': {'sccaf'},\n",
       " 'scalable toolkit for analyzing single cell gene expression data it includes preprocessing visualization clustering pseudotime and trajectory inference and differential expression testing the python based implementation efficiently deals with datasets of more than one million cells': {'scanpy'},\n",
       " 'pre processing quality control normalization and visualization of single cell rna seq data': {'scater'},\n",
       " 'sceasy is a package that helps easy conversion of different single cell data formats to each other': {'sceasy'},\n",
       " 'scikit image contains image processing algorithms for scipy including io morphology filtering warping color manipulation object detection etc': {'scikit-image'},\n",
       " 'scipio is a tool to determine the precise exon intron gene structure given a protein sequence and a genome it identifies splice sites and is able to cope with sequencing errors and genes spanning several contigs the output contains information about discrepancies that may result from sequencing errors scipio has also successfully been used to find homologous genes in related species webscipio allows to search for mutually exclusive spliced exons and tandemly arrayed gene duplicates': {'scipio'},\n",
       " 'implements a variety of low level analyses of single cell rna seq data methods are provided for normalization of cell specific biases assignment of cell cycle phase and detection of highly variable and significantly correlated genes': {'scran'},\n",
       " 'generalizing rna velocity to transient cell states through dynamical modeling single cell rna velocity generalized to transient cell states': {'scvelo'},\n",
       " 'command tool for metagenomic binning with semi supervised deep learning using information from reference genomes': {'semibin'},\n",
       " 'sepp stands for sat enabled phylogenetic placement and addresses the problem of phylogenetic placement for meta genomic short reads': {'sepp'},\n",
       " 'seq2hla is a computational tool to determine human leukocyte antigen hla directly from existing and future short rna seq reads it takes standard rna seq sequence reads in fastq format as input uses a bowtie index comprising known hla alleles and outputs the most likely hla class i and class ii types a p value for each call and the expression of each class': {'seq2hla'},\n",
       " 'fasta and fastq are basic and ubiquitous formats for storing nucleotide and protein sequences common manipulations of fasta q file include converting searching filtering deduplication splitting shuffling and sampling existing tools only implement some of these manipulations and not particularly efficiently and some are only available for certain operating systems furthermore the complicated installation process of required packages and running environments can render these programs less user friendly seqkit demonstrates competitive performance in execution time and memory usage compared to similar tools the efficiency and usability of seqkit enable researchers to rapidly accomplish common fasta q file manipulations': {'seqkit'},\n",
       " 'a tool for processing sequences in the fasta or fastq format it parses both fasta and fastq files which can also be optionally compressed by gzip': {'seqtk'},\n",
       " 'seurat is an r package designed for qc analysis and exploration of single cell rna seq data seurat aims to enable users to identify and interpret sources of heterogeneity from single cell transcriptomic measurements and to integrate diverse types of single cell data': {'seurat'},\n",
       " 'de novo assembly from oxford nanopore reads': {'shasta'},\n",
       " 'shovill is a pipeline for assembly of bacterial isolate genomes from illumina paired end reads shovill uses spades at its core but alters the steps before and after the primary assembly step to get similar results in less time shovill also supports other assemblers like skesa velvet and megahit so you can take advantage of the pre and post processing the shovill provides with those too': {'shovill'},\n",
       " 'a clustering approach for identification of enriched domains from histone modification chip seq data': {'sicer'},\n",
       " 'a text mining framework for interactive analysis and visualization of similarities among biomedical entities for each search query pmids or abstracts from pubmed are saved git clone https github com dlal group simtext for all pmids in each row of a table the according abstracts are saved in additional columns': {'simtext'},\n",
       " 'novelty inclusive microbial community profiling of shotgun metagenomes': {'singlem'},\n",
       " 'sinto is a toolkit for processing aligned single cell data': {'sinto'},\n",
       " 'the salmonella in silico typing resource sistr is an open source and freely available web application for rapid in silico typing and serovar prediction from salmonella genome assemblies using cgmlst and o and h antigen gene searching': {'sistr'},\n",
       " 'slow5tools is a simple toolkit for converting fast5 slow5 compressing viewing indexing and manipulating data in slow5 format about slow5 format slow5 is a new file format for storing signal data from oxford nanopore technologies ont devices slow5 was developed to overcome inherent limitations in the standard fast5 signal data format that prevent efficient scalable analysis and cause many headaches for developers slow5 can be encoded in human readable ascii format or a more compact and efficient binary format blow5 this is analogous to the seminal sam bam format for storing dna sequence alignments the blow5 binary format supports zlib deflate compression or other compression methods thereby minimising the data storage footprint while still permitting efficient parallel access detailed benchmarking experiments have shown that slow5 format is an order of magnitude faster and significantly smaller than fast5': {'slow5tools'},\n",
       " 'reference free profiling of polyploid genomes inference of ploidy and heterozygosity structure using whole genome sequencing data smudgeplots are computed from raw or even better from trimmed reads and show the haplotype structure using heterozygous kmer pairs for example this tool extracts heterozygous kmer pairs from kmer dump files and performs gymnastics with them we are able to disentangle genome structure by comparing the sum of kmer pair coverages cova covb to their relative coverage cova cova covb such an approach also allows us to analyze obscure genomes with duplications various ploidy levels etc genomescope 2 0 and smudgeplots reference free profiling of polyploid genomes timothy rhyker ranallo benavidez kamil s jaron michael c schatz biorxiv 747568 doi https doi org 10 1101 747568': {'Smudgeplots'},\n",
       " 'workflow engine and language it aims to reduce the complexity of creating workflows by providing a fast and comfortable execution environment together with a clean and modern domain specific specification language dsl in python style': {'snakemake'},\n",
       " 'the semi hmm based nucleic acid parser is a gene prediction tool': {'snap'},\n",
       " 'snapatac single nucleus analysis pipeline for atac seq is a fast accurate and comprehensive method for analyzing single cell atac seq datasets': {'snapatac'},\n",
       " 'a snakemake pipeline for scalable hiv 1 subtyping by phylogenetic pairing snappy is a snakemake pipeline for hiv 1 subtyping by phylogenetic pairing this is the repository for snappy a snakemake pipeline for hiv 1 subtyping by phylogenetic pairing snappy allows high throughput hiv 1 subtyping locally while being resource efficient and scalable this pipeline was constructed using snakemake and it uses mafft and for multiple sequence alignment blast for similarirty querys iq tree for phylogenetic inference and several biopython modules for data parsing an analysis for in depth information on how the tool works please visit the documentation page snappy was design for linux based operative systems welcome to snappy s documentation snappy hiv1 subtyping 1 0 0 documentation free document hosting provided by read the docs': {'snappy'},\n",
       " 'an algorithm for structural variation detection from third generation sequencing alignment': {'sniffles'},\n",
       " 'rapid haploid variant calling and core snp phylogeny generation': {'snippy'},\n",
       " 'variant annotation and effect prediction tool it annotates and predicts the effects of variants on genes and proteins such as amino acid changes': {'snpeff'},\n",
       " 'toolbox that allows you to filter and manipulate annotated vcf files': {'snpsift'},\n",
       " 'sequence analysis tool for filtering mapping and otu picking ngs reads': {'sortmerna'},\n",
       " 'st petersburg genome assembler is intended for both standard isolates and single cell mda bacteria assemblies spades 3 9 works with illumina or iontorrent reads and is capable of providing hybrid assemblies using pacbio oxford nanopore and sanger reads additional contigs can be provided and can be used as long reads': {'spades'},\n",
       " 'improved mass spectral similarity scoring through learning of structural relationships spec2vec is a novel spectral similarity score inspired by a natural language processing algorithm word2vec where word2vec learns relationships between words in sentences spec2vec does so for mass fragments and neutral losses in ms ms spectra the spectral similarity score is based on spectral embeddings learnt from the fragmental relationships within a large set of spectral data analysis and benchmarking of mass spectra similarity measures using gnps data set': {'spec2vec'},\n",
       " 'spectral repeat finder srf is a program to find repeats through an analysis of the power spectrum of a given dna sequence': {'srf'},\n",
       " 'squidpy spatial single cell analysis in python squidpy is a tool for the analysis and visualization of spatial molecular data it builds on top of scanpy and anndata from which it inherits modularity and scalability it provides analysis tools that leverages the spatial coordinates of the data as well as tissue images if available': {'squidpy'},\n",
       " 'some quick reconstruction to resolve evolutionary links squirrel provides a rapid way of producing reliable alignments for mpxv and also enable maximum likelihood phylogenetics pipeline tree estimation': {'squirrel'},\n",
       " 'the sra toolkit and sdk from ncbi is a collection of tools and libraries for using data in the insdc sequence read archives': {'sra-tools'},\n",
       " 'a fast implementation of the smith waterman algorithm whose api that can be flexibly used by programs written in c c and other languages': {'ssw'},\n",
       " 'developed to work with restriction enzyme based sequence data such as radseq for building genetic maps and conducting population genomics and phylogeography analysis': {'stacks'},\n",
       " 'ultrafast universal rna seq data aligner': {'star'},\n",
       " 'staramr amr scans bacterial genome contigs against the resfinder pointfinder and plasmidfinder databases used by the resfinder webservice and other webservices offered by the center for genomic epidemiology and compiles a summary report of detected antimicrobial resistance genes the star in staramr indicates that it can handle all of the resfinder pointfinder and plasmidfinder databases': {'staramr'},\n",
       " 'star fusion a method that is both fast and accurate in identifying fusion transcripts from rna seq data': {'star_fusion'},\n",
       " 'fast and highly efficient assembler of rna seq alignments into potential transcripts it uses a novel network flow algorithm as well as an optional de novo assembly step to assemble and quantitate full length transcripts representing multiple splice variants for each gene locus': {'stringtie'},\n",
       " 'subread is a general purpose read aligner which can be used to map both genomic dna seq reads and rna seq reads it uses a new mapping paradigm called seed and vote to achieve fast accurate and scalable read mapping it automatically determines if a read should be globally or locally aligned therefore particularly powerful in mapping rna seq reads it supports indel detection and can map reads with both fixed and variable lengths': {'subread'},\n",
       " 'an agile homology based approach using a reduced seed database to report the subsystems present in metagenomic samples and profile their abundances': {'superfocus'},\n",
       " 'this tool generates alternative splicing as events from an annotation and calculates the psi percentage spliced in value for each event exploiting fast quantification of transcript abundances from multiple samples': {'suppa'},\n",
       " 'structural variant detection from haploid and diploid genome assemblies svim asm structural variant identification method assembly edition svim asm pronounced swim assem is a structural variant caller for haploid or diploid genome genome alignments it analyzes a given sorted bam file preferably from minimap2 and detects five different variant classes between the query assembly and the reference deletions insertions tandem and interspersed duplications and inversions': {'svim-asm'},\n",
       " 'syri is tool for finding genomic rearrangements and local sequence differences from whole genome assemblies genomic differences range from single nucleotide differences to complex structural variations current methods typically annotate sequence differences ranging from snps to large indels accurately but do not unravel the full complexity of structural rearrangements including inversions translocations and duplications where highly similar sequence changes in location orientation or copy number here we present syri a pairwise whole genome comparison tool for chromosome level assemblies syri starts by finding rearranged regions and then searches for differences in the sequences which are distinguished for residing in syntenic or rearranged regions this distinction is important as rearranged regions are inherited differently compared to syntenic regions': {'syri'},\n",
       " 'targetfinder org provides a web based resource that allows users to find genes that have a similar expression to a query gene signature': {'targetfinder'},\n",
       " 'taxonkit is a practical and efficient ncbi taxonomy toolkit': {'taxonkit'},\n",
       " 'tbl2asn is a command line program that automates the creation of sequence records for submission to genbank it uses many of the same functions as genome workbench but is driven generally by data files tbl2asn generates sqn files for submission to genbank': {'tbl2asn'},\n",
       " 'a tool for drug resistance prediction from m tuberculosis genomic data sequencing reads alignments or variants': {'tbprofiler'},\n",
       " 'the combat tb workbench is an irida based module workbench for m tuberculosis bioinformatics it is designed to be easily deployed on a single server': {'tbvcfreport'},\n",
       " 'lineage level classification of transposable elements using conserved protein domains note do not move or hard link tesorter py alone to anywhere else as it rely on database and bin you can add the directory to path or soft link tesorter py to path': {'tesorter'},\n",
       " 'open source crossplatform tool that converts thermo raw files into open file formats such as mgf and to the hupo psi standard file format mzml': {'ThermoRawFileParser'},\n",
       " 'program that aligns rna seq reads to a genome in order to identify exon exon splice junctions it is built on the ultrafast short read mapping program bowtie a stable samtools version is now packaged with the program': {'tophat'},\n",
       " 'transdecoder identifies candidate coding regions within transcript sequences such as those generated by de novo rna seq transcript assembly using trinity or constructed based on rna seq alignments to the genome using tophat and cufflinks': {'transdecoder'},\n",
       " 'a tool for the analysis of tn seq data it provides an easy to use graphical interface and access to three different analysis methods that allow the user to determine essentiality in a single condition as well as between conditions': {'transit'},\n",
       " 'institute for systems biology trans proteomic pipeline': {'tpp'},\n",
       " 'tandem repeats finder find tandem repeats in dna sequences without the need to specify either the pattern or pattern size it uses the method of k tuple matching to avoid the need for full scale alignment matrix computations it requires no a priori knowledge of the pattern pattern size or number of copies there are no restrictions on the size of the repeats that can be detected it determines a consensus pattern for the smallest repetitive unit in the tandem repeat': {'trf'},\n",
       " 'tool for the automated removal of spurious sequences or poorly aligned regions from a multiple sequence alignment': {'trimal'},\n",
       " 'a wrapper tool around cutadapt and fastqc to consistently apply quality and adapter trimming to fastq files with some extra functionality for mspi digested rrbs type reduced representation bisufite seq libraries': {'trim_galore'},\n",
       " 'a flexible read trimming tool for illumina ngs data': {'trimmomatic'},\n",
       " 'trinity is a transcriptome assembler which relies on three different tools inchworm an assembler chrysalis which pools contigs and butterfly which amongst others compacts a graph resulting from butterfly with reads': {'trinity'},\n",
       " 'comprehensive annotation suite designed for automatic functional annotation of transcriptomes particularly de novo assembled transcriptomes from model or non model organisms': {'trinotate'},\n",
       " 'a program for improved detection of transfer rna genes in genomic sequence': {'trnascan-se'},\n",
       " 'trycycler consensus long read assemblies for bacterial genomes': {'trycycler'},\n",
       " 'utilities for handling sequences and assemblies from the ucsc genome browser project': {'ucsc'},\n",
       " 'tools for handling unique molecular identifiers in ngs data sets': {'umi_tools'},\n",
       " 'a tool for assembling bacterial genomes from a combination of short 2nd generation and long 3rd generation sequencing reads': {'unicycler'},\n",
       " 'metaproteomics data analysis with a focus on interactive data visualizations': {'unipept'},\n",
       " 'the universal protein knowledgebase in 2021 you are using a version of browser that may not display all the features of this website': {'UniProt_Downloader',\n",
       "  'uniprot_rest_interface'},\n",
       " 'tool for predicting effects of variants for any genome in ensembl or with genome annotation via gff this includes vertebrates and also plants fungi protists metazoa and bacteria there is a web and a rest api version but the most powerful is the perl script version see mclaren et al 2016 genome biology': {'vep'},\n",
       " 'vapor is a tool for classification of influenza samples from raw short read sequence data for downstream bioinformatics analysis vapor is provided with a fasta file of full length sequences 20 000 for a given segment a set of reads and attempts to retrieve a reference that is closest to the sample strain': {'vapor'},\n",
       " 'varscan an open source tool for variant detection that is compatible with several short read align ers': {'varscan'},\n",
       " 'api and command line utilities for the manipulation of vcf files': {'vcflib'},\n",
       " 'provide easily accessible methods for working with complex genetic variation data in the form of vcf files': {'vcftools'},\n",
       " 'estimating rna velocity in single cell rna sequencing datasets': {'velocyto'},\n",
       " 'a de novo genomic assembler specially designed for short read sequencing technologies such as solexa or 454 or solid': {'velvet'},\n",
       " 'high throughput search and clustering sequence analysis tool it supports de novo and reference based chimera detection clustering full length and prefix dereplication reverse complementation masking all vs all pairwise global alignment exact and global alignment searching shuffling subsampling and sorting it also supports fastq file analysis filtering and conversion': {'vsearch'},\n",
       " 'removal of batch effects for large scale untargeted metabolomics data based on wavelet transform': {'waveica'},\n",
       " 'software for phasing genomic variants using dna sequencing reads also called haplotype assembly it is especially suitable for long reads but works also well with short reads': {'whatshap'},\n",
       " 'windowmasker identifies and masks highly repetitive dna sequences in a genome using only the sequence of the genome itself': {'windowmasker'},\n",
       " 'winnowmap is a long read mapping algorithm optimized for mapping ont and pacbio reads to repetitive reference sequences winnowmap development began on top of minimap2 codebase and since then we have incorporated the following two ideas to improve mapping accuracy within repeats': {'winnowmap'},\n",
       " 'first fully open source and collaborative online platform for computational metabolomics it includes preprocessing normalization quality control statistical analysis of lc ms fia ms gc ms and nmr data': {'workflow4metabolomics'},\n",
       " 'caenorhabditis elegans genome database international consortium of biologists and computer scientists dedicated to providing the research community with accurate current accessible information concerning the genetics genomics and biology of c elegans and related nematodes founded in 2000 the consortium is led by paul sternberg of caltech paul kersey of the ebi matt berriman of the wellcome trust sanger institute and lincoln stein of the ontario institute for cancer research': {'wormbase'},\n",
       " 'framework for processing and visualization of chromatographically separated and single spectra mass spectral data the packages enables imports from aia andi netcdf mzxml mzdata and mzml files and preprocesses data for high throughput untargeted analyte profiling': {'xcms'},\n",
       " 'detection of differential rna modifications from direct rna sequencing of human cell lines python package for detection of differential rna modifications from direct rna sequencing': {'xpore'},\n",
       " 'matches tandem mass spectra with peptide sequences': {'xtandem'},\n",
       " 'the xtb molecular optimization tool based on the semiempirical tight binding method gfnn xtb is implemented in the xtb extended tight binding program package this tool handles molecular xyz input format and provides various levels of optimization to suit different computational needs the xtb method offers an efficient approach for molecular structure optimization making it valuable for large scale quantum chemical simulations': {'xtb'},\n",
       " 'yahs is scaffolding tool using hi c data it relies on a new algorithm for contig joining detection which considers the topological distribution of hi c signals aiming to distinguish real interaction signals from mapping noises': {'yahs'},\n",
       " 'search and retrieve s cerevisiae data populated by sgd and powered by intermine': {'yeastmine'},\n",
       " 'zebrafishmine is powered by the intermine data warehouse system and integrates biological data sets from multiple sources it currently includes updates of data from zfin the zebrafish model organism database there is also data from the panther database': {'zebrafishmine'}}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concepts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f65b3f2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bio-mcp (3.10.12)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
