{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4ca7e700",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "\n",
    "from bio_mcp.cache.load import load_biotools, load_galaxy_singularity\n",
    "from bio_mcp.mcp.server import search_containers, describe_container"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03258a05",
   "metadata": {},
   "source": [
    "## Loading snapshots\n",
    "\n",
    "Galaxy containers from the CVMFS and tool metadata from e.g. biotools are indexed by lowercased tool names so they can be mapped together."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4034b2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "bt = load_biotools(Path(\"/home/ubuntu/bio-mcp/data/scrnaseq_biotools.yaml\"))\n",
    "gx = load_galaxy_singularity(\n",
    "    Path(\"/home/ubuntu/bio-mcp/data/scrnaseq_galaxy_cvmfs.json\")\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "467ed279",
   "metadata": {},
   "outputs": [],
   "source": [
    "g = gx.get(\"cellbender\")\n",
    "g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e06ea05c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fff19532",
   "metadata": {},
   "outputs": [],
   "source": [
    "bt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ff9f1c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gx.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "376b6ea0",
   "metadata": {},
   "source": [
    "## `search_containers`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dad7a5a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"bcftools\", gx)  # Should return []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a50346f",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"seurat\", gx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa4624d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"cellranger\", gx)  # near match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5220f16c",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"multiqc\", gx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0229edf",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers([\"cellbender\", \"multiqc\"], gx)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41516606",
   "metadata": {},
   "source": [
    "## `describe_container()`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd78d312",
   "metadata": {},
   "outputs": [],
   "source": [
    "describe_container(gx, bt, \"cellbender\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41737a67",
   "metadata": {},
   "outputs": [],
   "source": [
    "for tool in [\"cellbender\"]:\n",
    "    print(tool)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20594da4",
   "metadata": {},
   "source": [
    "## Create metadata embeddings\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "471aac57",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sentence_transformers.cross_encoder import CrossEncoder\n",
    "from bio_mcp.cache.embeddings import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5614ea19",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load a pretrained ST model\n",
    "model = CrossEncoder(\"cross-encoder/stsb-distilroberta-base\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ea24cac",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_queries = [\n",
    "    \"I have fastq files from a single cell sequencing experiment, what is the first step i can do?\",\n",
    "    \"How do I qc, clustering and integration of my pre-processed scrna data?\",\n",
    "    \"what tools are available for scrna analysis\",\n",
    "    \"How can I remove technical artifacts (e.g. identify and remove doublets from my data)\",\n",
    "    \"I have just generated scrna data, how do I analyse it for publication?\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dfef745",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create corpus of tool metadata\n",
    "meta_text = []\n",
    "for tool, _ in bt.items():\n",
    "    text = make_embedding_text(bt.get(tool))\n",
    "    meta_text.append(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "80a20499",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Tool name: CellBender\\nDescription: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets.\\n\\nCellBender is a software package for eliminating technical artifacts from high-throughput single-cell RNA sequencing (scRNA-seq) data.\\n\\nWelcome to CellBender’s documentation! — CellBender documentation.\\n\\nFree document hosting provided by Read the Docs.\\n\\nStephen J Fleming, John C Marioni, and Mehrtash Babadi. CellBender remove-background: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets. bioRxiv 791699; doi: https://doi.org/10.1101/791699.\\nEDAM topics: Gene expression, RNA, RNA-Seq\\nEDAM operations: Quantification, Data retrieval, Expression analysis',\n",
       " 'Tool name: cellranger',\n",
       " 'Tool name: FASTQC\\nDescription: This tool aims to provide a QC report which can spot problems or biases which originate either in the sequencer or in the starting library material. It can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.\\nEDAM topics: Sequencing, Data quality management, Sequence analysis\\nEDAM operations: Sequence composition calculation, Sequencing quality control, Statistical calculation',\n",
       " \"Tool name: MultiQC\\nDescription: MultiQC aggregates results from multiple bioinformatics analyses across many samples into a single report. It searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.\\nEDAM topics: Sequencing, Bioinformatics, Sequence analysis, Genomics\\nEDAM operations: Validation, Sequencing quality control\",\n",
       " 'Tool name: scanpy\\nDescription: Scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.\\nEDAM topics: Gene expression, Cell biology, Genetics, Single-cell sequencing\\nEDAM operations: Differential gene expression analysis, Principal component visualisation, Expression profile clustering',\n",
       " 'Tool name: seurat\\nDescription: Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.\\nEDAM topics: RNA-Seq, Transcriptomics']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b106c8a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:02<00:00,  2.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5, 384)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# Rank all \n",
    "ranks = model.rank(example_queries[0], meta_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e5d9fa8",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(f\"Query: {example_queries[0]}\")\n",
    "for rank in ranks:\n",
    "    print(f\"{rank['score']:.2f}\\t{meta_text[rank['corpus_id']]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3e689a88",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches:   0%|          | 0/1 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:01<00:00,  1.59s/it]\n"
     ]
    }
   ],
   "source": [
    "rank2 = model.rank(\"What should I use for phylogenetics\", meta_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "26d3e139",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.61\tTool name: seurat\n",
      "Description: Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.\n",
      "EDAM topics: RNA-Seq, Transcriptomics\n",
      "0.52\tTool name: CellBender\n",
      "Description: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets.\n",
      "\n",
      "CellBender is a software package for eliminating technical artifacts from high-throughput single-cell RNA sequencing (scRNA-seq) data.\n",
      "\n",
      "Welcome to CellBender’s documentation! — CellBender documentation.\n",
      "\n",
      "Free document hosting provided by Read the Docs.\n",
      "\n",
      "Stephen J Fleming, John C Marioni, and Mehrtash Babadi. CellBender remove-background: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets. bioRxiv 791699; doi: https://doi.org/10.1101/791699.\n",
      "EDAM topics: Gene expression, RNA, RNA-Seq\n",
      "EDAM operations: Quantification, Data retrieval, Expression analysis\n",
      "0.49\tTool name: MultiQC\n",
      "Description: MultiQC aggregates results from multiple bioinformatics analyses across many samples into a single report. It searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.\n",
      "EDAM topics: Sequencing, Bioinformatics, Sequence analysis, Genomics\n",
      "EDAM operations: Validation, Sequencing quality control\n",
      "0.45\tTool name: scanpy\n",
      "Description: Scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.\n",
      "EDAM topics: Gene expression, Cell biology, Genetics, Single-cell sequencing\n",
      "EDAM operations: Differential gene expression analysis, Principal component visualisation, Expression profile clustering\n",
      "0.45\tTool name: FASTQC\n",
      "Description: This tool aims to provide a QC report which can spot problems or biases which originate either in the sequencer or in the starting library material. It can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.\n",
      "EDAM topics: Sequencing, Data quality management, Sequence analysis\n",
      "EDAM operations: Sequence composition calculation, Sequencing quality control, Statistical calculation\n",
      "0.07\tTool name: cellranger\n"
     ]
    }
   ],
   "source": [
    "for rank in rank2:\n",
    "    print(f\"{rank['score']:.2f}\\t{meta_text[rank['corpus_id']]}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd256422",
   "metadata": {},
   "source": [
    "## Aliasing edams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "685306fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_yaml = load_biotools(Path(\"/home/ubuntu/cache/johan.yaml\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7a6d8ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'biocontainers': 'samtools',\n",
       " 'biotools': 'samtools',\n",
       " 'bunya': ['1.13-gcc-10.3.0',\n",
       "  '1.13-gcc-11.3.0',\n",
       "  '1.16.1-gcc-11.3.0',\n",
       "  '1.18-gcc-12.3.0                  (D)'],\n",
       " 'description': 'SAMtools and BCFtools are widely used programs for processing and analysing high-throughput sequencing data. They include tools for file format conversion and manipulation, sorting, querying, statistics, variant calling, and effect analysis amongst other methods.',\n",
       " 'edam-inputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       " 'edam-operations': ['Data filtering',\n",
       "  'Visualisation',\n",
       "  'Data formatting',\n",
       "  'Indexing',\n",
       "  'Data parsing',\n",
       "  'Data sorting',\n",
       "  'Data editing'],\n",
       " 'edam-outputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       " 'edam-topics': ['Mapping',\n",
       "  'Sequence analysis',\n",
       "  'Sequencing',\n",
       "  'Rare diseases'],\n",
       " 'galaxy': [{'description': 'Samtools markdup: marks duplicate alignments',\n",
       "   'title': 'Samtools markdup 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_markdup%2Fsamtools_markdup%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools view: - reformat, filter, or subsample SAM, BAM or CRAM',\n",
       "   'title': 'Samtools view 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_view%2Fsamtools_view%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools split: BAM dataset on readgroups',\n",
       "   'title': 'Samtools split 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_split%2Fsamtools_split%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Slice: BAM by genomic regions',\n",
       "   'title': 'Slice 2.0.6',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_slice_bam%2Fsamtools_slice_bam%2F2.0.6'},\n",
       "  {'description': 'Samtools reheader: copy SAM/BAM header between datasets',\n",
       "   'title': 'Samtools reheader 2.0.6',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_reheader%2Fsamtools_reheader%2F2.0.6'},\n",
       "  {'description': 'Samtools phase: call and phase heterozygous SNPs',\n",
       "   'title': 'Samtools phase 2.0.2',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_phase%2Fsamtools_phase%2F2.0.2'},\n",
       "  {'description': 'Samtools mpileup: multi-way pileup of variants',\n",
       "   'title': 'Samtools mpileup 2.2.0',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_mpileup%2Fsamtools_mpileup%2F2.2.0'},\n",
       "  {'description': 'Samtools merge: merge multiple sorted alignment files',\n",
       "   'title': 'Samtools merge 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_merge%2Fsamtools_merge%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools idxstats: reports stats of the BAM index file',\n",
       "   'title': 'Samtools idxstats 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_idxstats%2Fsamtools_idxstats%2F2.0.8'},\n",
       "  {'description': 'Samtools fastx: extract FASTA or FASTQ from alignment files',\n",
       "   'title': 'Samtools fastx 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fastx%2Fsamtools_fastx%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools depth: compute the depth at each position or region',\n",
       "   'title': 'Samtools depth 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_depth%2Fsamtools_depth%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools coverage: Produces a histogram or table of coverage per chromosome',\n",
       "   'title': 'Samtools coverage 1.22+galaxy3',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_coverage%2Fsamtools_coverage%2F1.22%2Bgalaxy3'},\n",
       "  {'description': 'Samtools calmd: recalculate MD/NM tags',\n",
       "   'title': 'Samtools calmd 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_calmd%2Fsamtools_calmd%2F2.0.7'},\n",
       "  {'description': 'Samtools bedcov: calculate read depth for a set of genomic intervals',\n",
       "   'title': 'Samtools bedcov 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_bedcov%2Fsamtools_bedcov%2F2.0.7'},\n",
       "  {'description': 'BAM-to-SAM: convert BAM to SAM',\n",
       "   'title': 'BAM-to-SAM 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbam_to_sam%2Fbam_to_sam%2F2.0.7'},\n",
       "  {'description': 'Samtools stats: generate statistics for BAM dataset',\n",
       "   'title': 'Samtools stats 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_stats%2Fsamtools_stats%2F2.0.8'},\n",
       "  {'description': 'Samtools sort: order of storing aligned sequences',\n",
       "   'title': 'Samtools sort 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_sort%2Fsamtools_sort%2F2.0.8'},\n",
       "  {'description': 'SAM-to-BAM: convert SAM to BAM',\n",
       "   'title': 'SAM-to-BAM 2.1.5',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsam_to_bam%2Fsam_to_bam%2F2.1.5'},\n",
       "  {'description': 'Samtools flagstat: tabulate descriptive stats for BAM datset',\n",
       "   'title': 'Samtools flagstat 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_flagstat%2Fsamtools_flagstat%2F2.0.8'},\n",
       "  {'description': 'Samtools fixmate: fill mate coordinates, ISIZE and mate related flags',\n",
       "   'title': 'Samtools fixmate 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fixmate%2Fsamtools_fixmate%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Filter SAM or BAM, output SAM or BAM: files on FLAG MAPQ RG LN or by region',\n",
       "   'title': 'Filter SAM or BAM, output SAM or BAM 1.8+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtool_filter2%2Fsamtool_filter2%2F1.8%2Bgalaxy1'},\n",
       "  {'description': 'RmDup: remove PCR duplicates',\n",
       "   'title': 'RmDup 2.0.1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_rmdup%2Fsamtools_rmdup%2F2.0.1'},\n",
       "  {'description': 'samtools BAM to CRAM: convert BAM alignments to CRAM format',\n",
       "   'title': 'samtools BAM to CRAM 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_bam_to_cram%2Fsamtools_bam_to_cram%2F1.22%2Bgalaxy1'}],\n",
       " 'homepage': 'http://www.htslib.org/',\n",
       " 'id': 'samtools',\n",
       " 'license': 'MIT',\n",
       " 'name': 'SAMTools',\n",
       " 'nci-gadi': None,\n",
       " 'nci-if89': ['1.18', '1.19.2'],\n",
       " 'pawsey': ['1.15--h3843a85_0'],\n",
       " 'publications': [{'title': 'The Sequence Alignment/Map format and SAMtools',\n",
       "   'url': 'https://doi.org/10.1093/bioinformatics/btp352'},\n",
       "  {'title': 'Twelve years of SAMtools and BCFtools',\n",
       "   'url': 'https://doi.org/10.1093/gigascience/giab008'},\n",
       "  {'title': 'A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data',\n",
       "   'url': 'https://doi.org/10.1093/bioinformatics/btr509'}],\n",
       " 'resources': [{'title': nan, 'url': nan}]}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for \n",
    "all_yaml.get(\"samtools\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0ac64bb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "collected_data = []\n",
    "for tool_id, tool_data in all_yaml.items():\n",
    "    entry = {\n",
    "        \"id\": tool_id,\n",
    "        \"description\": tool_data.get(\"description\", \"\"),\n",
    "        \"edam_inputs\": tool_data.get(\"edam-inputs\", []),\n",
    "        \"edam_outputs\": tool_data.get(\"edam-outputs\", []),\n",
    "        \"edam_operations\": tool_data.get(\"edam-operations\", []),\n",
    "        \"edam_topics\": tool_data.get(\"edam-topics\", []),\n",
    "    }\n",
    "    collected_data.append(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "9d00e4a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'3D profile generation',\n",
       " 'Aggregation',\n",
       " 'Alignment',\n",
       " 'Alternative splicing prediction',\n",
       " 'Analysis',\n",
       " 'Ancestral reconstruction',\n",
       " 'Annotation',\n",
       " 'Antimicrobial resistance prediction',\n",
       " 'Backbone modelling',\n",
       " 'Base-calling',\n",
       " 'Bisulfite mapping',\n",
       " 'Blind peptide database search',\n",
       " 'Box-Whisker plot plotting',\n",
       " 'Calculation',\n",
       " 'Cell migration analysis',\n",
       " 'Chimera detection',\n",
       " 'Chromatographic alignment',\n",
       " 'Classification',\n",
       " 'Clustering',\n",
       " 'Coding region prediction',\n",
       " 'Comparison',\n",
       " 'Conversion',\n",
       " 'Copy number estimation',\n",
       " 'Correlation',\n",
       " 'Cross-assembly',\n",
       " 'DMR identification',\n",
       " 'DNA barcoding',\n",
       " 'DNA mapping',\n",
       " 'DNA transcription',\n",
       " 'DNA translation',\n",
       " 'Data editing',\n",
       " 'Data filtering',\n",
       " 'Data formatting',\n",
       " 'Data handling',\n",
       " 'Data parsing',\n",
       " 'Data retrieval',\n",
       " 'Data sorting',\n",
       " 'Database search',\n",
       " 'De-novo assembly',\n",
       " 'Deisotoping',\n",
       " 'Demultiplexing',\n",
       " 'Dendrogram visualisation',\n",
       " 'Deposition',\n",
       " 'Design',\n",
       " 'Differential binding analysis',\n",
       " 'Differential gene expression analysis',\n",
       " 'Differential gene expression profiling',\n",
       " 'Differential protein expression analysis',\n",
       " 'Differential protein expression profiling',\n",
       " 'Dimensionality reduction',\n",
       " 'Dot plot plotting',\n",
       " 'Editing',\n",
       " 'Enrichment analysis',\n",
       " 'Essential dynamics',\n",
       " 'Exonic splicing enhancer prediction',\n",
       " 'Expression analysis',\n",
       " 'Expression correlation analysis',\n",
       " 'Expression profile clustering',\n",
       " 'Feature extraction',\n",
       " 'Filtering',\n",
       " 'Fold recognition',\n",
       " 'Format validation',\n",
       " 'Formatting',\n",
       " 'Functional clustering',\n",
       " 'Gene expression analysis',\n",
       " 'Gene expression profiling',\n",
       " 'Gene functional annotation',\n",
       " 'Gene prediction',\n",
       " 'Gene regulatory network analysis',\n",
       " 'Gene set testing',\n",
       " 'Gene-set enrichment analysis',\n",
       " 'Generation',\n",
       " 'Genetic mapping',\n",
       " 'Genetic variation analysis',\n",
       " 'Genome alignment',\n",
       " 'Genome annotation',\n",
       " 'Genome assembly',\n",
       " 'Genome comparison',\n",
       " 'Genome indexing',\n",
       " 'Genome visualisation',\n",
       " 'Genotyping',\n",
       " 'Global alignment',\n",
       " 'Haplotype mapping',\n",
       " 'Heat map generation',\n",
       " 'Homology-based gene prediction',\n",
       " 'Image analysis',\n",
       " 'Image annotation',\n",
       " 'Imputation',\n",
       " 'Indexing',\n",
       " 'Information retrieval',\n",
       " 'Ion counting',\n",
       " 'Isotope-coded protein label',\n",
       " 'Label-free quantification',\n",
       " 'Labeled quantification',\n",
       " 'Linkage disequilibrium calculation',\n",
       " 'Loading',\n",
       " 'Local alignment',\n",
       " 'Loop modelling',\n",
       " 'Mapping',\n",
       " 'Mapping assembly',\n",
       " 'Mass spectra calibration',\n",
       " 'Mass spectrum visualisation',\n",
       " 'Metabolic labeling',\n",
       " 'Metabolic network modelling',\n",
       " 'Metabolic pathway prediction',\n",
       " 'Methylation analysis',\n",
       " 'Methylation calling',\n",
       " 'Microscope image visualisation',\n",
       " 'Modelling and simulation',\n",
       " 'Molecular docking',\n",
       " 'Molecular dynamics',\n",
       " 'Molecular dynamics simulation',\n",
       " 'Molecular surface analysis',\n",
       " 'Multilocus sequence typing',\n",
       " 'Multiple sequence alignment',\n",
       " 'Natural product identification',\n",
       " 'Network analysis',\n",
       " 'Neurite measurement',\n",
       " 'Nucleic acid feature detection',\n",
       " 'Nucleic acid sequence analysis',\n",
       " 'Nucleic acid structure prediction',\n",
       " 'Nucleosome formation or exclusion sequence prediction',\n",
       " 'Optimisation and refinement',\n",
       " 'PCR primer design',\n",
       " 'PTM identification',\n",
       " 'PTM localisation',\n",
       " 'PTM site prediction',\n",
       " 'Pairwise sequence alignment',\n",
       " 'Parsing',\n",
       " 'Pathway analysis',\n",
       " 'Pathway or network analysis',\n",
       " 'Pathway or network visualisation',\n",
       " 'Peak calling',\n",
       " 'Peak detection',\n",
       " 'Peptide database search',\n",
       " 'Peptide identification',\n",
       " 'Phasing',\n",
       " 'Phylogenetic analysis',\n",
       " 'Phylogenetic footprinting',\n",
       " 'Phylogenetic inference',\n",
       " 'Phylogenetic reconstruction',\n",
       " 'Phylogenetic tree analysis',\n",
       " 'Phylogenetic tree analysis (shape)',\n",
       " 'Phylogenetic tree bootstrapping',\n",
       " 'Phylogenetic tree distances calculation',\n",
       " 'Phylogenetic tree editing',\n",
       " 'Phylogenetic tree generation',\n",
       " 'Phylogenetic tree generation (from molecular sequences)',\n",
       " 'Phylogenetic tree generation (maximum likelihood and Bayesian methods)',\n",
       " 'Phylogenetic tree reconstruction',\n",
       " 'Phylogenetic tree topology analysis',\n",
       " 'Phylogenetic tree visualisation',\n",
       " 'Plasmid map drawing',\n",
       " 'PolyA signal detection',\n",
       " 'Polymorphism detection',\n",
       " 'Post-translation modification site prediction',\n",
       " 'Prediction and recognition',\n",
       " 'Prediction and recognition (protein)',\n",
       " 'Primer removal',\n",
       " 'Principal component plotting',\n",
       " 'Principal component visualisation',\n",
       " 'Protein disorder prediction',\n",
       " 'Protein feature detection',\n",
       " 'Protein folding analysis',\n",
       " 'Protein fragment weight comparison',\n",
       " 'Protein geometry calculation',\n",
       " 'Protein geometry validation',\n",
       " 'Protein identification',\n",
       " 'Protein property calculation',\n",
       " 'Protein quantification',\n",
       " 'Protein secondary structure prediction',\n",
       " 'Protein sequence analysis',\n",
       " 'Protein structure analysis',\n",
       " 'Protein structure prediction',\n",
       " 'Protein threading',\n",
       " 'Protein-ligand docking',\n",
       " 'Quantification',\n",
       " 'Query and retrieval',\n",
       " 'RNA-Seq analysis',\n",
       " 'RNA-Seq quantification',\n",
       " 'RNA-seq time series data analysis',\n",
       " 'Rarefaction',\n",
       " 'Read binning',\n",
       " 'Read depth analysis',\n",
       " 'Read mapping',\n",
       " 'Read pre-processing',\n",
       " 'Read summarisation',\n",
       " 'Regression analysis',\n",
       " 'Repeat sequence analysis',\n",
       " 'Repeat sequence detection',\n",
       " 'Retention time prediction',\n",
       " 'SNP annotation',\n",
       " 'SNP detection',\n",
       " 'Scaffold gap completion',\n",
       " 'Scaffolding',\n",
       " 'Scatter plot plotting',\n",
       " 'Sequence alignment',\n",
       " 'Sequence alignment analysis',\n",
       " 'Sequence alignment conversion',\n",
       " 'Sequence analysis',\n",
       " 'Sequence annotation',\n",
       " 'Sequence assembly',\n",
       " 'Sequence assembly validation',\n",
       " 'Sequence assembly visualisation',\n",
       " 'Sequence classification',\n",
       " 'Sequence cluster visualisation',\n",
       " 'Sequence clustering',\n",
       " 'Sequence comparison',\n",
       " 'Sequence composition calculation',\n",
       " 'Sequence contamination filtering',\n",
       " 'Sequence conversion',\n",
       " 'Sequence database search',\n",
       " 'Sequence distance matrix generation',\n",
       " 'Sequence editing',\n",
       " 'Sequence feature comparison',\n",
       " 'Sequence feature detection',\n",
       " 'Sequence file editing',\n",
       " 'Sequence generation',\n",
       " 'Sequence merging',\n",
       " 'Sequence motif recognition',\n",
       " 'Sequence read processing',\n",
       " 'Sequence similarity search',\n",
       " 'Sequence trimming',\n",
       " 'Sequence visualisation',\n",
       " 'Sequencing error detection',\n",
       " 'Sequencing quality control',\n",
       " 'Service composition',\n",
       " 'Service discovery',\n",
       " 'Service invocation',\n",
       " 'Service management',\n",
       " 'Side chain modelling',\n",
       " 'Simulation analysis',\n",
       " 'Single particle analysis',\n",
       " 'Species frequency estimation',\n",
       " 'Spectral analysis',\n",
       " 'Spectral library search',\n",
       " 'Spectrum calculation',\n",
       " 'Splice site prediction',\n",
       " 'Splice transcript prediction',\n",
       " 'Splicing analysis',\n",
       " 'Splicing model analysis',\n",
       " 'Split read mapping',\n",
       " 'Splitting',\n",
       " 'Standardisation and normalisation',\n",
       " 'Statistical calculation',\n",
       " 'Statistical inference',\n",
       " 'Statistical modelling',\n",
       " 'Structural variation detection',\n",
       " 'Structure alignment',\n",
       " 'Structure visualisation',\n",
       " 'Tag-based peptide identification',\n",
       " 'Target-Decoy',\n",
       " 'Taxonomic classification',\n",
       " 'Text annotation',\n",
       " 'Text mining',\n",
       " 'Trajectory visualization',\n",
       " 'Transcriptome assembly',\n",
       " 'Transposon prediction',\n",
       " 'Validation',\n",
       " 'Validation of peptide-spectrum matches',\n",
       " 'Variant calling',\n",
       " 'Variant classification',\n",
       " 'Variant effect prediction',\n",
       " 'Variant filtering',\n",
       " 'Variant pattern analysis',\n",
       " 'Variant prioritisation',\n",
       " 'Visualisation',\n",
       " 'de Novo sequencing',\n",
       " 'iTRAQ',\n",
       " 'k-mer counting',\n",
       " 'miRNA expression analysis',\n",
       " 'miRNA target prediction',\n",
       " 'tRNA gene prediction'}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generate set of unique edam operations\n",
    "x = { op\n",
    "        for tool in collected_data\n",
    "        for op in (tool.get(\"edam_operations\") or [])\n",
    "}\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "65fd7edd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "273"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed8dc539",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bio-mcp (3.10.12)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
