{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4ca7e700",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "\n",
    "from bio_mcp.cache.load import load_biotools, load_galaxy_singularity\n",
    "from bio_mcp.mcp.server import search_containers, describe_container"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03258a05",
   "metadata": {},
   "source": [
    "## Loading snapshots\n",
    "\n",
    "Galaxy containers from the CVMFS and tool metadata from e.g. biotools are indexed by lowercased tool names so they can be mapped together."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4034b2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "bt = load_biotools(Path(\"/home/ubuntu/bio-mcp/data/scrnaseq_biotools.yaml\"))\n",
    "gx = load_galaxy_singularity(\n",
    "    Path(\"/home/ubuntu/bio-mcp/data/scrnaseq_galaxy_cvmfs.json\")\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "467ed279",
   "metadata": {},
   "outputs": [],
   "source": [
    "g = gx.get(\"cellbender\")\n",
    "g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e06ea05c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fff19532",
   "metadata": {},
   "outputs": [],
   "source": [
    "bt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ff9f1c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gx.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "376b6ea0",
   "metadata": {},
   "source": [
    "## `search_containers`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dad7a5a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"bcftools\", gx)  # Should return []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a50346f",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"seurat\", gx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa4624d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"cellranger\", gx)  # near match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5220f16c",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers(\"multiqc\", gx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0229edf",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_containers([\"cellbender\", \"multiqc\"], gx)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41516606",
   "metadata": {},
   "source": [
    "## `describe_container()`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd78d312",
   "metadata": {},
   "outputs": [],
   "source": [
    "describe_container(gx, bt, \"cellbender\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41737a67",
   "metadata": {},
   "outputs": [],
   "source": [
    "for tool in [\"cellbender\"]:\n",
    "    print(tool)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20594da4",
   "metadata": {},
   "source": [
    "## Create metadata embeddings\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "471aac57",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sentence_transformers.cross_encoder import CrossEncoder\n",
    "from bio_mcp.cache.embeddings import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5614ea19",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load a pretrained ST model\n",
    "model = CrossEncoder(\"cross-encoder/stsb-distilroberta-base\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ea24cac",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_queries = [\n",
    "    \"I have fastq files from a single cell sequencing experiment, what is the first step i can do?\",\n",
    "    \"How do I qc, clustering and integration of my pre-processed scrna data?\",\n",
    "    \"what tools are available for scrna analysis\",\n",
    "    \"How can I remove technical artifacts (e.g. identify and remove doublets from my data)\",\n",
    "    \"I have just generated scrna data, how do I analyse it for publication?\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dfef745",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create corpus of tool metadata\n",
    "meta_text = []\n",
    "for tool, _ in bt.items():\n",
    "    text = make_embedding_text(bt.get(tool))\n",
    "    meta_text.append(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "80a20499",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Tool name: CellBender\\nDescription: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets.\\n\\nCellBender is a software package for eliminating technical artifacts from high-throughput single-cell RNA sequencing (scRNA-seq) data.\\n\\nWelcome to CellBender’s documentation! — CellBender documentation.\\n\\nFree document hosting provided by Read the Docs.\\n\\nStephen J Fleming, John C Marioni, and Mehrtash Babadi. CellBender remove-background: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets. bioRxiv 791699; doi: https://doi.org/10.1101/791699.\\nEDAM topics: Gene expression, RNA, RNA-Seq\\nEDAM operations: Quantification, Data retrieval, Expression analysis',\n",
       " 'Tool name: cellranger',\n",
       " 'Tool name: FASTQC\\nDescription: This tool aims to provide a QC report which can spot problems or biases which originate either in the sequencer or in the starting library material. It can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.\\nEDAM topics: Sequencing, Data quality management, Sequence analysis\\nEDAM operations: Sequence composition calculation, Sequencing quality control, Statistical calculation',\n",
       " \"Tool name: MultiQC\\nDescription: MultiQC aggregates results from multiple bioinformatics analyses across many samples into a single report. It searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.\\nEDAM topics: Sequencing, Bioinformatics, Sequence analysis, Genomics\\nEDAM operations: Validation, Sequencing quality control\",\n",
       " 'Tool name: scanpy\\nDescription: Scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.\\nEDAM topics: Gene expression, Cell biology, Genetics, Single-cell sequencing\\nEDAM operations: Differential gene expression analysis, Principal component visualisation, Expression profile clustering',\n",
       " 'Tool name: seurat\\nDescription: Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.\\nEDAM topics: RNA-Seq, Transcriptomics']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b106c8a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:02<00:00,  2.46s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5, 384)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# Rank all \n",
    "ranks = model.rank(example_queries[0], meta_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e5d9fa8",
   "metadata": {},
   "outputs": [],
   "source": [
    "print(f\"Query: {example_queries[0]}\")\n",
    "for rank in ranks:\n",
    "    print(f\"{rank['score']:.2f}\\t{meta_text[rank['corpus_id']]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3e689a88",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches:   0%|          | 0/1 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Batches: 100%|██████████| 1/1 [00:01<00:00,  1.59s/it]\n"
     ]
    }
   ],
   "source": [
    "rank2 = model.rank(\"What should I use for phylogenetics\", meta_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "26d3e139",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.61\tTool name: seurat\n",
      "Description: Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.\n",
      "EDAM topics: RNA-Seq, Transcriptomics\n",
      "0.52\tTool name: CellBender\n",
      "Description: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets.\n",
      "\n",
      "CellBender is a software package for eliminating technical artifacts from high-throughput single-cell RNA sequencing (scRNA-seq) data.\n",
      "\n",
      "Welcome to CellBender’s documentation! — CellBender documentation.\n",
      "\n",
      "Free document hosting provided by Read the Docs.\n",
      "\n",
      "Stephen J Fleming, John C Marioni, and Mehrtash Babadi. CellBender remove-background: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets. bioRxiv 791699; doi: https://doi.org/10.1101/791699.\n",
      "EDAM topics: Gene expression, RNA, RNA-Seq\n",
      "EDAM operations: Quantification, Data retrieval, Expression analysis\n",
      "0.49\tTool name: MultiQC\n",
      "Description: MultiQC aggregates results from multiple bioinformatics analyses across many samples into a single report. It searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.\n",
      "EDAM topics: Sequencing, Bioinformatics, Sequence analysis, Genomics\n",
      "EDAM operations: Validation, Sequencing quality control\n",
      "0.45\tTool name: scanpy\n",
      "Description: Scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.\n",
      "EDAM topics: Gene expression, Cell biology, Genetics, Single-cell sequencing\n",
      "EDAM operations: Differential gene expression analysis, Principal component visualisation, Expression profile clustering\n",
      "0.45\tTool name: FASTQC\n",
      "Description: This tool aims to provide a QC report which can spot problems or biases which originate either in the sequencer or in the starting library material. It can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.\n",
      "EDAM topics: Sequencing, Data quality management, Sequence analysis\n",
      "EDAM operations: Sequence composition calculation, Sequencing quality control, Statistical calculation\n",
      "0.07\tTool name: cellranger\n"
     ]
    }
   ],
   "source": [
    "for rank in rank2:\n",
    "    print(f\"{rank['score']:.2f}\\t{meta_text[rank['corpus_id']]}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd256422",
   "metadata": {},
   "source": [
    "## Aliasing edams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "685306fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_yaml = load_biotools(Path(\"/home/ubuntu/cache/johan.yaml\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7a6d8ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'biocontainers': 'samtools',\n",
       " 'biotools': 'samtools',\n",
       " 'bunya': ['1.13-gcc-10.3.0',\n",
       "  '1.13-gcc-11.3.0',\n",
       "  '1.16.1-gcc-11.3.0',\n",
       "  '1.18-gcc-12.3.0                  (D)'],\n",
       " 'description': 'SAMtools and BCFtools are widely used programs for processing and analysing high-throughput sequencing data. They include tools for file format conversion and manipulation, sorting, querying, statistics, variant calling, and effect analysis amongst other methods.',\n",
       " 'edam-inputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       " 'edam-operations': ['Data filtering',\n",
       "  'Visualisation',\n",
       "  'Data formatting',\n",
       "  'Indexing',\n",
       "  'Data parsing',\n",
       "  'Data sorting',\n",
       "  'Data editing'],\n",
       " 'edam-outputs': [{'formats': ['CRAM', 'SAM', 'BAM'], 'term': 'BAM'}],\n",
       " 'edam-topics': ['Mapping',\n",
       "  'Sequence analysis',\n",
       "  'Sequencing',\n",
       "  'Rare diseases'],\n",
       " 'galaxy': [{'description': 'Samtools markdup: marks duplicate alignments',\n",
       "   'title': 'Samtools markdup 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_markdup%2Fsamtools_markdup%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools view: - reformat, filter, or subsample SAM, BAM or CRAM',\n",
       "   'title': 'Samtools view 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_view%2Fsamtools_view%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools split: BAM dataset on readgroups',\n",
       "   'title': 'Samtools split 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_split%2Fsamtools_split%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Slice: BAM by genomic regions',\n",
       "   'title': 'Slice 2.0.6',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_slice_bam%2Fsamtools_slice_bam%2F2.0.6'},\n",
       "  {'description': 'Samtools reheader: copy SAM/BAM header between datasets',\n",
       "   'title': 'Samtools reheader 2.0.6',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_reheader%2Fsamtools_reheader%2F2.0.6'},\n",
       "  {'description': 'Samtools phase: call and phase heterozygous SNPs',\n",
       "   'title': 'Samtools phase 2.0.2',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_phase%2Fsamtools_phase%2F2.0.2'},\n",
       "  {'description': 'Samtools mpileup: multi-way pileup of variants',\n",
       "   'title': 'Samtools mpileup 2.2.0',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_mpileup%2Fsamtools_mpileup%2F2.2.0'},\n",
       "  {'description': 'Samtools merge: merge multiple sorted alignment files',\n",
       "   'title': 'Samtools merge 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_merge%2Fsamtools_merge%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools idxstats: reports stats of the BAM index file',\n",
       "   'title': 'Samtools idxstats 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_idxstats%2Fsamtools_idxstats%2F2.0.8'},\n",
       "  {'description': 'Samtools fastx: extract FASTA or FASTQ from alignment files',\n",
       "   'title': 'Samtools fastx 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fastx%2Fsamtools_fastx%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools depth: compute the depth at each position or region',\n",
       "   'title': 'Samtools depth 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_depth%2Fsamtools_depth%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Samtools coverage: Produces a histogram or table of coverage per chromosome',\n",
       "   'title': 'Samtools coverage 1.22+galaxy3',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_coverage%2Fsamtools_coverage%2F1.22%2Bgalaxy3'},\n",
       "  {'description': 'Samtools calmd: recalculate MD/NM tags',\n",
       "   'title': 'Samtools calmd 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_calmd%2Fsamtools_calmd%2F2.0.7'},\n",
       "  {'description': 'Samtools bedcov: calculate read depth for a set of genomic intervals',\n",
       "   'title': 'Samtools bedcov 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_bedcov%2Fsamtools_bedcov%2F2.0.7'},\n",
       "  {'description': 'BAM-to-SAM: convert BAM to SAM',\n",
       "   'title': 'BAM-to-SAM 2.0.7',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbam_to_sam%2Fbam_to_sam%2F2.0.7'},\n",
       "  {'description': 'Samtools stats: generate statistics for BAM dataset',\n",
       "   'title': 'Samtools stats 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_stats%2Fsamtools_stats%2F2.0.8'},\n",
       "  {'description': 'Samtools sort: order of storing aligned sequences',\n",
       "   'title': 'Samtools sort 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_sort%2Fsamtools_sort%2F2.0.8'},\n",
       "  {'description': 'SAM-to-BAM: convert SAM to BAM',\n",
       "   'title': 'SAM-to-BAM 2.1.5',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsam_to_bam%2Fsam_to_bam%2F2.1.5'},\n",
       "  {'description': 'Samtools flagstat: tabulate descriptive stats for BAM datset',\n",
       "   'title': 'Samtools flagstat 2.0.8',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_flagstat%2Fsamtools_flagstat%2F2.0.8'},\n",
       "  {'description': 'Samtools fixmate: fill mate coordinates, ISIZE and mate related flags',\n",
       "   'title': 'Samtools fixmate 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_fixmate%2Fsamtools_fixmate%2F1.22%2Bgalaxy1'},\n",
       "  {'description': 'Filter SAM or BAM, output SAM or BAM: files on FLAG MAPQ RG LN or by region',\n",
       "   'title': 'Filter SAM or BAM, output SAM or BAM 1.8+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtool_filter2%2Fsamtool_filter2%2F1.8%2Bgalaxy1'},\n",
       "  {'description': 'RmDup: remove PCR duplicates',\n",
       "   'title': 'RmDup 2.0.1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsamtools_rmdup%2Fsamtools_rmdup%2F2.0.1'},\n",
       "  {'description': 'samtools BAM to CRAM: convert BAM alignments to CRAM format',\n",
       "   'title': 'samtools BAM to CRAM 1.22+galaxy1',\n",
       "   'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsamtools_bam_to_cram%2Fsamtools_bam_to_cram%2F1.22%2Bgalaxy1'}],\n",
       " 'homepage': 'http://www.htslib.org/',\n",
       " 'id': 'samtools',\n",
       " 'license': 'MIT',\n",
       " 'name': 'SAMTools',\n",
       " 'nci-gadi': None,\n",
       " 'nci-if89': ['1.18', '1.19.2'],\n",
       " 'pawsey': ['1.15--h3843a85_0'],\n",
       " 'publications': [{'title': 'The Sequence Alignment/Map format and SAMtools',\n",
       "   'url': 'https://doi.org/10.1093/bioinformatics/btp352'},\n",
       "  {'title': 'Twelve years of SAMtools and BCFtools',\n",
       "   'url': 'https://doi.org/10.1093/gigascience/giab008'},\n",
       "  {'title': 'A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data',\n",
       "   'url': 'https://doi.org/10.1093/bioinformatics/btr509'}],\n",
       " 'resources': [{'title': nan, 'url': nan}]}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_yaml.get(\"samtools\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0ac64bb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "collected_data = []\n",
    "for tool_id, tool_data in all_yaml.items():\n",
    "    entry = {\n",
    "        \"id\": tool_id,\n",
    "        \"description\": tool_data.get(\"description\", \"\"),\n",
    "        \"edam_inputs\": tool_data.get(\"edam-inputs\", []),\n",
    "        \"edam_outputs\": tool_data.get(\"edam-outputs\", []),\n",
    "        \"edam_operations\": tool_data.get(\"edam-operations\", []),\n",
    "        \"edam_topics\": tool_data.get(\"edam-topics\", []),\n",
    "    }\n",
    "    collected_data.append(entry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "9d00e4a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'3D profile generation',\n",
       " 'Aggregation',\n",
       " 'Alignment',\n",
       " 'Alternative splicing prediction',\n",
       " 'Analysis',\n",
       " 'Ancestral reconstruction',\n",
       " 'Annotation',\n",
       " 'Antimicrobial resistance prediction',\n",
       " 'Backbone modelling',\n",
       " 'Base-calling',\n",
       " 'Bisulfite mapping',\n",
       " 'Blind peptide database search',\n",
       " 'Box-Whisker plot plotting',\n",
       " 'Calculation',\n",
       " 'Cell migration analysis',\n",
       " 'Chimera detection',\n",
       " 'Chromatographic alignment',\n",
       " 'Classification',\n",
       " 'Clustering',\n",
       " 'Coding region prediction',\n",
       " 'Comparison',\n",
       " 'Conversion',\n",
       " 'Copy number estimation',\n",
       " 'Correlation',\n",
       " 'Cross-assembly',\n",
       " 'DMR identification',\n",
       " 'DNA barcoding',\n",
       " 'DNA mapping',\n",
       " 'DNA transcription',\n",
       " 'DNA translation',\n",
       " 'Data editing',\n",
       " 'Data filtering',\n",
       " 'Data formatting',\n",
       " 'Data handling',\n",
       " 'Data parsing',\n",
       " 'Data retrieval',\n",
       " 'Data sorting',\n",
       " 'Database search',\n",
       " 'De-novo assembly',\n",
       " 'Deisotoping',\n",
       " 'Demultiplexing',\n",
       " 'Dendrogram visualisation',\n",
       " 'Deposition',\n",
       " 'Design',\n",
       " 'Differential binding analysis',\n",
       " 'Differential gene expression analysis',\n",
       " 'Differential gene expression profiling',\n",
       " 'Differential protein expression analysis',\n",
       " 'Differential protein expression profiling',\n",
       " 'Dimensionality reduction',\n",
       " 'Dot plot plotting',\n",
       " 'Editing',\n",
       " 'Enrichment analysis',\n",
       " 'Essential dynamics',\n",
       " 'Exonic splicing enhancer prediction',\n",
       " 'Expression analysis',\n",
       " 'Expression correlation analysis',\n",
       " 'Expression profile clustering',\n",
       " 'Feature extraction',\n",
       " 'Filtering',\n",
       " 'Fold recognition',\n",
       " 'Format validation',\n",
       " 'Formatting',\n",
       " 'Functional clustering',\n",
       " 'Gene expression analysis',\n",
       " 'Gene expression profiling',\n",
       " 'Gene functional annotation',\n",
       " 'Gene prediction',\n",
       " 'Gene regulatory network analysis',\n",
       " 'Gene set testing',\n",
       " 'Gene-set enrichment analysis',\n",
       " 'Generation',\n",
       " 'Genetic mapping',\n",
       " 'Genetic variation analysis',\n",
       " 'Genome alignment',\n",
       " 'Genome annotation',\n",
       " 'Genome assembly',\n",
       " 'Genome comparison',\n",
       " 'Genome indexing',\n",
       " 'Genome visualisation',\n",
       " 'Genotyping',\n",
       " 'Global alignment',\n",
       " 'Haplotype mapping',\n",
       " 'Heat map generation',\n",
       " 'Homology-based gene prediction',\n",
       " 'Image analysis',\n",
       " 'Image annotation',\n",
       " 'Imputation',\n",
       " 'Indexing',\n",
       " 'Information retrieval',\n",
       " 'Ion counting',\n",
       " 'Isotope-coded protein label',\n",
       " 'Label-free quantification',\n",
       " 'Labeled quantification',\n",
       " 'Linkage disequilibrium calculation',\n",
       " 'Loading',\n",
       " 'Local alignment',\n",
       " 'Loop modelling',\n",
       " 'Mapping',\n",
       " 'Mapping assembly',\n",
       " 'Mass spectra calibration',\n",
       " 'Mass spectrum visualisation',\n",
       " 'Metabolic labeling',\n",
       " 'Metabolic network modelling',\n",
       " 'Metabolic pathway prediction',\n",
       " 'Methylation analysis',\n",
       " 'Methylation calling',\n",
       " 'Microscope image visualisation',\n",
       " 'Modelling and simulation',\n",
       " 'Molecular docking',\n",
       " 'Molecular dynamics',\n",
       " 'Molecular dynamics simulation',\n",
       " 'Molecular surface analysis',\n",
       " 'Multilocus sequence typing',\n",
       " 'Multiple sequence alignment',\n",
       " 'Natural product identification',\n",
       " 'Network analysis',\n",
       " 'Neurite measurement',\n",
       " 'Nucleic acid feature detection',\n",
       " 'Nucleic acid sequence analysis',\n",
       " 'Nucleic acid structure prediction',\n",
       " 'Nucleosome formation or exclusion sequence prediction',\n",
       " 'Optimisation and refinement',\n",
       " 'PCR primer design',\n",
       " 'PTM identification',\n",
       " 'PTM localisation',\n",
       " 'PTM site prediction',\n",
       " 'Pairwise sequence alignment',\n",
       " 'Parsing',\n",
       " 'Pathway analysis',\n",
       " 'Pathway or network analysis',\n",
       " 'Pathway or network visualisation',\n",
       " 'Peak calling',\n",
       " 'Peak detection',\n",
       " 'Peptide database search',\n",
       " 'Peptide identification',\n",
       " 'Phasing',\n",
       " 'Phylogenetic analysis',\n",
       " 'Phylogenetic footprinting',\n",
       " 'Phylogenetic inference',\n",
       " 'Phylogenetic reconstruction',\n",
       " 'Phylogenetic tree analysis',\n",
       " 'Phylogenetic tree analysis (shape)',\n",
       " 'Phylogenetic tree bootstrapping',\n",
       " 'Phylogenetic tree distances calculation',\n",
       " 'Phylogenetic tree editing',\n",
       " 'Phylogenetic tree generation',\n",
       " 'Phylogenetic tree generation (from molecular sequences)',\n",
       " 'Phylogenetic tree generation (maximum likelihood and Bayesian methods)',\n",
       " 'Phylogenetic tree reconstruction',\n",
       " 'Phylogenetic tree topology analysis',\n",
       " 'Phylogenetic tree visualisation',\n",
       " 'Plasmid map drawing',\n",
       " 'PolyA signal detection',\n",
       " 'Polymorphism detection',\n",
       " 'Post-translation modification site prediction',\n",
       " 'Prediction and recognition',\n",
       " 'Prediction and recognition (protein)',\n",
       " 'Primer removal',\n",
       " 'Principal component plotting',\n",
       " 'Principal component visualisation',\n",
       " 'Protein disorder prediction',\n",
       " 'Protein feature detection',\n",
       " 'Protein folding analysis',\n",
       " 'Protein fragment weight comparison',\n",
       " 'Protein geometry calculation',\n",
       " 'Protein geometry validation',\n",
       " 'Protein identification',\n",
       " 'Protein property calculation',\n",
       " 'Protein quantification',\n",
       " 'Protein secondary structure prediction',\n",
       " 'Protein sequence analysis',\n",
       " 'Protein structure analysis',\n",
       " 'Protein structure prediction',\n",
       " 'Protein threading',\n",
       " 'Protein-ligand docking',\n",
       " 'Quantification',\n",
       " 'Query and retrieval',\n",
       " 'RNA-Seq analysis',\n",
       " 'RNA-Seq quantification',\n",
       " 'RNA-seq time series data analysis',\n",
       " 'Rarefaction',\n",
       " 'Read binning',\n",
       " 'Read depth analysis',\n",
       " 'Read mapping',\n",
       " 'Read pre-processing',\n",
       " 'Read summarisation',\n",
       " 'Regression analysis',\n",
       " 'Repeat sequence analysis',\n",
       " 'Repeat sequence detection',\n",
       " 'Retention time prediction',\n",
       " 'SNP annotation',\n",
       " 'SNP detection',\n",
       " 'Scaffold gap completion',\n",
       " 'Scaffolding',\n",
       " 'Scatter plot plotting',\n",
       " 'Sequence alignment',\n",
       " 'Sequence alignment analysis',\n",
       " 'Sequence alignment conversion',\n",
       " 'Sequence analysis',\n",
       " 'Sequence annotation',\n",
       " 'Sequence assembly',\n",
       " 'Sequence assembly validation',\n",
       " 'Sequence assembly visualisation',\n",
       " 'Sequence classification',\n",
       " 'Sequence cluster visualisation',\n",
       " 'Sequence clustering',\n",
       " 'Sequence comparison',\n",
       " 'Sequence composition calculation',\n",
       " 'Sequence contamination filtering',\n",
       " 'Sequence conversion',\n",
       " 'Sequence database search',\n",
       " 'Sequence distance matrix generation',\n",
       " 'Sequence editing',\n",
       " 'Sequence feature comparison',\n",
       " 'Sequence feature detection',\n",
       " 'Sequence file editing',\n",
       " 'Sequence generation',\n",
       " 'Sequence merging',\n",
       " 'Sequence motif recognition',\n",
       " 'Sequence read processing',\n",
       " 'Sequence similarity search',\n",
       " 'Sequence trimming',\n",
       " 'Sequence visualisation',\n",
       " 'Sequencing error detection',\n",
       " 'Sequencing quality control',\n",
       " 'Service composition',\n",
       " 'Service discovery',\n",
       " 'Service invocation',\n",
       " 'Service management',\n",
       " 'Side chain modelling',\n",
       " 'Simulation analysis',\n",
       " 'Single particle analysis',\n",
       " 'Species frequency estimation',\n",
       " 'Spectral analysis',\n",
       " 'Spectral library search',\n",
       " 'Spectrum calculation',\n",
       " 'Splice site prediction',\n",
       " 'Splice transcript prediction',\n",
       " 'Splicing analysis',\n",
       " 'Splicing model analysis',\n",
       " 'Split read mapping',\n",
       " 'Splitting',\n",
       " 'Standardisation and normalisation',\n",
       " 'Statistical calculation',\n",
       " 'Statistical inference',\n",
       " 'Statistical modelling',\n",
       " 'Structural variation detection',\n",
       " 'Structure alignment',\n",
       " 'Structure visualisation',\n",
       " 'Tag-based peptide identification',\n",
       " 'Target-Decoy',\n",
       " 'Taxonomic classification',\n",
       " 'Text annotation',\n",
       " 'Text mining',\n",
       " 'Trajectory visualization',\n",
       " 'Transcriptome assembly',\n",
       " 'Transposon prediction',\n",
       " 'Validation',\n",
       " 'Validation of peptide-spectrum matches',\n",
       " 'Variant calling',\n",
       " 'Variant classification',\n",
       " 'Variant effect prediction',\n",
       " 'Variant filtering',\n",
       " 'Variant pattern analysis',\n",
       " 'Variant prioritisation',\n",
       " 'Visualisation',\n",
       " 'de Novo sequencing',\n",
       " 'iTRAQ',\n",
       " 'k-mer counting',\n",
       " 'miRNA expression analysis',\n",
       " 'miRNA target prediction',\n",
       " 'tRNA gene prediction'}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generate set of unique edam operations\n",
    "x = { op\n",
    "        for tool in collected_data\n",
    "        for op in (tool.get(\"edam_operations\") or [])\n",
    "}\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "65fd7edd",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pronto import Ontology\n",
    "from bio_mcp.cache.embeddings import load_edam_obo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "06d83a6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "obo = load_edam_obo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "b04fbb0a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'EDAM_data:0005': {'id': 'EDAM_data:0005',\n",
       "  'name': 'Resource type',\n",
       "  'def': '\"A type of computational resource used in bioinformatics.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0006': {'id': 'EDAM_data:0006',\n",
       "  'name': 'Data',\n",
       "  'def': '\"Information, represented in an information artefact (data record) that is \\'understandable\\' by dedicated computational tools that can use the data as input or produce it as output.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0007': {'id': 'EDAM_data:0007',\n",
       "  'name': 'Tool',\n",
       "  'def': '\"A bioinformatics package or tool, e.g. a standalone application or web service.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0581': {'id': 'EDAM_data:0581',\n",
       "  'name': 'Database',\n",
       "  'def': '\"A digital data archive typically based around a relational model but sometimes using an object-oriented, tree or graph-based model.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0582': {'id': 'EDAM_data:0582',\n",
       "  'name': 'Ontology',\n",
       "  'def': '\"An ontology of biological or bioinformatics concepts and relations, a controlled vocabulary, structured glossary etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0583': {'id': 'EDAM_data:0583',\n",
       "  'name': 'Directory metadata',\n",
       "  'def': '\"A directory on disk from which files are read.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0831': {'id': 'EDAM_data:0831',\n",
       "  'name': 'MeSH vocabulary',\n",
       "  'def': '\"Controlled vocabulary from National Library of Medicine. The MeSH thesaurus is used to index articles in biomedical journals for the Medline/PubMED databases.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0832': {'id': 'EDAM_data:0832',\n",
       "  'name': 'HGNC vocabulary',\n",
       "  'def': '\"Controlled vocabulary for gene names (symbols) from HUGO Gene Nomenclature Committee.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0835': {'id': 'EDAM_data:0835',\n",
       "  'name': 'UMLS vocabulary',\n",
       "  'def': '\"Compendium of controlled vocabularies for the biomedical domain (Unified Medical Language System).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0842': {'id': 'EDAM_data:0842',\n",
       "  'name': 'Identifier',\n",
       "  'def': '\"A text token, number or something else which identifies an entity, but which may not be persistent (stable) or unique (the same identifier may identify multiple things).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0843': {'id': 'EDAM_data:0843',\n",
       "  'name': 'Database entry',\n",
       "  'def': '\"An entry (retrievable via URL) from a biological database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0844': {'id': 'EDAM_data:0844',\n",
       "  'name': 'Molecular mass',\n",
       "  'def': '\"Mass of a molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0845': {'id': 'EDAM_data:0845',\n",
       "  'name': 'Molecular charge',\n",
       "  'def': '\"Net charge of a molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0846': {'id': 'EDAM_data:0846',\n",
       "  'name': 'Chemical formula',\n",
       "  'def': '\"A specification of a chemical structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0847': {'id': 'EDAM_data:0847',\n",
       "  'name': 'QSAR descriptor',\n",
       "  'def': '\"A QSAR quantitative descriptor (name-value pair) of chemical structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0848': {'id': 'EDAM_data:0848',\n",
       "  'name': 'Raw sequence',\n",
       "  'def': '\"A raw molecular sequence (string of characters) which might include ambiguity, unknown positions and non-sequence characters.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0849': {'id': 'EDAM_data:0849',\n",
       "  'name': 'Sequence record',\n",
       "  'def': '\"A molecular sequence and associated metadata.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0850': {'id': 'EDAM_data:0850',\n",
       "  'name': 'Sequence set',\n",
       "  'def': '\"A collection of multiple molecular sequences and associated metadata that do not (typically) correspond to molecular sequence database records or entries and which (typically) are derived from some analytical method.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0851': {'id': 'EDAM_data:0851',\n",
       "  'name': 'Sequence mask character',\n",
       "  'def': '\"A character used to replace (mask) other characters in a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0852': {'id': 'EDAM_data:0852',\n",
       "  'name': 'Sequence mask type',\n",
       "  'def': '\"A label (text token) describing the type of sequence masking to perform.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0853': {'id': 'EDAM_data:0853',\n",
       "  'name': 'DNA sense specification',\n",
       "  'def': '\"The strand of a DNA sequence (forward or reverse).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0854': {'id': 'EDAM_data:0854',\n",
       "  'name': 'Sequence length specification',\n",
       "  'def': '\"A specification of sequence length(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0855': {'id': 'EDAM_data:0855',\n",
       "  'name': 'Sequence metadata',\n",
       "  'def': '\"Basic or general information concerning molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0856': {'id': 'EDAM_data:0856',\n",
       "  'name': 'Sequence feature source',\n",
       "  'def': '\"How the annotation of a sequence feature (for example in EMBL or Swiss-Prot) was derived.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0857': {'id': 'EDAM_data:0857',\n",
       "  'name': 'Database hits (sequence)',\n",
       "  'def': '\"A report of sequence hits and associated data from searching a sequence database (for example a BLAST search). This will typically include a list of scores (often with statistical evaluation) and a set of alignments for the hits.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0858': {'id': 'EDAM_data:0858',\n",
       "  'name': 'Database hits (secondary)',\n",
       "  'def': '\"A report of hits from a search of a protein secondary or domain database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0859': {'id': 'EDAM_data:0859',\n",
       "  'name': 'Sequence signature model',\n",
       "  'def': '\"Data files used by motif or profile methods.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0860': {'id': 'EDAM_data:0860',\n",
       "  'name': 'Sequence signature',\n",
       "  'def': '\"A classifier of sequences such as a sequence motif, profile or other diagnostic element.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0861': {'id': 'EDAM_data:0861',\n",
       "  'name': 'Sequence alignment (words)',\n",
       "  'def': '\"Alignment of exact matches between subsequences (words) within two or more molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0862': {'id': 'EDAM_data:0862',\n",
       "  'name': 'Dotplot',\n",
       "  'def': '\"A dotplot of sequence similarities identified from word-matching or character comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0863': {'id': 'EDAM_data:0863',\n",
       "  'name': 'Sequence alignment',\n",
       "  'def': '\"Alignment of multiple molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0864': {'id': 'EDAM_data:0864',\n",
       "  'name': 'Sequence alignment parameter',\n",
       "  'def': '\"Some simple value controlling a sequence alignment (or similar \\'match\\') operation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0865': {'id': 'EDAM_data:0865',\n",
       "  'name': 'Sequence similarity score',\n",
       "  'def': '\"A value representing molecular sequence similarity.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0866': {'id': 'EDAM_data:0866',\n",
       "  'name': 'Sequence alignment metadata',\n",
       "  'def': '\"Report of general information on a sequence alignment, typically include a description, sequence identifiers and alignment score.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0867': {'id': 'EDAM_data:0867',\n",
       "  'name': 'Sequence alignment report',\n",
       "  'def': '\"An informative report of molecular sequence alignment-derived data or metadata.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0868': {'id': 'EDAM_data:0868',\n",
       "  'name': 'Sequence profile alignment',\n",
       "  'def': '\"A profile-profile alignment (each profile typically representing a sequence alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0869': {'id': 'EDAM_data:0869',\n",
       "  'name': 'Sequence-profile alignment',\n",
       "  'def': '\"Alignment of one or more molecular sequence(s) to one or more sequence profile(s) (each profile typically representing a sequence alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0870': {'id': 'EDAM_data:0870',\n",
       "  'name': 'Sequence distance matrix',\n",
       "  'def': '\"A matrix of estimated evolutionary distance between molecular sequences, such as is suitable for phylogenetic tree calculation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0871': {'id': 'EDAM_data:0871',\n",
       "  'name': 'Phylogenetic character data',\n",
       "  'def': '\"Basic character data from which a phylogenetic tree may be generated.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0872': {'id': 'EDAM_data:0872',\n",
       "  'name': 'Phylogenetic tree',\n",
       "  'def': '\"The raw data (not just an image) from which a phylogenetic tree is directly generated or plotted, such as topology, lengths (in time or in expected amounts of variance) and a confidence interval for each length.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0874': {'id': 'EDAM_data:0874',\n",
       "  'name': 'Comparison matrix',\n",
       "  'def': '\"Matrix of integer or floating point numbers for amino acid or nucleotide sequence comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0875': {'id': 'EDAM_data:0875',\n",
       "  'name': 'Protein topology',\n",
       "  'def': '\"Predicted or actual protein topology represented as a string of protein secondary structure elements.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0876': {'id': 'EDAM_data:0876',\n",
       "  'name': 'Protein features (secondary structure)',\n",
       "  'def': '\"Secondary structure assignment (predicted or real) of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0877': {'id': 'EDAM_data:0877',\n",
       "  'name': 'Protein features (super-secondary)',\n",
       "  'def': '\"A report of predicted or actual super-secondary structure of protein sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0878': {'id': 'EDAM_data:0878',\n",
       "  'name': 'Secondary structure alignment (protein)',\n",
       "  'def': '\"Alignment of the (1D representations of) secondary structure of two or more proteins.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0879': {'id': 'EDAM_data:0879',\n",
       "  'name': 'Secondary structure alignment metadata (protein)',\n",
       "  'def': '\"An informative report on protein secondary structure alignment-derived data or metadata.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0880': {'id': 'EDAM_data:0880',\n",
       "  'name': 'RNA secondary structure record',\n",
       "  'def': '\"An informative report of secondary structure (predicted or real) of an RNA molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0881': {'id': 'EDAM_data:0881',\n",
       "  'name': 'Secondary structure alignment (RNA)',\n",
       "  'def': '\"Alignment of the (1D representations of) secondary structure of two or more RNA molecules.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0882': {'id': 'EDAM_data:0882',\n",
       "  'name': 'Secondary structure alignment metadata (RNA)',\n",
       "  'def': '\"An informative report of RNA secondary structure alignment-derived data or metadata.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0883': {'id': 'EDAM_data:0883',\n",
       "  'name': 'Structure',\n",
       "  'def': '\"3D coordinate and associated data for a macromolecular tertiary (3D) structure or part of a structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0884': {'id': 'EDAM_data:0884',\n",
       "  'name': 'Tertiary structure record',\n",
       "  'def': '\"An entry from a molecular tertiary (3D) structure database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0885': {'id': 'EDAM_data:0885',\n",
       "  'name': 'Database hits (structure)',\n",
       "  'def': '\"Results (hits) from searching a database of tertiary structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0886': {'id': 'EDAM_data:0886',\n",
       "  'name': 'Structure alignment',\n",
       "  'def': '\"Alignment (superimposition) of molecular tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0887': {'id': 'EDAM_data:0887',\n",
       "  'name': 'Structure alignment report',\n",
       "  'def': '\"An informative report on molecular tertiary structure alignment-derived data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0888': {'id': 'EDAM_data:0888',\n",
       "  'name': 'Structure similarity score',\n",
       "  'def': '\"A value representing molecular structure similarity, measured from structure alignment or some other type of structure comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0889': {'id': 'EDAM_data:0889',\n",
       "  'name': 'Structural (3D) profile',\n",
       "  'def': '\"Some type of structural (3D) profile or template (representing a structure or structure alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0890': {'id': 'EDAM_data:0890',\n",
       "  'name': 'Structural (3D) profile alignment',\n",
       "  'def': '\"A 3D profile-3D profile alignment (each profile representing structures or a structure alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0891': {'id': 'EDAM_data:0891',\n",
       "  'name': 'Sequence-3D profile alignment',\n",
       "  'def': '\"An alignment of a sequence to a 3D profile (representing structures or a structure alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0892': {'id': 'EDAM_data:0892',\n",
       "  'name': 'Protein sequence-structure scoring matrix',\n",
       "  'def': '\"Matrix of values used for scoring sequence-structure compatibility.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0893': {'id': 'EDAM_data:0893',\n",
       "  'name': 'Sequence-structure alignment',\n",
       "  'def': '\"An alignment of molecular sequence to structure (from threading sequence(s) through 3D structure or representation of structure(s)).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0894': {'id': 'EDAM_data:0894',\n",
       "  'name': 'Amino acid annotation',\n",
       "  'def': '\"An informative report about a specific amino acid.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0895': {'id': 'EDAM_data:0895',\n",
       "  'name': 'Peptide annotation',\n",
       "  'def': '\"An informative report about a specific peptide.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0896': {'id': 'EDAM_data:0896',\n",
       "  'name': 'Protein report',\n",
       "  'def': '\"An informative report about one or more specific protein molecules or protein structural domains, derived from analysis of primary (sequence or structural) data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0897': {'id': 'EDAM_data:0897',\n",
       "  'name': 'Protein property',\n",
       "  'def': '\"A report of primarily non-positional data describing intrinsic physical, chemical or other properties of a protein molecule or model.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0899': {'id': 'EDAM_data:0899',\n",
       "  'name': 'Protein features (3D motif)',\n",
       "  'def': '\"An informative report on the 3D structural motifs in a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0900': {'id': 'EDAM_data:0900',\n",
       "  'name': 'Protein domain classification',\n",
       "  'def': '\"Data concerning the classification of the sequences and/or structures of protein structural domain(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0901': {'id': 'EDAM_data:0901',\n",
       "  'name': 'Protein features (domains)',\n",
       "  'def': '\"Summary of structural domains or 3D folds in a protein or polypeptide chain.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0902': {'id': 'EDAM_data:0902',\n",
       "  'name': 'Protein architecture report',\n",
       "  'def': '\"An informative report on architecture (spatial arrangement of secondary structure) of a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0903': {'id': 'EDAM_data:0903',\n",
       "  'name': 'Protein folding report',\n",
       "  'def': '\"A report on an analysis or model of protein folding properties, folding pathways, residues or sites that are key to protein folding, nucleation or stabilization centers etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0904': {'id': 'EDAM_data:0904',\n",
       "  'name': 'Protein features (mutation)',\n",
       "  'def': '\"Data on the effect of (typically point) mutation on protein folding, stability, structure and function.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0905': {'id': 'EDAM_data:0905',\n",
       "  'name': 'Protein interaction raw data',\n",
       "  'def': '\"Protein-protein interaction data from for example yeast two-hybrid analysis, protein microarrays, immunoaffinity chromatography followed by mass spectrometry, phage display etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0906': {'id': 'EDAM_data:0906',\n",
       "  'name': 'Protein interaction',\n",
       "  'def': '\"An informative report on the interactions (predicted or known) of a protein, protein domain or part of a protein with itself or some other molecule(s), which might be another protein, nucleic acid or some other ligand.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0907': {'id': 'EDAM_data:0907',\n",
       "  'name': 'Protein family',\n",
       "  'def': '\"An informative report on a specific protein family or other group of classified proteins.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0909': {'id': 'EDAM_data:0909',\n",
       "  'name': 'Vmax',\n",
       "  'def': '\"The maximum initial velocity or rate of a reaction. It is the limiting velocity as substrate concentrations get very large.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0910': {'id': 'EDAM_data:0910',\n",
       "  'name': 'Km',\n",
       "  'def': '\"Km is the concentration (usually in Molar units) of substrate that leads to half-maximal velocity of an enzyme-catalysed reaction.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0911': {'id': 'EDAM_data:0911',\n",
       "  'name': 'Nucleotide base annotation',\n",
       "  'def': '\"An informative report about a specific nucleotide base.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0912': {'id': 'EDAM_data:0912',\n",
       "  'name': 'Nucleic acid property',\n",
       "  'def': '\"A report of primarily non-positional data describing intrinsic physical, chemical or other properties of a nucleic acid molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0914': {'id': 'EDAM_data:0914',\n",
       "  'name': 'Codon usage report',\n",
       "  'def': '\"Data derived from analysis of codon usage (typically a codon usage table) of DNA sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0916': {'id': 'EDAM_data:0916',\n",
       "  'name': 'Gene annotation',\n",
       "  'def': '\"An informative report on a particular locus, gene, gene system or groups of genes.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0917': {'id': 'EDAM_data:0917',\n",
       "  'name': 'Gene classification',\n",
       "  'def': '\"A report on the classification of nucleic acid / gene sequences according to the functional classification of their gene products.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0918': {'id': 'EDAM_data:0918',\n",
       "  'name': 'Nucleic acid features (variation)',\n",
       "  'def': '\"A report on stable, naturally occuring mutations in a nucleotide sequence including alleles, naturally occurring mutations such as single base nucleotide substitutions, deletions and insertions, RFLPs and other polymorphisms.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0919': {'id': 'EDAM_data:0919',\n",
       "  'name': 'Gene annotation (chromosome)',\n",
       "  'def': '\"An informative report on a specific chromosome.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0920': {'id': 'EDAM_data:0920',\n",
       "  'name': 'Genotype/phenotype annotation',\n",
       "  'def': '\"An informative report on the set of genes (or allelic forms) present in an individual, organism or cell and associated with a specific physical characteristic, or a report concerning an organisms traits and phenotypes.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0922': {'id': 'EDAM_data:0922',\n",
       "  'name': 'Nucleic acid features (primers)',\n",
       "  'def': '\"Report on matches to PCR primers and hybridization oligos in a nucleic acid sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0923': {'id': 'EDAM_data:0923',\n",
       "  'name': 'Experiment annotation (PCR assay data)',\n",
       "  'def': '\"Data on a PCR assay or electronic / virtual PCR.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0924': {'id': 'EDAM_data:0924',\n",
       "  'name': 'Sequence trace',\n",
       "  'def': '\"Fluorescence trace data generated by an automated DNA sequencer, which can be interprted as a molecular sequence (reads), given associated sequencing metadata such as base-call quality scores.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0925': {'id': 'EDAM_data:0925',\n",
       "  'name': 'Sequence assembly',\n",
       "  'def': '\"An assembly of fragments of a (typically genomic) DNA sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0926': {'id': 'EDAM_data:0926',\n",
       "  'name': 'Radiation Hybrid (RH) scores',\n",
       "  'def': '\"Radiation hybrid scores (RH) scores for one or more markers.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0927': {'id': 'EDAM_data:0927',\n",
       "  'name': 'Gene annotation (linkage)',\n",
       "  'def': '\"An informative report on the linkage of alleles.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0928': {'id': 'EDAM_data:0928',\n",
       "  'name': 'Gene expression profile',\n",
       "  'def': '\"Data quantifying the level of expression of (typically) multiple genes, derived for example from microarray experiments.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0931': {'id': 'EDAM_data:0931',\n",
       "  'name': 'Experiment annotation (microarray)',\n",
       "  'def': '\"Information on a microarray experiment such as conditions, protocol, sample:data relationships etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0932': {'id': 'EDAM_data:0932',\n",
       "  'name': 'Oligonucleotide probe data',\n",
       "  'def': '\"Data on oligonucleotide probes (typically for use with DNA microarrays).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0933': {'id': 'EDAM_data:0933',\n",
       "  'name': 'SAGE experimental data',\n",
       "  'def': '\"Output from a serial analysis of gene expression (SAGE) experiment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0934': {'id': 'EDAM_data:0934',\n",
       "  'name': 'MPSS experimental data',\n",
       "  'def': '\"Massively parallel signature sequencing (MPSS) data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0935': {'id': 'EDAM_data:0935',\n",
       "  'name': 'SBS experimental data',\n",
       "  'def': '\"Sequencing by synthesis (SBS) data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0936': {'id': 'EDAM_data:0936',\n",
       "  'name': 'Sequence tag profile (with gene assignment)',\n",
       "  'def': '\"Tag to gene assignments (tag mapping) of SAGE, MPSS and SBS data. Typically this is the sequencing-based expression profile annotated with gene identifiers.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0937': {'id': 'EDAM_data:0937',\n",
       "  'name': 'Protein X-ray crystallographic data',\n",
       "  'def': '\"X-ray crystallography data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0938': {'id': 'EDAM_data:0938',\n",
       "  'name': 'Protein NMR data',\n",
       "  'def': '\"Protein nuclear magnetic resonance (NMR) raw data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0939': {'id': 'EDAM_data:0939',\n",
       "  'name': 'Protein circular dichroism (CD) spectroscopic data',\n",
       "  'def': '\"Protein secondary structure from protein coordinate or circular dichroism (CD) spectroscopic data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0940': {'id': 'EDAM_data:0940',\n",
       "  'name': 'Electron microscopy volume map',\n",
       "  'def': '\"Volume map data from electron microscopy.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0941': {'id': 'EDAM_data:0941',\n",
       "  'name': 'Electron microscopy model',\n",
       "  'def': '\"Annotation on a structural 3D model (volume map) from electron microscopy.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0942': {'id': 'EDAM_data:0942',\n",
       "  'name': '2D PAGE image',\n",
       "  'def': '\"Two-dimensional gel electrophoresis image.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0943': {'id': 'EDAM_data:0943',\n",
       "  'name': 'Mass spectrometry spectra',\n",
       "  'def': '\"Spectra from mass spectrometry.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0944': {'id': 'EDAM_data:0944',\n",
       "  'name': 'Peptide mass fingerprint',\n",
       "  'def': '\"A set of peptide masses (peptide mass fingerprint) from mass spectrometry.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0945': {'id': 'EDAM_data:0945',\n",
       "  'name': 'Peptide identification',\n",
       "  'def': '\"Protein or peptide identifications with evidence supporting the identifications, typically from comparing a peptide mass fingerprint (from mass spectrometry) to a sequence database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0946': {'id': 'EDAM_data:0946',\n",
       "  'name': 'Pathway or network annotation',\n",
       "  'def': '\"An informative report about a specific biological pathway or network, typically including a map (diagram) of the pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0947': {'id': 'EDAM_data:0947',\n",
       "  'name': 'Biological pathway map',\n",
       "  'def': '\"A map (typically a diagram) of a biological pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0948': {'id': 'EDAM_data:0948',\n",
       "  'name': 'Data resource definition',\n",
       "  'def': '\"A definition of a data resource serving one or more types of data, including metadata and links to the resource or data proper.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0949': {'id': 'EDAM_data:0949',\n",
       "  'name': 'Workflow metadata',\n",
       "  'def': '\"Basic information, annotation or documentation concerning a workflow (but not the workflow itself).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0950': {'id': 'EDAM_data:0950',\n",
       "  'name': 'Biological model',\n",
       "  'def': '\"A biological model which can be represented in mathematical terms.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0951': {'id': 'EDAM_data:0951',\n",
       "  'name': 'Statistical estimate score',\n",
       "  'def': '\"A value representing estimated statistical significance of some observed data; typically sequence database hits.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0952': {'id': 'EDAM_data:0952',\n",
       "  'name': 'EMBOSS database resource definition',\n",
       "  'def': '\"Resource definition for an EMBOSS database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0953': {'id': 'EDAM_data:0953',\n",
       "  'name': 'Version information',\n",
       "  'def': '\"Information on a version of software or data, for example name, version number and release date.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0954': {'id': 'EDAM_data:0954',\n",
       "  'name': 'Database cross-mapping',\n",
       "  'def': '\"A mapping of the accession numbers (or other database identifier) of entries between (typically) two biological or biomedical databases.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0955': {'id': 'EDAM_data:0955',\n",
       "  'name': 'Data index',\n",
       "  'def': '\"An index of data of biological relevance.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0956': {'id': 'EDAM_data:0956',\n",
       "  'name': 'Data index report',\n",
       "  'def': '\"A report of an analysis of an index of biological data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0957': {'id': 'EDAM_data:0957',\n",
       "  'name': 'Database metadata',\n",
       "  'def': '\"Basic information on bioinformatics database(s) or other data sources such as name, type, description, URL etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0958': {'id': 'EDAM_data:0958',\n",
       "  'name': 'Tool metadata',\n",
       "  'def': '\"Basic information about one or more bioinformatics applications or packages, such as name, type, description, or other documentation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0959': {'id': 'EDAM_data:0959',\n",
       "  'name': 'Job metadata',\n",
       "  'def': '\"Textual metadata on a submitted or completed job.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0960': {'id': 'EDAM_data:0960',\n",
       "  'name': 'User metadata',\n",
       "  'def': '\"Textual metadata on a software author or end-user, for example a person or other software.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0962': {'id': 'EDAM_data:0962',\n",
       "  'name': 'Small molecule annotation',\n",
       "  'def': '\"An informative report on a specific chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0963': {'id': 'EDAM_data:0963',\n",
       "  'name': 'Cell line annotation',\n",
       "  'def': '\"Report on a particular strain of organism cell line including plants, virus, fungi and bacteria. The data typically includes strain number, organism type, growth conditions, source and so on.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0964': {'id': 'EDAM_data:0964',\n",
       "  'name': 'Scent annotation',\n",
       "  'def': '\"An informative report about a specific scent.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0966': {'id': 'EDAM_data:0966',\n",
       "  'name': 'Ontology term',\n",
       "  'def': '\"A term (name) from an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0967': {'id': 'EDAM_data:0967',\n",
       "  'name': 'Ontology concept metadata',\n",
       "  'def': '\"Data concerning or derived from a concept from a biological ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0968': {'id': 'EDAM_data:0968',\n",
       "  'name': 'Keyword',\n",
       "  'def': '\"Keyword(s) or phrase(s) used (typically) for text-searching purposes.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0970': {'id': 'EDAM_data:0970',\n",
       "  'name': 'Bibliographic reference',\n",
       "  'def': '\"Bibliographic data that uniquely identifies a scientific article, book or other published material.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0971': {'id': 'EDAM_data:0971',\n",
       "  'name': 'Article',\n",
       "  'def': '\"A body of scientific text, typically a full text article from a scientific journal.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0972': {'id': 'EDAM_data:0972',\n",
       "  'name': 'Text mining report',\n",
       "  'def': '\"An abstract of the results of text mining.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0974': {'id': 'EDAM_data:0974',\n",
       "  'name': 'Entity identifier',\n",
       "  'def': '\"An identifier of a biological entity or phenomenon.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0975': {'id': 'EDAM_data:0975',\n",
       "  'name': 'Data resource identifier',\n",
       "  'def': '\"An identifier of a data resource.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0976': {'id': 'EDAM_data:0976',\n",
       "  'name': 'Identifier (typed)',\n",
       "  'def': '\"An identifier that identifies a particular type of data.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0977': {'id': 'EDAM_data:0977',\n",
       "  'name': 'Tool identifier',\n",
       "  'def': '\"An identifier of a bioinformatics tool, e.g. an application or web service.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0978': {'id': 'EDAM_data:0978',\n",
       "  'name': 'Discrete entity identifier',\n",
       "  'def': '\"Name or other identifier of a discrete entity (any biological thing with a distinct, discrete physical existence).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0979': {'id': 'EDAM_data:0979',\n",
       "  'name': 'Entity feature identifier',\n",
       "  'def': '\"Name or other identifier of an entity feature (a physical part or region of a discrete biological entity, or a feature that can be mapped to such a thing).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0980': {'id': 'EDAM_data:0980',\n",
       "  'name': 'Entity collection identifier',\n",
       "  'def': '\"Name or other identifier of a collection of discrete biological entities.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0981': {'id': 'EDAM_data:0981',\n",
       "  'name': 'Phenomenon identifier',\n",
       "  'def': '\"Name or other identifier of a physical, observable biological occurrence or event.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0982': {'id': 'EDAM_data:0982',\n",
       "  'name': 'Molecule identifier',\n",
       "  'def': '\"Name or other identifier of a molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0983': {'id': 'EDAM_data:0983',\n",
       "  'name': 'Atom identifier',\n",
       "  'def': '\"Identifier (e.g. character symbol) of a specific atom.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0984': {'id': 'EDAM_data:0984',\n",
       "  'name': 'Molecule name',\n",
       "  'def': '\"Name of a specific molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0985': {'id': 'EDAM_data:0985',\n",
       "  'name': 'Molecule type',\n",
       "  'def': '\"A label (text token) describing the type a molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:0986': {'id': 'EDAM_data:0986',\n",
       "  'name': 'Chemical identifier',\n",
       "  'def': '\"Unique identifier of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0987': {'id': 'EDAM_data:0987',\n",
       "  'name': 'Chromosome name',\n",
       "  'def': '\"Name of a chromosome.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0988': {'id': 'EDAM_data:0988',\n",
       "  'name': 'Peptide identifier',\n",
       "  'def': '\"Identifier of a peptide chain.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0989': {'id': 'EDAM_data:0989',\n",
       "  'name': 'Protein identifier',\n",
       "  'def': '\"Identifier of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0990': {'id': 'EDAM_data:0990',\n",
       "  'name': 'Compound name',\n",
       "  'def': '\"Unique name of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0991': {'id': 'EDAM_data:0991',\n",
       "  'name': 'Chemical registry number',\n",
       "  'def': '\"Unique registry number of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0992': {'id': 'EDAM_data:0992',\n",
       "  'name': 'Ligand identifier',\n",
       "  'def': '\"Code word for a ligand, for example from a PDB file.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0993': {'id': 'EDAM_data:0993',\n",
       "  'name': 'Drug identifier',\n",
       "  'def': '\"Identifier of a drug.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0994': {'id': 'EDAM_data:0994',\n",
       "  'name': 'Amino acid identifier',\n",
       "  'def': '\"Identifier of an amino acid.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0995': {'id': 'EDAM_data:0995',\n",
       "  'name': 'Nucleotide identifier',\n",
       "  'def': '\"Name or other identifier of a nucleotide.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0996': {'id': 'EDAM_data:0996',\n",
       "  'name': 'Monosaccharide identifier',\n",
       "  'def': '\"Identifier of a monosaccharide.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0997': {'id': 'EDAM_data:0997',\n",
       "  'name': 'Chemical name (ChEBI)',\n",
       "  'def': '\"Unique name from Chemical Entities of Biological Interest (ChEBI) of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0998': {'id': 'EDAM_data:0998',\n",
       "  'name': 'Chemical name (IUPAC)',\n",
       "  'def': '\"IUPAC recommended name of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:0999': {'id': 'EDAM_data:0999',\n",
       "  'name': 'Chemical name (INN)',\n",
       "  'def': '\"International Non-proprietary Name (INN or \\'generic name\\') of a chemical compound, assigned by the World Health Organization (WHO).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1000': {'id': 'EDAM_data:1000',\n",
       "  'name': 'Chemical name (brand)',\n",
       "  'def': '\"Brand name of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1001': {'id': 'EDAM_data:1001',\n",
       "  'name': 'Chemical name (synonymous)',\n",
       "  'def': '\"Synonymous name of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1002': {'id': 'EDAM_data:1002',\n",
       "  'name': 'Chemical registry number (CAS)',\n",
       "  'def': '\"CAS registry number of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1003': {'id': 'EDAM_data:1003',\n",
       "  'name': 'Chemical registry number (Beilstein)',\n",
       "  'def': '\"Beilstein registry number of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1004': {'id': 'EDAM_data:1004',\n",
       "  'name': 'Chemical registry number (Gmelin)',\n",
       "  'def': '\"Gmelin registry number of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1005': {'id': 'EDAM_data:1005',\n",
       "  'name': 'HET group name',\n",
       "  'def': '\"3-letter code word for a ligand (HET group) from a PDB file, for example ATP.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1006': {'id': 'EDAM_data:1006',\n",
       "  'name': 'Amino acid name',\n",
       "  'def': '\"String of one or more ASCII characters representing an amino acid.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1007': {'id': 'EDAM_data:1007',\n",
       "  'name': 'Nucleotide code',\n",
       "  'def': '\"String of one or more ASCII characters representing a nucleotide.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1008': {'id': 'EDAM_data:1008',\n",
       "  'name': 'Polypeptide chain ID',\n",
       "  'def': '\"Identifier of a polypeptide chain from a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1009': {'id': 'EDAM_data:1009',\n",
       "  'name': 'Protein name',\n",
       "  'def': '\"Name of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1010': {'id': 'EDAM_data:1010',\n",
       "  'name': 'Enzyme identifier',\n",
       "  'def': '\"Name or other identifier of an enzyme or record from a database of enzymes.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1011': {'id': 'EDAM_data:1011',\n",
       "  'name': 'EC number',\n",
       "  'def': '\"An Enzyme Commission (EC) number of an enzyme.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1012': {'id': 'EDAM_data:1012',\n",
       "  'name': 'Enzyme name',\n",
       "  'def': '\"Name of an enzyme.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1013': {'id': 'EDAM_data:1013',\n",
       "  'name': 'Restriction enzyme name',\n",
       "  'def': '\"Name of a restriction enzyme.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1014': {'id': 'EDAM_data:1014',\n",
       "  'name': 'Sequence position specification',\n",
       "  'def': '\"A specification (partial or complete) of one or more positions or regions of a molecular sequence or map.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1015': {'id': 'EDAM_data:1015',\n",
       "  'name': 'Sequence feature ID',\n",
       "  'def': '\"A unique identifier of molecular sequence feature, for example an ID of a feature that is unique within the scope of the GFF file.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1016': {'id': 'EDAM_data:1016',\n",
       "  'name': 'Sequence position',\n",
       "  'def': '\"A position of a single point (base or residue) in a sequence, or part of such a specification.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1017': {'id': 'EDAM_data:1017',\n",
       "  'name': 'Sequence range',\n",
       "  'def': '\"Specification of range(s) of sequence positions.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1018': {'id': 'EDAM_data:1018',\n",
       "  'name': 'Nucleic acid feature identifier',\n",
       "  'def': '\"Name or other identifier of an nucleic acid feature.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1019': {'id': 'EDAM_data:1019',\n",
       "  'name': 'Protein feature identifier',\n",
       "  'def': '\"Name or other identifier of a protein feature.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1020': {'id': 'EDAM_data:1020',\n",
       "  'name': 'Sequence feature key',\n",
       "  'def': '\"The type of a sequence feature, typically a term or accession from the Sequence Ontology, for example an EMBL or Swiss-Prot sequence feature key.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1021': {'id': 'EDAM_data:1021',\n",
       "  'name': 'Sequence feature qualifier',\n",
       "  'def': '\"Typically one of the EMBL or Swiss-Prot feature qualifiers.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1022': {'id': 'EDAM_data:1022',\n",
       "  'name': 'Sequence feature label',\n",
       "  'def': '\"Typically an EMBL or Swiss-Prot feature label.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1023': {'id': 'EDAM_data:1023',\n",
       "  'name': 'EMBOSS Uniform Feature Object',\n",
       "  'def': '\"The name of a sequence feature-containing entity adhering to the standard feature naming scheme used by all EMBOSS applications.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1024': {'id': 'EDAM_data:1024',\n",
       "  'name': 'Codon name',\n",
       "  'def': '\"String of one or more ASCII characters representing a codon.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1025': {'id': 'EDAM_data:1025',\n",
       "  'name': 'Gene identifier',\n",
       "  'def': '\"An identifier of a gene, such as a name/symbol or a unique identifier of a gene in a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1026': {'id': 'EDAM_data:1026',\n",
       "  'name': 'Gene symbol',\n",
       "  'def': '\"The short name of a gene; a single word that does not contain white space characters. It is typically derived from the gene name.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1027': {'id': 'EDAM_data:1027',\n",
       "  'name': 'Gene ID (NCBI)',\n",
       "  'def': '\"An NCBI unique identifier of a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1028': {'id': 'EDAM_data:1028',\n",
       "  'name': 'Gene identifier (NCBI RefSeq)',\n",
       "  'def': '\"An NCBI RefSeq unique identifier of a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1029': {'id': 'EDAM_data:1029',\n",
       "  'name': 'Gene identifier (NCBI UniGene)',\n",
       "  'def': '\"An NCBI UniGene unique identifier of a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1030': {'id': 'EDAM_data:1030',\n",
       "  'name': 'Gene identifier (Entrez)',\n",
       "  'def': '\"An Entrez unique identifier of a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1031': {'id': 'EDAM_data:1031',\n",
       "  'name': 'Gene ID (CGD)',\n",
       "  'def': '\"Identifier of a gene or feature from the CGD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1032': {'id': 'EDAM_data:1032',\n",
       "  'name': 'Gene ID (DictyBase)',\n",
       "  'def': '\"Identifier of a gene from DictyBase.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1033': {'id': 'EDAM_data:1033',\n",
       "  'name': 'Gene ID (Ensembl)',\n",
       "  'def': '\"Unique identifier for a gene (or other feature) from the Ensembl database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1034': {'id': 'EDAM_data:1034',\n",
       "  'name': 'Gene ID (SGD)',\n",
       "  'def': '\"Identifier of an entry from the SGD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1035': {'id': 'EDAM_data:1035',\n",
       "  'name': 'Gene ID (GeneDB)',\n",
       "  'def': '\"Identifier of a gene from the GeneDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1036': {'id': 'EDAM_data:1036',\n",
       "  'name': 'TIGR identifier',\n",
       "  'def': '\"Identifier of an entry from the TIGR database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1037': {'id': 'EDAM_data:1037',\n",
       "  'name': 'TAIR accession (gene)',\n",
       "  'def': '\"Identifier of an gene from the TAIR database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1038': {'id': 'EDAM_data:1038',\n",
       "  'name': 'Protein domain ID',\n",
       "  'def': '\"Identifier of a protein structural domain.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1039': {'id': 'EDAM_data:1039',\n",
       "  'name': 'SCOP domain identifier',\n",
       "  'def': '\"Identifier of a protein domain (or other node) from the SCOP database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1040': {'id': 'EDAM_data:1040',\n",
       "  'name': 'CATH domain ID',\n",
       "  'def': '\"Identifier of a protein domain from CATH.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1041': {'id': 'EDAM_data:1041',\n",
       "  'name': 'SCOP concise classification string (sccs)',\n",
       "  'def': '\"A SCOP concise classification string (sccs) is a compact representation of a SCOP domain classification.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1042': {'id': 'EDAM_data:1042',\n",
       "  'name': 'SCOP sunid',\n",
       "  'def': '\"Unique identifier (number) of an entry in the SCOP hierarchy, for example 33229.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1043': {'id': 'EDAM_data:1043',\n",
       "  'name': 'CATH node ID',\n",
       "  'def': '\"A code number identifying a node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1044': {'id': 'EDAM_data:1044',\n",
       "  'name': 'Kingdom name',\n",
       "  'def': '\"The name of a biological kingdom (Bacteria, Archaea, or Eukaryotes).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1045': {'id': 'EDAM_data:1045',\n",
       "  'name': 'Species name',\n",
       "  'def': '\"The name of a species (typically a taxonomic group) of organism.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1046': {'id': 'EDAM_data:1046',\n",
       "  'name': 'Strain name',\n",
       "  'def': '\"The name of a strain of an organism variant, typically a plant, virus or bacterium.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1047': {'id': 'EDAM_data:1047',\n",
       "  'name': 'URI',\n",
       "  'def': '\"A string of characters that name or otherwise identify a resource on the Internet.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1048': {'id': 'EDAM_data:1048',\n",
       "  'name': 'Database identifier',\n",
       "  'def': '\"An identifier of a biological or bioinformatics database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1049': {'id': 'EDAM_data:1049',\n",
       "  'name': 'Directory name',\n",
       "  'def': '\"The name of a directory.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1050': {'id': 'EDAM_data:1050',\n",
       "  'name': 'File name',\n",
       "  'def': '\"The name (or part of a name) of a file (of any type).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1051': {'id': 'EDAM_data:1051',\n",
       "  'name': 'Ontology name',\n",
       "  'def': '\"Name of an ontology of biological or bioinformatics concepts and relations.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1052': {'id': 'EDAM_data:1052',\n",
       "  'name': 'URL',\n",
       "  'def': '\"A Uniform Resource Locator (URL).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1053': {'id': 'EDAM_data:1053',\n",
       "  'name': 'URN',\n",
       "  'def': '\"A Uniform Resource Name (URN).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1055': {'id': 'EDAM_data:1055',\n",
       "  'name': 'LSID',\n",
       "  'def': '\"A Life Science Identifier (LSID) - a unique identifier of some data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1056': {'id': 'EDAM_data:1056',\n",
       "  'name': 'Database name',\n",
       "  'def': '\"The name of a biological or bioinformatics database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1057': {'id': 'EDAM_data:1057',\n",
       "  'name': 'Sequence database name',\n",
       "  'def': '\"The name of a molecular sequence database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1058': {'id': 'EDAM_data:1058',\n",
       "  'name': 'Enumerated file name',\n",
       "  'def': '\"The name of a file (of any type) with restricted possible values.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1059': {'id': 'EDAM_data:1059',\n",
       "  'name': 'File name extension',\n",
       "  'def': '\"The extension of a file name.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1060': {'id': 'EDAM_data:1060',\n",
       "  'name': 'File base name',\n",
       "  'def': '\"The base name of a file.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1061': {'id': 'EDAM_data:1061',\n",
       "  'name': 'QSAR descriptor name',\n",
       "  'def': '\"Name of a QSAR descriptor.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1062': {'id': 'EDAM_data:1062',\n",
       "  'name': 'Database entry identifier',\n",
       "  'def': '\"An identifier of an entry from a database where the same type of identifier is used for objects (data) of different semantic type.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1063': {'id': 'EDAM_data:1063',\n",
       "  'name': 'Sequence identifier',\n",
       "  'def': '\"An identifier of molecular sequence(s) or entries from a molecular sequence database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1064': {'id': 'EDAM_data:1064',\n",
       "  'name': 'Sequence set ID',\n",
       "  'def': '\"An identifier of a set of molecular sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1065': {'id': 'EDAM_data:1065',\n",
       "  'name': 'Sequence signature identifier',\n",
       "  'def': '\"Identifier of a sequence signature (motif or profile) for example from a database of sequence patterns.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1066': {'id': 'EDAM_data:1066',\n",
       "  'name': 'Sequence alignment ID',\n",
       "  'def': '\"Identifier of a molecular sequence alignment, for example a record from an alignment database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1067': {'id': 'EDAM_data:1067',\n",
       "  'name': 'Phylogenetic distance matrix identifier',\n",
       "  'def': '\"Identifier of a phylogenetic distance matrix.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1068': {'id': 'EDAM_data:1068',\n",
       "  'name': 'Phylogenetic tree ID',\n",
       "  'def': '\"Identifier of a phylogenetic tree for example from a phylogenetic tree database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1069': {'id': 'EDAM_data:1069',\n",
       "  'name': 'Comparison matrix identifier',\n",
       "  'def': '\"An identifier of a comparison matrix.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1070': {'id': 'EDAM_data:1070',\n",
       "  'name': 'Structure ID',\n",
       "  'def': '\"A unique and persistent identifier of a molecular tertiary structure, typically an entry from a structure database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1071': {'id': 'EDAM_data:1071',\n",
       "  'name': 'Structural (3D) profile ID',\n",
       "  'def': '\"Identifier or name of a structural (3D) profile or template (representing a structure or structure alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1072': {'id': 'EDAM_data:1072',\n",
       "  'name': 'Structure alignment ID',\n",
       "  'def': '\"Identifier of an entry from a database of tertiary structure alignments.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1073': {'id': 'EDAM_data:1073',\n",
       "  'name': 'Amino acid index ID',\n",
       "  'def': '\"Identifier of an index of amino acid physicochemical and biochemical property data.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1074': {'id': 'EDAM_data:1074',\n",
       "  'name': 'Protein interaction ID',\n",
       "  'def': '\"Identifier of a report of protein interactions from a protein interaction database (typically).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1075': {'id': 'EDAM_data:1075',\n",
       "  'name': 'Protein family identifier',\n",
       "  'def': '\"Identifier of a protein family.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1076': {'id': 'EDAM_data:1076',\n",
       "  'name': 'Codon usage table name',\n",
       "  'def': '\"Unique name of a codon usage table.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1077': {'id': 'EDAM_data:1077',\n",
       "  'name': 'Transcription factor identifier',\n",
       "  'def': '\"Identifier of a transcription factor (or a TF binding site).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1078': {'id': 'EDAM_data:1078',\n",
       "  'name': 'Microarray experiment annotation ID',\n",
       "  'def': '\"Identifier of an entry from a database of microarray data.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1079': {'id': 'EDAM_data:1079',\n",
       "  'name': 'Electron microscopy model ID',\n",
       "  'def': '\"Identifier of an entry from a database of electron microscopy data.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1080': {'id': 'EDAM_data:1080',\n",
       "  'name': 'Gene expression report ID',\n",
       "  'def': '\"Accession of a report of gene expression (e.g. a gene expression profile) from a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1081': {'id': 'EDAM_data:1081',\n",
       "  'name': 'Genotype and phenotype annotation ID',\n",
       "  'def': '\"Identifier of an entry from a database of genotypes and phenotypes.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1082': {'id': 'EDAM_data:1082',\n",
       "  'name': 'Pathway or network identifier',\n",
       "  'def': '\"Identifier of an entry from a database of biological pathways or networks.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1083': {'id': 'EDAM_data:1083',\n",
       "  'name': 'Workflow ID',\n",
       "  'def': '\"Identifier of a biological or biomedical workflow, typically from a database of workflows.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1084': {'id': 'EDAM_data:1084',\n",
       "  'name': 'Data resource definition identifier',\n",
       "  'def': '\"Identifier of a data type definition from some provider.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1085': {'id': 'EDAM_data:1085',\n",
       "  'name': 'Biological model identifier',\n",
       "  'def': '\"Identifier of a mathematical model, typically an entry from a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1086': {'id': 'EDAM_data:1086',\n",
       "  'name': 'Compound identifier',\n",
       "  'def': '\"Identifier of an entry from a database of chemicals.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1087': {'id': 'EDAM_data:1087',\n",
       "  'name': 'Ontology concept ID',\n",
       "  'def': '\"A unique (typically numerical) identifier of a concept in an ontology of biological or bioinformatics concepts and relations.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1088': {'id': 'EDAM_data:1088',\n",
       "  'name': 'Article ID',\n",
       "  'def': '\"Unique identifier of a scientific article.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1089': {'id': 'EDAM_data:1089',\n",
       "  'name': 'FlyBase ID',\n",
       "  'def': '\"Identifier of an object from the FlyBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1091': {'id': 'EDAM_data:1091',\n",
       "  'name': 'WormBase name',\n",
       "  'def': '\"Name of an object from the WormBase database, usually a human-readable name.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1092': {'id': 'EDAM_data:1092',\n",
       "  'name': 'WormBase class',\n",
       "  'def': '\"Class of an object from the WormBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1093': {'id': 'EDAM_data:1093',\n",
       "  'name': 'Sequence accession',\n",
       "  'def': '\"A persistent, unique identifier of a molecular sequence database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1094': {'id': 'EDAM_data:1094',\n",
       "  'name': 'Sequence type',\n",
       "  'def': '\"A label (text token) describing a type of molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1095': {'id': 'EDAM_data:1095',\n",
       "  'name': 'EMBOSS Uniform Sequence Address',\n",
       "  'def': '\"The name of a sequence-based entity adhering to the standard sequence naming scheme used by all EMBOSS applications.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1096': {'id': 'EDAM_data:1096',\n",
       "  'name': 'Sequence accession (protein)',\n",
       "  'def': '\"Accession number of a protein sequence database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1097': {'id': 'EDAM_data:1097',\n",
       "  'name': 'Sequence accession (nucleic acid)',\n",
       "  'def': '\"Accession number of a nucleotide sequence database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1098': {'id': 'EDAM_data:1098',\n",
       "  'name': 'RefSeq accession',\n",
       "  'def': '\"Accession number of a RefSeq database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1099': {'id': 'EDAM_data:1099',\n",
       "  'name': 'UniProt accession (extended)',\n",
       "  'def': '\"Accession number of a UniProt (protein sequence) database entry. May contain version or isoform number.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1100': {'id': 'EDAM_data:1100',\n",
       "  'name': 'PIR identifier',\n",
       "  'def': '\"An identifier of PIR sequence database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1101': {'id': 'EDAM_data:1101',\n",
       "  'name': 'TREMBL accession',\n",
       "  'def': '\"Identifier of a TREMBL sequence database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1102': {'id': 'EDAM_data:1102',\n",
       "  'name': 'Gramene primary identifier',\n",
       "  'def': '\"Primary identifier of a Gramene database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1103': {'id': 'EDAM_data:1103',\n",
       "  'name': 'EMBL/GenBank/DDBJ ID',\n",
       "  'def': '\"Identifier of a (nucleic acid) entry from the EMBL/GenBank/DDBJ databases.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1104': {'id': 'EDAM_data:1104',\n",
       "  'name': 'Sequence cluster ID (UniGene)',\n",
       "  'def': '\"A unique identifier of an entry (gene cluster) from the NCBI UniGene database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1105': {'id': 'EDAM_data:1105',\n",
       "  'name': 'dbEST accession',\n",
       "  'def': '\"Identifier of a dbEST database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1106': {'id': 'EDAM_data:1106',\n",
       "  'name': 'dbSNP ID',\n",
       "  'def': '\"Identifier of a dbSNP database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1110': {'id': 'EDAM_data:1110',\n",
       "  'name': 'EMBOSS sequence type',\n",
       "  'def': '\"The EMBOSS type of a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1111': {'id': 'EDAM_data:1111',\n",
       "  'name': 'EMBOSS listfile',\n",
       "  'def': '\"List of EMBOSS Uniform Sequence Addresses (EMBOSS listfile).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1112': {'id': 'EDAM_data:1112',\n",
       "  'name': 'Sequence cluster ID',\n",
       "  'def': '\"An identifier of a cluster of molecular sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1113': {'id': 'EDAM_data:1113',\n",
       "  'name': 'Sequence cluster ID (COG)',\n",
       "  'def': '\"Unique identifier of an entry from the COG database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1114': {'id': 'EDAM_data:1114',\n",
       "  'name': 'Sequence motif identifier',\n",
       "  'def': '\"Identifier of a sequence motif, for example an entry from a motif database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1115': {'id': 'EDAM_data:1115',\n",
       "  'name': 'Sequence profile ID',\n",
       "  'def': '\"Identifier of a sequence profile.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1116': {'id': 'EDAM_data:1116',\n",
       "  'name': 'ELM ID',\n",
       "  'def': '\"Identifier of an entry from the ELMdb database of protein functional sites.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1117': {'id': 'EDAM_data:1117',\n",
       "  'name': 'Prosite accession number',\n",
       "  'def': '\"Accession number of an entry from the Prosite database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1118': {'id': 'EDAM_data:1118',\n",
       "  'name': 'HMMER hidden Markov model ID',\n",
       "  'def': '\"Unique identifier or name of a HMMER hidden Markov model.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1119': {'id': 'EDAM_data:1119',\n",
       "  'name': 'JASPAR profile ID',\n",
       "  'def': '\"Unique identifier or name of a profile from the JASPAR database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1120': {'id': 'EDAM_data:1120',\n",
       "  'name': 'Sequence alignment type',\n",
       "  'def': '\"A label (text token) describing the type of a sequence alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1121': {'id': 'EDAM_data:1121',\n",
       "  'name': 'BLAST sequence alignment type',\n",
       "  'def': '\"The type of a BLAST sequence alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1122': {'id': 'EDAM_data:1122',\n",
       "  'name': 'Phylogenetic tree type',\n",
       "  'def': '\"A label (text token) describing the type of a phylogenetic tree.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1123': {'id': 'EDAM_data:1123',\n",
       "  'name': 'TreeBASE study accession number',\n",
       "  'def': '\"Accession number of an entry from the TreeBASE database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1124': {'id': 'EDAM_data:1124',\n",
       "  'name': 'TreeFam accession number',\n",
       "  'def': '\"Accession number of an entry from the TreeFam database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1125': {'id': 'EDAM_data:1125',\n",
       "  'name': 'Comparison matrix type',\n",
       "  'def': '\"A label (text token) describing the type of a comparison matrix.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1126': {'id': 'EDAM_data:1126',\n",
       "  'name': 'Comparison matrix name',\n",
       "  'def': '\"Unique name or identifier of a comparison matrix.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1127': {'id': 'EDAM_data:1127',\n",
       "  'name': 'PDB ID',\n",
       "  'def': '\"An identifier of an entry from the PDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1128': {'id': 'EDAM_data:1128',\n",
       "  'name': 'AAindex ID',\n",
       "  'def': '\"Identifier of an entry from the AAindex database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1129': {'id': 'EDAM_data:1129',\n",
       "  'name': 'BIND accession number',\n",
       "  'def': '\"Accession number of an entry from the BIND database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1130': {'id': 'EDAM_data:1130',\n",
       "  'name': 'IntAct accession number',\n",
       "  'def': '\"Accession number of an entry from the IntAct database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1131': {'id': 'EDAM_data:1131',\n",
       "  'name': 'Protein family name',\n",
       "  'def': '\"Name of a protein family.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1132': {'id': 'EDAM_data:1132',\n",
       "  'name': 'InterPro entry name',\n",
       "  'def': '\"Name of an InterPro entry, usually indicating the type of protein matches for that entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1133': {'id': 'EDAM_data:1133',\n",
       "  'name': 'InterPro accession',\n",
       "  'def': '\"Primary accession number of an InterPro entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1134': {'id': 'EDAM_data:1134',\n",
       "  'name': 'InterPro secondary accession',\n",
       "  'def': '\"Secondary accession number of an InterPro entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1135': {'id': 'EDAM_data:1135',\n",
       "  'name': 'Gene3D ID',\n",
       "  'def': '\"Unique identifier of an entry from the Gene3D database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1136': {'id': 'EDAM_data:1136',\n",
       "  'name': 'PIRSF ID',\n",
       "  'def': '\"Unique identifier of an entry from the PIRSF database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1137': {'id': 'EDAM_data:1137',\n",
       "  'name': 'PRINTS code',\n",
       "  'def': '\"The unique identifier of an entry in the PRINTS database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1138': {'id': 'EDAM_data:1138',\n",
       "  'name': 'Pfam accession number',\n",
       "  'def': '\"Accession number of a Pfam entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1139': {'id': 'EDAM_data:1139',\n",
       "  'name': 'SMART accession number',\n",
       "  'def': '\"Accession number of an entry from the SMART database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1140': {'id': 'EDAM_data:1140',\n",
       "  'name': 'Superfamily hidden Markov model number',\n",
       "  'def': '\"Unique identifier (number) of a hidden Markov model from the Superfamily database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1141': {'id': 'EDAM_data:1141',\n",
       "  'name': 'TIGRFam ID',\n",
       "  'def': '\"Accession number of an entry (family) from the TIGRFam database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1142': {'id': 'EDAM_data:1142',\n",
       "  'name': 'ProDom accession number',\n",
       "  'def': '\"A ProDom domain family accession number.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1143': {'id': 'EDAM_data:1143',\n",
       "  'name': 'TRANSFAC accession number',\n",
       "  'def': '\"Identifier of an entry from the TRANSFAC database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1144': {'id': 'EDAM_data:1144',\n",
       "  'name': 'ArrayExpress accession number',\n",
       "  'def': '\"Accession number of an entry from the ArrayExpress database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1145': {'id': 'EDAM_data:1145',\n",
       "  'name': 'PRIDE experiment accession number',\n",
       "  'def': '\"PRIDE experiment accession number.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1146': {'id': 'EDAM_data:1146',\n",
       "  'name': 'EMDB ID',\n",
       "  'def': '\"Identifier of an entry from the EMDB electron microscopy database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1147': {'id': 'EDAM_data:1147',\n",
       "  'name': 'GEO accession number',\n",
       "  'def': '\"Accession number of an entry from the GEO database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1148': {'id': 'EDAM_data:1148',\n",
       "  'name': 'GermOnline ID',\n",
       "  'def': '\"Identifier of an entry from the GermOnline database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1149': {'id': 'EDAM_data:1149',\n",
       "  'name': 'EMAGE ID',\n",
       "  'def': '\"Identifier of an entry from the EMAGE database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1150': {'id': 'EDAM_data:1150',\n",
       "  'name': 'Disease ID',\n",
       "  'def': '\"Identifier of an entry from a database of disease.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1151': {'id': 'EDAM_data:1151',\n",
       "  'name': 'HGVbase ID',\n",
       "  'def': '\"Identifier of an entry from the HGVbase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1152': {'id': 'EDAM_data:1152',\n",
       "  'name': 'HIVDB identifier',\n",
       "  'def': '\"Identifier of an entry from the HIVDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1153': {'id': 'EDAM_data:1153',\n",
       "  'name': 'OMIM ID',\n",
       "  'def': '\"Identifier of an entry from the OMIM database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1154': {'id': 'EDAM_data:1154',\n",
       "  'name': 'KEGG object identifier',\n",
       "  'def': '\"Unique identifier of an object from one of the KEGG databases (excluding the GENES division).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1155': {'id': 'EDAM_data:1155',\n",
       "  'name': 'Pathway ID (reactome)',\n",
       "  'def': '\"Identifier of an entry from the Reactome database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1156': {'id': 'EDAM_data:1156',\n",
       "  'name': 'Pathway ID (aMAZE)',\n",
       "  'def': '\"Identifier of an entry from the aMAZE database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1157': {'id': 'EDAM_data:1157',\n",
       "  'name': 'Pathway ID (BioCyc)',\n",
       "  'def': '\"Identifier of an pathway from the BioCyc biological pathways database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1158': {'id': 'EDAM_data:1158',\n",
       "  'name': 'Pathway ID (INOH)',\n",
       "  'def': '\"Identifier of an entry from the INOH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1159': {'id': 'EDAM_data:1159',\n",
       "  'name': 'Pathway ID (PATIKA)',\n",
       "  'def': '\"Identifier of an entry from the PATIKA database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1160': {'id': 'EDAM_data:1160',\n",
       "  'name': 'Pathway ID (CPDB)',\n",
       "  'def': '\"Identifier of an entry from the CPDB (ConsensusPathDB) biological pathways database, which is an identifier from an external database integrated into CPDB.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1161': {'id': 'EDAM_data:1161',\n",
       "  'name': 'Pathway ID (Panther)',\n",
       "  'def': '\"Identifier of a biological pathway from the Panther Pathways database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1162': {'id': 'EDAM_data:1162',\n",
       "  'name': 'MIRIAM identifier',\n",
       "  'def': '\"Unique identifier of a MIRIAM data resource.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1163': {'id': 'EDAM_data:1163',\n",
       "  'name': 'MIRIAM data type name',\n",
       "  'def': '\"The name of a data type from the MIRIAM database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1164': {'id': 'EDAM_data:1164',\n",
       "  'name': 'MIRIAM URI',\n",
       "  'def': '\"The URI (URL or URN) of a data entity from the MIRIAM database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1165': {'id': 'EDAM_data:1165',\n",
       "  'name': 'MIRIAM data type primary name',\n",
       "  'def': '\"The primary name of a data type from the MIRIAM database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1166': {'id': 'EDAM_data:1166',\n",
       "  'name': 'MIRIAM data type synonymous name',\n",
       "  'def': '\"A synonymous name of a data type from the MIRIAM database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1167': {'id': 'EDAM_data:1167',\n",
       "  'name': 'Taverna workflow ID',\n",
       "  'def': '\"Unique identifier of a Taverna workflow.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1170': {'id': 'EDAM_data:1170',\n",
       "  'name': 'Biological model name',\n",
       "  'def': '\"Name of a biological (mathematical) model.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1171': {'id': 'EDAM_data:1171',\n",
       "  'name': 'BioModel ID',\n",
       "  'def': '\"Unique identifier of an entry from the BioModel database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1172': {'id': 'EDAM_data:1172',\n",
       "  'name': 'PubChem CID',\n",
       "  'def': '\"Chemical structure specified in PubChem Compound Identification (CID), a non-zero integer identifier for a unique chemical structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1173': {'id': 'EDAM_data:1173',\n",
       "  'name': 'ChemSpider ID',\n",
       "  'def': '\"Identifier of an entry from the ChemSpider database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1174': {'id': 'EDAM_data:1174',\n",
       "  'name': 'ChEBI ID',\n",
       "  'def': '\"Identifier of an entry from the ChEBI database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1175': {'id': 'EDAM_data:1175',\n",
       "  'name': 'BioPax concept ID',\n",
       "  'def': '\"An identifier of a concept from the BioPax ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1176': {'id': 'EDAM_data:1176',\n",
       "  'name': 'GO concept ID',\n",
       "  'def': '\"An identifier of a concept from The Gene Ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1177': {'id': 'EDAM_data:1177',\n",
       "  'name': 'MeSH concept ID',\n",
       "  'def': '\"An identifier of a concept from the MeSH vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1178': {'id': 'EDAM_data:1178',\n",
       "  'name': 'HGNC concept ID',\n",
       "  'def': '\"An identifier of a concept from the HGNC controlled vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1179': {'id': 'EDAM_data:1179',\n",
       "  'name': 'NCBI taxonomy ID',\n",
       "  'def': '\"A stable unique identifier for each taxon (for a species, a family, an order, or any other group in the NCBI taxonomy database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1180': {'id': 'EDAM_data:1180',\n",
       "  'name': 'Plant Ontology concept ID',\n",
       "  'def': '\"An identifier of a concept from the Plant Ontology (PO).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1181': {'id': 'EDAM_data:1181',\n",
       "  'name': 'UMLS concept ID',\n",
       "  'def': '\"An identifier of a concept from the UMLS vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1182': {'id': 'EDAM_data:1182',\n",
       "  'name': 'FMA concept ID',\n",
       "  'def': '\"An identifier of a concept from Foundational Model of Anatomy.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1183': {'id': 'EDAM_data:1183',\n",
       "  'name': 'EMAP concept ID',\n",
       "  'def': '\"An identifier of a concept from the EMAP mouse ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1184': {'id': 'EDAM_data:1184',\n",
       "  'name': 'ChEBI concept ID',\n",
       "  'def': '\"An identifier of a concept from the ChEBI ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1185': {'id': 'EDAM_data:1185',\n",
       "  'name': 'MGED concept ID',\n",
       "  'def': '\"An identifier of a concept from the MGED ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1186': {'id': 'EDAM_data:1186',\n",
       "  'name': 'myGrid concept ID',\n",
       "  'def': '\"An identifier of a concept from the myGrid ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1187': {'id': 'EDAM_data:1187',\n",
       "  'name': 'PubMed ID',\n",
       "  'def': '\"PubMed unique identifier of an article.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1188': {'id': 'EDAM_data:1188',\n",
       "  'name': 'Digital Object Identifier',\n",
       "  'def': '\"Digital Object Identifier (DOI) of a published article.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1189': {'id': 'EDAM_data:1189',\n",
       "  'name': 'Medline UI',\n",
       "  'def': '\"Medline UI (unique identifier) of an article.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1190': {'id': 'EDAM_data:1190',\n",
       "  'name': 'Tool name',\n",
       "  'def': '\"The name of a computer package, application, method or function.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1191': {'id': 'EDAM_data:1191',\n",
       "  'name': 'Tool name (signature)',\n",
       "  'def': '\"The unique name of a signature (sequence classifier) method.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1192': {'id': 'EDAM_data:1192',\n",
       "  'name': 'Tool name (BLAST)',\n",
       "  'def': '\"The name of a BLAST tool.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1193': {'id': 'EDAM_data:1193',\n",
       "  'name': 'Tool name (FASTA)',\n",
       "  'def': '\"The name of a FASTA tool.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1194': {'id': 'EDAM_data:1194',\n",
       "  'name': 'Tool name (EMBOSS)',\n",
       "  'def': '\"The name of an EMBOSS application.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1195': {'id': 'EDAM_data:1195',\n",
       "  'name': 'Tool name (EMBASSY package)',\n",
       "  'def': '\"The name of an EMBASSY package.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1201': {'id': 'EDAM_data:1201',\n",
       "  'name': 'QSAR descriptor (constitutional)',\n",
       "  'def': '\"A QSAR constitutional descriptor.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1202': {'id': 'EDAM_data:1202',\n",
       "  'name': 'QSAR descriptor (electronic)',\n",
       "  'def': '\"A QSAR electronic descriptor.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1203': {'id': 'EDAM_data:1203',\n",
       "  'name': 'QSAR descriptor (geometrical)',\n",
       "  'def': '\"A QSAR geometrical descriptor.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1204': {'id': 'EDAM_data:1204',\n",
       "  'name': 'QSAR descriptor (topological)',\n",
       "  'def': '\"A QSAR topological descriptor.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1205': {'id': 'EDAM_data:1205',\n",
       "  'name': 'QSAR descriptor (molecular)',\n",
       "  'def': '\"A QSAR molecular descriptor.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1233': {'id': 'EDAM_data:1233',\n",
       "  'name': 'Sequence set (protein)',\n",
       "  'def': '\"Any collection of multiple protein sequences and associated metadata that do not (typically) correspond to common sequence database records or database entries.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1234': {'id': 'EDAM_data:1234',\n",
       "  'name': 'Sequence set (nucleic acid)',\n",
       "  'def': '\"Any collection of multiple nucleotide sequences and associated metadata that do not (typically) correspond to common sequence database records or database entries.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1235': {'id': 'EDAM_data:1235',\n",
       "  'name': 'Sequence cluster',\n",
       "  'def': '\"A set of sequences that have been clustered or otherwise classified as belonging to a group including (typically) sequence cluster information.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1236': {'id': 'EDAM_data:1236',\n",
       "  'name': 'Psiblast checkpoint file',\n",
       "  'def': '\"A file of intermediate results from a PSIBLAST search that is used for priming the search in the next PSIBLAST iteration.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1237': {'id': 'EDAM_data:1237',\n",
       "  'name': 'HMMER synthetic sequences set',\n",
       "  'def': '\"Sequences generated by HMMER package in FASTA-style format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1238': {'id': 'EDAM_data:1238',\n",
       "  'name': 'Proteolytic digest',\n",
       "  'def': '\"A protein sequence cleaved into peptide fragments (by enzymatic or chemical cleavage) with fragment masses.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1239': {'id': 'EDAM_data:1239',\n",
       "  'name': 'Restriction digest',\n",
       "  'def': '\"Restriction digest fragments from digesting a nucleotide sequence with restriction sites using a restriction endonuclease.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1240': {'id': 'EDAM_data:1240',\n",
       "  'name': 'PCR primers',\n",
       "  'def': '\"Oligonucleotide primer(s) for PCR and DNA amplification, for example a minimal primer set.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1241': {'id': 'EDAM_data:1241',\n",
       "  'name': 'vectorstrip cloning vector definition file',\n",
       "  'def': '\"File of sequence vectors used by EMBOSS vectorstrip application, or any file in same format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1242': {'id': 'EDAM_data:1242',\n",
       "  'name': 'Primer3 internal oligo mishybridizing library',\n",
       "  'def': '\"A library of nucleotide sequences to avoid during hybridization events. Hybridization of the internal oligo to sequences in this library is avoided, rather than priming from them. The file is in a restricted FASTA format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1243': {'id': 'EDAM_data:1243',\n",
       "  'name': 'Primer3 mispriming library file',\n",
       "  'def': '\"A nucleotide sequence library of sequences to avoid during amplification (for example repetitive sequences, or possibly the sequences of genes in a gene family that should not be amplified. The file must is in a restricted FASTA format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1244': {'id': 'EDAM_data:1244',\n",
       "  'name': 'primersearch primer pairs sequence record',\n",
       "  'def': '\"File of one or more pairs of primer sequences, as used by EMBOSS primersearch application.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1245': {'id': 'EDAM_data:1245',\n",
       "  'name': 'Sequence cluster (protein)',\n",
       "  'def': '\"A cluster of protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1246': {'id': 'EDAM_data:1246',\n",
       "  'name': 'Sequence cluster (nucleic acid)',\n",
       "  'def': '\"A cluster of nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1249': {'id': 'EDAM_data:1249',\n",
       "  'name': 'Sequence length',\n",
       "  'def': '\"The size (length) of a sequence, subsequence or region in a sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1250': {'id': 'EDAM_data:1250',\n",
       "  'name': 'Word size',\n",
       "  'def': '\"Size of a sequence word.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1251': {'id': 'EDAM_data:1251',\n",
       "  'name': 'Window size',\n",
       "  'def': '\"Size of a sequence window.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1252': {'id': 'EDAM_data:1252',\n",
       "  'name': 'Sequence length range',\n",
       "  'def': '\"Specification of range(s) of length of sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1253': {'id': 'EDAM_data:1253',\n",
       "  'name': 'Sequence information report',\n",
       "  'def': '\"Report on basic information about a molecular sequence such as name, accession number, type (nucleic or protein), length, description etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1254': {'id': 'EDAM_data:1254',\n",
       "  'name': 'Sequence property',\n",
       "  'def': '\"An informative report about non-positional sequence features, typically a report on general molecular sequence properties derived from sequence analysis.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1255': {'id': 'EDAM_data:1255',\n",
       "  'name': 'Feature record',\n",
       "  'def': '\"Annotation of positional features of molecular sequence(s), i.e. that can be mapped to position(s) in the sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1256': {'id': 'EDAM_data:1256',\n",
       "  'name': 'Sequence features (comparative)',\n",
       "  'def': '\"Comparative data on sequence features such as statistics, intersections (and data on intersections), differences etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1257': {'id': 'EDAM_data:1257',\n",
       "  'name': 'Sequence property (protein)',\n",
       "  'def': '\"A report of general sequence properties derived from protein sequence data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1258': {'id': 'EDAM_data:1258',\n",
       "  'name': 'Sequence property (nucleic acid)',\n",
       "  'def': '\"A report of general sequence properties derived from nucleotide sequence data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1259': {'id': 'EDAM_data:1259',\n",
       "  'name': 'Sequence complexity',\n",
       "  'def': '\"A report on sequence complexity, for example low-complexity or repeat regions in sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1260': {'id': 'EDAM_data:1260',\n",
       "  'name': 'Sequence ambiguity',\n",
       "  'def': '\"A report on ambiguity in molecular sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1261': {'id': 'EDAM_data:1261',\n",
       "  'name': 'Sequence composition',\n",
       "  'def': '\"A report (typically a table) on character or word composition / frequency of a molecular sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1262': {'id': 'EDAM_data:1262',\n",
       "  'name': 'Peptide molecular weight hits',\n",
       "  'def': '\"A report on peptide fragments of certain molecular weight(s) in one or more protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1263': {'id': 'EDAM_data:1263',\n",
       "  'name': 'Sequence composition (base position variability)',\n",
       "  'def': '\"Report on or plot of third base position variability in a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1264': {'id': 'EDAM_data:1264',\n",
       "  'name': 'Sequence composition table',\n",
       "  'def': '\"A table of character or word composition / frequency of a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1265': {'id': 'EDAM_data:1265',\n",
       "  'name': 'Sequence composition (base frequencies)',\n",
       "  'def': '\"A table of base frequencies of a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1266': {'id': 'EDAM_data:1266',\n",
       "  'name': 'Sequence composition (base words)',\n",
       "  'def': '\"A table of word composition of a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1267': {'id': 'EDAM_data:1267',\n",
       "  'name': 'Amino acid frequencies',\n",
       "  'def': '\"A table of amino acid frequencies of a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1268': {'id': 'EDAM_data:1268',\n",
       "  'name': 'Amino acid word frequencies',\n",
       "  'def': '\"A table of amino acid word composition of a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1269': {'id': 'EDAM_data:1269',\n",
       "  'name': 'DAS sequence feature annotation',\n",
       "  'def': '\"Annotation of a molecular sequence in DAS format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1270': {'id': 'EDAM_data:1270',\n",
       "  'name': 'Sequence feature table',\n",
       "  'def': '\"Annotation of positional sequence features, organized into a standard feature table.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1274': {'id': 'EDAM_data:1274',\n",
       "  'name': 'Map',\n",
       "  'def': '\"A map of (typically one) DNA sequence annotated with positional or non-positional features.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1276': {'id': 'EDAM_data:1276',\n",
       "  'name': 'Nucleic acid features',\n",
       "  'def': '\"Nucleotide sequence-specific feature annotation (positional features of a nucleotide sequence).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1277': {'id': 'EDAM_data:1277',\n",
       "  'name': 'Protein features',\n",
       "  'def': '\"Protein sequence-specific feature annotation (positional features of a protein sequence).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1278': {'id': 'EDAM_data:1278',\n",
       "  'name': 'Genetic map',\n",
       "  'def': '\"A map showing the relative positions of genetic markers in a nucleic acid sequence, based on estimation of non-physical distance such as recombination frequencies.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1279': {'id': 'EDAM_data:1279',\n",
       "  'name': 'Sequence map',\n",
       "  'def': '\"A map of genetic markers in a contiguous, assembled genomic sequence, with the sizes and separation of markers measured in base pairs.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1280': {'id': 'EDAM_data:1280',\n",
       "  'name': 'Physical map',\n",
       "  'def': '\"A map of DNA (linear or circular) annotated with physical features or landmarks such as restriction sites, cloned DNA fragments, genes or genetic markers, along with the physical distances between them.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1281': {'id': 'EDAM_data:1281',\n",
       "  'name': 'Sequence signature map',\n",
       "  'def': '\"Image of a sequence with matches to signatures, motifs or profiles.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1283': {'id': 'EDAM_data:1283',\n",
       "  'name': 'Cytogenetic map',\n",
       "  'def': '\"A map showing banding patterns derived from direct observation of a stained chromosome.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1284': {'id': 'EDAM_data:1284',\n",
       "  'name': 'DNA transduction map',\n",
       "  'def': '\"A gene map showing distances between loci based on relative cotransduction frequencies.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1285': {'id': 'EDAM_data:1285',\n",
       "  'name': 'Gene map',\n",
       "  'def': '\"Sequence map of a single gene annotated with genetic features such as introns, exons, untranslated regions, polyA signals, promoters, enhancers and (possibly) mutations defining alleles of a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1286': {'id': 'EDAM_data:1286',\n",
       "  'name': 'Plasmid map',\n",
       "  'def': '\"Sequence map of a plasmid (circular DNA).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1288': {'id': 'EDAM_data:1288',\n",
       "  'name': 'Genome map',\n",
       "  'def': '\"Sequence map of a whole genome.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1289': {'id': 'EDAM_data:1289',\n",
       "  'name': 'Restriction map',\n",
       "  'def': '\"Image of the restriction enzyme cleavage sites (restriction sites) in a nucleic acid sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1290': {'id': 'EDAM_data:1290',\n",
       "  'name': 'InterPro compact match image',\n",
       "  'def': '\"Image showing matches between protein sequence(s) and InterPro Entries.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1291': {'id': 'EDAM_data:1291',\n",
       "  'name': 'InterPro detailed match image',\n",
       "  'def': '\"Image showing detailed information on matches between protein sequence(s) and InterPro Entries.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1292': {'id': 'EDAM_data:1292',\n",
       "  'name': 'InterPro architecture image',\n",
       "  'def': '\"Image showing the architecture of InterPro domains in a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1293': {'id': 'EDAM_data:1293',\n",
       "  'name': 'SMART protein schematic',\n",
       "  'def': '\"SMART protein schematic in PNG format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1294': {'id': 'EDAM_data:1294',\n",
       "  'name': 'GlobPlot domain image',\n",
       "  'def': '\"Images based on GlobPlot prediction of intrinsic disordered regions and globular domains in protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1298': {'id': 'EDAM_data:1298',\n",
       "  'name': 'Sequence features (motifs)',\n",
       "  'def': '\"Report on the location of matches to profiles, motifs (conserved or functional patterns) or other signatures in one or more sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1299': {'id': 'EDAM_data:1299',\n",
       "  'name': 'Sequence features (repeats)',\n",
       "  'def': '\"Location of short repetitive subsequences (repeat sequences) in (typically nucleotide) sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1300': {'id': 'EDAM_data:1300',\n",
       "  'name': 'Nucleic acid features (gene and transcript structure)',\n",
       "  'def': '\"A report on predicted or actual gene structure, regions which make an RNA product and features such as promoters, coding regions, splice sites etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1301': {'id': 'EDAM_data:1301',\n",
       "  'name': 'Nucleic acid features (mobile genetic elements)',\n",
       "  'def': '\"A report on a region of a nucleic acid sequence containin mobile genetic elements.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1302': {'id': 'EDAM_data:1302',\n",
       "  'name': 'Nucleic acid features (PolyA signal or site)',\n",
       "  'def': '\"A region or site in a eukaryotic and eukaryotic viral RNA sequence which directs endonuclease cleavage or polyadenylation of an RNA transcript.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1303': {'id': 'EDAM_data:1303',\n",
       "  'name': 'Nucleic acid features (quadruplexes)',\n",
       "  'def': '\"A report on quadruplex-forming motifs in a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1304': {'id': 'EDAM_data:1304',\n",
       "  'name': 'Nucleic acid features (CpG island and isochore)',\n",
       "  'def': '\"A report or plot of CpG rich regions (isochores) in a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1305': {'id': 'EDAM_data:1305',\n",
       "  'name': 'Nucleic acid features (restriction sites)',\n",
       "  'def': '\"Report on restriction enzyme recognition sites (restriction sites) in a nucleic acid sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1306': {'id': 'EDAM_data:1306',\n",
       "  'name': 'Nucleic acid features (nucleosome exclusion sequences)',\n",
       "  'def': '\"A report on nucleosome formation potential or exclusion sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1307': {'id': 'EDAM_data:1307',\n",
       "  'name': 'Nucleic acid features (splice sites)',\n",
       "  'def': '\"A report on splice sites in a nucleotide sequence or alternative RNA splicing events.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1308': {'id': 'EDAM_data:1308',\n",
       "  'name': 'Nucleic acid features (matrix/scaffold attachment sites)',\n",
       "  'def': '\"A report on matrix/scaffold attachment regions (MARs/SARs) in a DNA sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1309': {'id': 'EDAM_data:1309',\n",
       "  'name': 'Gene features (exonic splicing enhancer)',\n",
       "  'def': '\"A report on exonic splicing enhancers (ESE) in an exon.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1310': {'id': 'EDAM_data:1310',\n",
       "  'name': 'Nucleic acid features (microRNA)',\n",
       "  'def': '\"A report on microRNA sequence (miRNA) or precursor, microRNA targets, miRNA binding sites in an RNA sequence etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1311': {'id': 'EDAM_data:1311',\n",
       "  'name': 'Nucleic acid features (operon)',\n",
       "  'def': '\"A report on operons (operators, promoters and genes) from a bacterial genome.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1312': {'id': 'EDAM_data:1312',\n",
       "  'name': 'Gene features (promoter)',\n",
       "  'def': '\"A report on whole promoters or promoter elements (transcription start sites, RNA polymerase binding site, transcription factor binding sites, promoter enhancers etc) in a DNA sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1313': {'id': 'EDAM_data:1313',\n",
       "  'name': 'Nucleic acid features (coding sequence)',\n",
       "  'def': '\"A report on protein-coding regions including coding sequences (CDS), exons, translation initiation sites and open reading frames.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1314': {'id': 'EDAM_data:1314',\n",
       "  'name': 'Gene features (SECIS element)',\n",
       "  'def': '\"A report on selenocysteine insertion sequence (SECIS) element in a DNA sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1315': {'id': 'EDAM_data:1315',\n",
       "  'name': 'Gene features (TFBS)',\n",
       "  'def': '\"A report on the transcription factor binding sites (TFBS) in a DNA sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1321': {'id': 'EDAM_data:1321',\n",
       "  'name': 'Protein features (sites)',\n",
       "  'def': '\"A report on predicted or known key residue positions (sites) in a protein sequence, such as binding or functional sites.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1322': {'id': 'EDAM_data:1322',\n",
       "  'name': 'Protein features (signal peptides)',\n",
       "  'def': '\"A report on the location of signal peptides or signal peptide cleavage sites in protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1323': {'id': 'EDAM_data:1323',\n",
       "  'name': 'Protein features (cleavage sites)',\n",
       "  'def': '\"A report on cleavage sites (for a proteolytic enzyme or agent) in a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1324': {'id': 'EDAM_data:1324',\n",
       "  'name': 'Protein features (post-translation modifications)',\n",
       "  'def': '\"A report on post-translation modifications in a protein sequence, typically describing the specific sites involved.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1325': {'id': 'EDAM_data:1325',\n",
       "  'name': 'Protein features (active sites)',\n",
       "  'def': '\"A report on catalytic residues (active site) of an enzyme.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1326': {'id': 'EDAM_data:1326',\n",
       "  'name': 'Protein features (binding sites)',\n",
       "  'def': '\"A report on ligand-binding (non-catalytic) residues of a protein, such as sites that bind metal, prosthetic groups or lipids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1327': {'id': 'EDAM_data:1327',\n",
       "  'name': 'Protein features (epitopes)',\n",
       "  'def': '\"A report on antigenic determinant sites (epitopes) in proteins, from sequence and / or structural data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1328': {'id': 'EDAM_data:1328',\n",
       "  'name': 'Protein features (nucleic acid binding sites)',\n",
       "  'def': '\"A report on RNA and DNA-binding proteins and binding sites in protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1329': {'id': 'EDAM_data:1329',\n",
       "  'name': 'MHC Class I epitopes report',\n",
       "  'def': '\"A report on epitopes that bind to MHC class I molecules.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1330': {'id': 'EDAM_data:1330',\n",
       "  'name': 'MHC Class II epitopes report',\n",
       "  'def': '\"A report on predicted epitopes that bind to MHC class II molecules.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1331': {'id': 'EDAM_data:1331',\n",
       "  'name': 'Protein features (PEST sites)',\n",
       "  'def': '\"A report or plot of PEST sites in a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1338': {'id': 'EDAM_data:1338',\n",
       "  'name': 'Sequence database hits scores list',\n",
       "  'def': '\"Scores from a sequence database search (for example a BLAST search).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1339': {'id': 'EDAM_data:1339',\n",
       "  'name': 'Sequence database hits alignments list',\n",
       "  'def': '\"Alignments from a sequence database search (for example a BLAST search).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1340': {'id': 'EDAM_data:1340',\n",
       "  'name': 'Sequence database hits evaluation data',\n",
       "  'def': '\"A report on the evaluation of the significance of sequence similarity scores from a sequence database search (for example a BLAST search).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1344': {'id': 'EDAM_data:1344',\n",
       "  'name': 'MEME motif alphabet',\n",
       "  'def': '\"Alphabet for the motifs (patterns) that MEME will search for.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1345': {'id': 'EDAM_data:1345',\n",
       "  'name': 'MEME background frequencies file',\n",
       "  'def': '\"MEME background frequencies file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1346': {'id': 'EDAM_data:1346',\n",
       "  'name': 'MEME motifs directive file',\n",
       "  'def': '\"File of directives for ordering and spacing of MEME motifs.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1347': {'id': 'EDAM_data:1347',\n",
       "  'name': 'Dirichlet distribution',\n",
       "  'def': '\"Dirichlet distribution used by hidden Markov model analysis programs.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1348': {'id': 'EDAM_data:1348',\n",
       "  'name': 'HMM emission and transition counts',\n",
       "  'def': '\"Emission and transition counts of a hidden Markov model, generated once HMM has been determined, for example after residues/gaps have been assigned to match, delete and insert states.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1352': {'id': 'EDAM_data:1352',\n",
       "  'name': 'Regular expression',\n",
       "  'def': '\"Regular expression pattern.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1353': {'id': 'EDAM_data:1353',\n",
       "  'name': 'Sequence motif',\n",
       "  'def': '\"Any specific or conserved pattern (typically expressed as a regular expression) in a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1354': {'id': 'EDAM_data:1354',\n",
       "  'name': 'Sequence profile',\n",
       "  'def': '\"Some type of statistical model representing a (typically multiple) sequence alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1355': {'id': 'EDAM_data:1355',\n",
       "  'name': 'Protein signature',\n",
       "  'def': '\"An entry (sequence classifier and associated data) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1358': {'id': 'EDAM_data:1358',\n",
       "  'name': 'Prosite nucleotide pattern',\n",
       "  'def': '\"A nucleotide regular expression pattern from the Prosite database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1359': {'id': 'EDAM_data:1359',\n",
       "  'name': 'Prosite protein pattern',\n",
       "  'def': '\"A protein regular expression pattern from the Prosite database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1361': {'id': 'EDAM_data:1361',\n",
       "  'name': 'Position frequency matrix',\n",
       "  'def': '\"A profile (typically representing a sequence alignment) that is a simple matrix of nucleotide (or amino acid) counts per position.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1362': {'id': 'EDAM_data:1362',\n",
       "  'name': 'Position weight matrix',\n",
       "  'def': '\"A profile (typically representing a sequence alignment) that is weighted matrix of nucleotide (or amino acid) counts per position.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1363': {'id': 'EDAM_data:1363',\n",
       "  'name': 'Information content matrix',\n",
       "  'def': '\"A profile (typically representing a sequence alignment) derived from a matrix of nucleotide (or amino acid) counts per position that reflects information content at each position.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1364': {'id': 'EDAM_data:1364',\n",
       "  'name': 'Hidden Markov model',\n",
       "  'def': '\"A hidden Markov model representation of a set or alignment of sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1365': {'id': 'EDAM_data:1365',\n",
       "  'name': 'Fingerprint',\n",
       "  'def': '\"One or more fingerprints (sequence classifiers) as used in the PRINTS database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1368': {'id': 'EDAM_data:1368',\n",
       "  'name': 'Domainatrix signature',\n",
       "  'def': '\"A protein signature of the type used in the EMBASSY Signature package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1371': {'id': 'EDAM_data:1371',\n",
       "  'name': 'HMMER NULL hidden Markov model',\n",
       "  'def': '\"NULL hidden Markov model representation used by the HMMER package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1372': {'id': 'EDAM_data:1372',\n",
       "  'name': 'Protein family signature',\n",
       "  'def': '\"A protein family signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1373': {'id': 'EDAM_data:1373',\n",
       "  'name': 'Protein domain signature',\n",
       "  'def': '\"A protein domain signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1374': {'id': 'EDAM_data:1374',\n",
       "  'name': 'Protein region signature',\n",
       "  'def': '\"A protein region signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1375': {'id': 'EDAM_data:1375',\n",
       "  'name': 'Protein repeat signature',\n",
       "  'def': '\"A protein repeat signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1376': {'id': 'EDAM_data:1376',\n",
       "  'name': 'Protein site signature',\n",
       "  'def': '\"A protein site signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1377': {'id': 'EDAM_data:1377',\n",
       "  'name': 'Protein conserved site signature',\n",
       "  'def': '\"A protein conserved site signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1378': {'id': 'EDAM_data:1378',\n",
       "  'name': 'Protein active site signature',\n",
       "  'def': '\"A protein active site signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1379': {'id': 'EDAM_data:1379',\n",
       "  'name': 'Protein binding site signature',\n",
       "  'def': '\"A protein binding site signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1380': {'id': 'EDAM_data:1380',\n",
       "  'name': 'Protein post-translational modification signature',\n",
       "  'def': '\"A protein post-translational modification signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1381': {'id': 'EDAM_data:1381',\n",
       "  'name': 'Sequence alignment (pair)',\n",
       "  'def': '\"Alignment of exactly two molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1382': {'id': 'EDAM_data:1382',\n",
       "  'name': 'Sequence alignment (multiple)',\n",
       "  'def': '\"Alignment of more than two molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1383': {'id': 'EDAM_data:1383',\n",
       "  'name': 'Sequence alignment (nucleic acid)',\n",
       "  'def': '\"Alignment of multiple nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1384': {'id': 'EDAM_data:1384',\n",
       "  'name': 'Sequence alignment (protein)',\n",
       "  'def': '\"Alignment of multiple protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1385': {'id': 'EDAM_data:1385',\n",
       "  'name': 'Sequence alignment (hybrid)',\n",
       "  'def': '\"Alignment of multiple molecular sequences of different types.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1386': {'id': 'EDAM_data:1386',\n",
       "  'name': 'Sequence alignment (nucleic acid pair)',\n",
       "  'def': '\"Alignment of exactly two nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1387': {'id': 'EDAM_data:1387',\n",
       "  'name': 'Sequence alignment (protein pair)',\n",
       "  'def': '\"Alignment of exactly two protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1388': {'id': 'EDAM_data:1388',\n",
       "  'name': 'Hybrid sequence alignment (pair)',\n",
       "  'def': '\"Alignment of exactly two molecular sequences of different types.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1389': {'id': 'EDAM_data:1389',\n",
       "  'name': 'Multiple nucleotide sequence alignment',\n",
       "  'def': '\"Alignment of more than two nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1390': {'id': 'EDAM_data:1390',\n",
       "  'name': 'Multiple protein sequence alignment',\n",
       "  'def': '\"Alignment of more than two protein sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1394': {'id': 'EDAM_data:1394',\n",
       "  'name': 'Alignment score or penalty',\n",
       "  'def': '\"A simple floating point number defining the penalty for opening or extending a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1395': {'id': 'EDAM_data:1395',\n",
       "  'name': 'Score end gaps control',\n",
       "  'def': '\"Whether end gaps are scored or not.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1396': {'id': 'EDAM_data:1396',\n",
       "  'name': 'Aligned sequence order',\n",
       "  'def': '\"Controls the order of sequences in an output sequence alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1397': {'id': 'EDAM_data:1397',\n",
       "  'name': 'Gap opening penalty',\n",
       "  'def': '\"A penalty for opening a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1398': {'id': 'EDAM_data:1398',\n",
       "  'name': 'Gap extension penalty',\n",
       "  'def': '\"A penalty for extending a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1399': {'id': 'EDAM_data:1399',\n",
       "  'name': 'Gap separation penalty',\n",
       "  'def': '\"A penalty for gaps that are close together in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1400': {'id': 'EDAM_data:1400',\n",
       "  'name': 'Terminal gap penalty',\n",
       "  'def': '\"A penalty for gaps at the termini of an alignment, either from the N/C terminal of protein or 5\\'/3\\' terminal of nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1401': {'id': 'EDAM_data:1401',\n",
       "  'name': 'Match reward score',\n",
       "  'def': '\"The score for a \\'match\\' used in various sequence database search applications with simple scoring schemes.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1402': {'id': 'EDAM_data:1402',\n",
       "  'name': 'Mismatch penalty score',\n",
       "  'def': '\"The score (penalty) for a \\'mismatch\\' used in various alignment and sequence database search applications with simple scoring schemes.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1403': {'id': 'EDAM_data:1403',\n",
       "  'name': 'Drop off score',\n",
       "  'def': '\"This is the threshold drop in score at which extension of word alignment is halted.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1404': {'id': 'EDAM_data:1404',\n",
       "  'name': 'Gap opening penalty (integer)',\n",
       "  'def': '\"A simple floating point number defining the penalty for opening a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1405': {'id': 'EDAM_data:1405',\n",
       "  'name': 'Gap opening penalty (float)',\n",
       "  'def': '\"A simple floating point number defining the penalty for opening a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1406': {'id': 'EDAM_data:1406',\n",
       "  'name': 'Gap extension penalty (integer)',\n",
       "  'def': '\"A simple floating point number defining the penalty for extending a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1407': {'id': 'EDAM_data:1407',\n",
       "  'name': 'Gap extension penalty (float)',\n",
       "  'def': '\"A simple floating point number defining the penalty for extending a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1408': {'id': 'EDAM_data:1408',\n",
       "  'name': 'Gap separation penalty (integer)',\n",
       "  'def': '\"A simple floating point number defining the penalty for gaps that are close together in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1409': {'id': 'EDAM_data:1409',\n",
       "  'name': 'Gap separation penalty (float)',\n",
       "  'def': '\"A simple floating point number defining the penalty for gaps that are close together in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1410': {'id': 'EDAM_data:1410',\n",
       "  'name': 'Terminal gap opening penalty',\n",
       "  'def': '\"A number defining the penalty for opening gaps at the termini of an alignment, either from the N/C terminal of protein or 5\\'/3\\' terminal of nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1411': {'id': 'EDAM_data:1411',\n",
       "  'name': 'Terminal gap extension penalty',\n",
       "  'def': '\"A number defining the penalty for extending gaps at the termini of an alignment, either from the N/C terminal of protein or 5\\'/3\\' terminal of nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1412': {'id': 'EDAM_data:1412',\n",
       "  'name': 'Sequence identity',\n",
       "  'def': '\"Sequence identity is the number (%) of matches (identical characters) in positions from an alignment of two molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1413': {'id': 'EDAM_data:1413',\n",
       "  'name': 'Sequence similarity',\n",
       "  'def': '\"Sequence similarity is the similarity (expressed as a percentage) of two molecular sequences calculated from their alignment, a scoring matrix for scoring characters substitutions and penalties for gap insertion and extension.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1414': {'id': 'EDAM_data:1414',\n",
       "  'name': 'Sequence alignment metadata (quality report)',\n",
       "  'def': '\"Data on molecular sequence alignment quality (estimated accuracy).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1415': {'id': 'EDAM_data:1415',\n",
       "  'name': 'Sequence alignment report (site conservation)',\n",
       "  'def': '\"Data on character conservation in a molecular sequence alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1416': {'id': 'EDAM_data:1416',\n",
       "  'name': 'Sequence alignment report (site correlation)',\n",
       "  'def': '\"Data on correlations between sites in a molecular sequence alignment, typically to identify possible covarying positions and predict contacts or structural constraints in protein structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1417': {'id': 'EDAM_data:1417',\n",
       "  'name': 'Sequence-profile alignment (Domainatrix signature)',\n",
       "  'def': '\"Alignment of molecular sequences to a Domainatrix signature (representing a sequence alignment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1418': {'id': 'EDAM_data:1418',\n",
       "  'name': 'Sequence-profile alignment (HMM)',\n",
       "  'def': '\"Alignment of molecular sequence(s) to a hidden Markov model(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1420': {'id': 'EDAM_data:1420',\n",
       "  'name': 'Sequence-profile alignment (fingerprint)',\n",
       "  'def': '\"Alignment of molecular sequences to a protein fingerprint from the PRINTS database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1426': {'id': 'EDAM_data:1426',\n",
       "  'name': 'Phylogenetic continuous quantitative data',\n",
       "  'def': '\"Continuous quantitative data that may be read during phylogenetic tree calculation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1427': {'id': 'EDAM_data:1427',\n",
       "  'name': 'Phylogenetic discrete data',\n",
       "  'def': '\"Character data with discrete states that may be read during phylogenetic tree calculation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1428': {'id': 'EDAM_data:1428',\n",
       "  'name': 'Phylogenetic character cliques',\n",
       "  'def': '\"One or more cliques of mutually compatible characters that are generated, for example from analysis of discrete character data, and are used to generate a phylogeny.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1429': {'id': 'EDAM_data:1429',\n",
       "  'name': 'Phylogenetic invariants',\n",
       "  'def': '\"Phylogenetic invariants data for testing alternative tree topologies.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1438': {'id': 'EDAM_data:1438',\n",
       "  'name': 'Phylogenetic tree report',\n",
       "  'def': '\"A report of data concerning or derived from a phylogenetic tree, or from comparing two or more phylogenetic trees.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1439': {'id': 'EDAM_data:1439',\n",
       "  'name': 'DNA substitution model',\n",
       "  'def': '\"A model of DNA substitution that explains a DNA sequence alignment, derived from phylogenetic tree analysis.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1440': {'id': 'EDAM_data:1440',\n",
       "  'name': 'Phylogenetic tree report (tree shape)',\n",
       "  'def': '\"Data about the shape of a phylogenetic tree.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1441': {'id': 'EDAM_data:1441',\n",
       "  'name': 'Phylogenetic tree report (tree evaluation)',\n",
       "  'def': '\"Data on the confidence of a phylogenetic tree.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1442': {'id': 'EDAM_data:1442',\n",
       "  'name': 'Phylogenetic tree report (tree distances)',\n",
       "  'def': '\"Distances, such as Branch Score distance, between two or more phylogenetic trees.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1443': {'id': 'EDAM_data:1443',\n",
       "  'name': 'Phylogenetic tree report (tree stratigraphic)',\n",
       "  'def': '\"Molecular clock and stratigraphic (age) data derived from phylogenetic tree analysis.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1444': {'id': 'EDAM_data:1444',\n",
       "  'name': 'Phylogenetic character contrasts',\n",
       "  'def': '\"Independent contrasts for characters used in a phylogenetic tree, or covariances, regressions and correlations between characters for those contrasts.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1446': {'id': 'EDAM_data:1446',\n",
       "  'name': 'Comparison matrix (integers)',\n",
       "  'def': '\"Matrix of integer numbers for sequence comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1447': {'id': 'EDAM_data:1447',\n",
       "  'name': 'Comparison matrix (floats)',\n",
       "  'def': '\"Matrix of floating point numbers for sequence comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1448': {'id': 'EDAM_data:1448',\n",
       "  'name': 'Comparison matrix (nucleotide)',\n",
       "  'def': '\"Matrix of integer or floating point numbers for nucleotide comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1449': {'id': 'EDAM_data:1449',\n",
       "  'name': 'Comparison matrix (amino acid)',\n",
       "  'def': '\"Matrix of integer or floating point numbers for amino acid comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1450': {'id': 'EDAM_data:1450',\n",
       "  'name': 'Nucleotide comparison matrix (integers)',\n",
       "  'def': '\"Matrix of integer numbers for nucleotide comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1451': {'id': 'EDAM_data:1451',\n",
       "  'name': 'Nucleotide comparison matrix (floats)',\n",
       "  'def': '\"Matrix of floating point numbers for nucleotide comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1452': {'id': 'EDAM_data:1452',\n",
       "  'name': 'Amino acid comparison matrix (integers)',\n",
       "  'def': '\"Matrix of integer numbers for amino acid comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1453': {'id': 'EDAM_data:1453',\n",
       "  'name': 'Amino acid comparison matrix (floats)',\n",
       "  'def': '\"Matrix of floating point numbers for amino acid comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1456': {'id': 'EDAM_data:1456',\n",
       "  'name': 'Protein features (membrane regions)',\n",
       "  'def': '\"An informative report on trans- or intra-membrane regions of a protein, typically describing physicochemical properties of the secondary structure elements.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1459': {'id': 'EDAM_data:1459',\n",
       "  'name': 'Nucleic acid structure',\n",
       "  'def': '\"3D coordinate and associated data for a nucleic acid tertiary (3D) structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1460': {'id': 'EDAM_data:1460',\n",
       "  'name': 'Protein structure',\n",
       "  'def': '\"3D coordinate and associated data for a protein tertiary (3D) structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1461': {'id': 'EDAM_data:1461',\n",
       "  'name': 'Protein-ligand complex',\n",
       "  'def': '\"The structure of a protein in complex with a ligand, typically a small molecule such as an enzyme substrate or cofactor, but possibly another macromolecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1462': {'id': 'EDAM_data:1462',\n",
       "  'name': 'Carbohydrate structure',\n",
       "  'def': '\"3D coordinate and associated data for a carbohydrate (3D) structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1463': {'id': 'EDAM_data:1463',\n",
       "  'name': 'Small molecule structure',\n",
       "  'def': '\"3D coordinate and associated data for the (3D) structure of a small molecule, such as any common chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1464': {'id': 'EDAM_data:1464',\n",
       "  'name': 'DNA structure',\n",
       "  'def': '\"3D coordinate and associated data for a DNA tertiary (3D) structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1465': {'id': 'EDAM_data:1465',\n",
       "  'name': 'RNA structure record',\n",
       "  'def': '\"3D coordinate and associated data for an RNA tertiary (3D) structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1466': {'id': 'EDAM_data:1466',\n",
       "  'name': 'tRNA structure record',\n",
       "  'def': '\"3D coordinate and associated data for a tRNA tertiary (3D) structure, including tmRNA, snoRNAs etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1467': {'id': 'EDAM_data:1467',\n",
       "  'name': 'Protein chain',\n",
       "  'def': '\"3D coordinate and associated data for the tertiary (3D) structure of a polypeptide chain.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1468': {'id': 'EDAM_data:1468',\n",
       "  'name': 'Protein domain',\n",
       "  'def': '\"3D coordinate and associated data for the tertiary (3D) structure of a protein domain.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1469': {'id': 'EDAM_data:1469',\n",
       "  'name': 'Protein structure (all atoms)',\n",
       "  'def': '\"3D coordinate and associated data for a protein tertiary (3D) structure (all atoms).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1470': {'id': 'EDAM_data:1470',\n",
       "  'name': 'Protein structure (C-alpha atoms)',\n",
       "  'def': '\"3D coordinate and associated data for a protein tertiary (3D) structure (typically C-alpha atoms only).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1471': {'id': 'EDAM_data:1471',\n",
       "  'name': 'Protein chain (all atoms)',\n",
       "  'def': '\"3D coordinate and associated data for a polypeptide chain tertiary (3D) structure (all atoms).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1472': {'id': 'EDAM_data:1472',\n",
       "  'name': 'Protein chain (C-alpha atoms)',\n",
       "  'def': '\"3D coordinate and associated data for a polypeptide chain tertiary (3D) structure (typically C-alpha atoms only).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1473': {'id': 'EDAM_data:1473',\n",
       "  'name': 'Protein domain (all atoms)',\n",
       "  'def': '\"3D coordinate and associated data for a protein domain tertiary (3D) structure (all atoms).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1474': {'id': 'EDAM_data:1474',\n",
       "  'name': 'Protein domain (C-alpha atoms)',\n",
       "  'def': '\"3D coordinate and associated data for a protein domain tertiary (3D) structure (typically C-alpha atoms only).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1479': {'id': 'EDAM_data:1479',\n",
       "  'name': 'Structure alignment (pair)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two molecular tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1480': {'id': 'EDAM_data:1480',\n",
       "  'name': 'Structure alignment (multiple)',\n",
       "  'def': '\"Alignment (superimposition) of more than two molecular tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1481': {'id': 'EDAM_data:1481',\n",
       "  'name': 'Structure alignment (protein)',\n",
       "  'def': '\"Alignment (superimposition) of protein tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1482': {'id': 'EDAM_data:1482',\n",
       "  'name': 'Structure alignment (nucleic acid)',\n",
       "  'def': '\"Alignment (superimposition) of nucleic acid tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1483': {'id': 'EDAM_data:1483',\n",
       "  'name': 'Structure alignment (protein pair)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two protein tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1484': {'id': 'EDAM_data:1484',\n",
       "  'name': 'Multiple protein tertiary structure alignment',\n",
       "  'def': '\"Alignment (superimposition) of more than two protein tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1485': {'id': 'EDAM_data:1485',\n",
       "  'name': 'Structure alignment (protein all atoms)',\n",
       "  'def': '\"Alignment (superimposition) of protein tertiary (3D) structures (all atoms considered).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1486': {'id': 'EDAM_data:1486',\n",
       "  'name': 'Structure alignment (protein C-alpha atoms)',\n",
       "  'def': '\"Alignment (superimposition) of protein tertiary (3D) structures (typically C-alpha atoms only considered).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1487': {'id': 'EDAM_data:1487',\n",
       "  'name': 'Pairwise protein tertiary structure alignment (all atoms)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two protein tertiary (3D) structures (all atoms considered).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1488': {'id': 'EDAM_data:1488',\n",
       "  'name': 'Pairwise protein tertiary structure alignment (C-alpha atoms)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two protein tertiary (3D) structures (typically C-alpha atoms only considered).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1489': {'id': 'EDAM_data:1489',\n",
       "  'name': 'Multiple protein tertiary structure alignment (all atoms)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two protein tertiary (3D) structures (all atoms considered).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1490': {'id': 'EDAM_data:1490',\n",
       "  'name': 'Multiple protein tertiary structure alignment (C-alpha atoms)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two protein tertiary (3D) structures (typically C-alpha atoms only considered).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1491': {'id': 'EDAM_data:1491',\n",
       "  'name': 'Structure alignment (nucleic acid pair)',\n",
       "  'def': '\"Alignment (superimposition) of exactly two nucleic acid tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1492': {'id': 'EDAM_data:1492',\n",
       "  'name': 'Multiple nucleic acid tertiary structure alignment',\n",
       "  'def': '\"Alignment (superimposition) of more than two nucleic acid tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1493': {'id': 'EDAM_data:1493',\n",
       "  'name': 'Structure alignment (RNA)',\n",
       "  'def': '\"Alignment (superimposition) of RNA tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1494': {'id': 'EDAM_data:1494',\n",
       "  'name': 'Structural transformation matrix',\n",
       "  'def': '\"Matrix to transform (rotate/translate) 3D coordinates, typically the transformation necessary to superimpose two molecular structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1495': {'id': 'EDAM_data:1495',\n",
       "  'name': 'DaliLite hit table',\n",
       "  'def': '\"DaliLite hit table of protein chain tertiary structure alignment data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1496': {'id': 'EDAM_data:1496',\n",
       "  'name': 'Molecular similarity score',\n",
       "  'def': '\"A score reflecting structural similarities of two molecules.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1497': {'id': 'EDAM_data:1497',\n",
       "  'name': 'Root-mean-square deviation',\n",
       "  'def': '\"Root-mean-square deviation (RMSD) is calculated to measure the average distance between superimposed macromolecular coordinates.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1498': {'id': 'EDAM_data:1498',\n",
       "  'name': 'Tanimoto similarity score',\n",
       "  'def': '\"A measure of the similarity between two ligand fingerprints.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1499': {'id': 'EDAM_data:1499',\n",
       "  'name': '3D-1D scoring matrix',\n",
       "  'def': '\"A matrix of 3D-1D scores reflecting the probability of amino acids to occur in different tertiary structural environments.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1501': {'id': 'EDAM_data:1501',\n",
       "  'name': 'Amino acid index',\n",
       "  'def': '\"A table of 20 numerical values which quantify a property (e.g. physicochemical or biochemical) of the common amino acids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1502': {'id': 'EDAM_data:1502',\n",
       "  'name': 'Amino acid index (chemical classes)',\n",
       "  'def': '\"Chemical classification (small, aliphatic, aromatic, polar, charged etc) of amino acids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1503': {'id': 'EDAM_data:1503',\n",
       "  'name': 'Amino acid pair-wise contact potentials',\n",
       "  'def': '\"Statistical protein contact potentials.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1505': {'id': 'EDAM_data:1505',\n",
       "  'name': 'Amino acid index (molecular weight)',\n",
       "  'def': '\"Molecular weights of amino acids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1506': {'id': 'EDAM_data:1506',\n",
       "  'name': 'Amino acid index (hydropathy)',\n",
       "  'def': '\"Hydrophobic, hydrophilic or charge properties of amino acids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1507': {'id': 'EDAM_data:1507',\n",
       "  'name': 'Amino acid index (White-Wimley data)',\n",
       "  'def': '\"Experimental free energy values for the water-interface and water-octanol transitions for the amino acids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1508': {'id': 'EDAM_data:1508',\n",
       "  'name': 'Amino acid index (van der Waals radii)',\n",
       "  'def': '\"Van der Waals radii of atoms for different amino acid residues.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1509': {'id': 'EDAM_data:1509',\n",
       "  'name': 'Enzyme property',\n",
       "  'def': '\"An informative report on a specific enzyme.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1517': {'id': 'EDAM_data:1517',\n",
       "  'name': 'Restriction enzyme property',\n",
       "  'def': '\"An informative report on a specific restriction enzyme such as enzyme reference data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1519': {'id': 'EDAM_data:1519',\n",
       "  'name': 'Peptide molecular weights',\n",
       "  'def': '\"List of molecular weight(s) of one or more proteins or peptides, for example cut by proteolytic enzymes or reagents.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1520': {'id': 'EDAM_data:1520',\n",
       "  'name': 'Peptide hydrophobic moment',\n",
       "  'def': '\"Report on the hydrophobic moment of a polypeptide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1521': {'id': 'EDAM_data:1521',\n",
       "  'name': 'Protein aliphatic index',\n",
       "  'def': '\"The aliphatic index of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1522': {'id': 'EDAM_data:1522',\n",
       "  'name': 'Protein sequence hydropathy plot',\n",
       "  'def': '\"A protein sequence with annotation on hydrophobic or hydrophilic / charged regions, hydrophobicity plot etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1523': {'id': 'EDAM_data:1523',\n",
       "  'name': 'Protein charge plot',\n",
       "  'def': '\"A plot of the mean charge of the amino acids within a window of specified length as the window is moved along a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1524': {'id': 'EDAM_data:1524',\n",
       "  'name': 'Protein solubility',\n",
       "  'def': '\"The solubility or atomic solvation energy of a protein sequence or structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1525': {'id': 'EDAM_data:1525',\n",
       "  'name': 'Protein crystallizability',\n",
       "  'def': '\"Data on the crystallizability of a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1526': {'id': 'EDAM_data:1526',\n",
       "  'name': 'Protein globularity',\n",
       "  'def': '\"Data on the stability, intrinsic disorder or globularity of a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1527': {'id': 'EDAM_data:1527',\n",
       "  'name': 'Protein titration curve',\n",
       "  'def': '\"The titration curve of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1528': {'id': 'EDAM_data:1528',\n",
       "  'name': 'Protein isoelectric point',\n",
       "  'def': '\"The isoelectric point of one proteins.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1529': {'id': 'EDAM_data:1529',\n",
       "  'name': 'Protein pKa value',\n",
       "  'def': '\"The pKa value of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1530': {'id': 'EDAM_data:1530',\n",
       "  'name': 'Protein hydrogen exchange rate',\n",
       "  'def': '\"The hydrogen exchange rate of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1531': {'id': 'EDAM_data:1531',\n",
       "  'name': 'Protein extinction coefficient',\n",
       "  'def': '\"The extinction coefficient of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1532': {'id': 'EDAM_data:1532',\n",
       "  'name': 'Protein optical density',\n",
       "  'def': '\"The optical density of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1533': {'id': 'EDAM_data:1533',\n",
       "  'name': 'Protein subcellular localization',\n",
       "  'def': '\"An informative report on protein subcellular localization (nuclear, cytoplasmic, mitochondrial, chloroplast, plastid, membrane etc) or destination (exported / extracellular proteins).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1534': {'id': 'EDAM_data:1534',\n",
       "  'name': 'Peptide immunogenicity data',\n",
       "  'def': '\"An report on allergenicity / immunogenicity of peptides and proteins.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1536': {'id': 'EDAM_data:1536',\n",
       "  'name': 'MHC peptide immunogenicity report',\n",
       "  'def': '\"A report on the immunogenicity of MHC class I or class II binding peptides.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1537': {'id': 'EDAM_data:1537',\n",
       "  'name': 'Protein structure report',\n",
       "  'def': '\"Annotation on or structural information derived from one or more specific protein 3D structure(s) or structural domains.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1539': {'id': 'EDAM_data:1539',\n",
       "  'name': 'Protein structural quality report',\n",
       "  'def': '\"Report on the quality of a protein three-dimensional model.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1540': {'id': 'EDAM_data:1540',\n",
       "  'name': 'Protein residue interactions',\n",
       "  'def': '\"Data on inter-atomic or inter-residue contacts, distances and interactions in protein structure(s) or on the interactions of protein atoms or residues with non-protein groups.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1541': {'id': 'EDAM_data:1541',\n",
       "  'name': 'Protein flexibility or motion report',\n",
       "  'def': '\"Informative report on flexibility or motion of a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1542': {'id': 'EDAM_data:1542',\n",
       "  'name': 'Protein solvent accessibility',\n",
       "  'def': '\"Data on the solvent accessible or buried surface area of a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1543': {'id': 'EDAM_data:1543',\n",
       "  'name': 'Protein surface report',\n",
       "  'def': '\"Data on the surface properties (shape, hydropathy, electrostatic patches etc) of a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1544': {'id': 'EDAM_data:1544',\n",
       "  'name': 'Ramachandran plot',\n",
       "  'def': '\"Phi/psi angle data or a Ramachandran plot of a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1545': {'id': 'EDAM_data:1545',\n",
       "  'name': 'Protein dipole moment',\n",
       "  'def': '\"Data on the net charge distribution (dipole moment) of a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1546': {'id': 'EDAM_data:1546',\n",
       "  'name': 'Protein distance matrix',\n",
       "  'def': '\"A matrix of distances between amino acid residues (for example the C-alpha atoms) in a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1547': {'id': 'EDAM_data:1547',\n",
       "  'name': 'Protein contact map',\n",
       "  'def': '\"An amino acid residue contact map for a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1548': {'id': 'EDAM_data:1548',\n",
       "  'name': 'Protein residue 3D cluster',\n",
       "  'def': '\"Report on clusters of contacting residues in protein structures such as a key structural residue network.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1549': {'id': 'EDAM_data:1549',\n",
       "  'name': 'Protein hydrogen bonds',\n",
       "  'def': '\"Patterns of hydrogen bonding in protein structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1550': {'id': 'EDAM_data:1550',\n",
       "  'name': 'Protein non-canonical interactions',\n",
       "  'def': '\"Non-canonical atomic interactions in protein structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1553': {'id': 'EDAM_data:1553',\n",
       "  'name': 'CATH node',\n",
       "  'def': '\"Information on a node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1554': {'id': 'EDAM_data:1554',\n",
       "  'name': 'SCOP node',\n",
       "  'def': '\"Information on a node from the SCOP database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1555': {'id': 'EDAM_data:1555',\n",
       "  'name': 'EMBASSY domain classification',\n",
       "  'def': '\"An EMBASSY domain classification file (DCF) of classification and other data for domains from SCOP or CATH, in EMBL-like format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1556': {'id': 'EDAM_data:1556',\n",
       "  'name': 'CATH class',\n",
       "  'def': '\"Information on a protein \\'class\\' node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1557': {'id': 'EDAM_data:1557',\n",
       "  'name': 'CATH architecture',\n",
       "  'def': '\"Information on a protein \\'architecture\\' node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1558': {'id': 'EDAM_data:1558',\n",
       "  'name': 'CATH topology',\n",
       "  'def': '\"Information on a protein \\'topology\\' node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1559': {'id': 'EDAM_data:1559',\n",
       "  'name': 'CATH homologous superfamily',\n",
       "  'def': '\"Information on a protein \\'homologous superfamily\\' node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1560': {'id': 'EDAM_data:1560',\n",
       "  'name': 'CATH structurally similar group',\n",
       "  'def': '\"Information on a protein \\'structurally similar group\\' node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1561': {'id': 'EDAM_data:1561',\n",
       "  'name': 'CATH functional category',\n",
       "  'def': '\"Information on a protein \\'functional category\\' node from the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1564': {'id': 'EDAM_data:1564',\n",
       "  'name': 'Protein fold recognition report',\n",
       "  'def': '\"A report on known protein structural domains or folds that are recognized (identified) in protein sequence(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1565': {'id': 'EDAM_data:1565',\n",
       "  'name': 'Protein-protein interaction',\n",
       "  'def': '\"Data on protein-protein interaction(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1566': {'id': 'EDAM_data:1566',\n",
       "  'name': 'Protein-ligand interaction',\n",
       "  'def': '\"Data on protein-ligand (small molecule) interaction(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1567': {'id': 'EDAM_data:1567',\n",
       "  'name': 'Protein-nucleic acid interaction',\n",
       "  'def': '\"Data on protein-DNA/RNA interaction(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1583': {'id': 'EDAM_data:1583',\n",
       "  'name': 'Nucleic acid melting profile',\n",
       "  'def': '\"Data on the dissociation characteristics of a double-stranded nucleic acid molecule (DNA or a DNA/RNA hybrid) during heating.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1584': {'id': 'EDAM_data:1584',\n",
       "  'name': 'Nucleic acid enthalpy',\n",
       "  'def': '\"Enthalpy of hybridized or double stranded nucleic acid (DNA or RNA/DNA).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1585': {'id': 'EDAM_data:1585',\n",
       "  'name': 'Nucleic acid entropy',\n",
       "  'def': '\"Entropy of hybridized or double stranded nucleic acid (DNA or RNA/DNA).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1586': {'id': 'EDAM_data:1586',\n",
       "  'name': 'Nucleic acid melting temperature',\n",
       "  'def': '\"Melting temperature of hybridized or double stranded nucleic acid (DNA or RNA/DNA).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1587': {'id': 'EDAM_data:1587',\n",
       "  'name': 'Nucleic acid stitch profile',\n",
       "  'def': '\"Stitch profile of hybridized or double stranded nucleic acid (DNA or RNA/DNA).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1588': {'id': 'EDAM_data:1588',\n",
       "  'name': 'DNA base pair stacking energies data',\n",
       "  'def': '\"DNA base pair stacking energies data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1589': {'id': 'EDAM_data:1589',\n",
       "  'name': 'DNA base pair twist angle data',\n",
       "  'def': '\"DNA base pair twist angle data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1590': {'id': 'EDAM_data:1590',\n",
       "  'name': 'DNA base trimer roll angles data',\n",
       "  'def': '\"DNA base trimer roll angles data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1591': {'id': 'EDAM_data:1591',\n",
       "  'name': 'Vienna RNA parameters',\n",
       "  'def': '\"RNA parameters used by the Vienna package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1592': {'id': 'EDAM_data:1592',\n",
       "  'name': 'Vienna RNA structure constraints',\n",
       "  'def': '\"Structure constraints used by the Vienna package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1593': {'id': 'EDAM_data:1593',\n",
       "  'name': 'Vienna RNA concentration data',\n",
       "  'def': '\"RNA concentration data used by the Vienna package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1594': {'id': 'EDAM_data:1594',\n",
       "  'name': 'Vienna RNA calculated energy',\n",
       "  'def': '\"RNA calculated energy data generated by the Vienna package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1595': {'id': 'EDAM_data:1595',\n",
       "  'name': 'Base pairing probability matrix dotplot',\n",
       "  'def': '\"Dotplot of RNA base pairing probability matrix.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1596': {'id': 'EDAM_data:1596',\n",
       "  'name': 'Nucleic acid folding report',\n",
       "  'def': '\"A report on an analysis of RNA/DNA folding, minimum folding energies for DNA or RNA sequences, energy landscape of RNA mutants etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1597': {'id': 'EDAM_data:1597',\n",
       "  'name': 'Codon usage table',\n",
       "  'def': '\"Table of codon usage data calculated from one or more nucleic acid sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1598': {'id': 'EDAM_data:1598',\n",
       "  'name': 'Genetic code',\n",
       "  'def': '\"A genetic code for an organism.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1599': {'id': 'EDAM_data:1599',\n",
       "  'name': 'Codon adaptation index',\n",
       "  'def': '\"A simple measure of synonymous codon usage bias often used to predict gene expression levels.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1600': {'id': 'EDAM_data:1600',\n",
       "  'name': 'Codon usage bias plot',\n",
       "  'def': '\"A plot of the synonymous codon usage calculated for windows over a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1601': {'id': 'EDAM_data:1601',\n",
       "  'name': 'Nc statistic',\n",
       "  'def': '\"The effective number of codons used in a gene sequence.  This reflects how far codon usage of a gene departs from equal usage of synonymous codons.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1602': {'id': 'EDAM_data:1602',\n",
       "  'name': 'Codon usage fraction difference',\n",
       "  'def': '\"The differences in codon usage fractions between two codon usage tables.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1621': {'id': 'EDAM_data:1621',\n",
       "  'name': 'Pharmacogenomic annotation',\n",
       "  'def': '\"Data on the influence of genotype on drug response.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1622': {'id': 'EDAM_data:1622',\n",
       "  'name': 'Disease annotation',\n",
       "  'def': '\"An informative report on a specific disease.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1634': {'id': 'EDAM_data:1634',\n",
       "  'name': 'Gene annotation (linkage disequilibrium)',\n",
       "  'def': '\"A report on linkage disequilibrium; the non-random association of alleles or polymorphisms at two or more loci (not necessarily on the same chromosome).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1636': {'id': 'EDAM_data:1636',\n",
       "  'name': 'Heat map',\n",
       "  'def': '\"A graphical 2D tabular representation of gene expression data, typically derived from a DNA microarray experiment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1642': {'id': 'EDAM_data:1642',\n",
       "  'name': 'Affymetrix probe sets library file',\n",
       "  'def': '\"Affymetrix library file of information about which probes belong to which probe set.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1643': {'id': 'EDAM_data:1643',\n",
       "  'name': 'Affymetrix probe sets information library file',\n",
       "  'def': '\"Affymetrix library file of information about the probe sets such as the gene name with which the probe set is associated.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1646': {'id': 'EDAM_data:1646',\n",
       "  'name': 'Molecular weights standard fingerprint',\n",
       "  'def': '\"Standard protonated molecular masses from trypsin (modified porcine trypsin, Promega) and keratin peptides, used in EMBOSS.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1656': {'id': 'EDAM_data:1656',\n",
       "  'name': 'Pathway or network (metabolic)',\n",
       "  'def': '\"A report typically including a map (diagram) of a metabolic pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1657': {'id': 'EDAM_data:1657',\n",
       "  'name': 'Pathway or network (genetic information processing)',\n",
       "  'def': '\"A report typically including a map (diagram) of a genetic information processing pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1658': {'id': 'EDAM_data:1658',\n",
       "  'name': 'Pathway or network (environmental information processing)',\n",
       "  'def': '\"A report typically including a map (diagram) of an environmental information processing pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1659': {'id': 'EDAM_data:1659',\n",
       "  'name': 'Pathway or network (signal transduction)',\n",
       "  'def': '\"A report typically including a map (diagram) of a signal transduction pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1660': {'id': 'EDAM_data:1660',\n",
       "  'name': 'Pathway or network (cellular process)',\n",
       "  'def': '\"A report typically including a map (diagram) of a cellular process pathway.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1661': {'id': 'EDAM_data:1661',\n",
       "  'name': 'Pathway or network (disease)',\n",
       "  'def': '\"A report typically including a map (diagram) of a disease pathway, typically a diagram for a human disease.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1662': {'id': 'EDAM_data:1662',\n",
       "  'name': 'Pathway or network (drug structure relationship)',\n",
       "  'def': '\"A report typically including a map (diagram) of drug structure relationships.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1663': {'id': 'EDAM_data:1663',\n",
       "  'name': 'Pathway or network (protein-protein interaction)',\n",
       "  'def': '\"A report typically including a map (diagram) of a network of protein interactions.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1664': {'id': 'EDAM_data:1664',\n",
       "  'name': 'MIRIAM datatype',\n",
       "  'def': '\"An entry (data type) from the Minimal Information Requested in the Annotation of Biochemical Models (MIRIAM) database of data resources.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1667': {'id': 'EDAM_data:1667',\n",
       "  'name': 'E-value',\n",
       "  'def': '\"A simple floating point number defining the lower or upper limit of an expectation value (E-value).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1668': {'id': 'EDAM_data:1668',\n",
       "  'name': 'Z-value',\n",
       "  'def': '\"The z-value is the number of standard deviations a data value is above or below a mean value.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1669': {'id': 'EDAM_data:1669',\n",
       "  'name': 'P-value',\n",
       "  'def': '\"The P-value is the probability of obtaining by random chance a result that is at least as extreme as an observed result, assuming a NULL hypothesis is true.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1670': {'id': 'EDAM_data:1670',\n",
       "  'name': 'Database version information',\n",
       "  'def': '\"Information on a database (or ontology) version, for example name, version number and release date.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1671': {'id': 'EDAM_data:1671',\n",
       "  'name': 'Tool version information',\n",
       "  'def': '\"Information on an application version, for example name, version number and release date.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1672': {'id': 'EDAM_data:1672',\n",
       "  'name': 'CATH version information',\n",
       "  'def': '\"Information on a version of the CATH database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1673': {'id': 'EDAM_data:1673',\n",
       "  'name': 'Swiss-Prot to PDB mapping',\n",
       "  'def': '\"Cross-mapping of Swiss-Prot codes to PDB identifiers.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1674': {'id': 'EDAM_data:1674',\n",
       "  'name': 'Sequence database cross-references',\n",
       "  'def': '\"Cross-references from a sequence record to other databases.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1675': {'id': 'EDAM_data:1675',\n",
       "  'name': 'Job status',\n",
       "  'def': '\"Metadata on the status of a submitted job.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1676': {'id': 'EDAM_data:1676',\n",
       "  'name': 'Job ID',\n",
       "  'def': '\"The (typically numeric) unique identifier of a submitted job.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1677': {'id': 'EDAM_data:1677',\n",
       "  'name': 'Job type',\n",
       "  'def': '\"A label (text token) describing the type of job, for example interactive or non-interactive.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1678': {'id': 'EDAM_data:1678',\n",
       "  'name': 'Tool log',\n",
       "  'def': '\"A report of tool-specific metadata on some analysis or process performed, for example a log of diagnostic or error messages.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1679': {'id': 'EDAM_data:1679',\n",
       "  'name': 'DaliLite log file',\n",
       "  'def': '\"DaliLite log file describing all the steps taken by a DaliLite alignment of two protein structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1680': {'id': 'EDAM_data:1680',\n",
       "  'name': 'STRIDE log file',\n",
       "  'def': '\"STRIDE log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1681': {'id': 'EDAM_data:1681',\n",
       "  'name': 'NACCESS log file',\n",
       "  'def': '\"NACCESS log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1682': {'id': 'EDAM_data:1682',\n",
       "  'name': 'EMBOSS wordfinder log file',\n",
       "  'def': '\"EMBOSS wordfinder log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1683': {'id': 'EDAM_data:1683',\n",
       "  'name': 'EMBOSS domainatrix log file',\n",
       "  'def': '\"EMBOSS (EMBASSY) domainatrix application log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1684': {'id': 'EDAM_data:1684',\n",
       "  'name': 'EMBOSS sites log file',\n",
       "  'def': '\"EMBOSS (EMBASSY) sites application log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1685': {'id': 'EDAM_data:1685',\n",
       "  'name': 'EMBOSS supermatcher error file',\n",
       "  'def': '\"EMBOSS (EMBASSY) supermatcher error file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1686': {'id': 'EDAM_data:1686',\n",
       "  'name': 'EMBOSS megamerger log file',\n",
       "  'def': '\"EMBOSS megamerger log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1687': {'id': 'EDAM_data:1687',\n",
       "  'name': 'EMBOSS whichdb log file',\n",
       "  'def': '\"EMBOSS megamerger log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1688': {'id': 'EDAM_data:1688',\n",
       "  'name': 'EMBOSS vectorstrip log file',\n",
       "  'def': '\"EMBOSS vectorstrip log file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1689': {'id': 'EDAM_data:1689',\n",
       "  'name': 'Username',\n",
       "  'def': '\"A username on a computer system.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1690': {'id': 'EDAM_data:1690',\n",
       "  'name': 'Password',\n",
       "  'def': '\"A password on a computer system.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1691': {'id': 'EDAM_data:1691',\n",
       "  'name': 'Email address',\n",
       "  'def': '\"A valid email address of an end-user.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1692': {'id': 'EDAM_data:1692',\n",
       "  'name': 'Person name',\n",
       "  'def': '\"The name of a person.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1693': {'id': 'EDAM_data:1693',\n",
       "  'name': 'Number of iterations',\n",
       "  'def': '\"Number of iterations of an algorithm.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1694': {'id': 'EDAM_data:1694',\n",
       "  'name': 'Number of output entities',\n",
       "  'def': '\"Number of entities (for example database hits, sequences, alignments etc) to write to an output file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1695': {'id': 'EDAM_data:1695',\n",
       "  'name': 'Hit sort order',\n",
       "  'def': '\"Controls the order of hits (reported matches) in an output file from a database search.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1696': {'id': 'EDAM_data:1696',\n",
       "  'name': 'Drug annotation',\n",
       "  'def': '\"An informative report on a specific drug.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1707': {'id': 'EDAM_data:1707',\n",
       "  'name': 'Phylogenetic tree image',\n",
       "  'def': '\"An image (for viewing or printing) of a phylogenetic tree including (typically) a plot of rooted or unrooted phylogenies, cladograms, circular trees or phenograms and associated information.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1708': {'id': 'EDAM_data:1708',\n",
       "  'name': 'RNA secondary structure image',\n",
       "  'def': '\"Image of RNA secondary structure, knots, pseudoknots etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1709': {'id': 'EDAM_data:1709',\n",
       "  'name': 'Protein secondary structure image',\n",
       "  'def': '\"Image of protein secondary structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1710': {'id': 'EDAM_data:1710',\n",
       "  'name': 'Structure image',\n",
       "  'def': '\"Image of one or more molecular tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1711': {'id': 'EDAM_data:1711',\n",
       "  'name': 'Sequence alignment image',\n",
       "  'def': '\"Image of two or more aligned molecular sequences possibly annotated with alignment features.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1712': {'id': 'EDAM_data:1712',\n",
       "  'name': 'Structure image (small molecule)',\n",
       "  'def': '\"An image of the structure of a small chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1713': {'id': 'EDAM_data:1713',\n",
       "  'name': 'Fate map',\n",
       "  'def': '\" A fate map is a plan of early stage of an embryo such as a blastula, showing areas that are significance to development.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1714': {'id': 'EDAM_data:1714',\n",
       "  'name': 'Microarray spots image',\n",
       "  'def': '\"An image of spots from a microarray experiment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1715': {'id': 'EDAM_data:1715',\n",
       "  'name': 'BioPax',\n",
       "  'def': '\"A term from the BioPax ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1716': {'id': 'EDAM_data:1716',\n",
       "  'name': 'GO',\n",
       "  'def': '\"A term definition from The Gene Ontology (GO).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1717': {'id': 'EDAM_data:1717',\n",
       "  'name': 'MeSH',\n",
       "  'def': '\"A term from the MeSH vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1718': {'id': 'EDAM_data:1718',\n",
       "  'name': 'HGNC',\n",
       "  'def': '\"A term from the HGNC controlled vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1719': {'id': 'EDAM_data:1719',\n",
       "  'name': 'NCBI taxonomy vocabulary',\n",
       "  'def': '\"A term from the NCBI taxonomy vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1720': {'id': 'EDAM_data:1720',\n",
       "  'name': 'Plant ontology term',\n",
       "  'def': '\"A term from the Plant Ontology (PO).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1721': {'id': 'EDAM_data:1721',\n",
       "  'name': 'UMLS',\n",
       "  'def': '\"A term from the UMLS vocabulary.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1722': {'id': 'EDAM_data:1722',\n",
       "  'name': 'FMA',\n",
       "  'def': '\"A term from Foundational Model of Anatomy.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1723': {'id': 'EDAM_data:1723',\n",
       "  'name': 'EMAP',\n",
       "  'def': '\"A term from the EMAP mouse ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1724': {'id': 'EDAM_data:1724',\n",
       "  'name': 'ChEBI',\n",
       "  'def': '\"A term from the ChEBI ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1725': {'id': 'EDAM_data:1725',\n",
       "  'name': 'MGED',\n",
       "  'def': '\"A term from the MGED ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1726': {'id': 'EDAM_data:1726',\n",
       "  'name': 'myGrid',\n",
       "  'def': '\"A term from the myGrid ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1727': {'id': 'EDAM_data:1727',\n",
       "  'name': 'GO (biological process)',\n",
       "  'def': '\"A term definition for a biological process from the Gene Ontology (GO).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1728': {'id': 'EDAM_data:1728',\n",
       "  'name': 'GO (molecular function)',\n",
       "  'def': '\"A term definition for a molecular function from the Gene Ontology (GO).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1729': {'id': 'EDAM_data:1729',\n",
       "  'name': 'GO (cellular component)',\n",
       "  'def': '\"A term definition for a cellular component from the Gene Ontology (GO).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1730': {'id': 'EDAM_data:1730',\n",
       "  'name': 'Ontology relation type',\n",
       "  'def': '\"A relation type defined in an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1731': {'id': 'EDAM_data:1731',\n",
       "  'name': 'Ontology concept definition',\n",
       "  'def': '\"The definition of a concept from an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1732': {'id': 'EDAM_data:1732',\n",
       "  'name': 'Ontology concept comment',\n",
       "  'def': '\"A comment on a concept from an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1733': {'id': 'EDAM_data:1733',\n",
       "  'name': 'Ontology concept reference',\n",
       "  'def': '\"Reference for a concept from an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1738': {'id': 'EDAM_data:1738',\n",
       "  'name': 'doc2loc document information',\n",
       "  'def': '\"Information on a published article provided by the doc2loc program.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1742': {'id': 'EDAM_data:1742',\n",
       "  'name': 'PDB residue number',\n",
       "  'def': '\"A residue identifier (a string) from a PDB file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1743': {'id': 'EDAM_data:1743',\n",
       "  'name': 'Atomic coordinate',\n",
       "  'def': '\"Cartesian coordinate of an atom (in a molecular structure).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1744': {'id': 'EDAM_data:1744',\n",
       "  'name': 'Atomic x coordinate',\n",
       "  'def': '\"Cartesian x coordinate of an atom (in a molecular structure).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1745': {'id': 'EDAM_data:1745',\n",
       "  'name': 'Atomic y coordinate',\n",
       "  'def': '\"Cartesian y coordinate of an atom (in a molecular structure).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1746': {'id': 'EDAM_data:1746',\n",
       "  'name': 'Atomic z coordinate',\n",
       "  'def': '\"Cartesian z coordinate of an atom (in a molecular structure).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1748': {'id': 'EDAM_data:1748',\n",
       "  'name': 'PDB atom name',\n",
       "  'def': '\"Identifier (a string) of a specific atom from a PDB file for a molecular structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1755': {'id': 'EDAM_data:1755',\n",
       "  'name': 'Protein atom',\n",
       "  'def': '\"Data on a single atom from a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1756': {'id': 'EDAM_data:1756',\n",
       "  'name': 'Protein residue',\n",
       "  'def': '\"Data on a single amino acid residue position in a protein structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1757': {'id': 'EDAM_data:1757',\n",
       "  'name': 'Atom name',\n",
       "  'def': '\"Name of an atom.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1758': {'id': 'EDAM_data:1758',\n",
       "  'name': 'PDB residue name',\n",
       "  'def': '\"Three-letter amino acid residue names as used in PDB files.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1759': {'id': 'EDAM_data:1759',\n",
       "  'name': 'PDB model number',\n",
       "  'def': '\"Identifier of a model structure from a PDB file.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1762': {'id': 'EDAM_data:1762',\n",
       "  'name': 'CATH domain report',\n",
       "  'def': '\"Summary of domain classification information for a CATH domain.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1764': {'id': 'EDAM_data:1764',\n",
       "  'name': 'CATH representative domain sequences (ATOM)',\n",
       "  'def': '\"FASTA sequence database (based on ATOM records in PDB) for CATH domains (clustered at different levels of sequence identity).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1765': {'id': 'EDAM_data:1765',\n",
       "  'name': 'CATH representative domain sequences (COMBS)',\n",
       "  'def': '\"FASTA sequence database (based on COMBS sequence data) for CATH domains (clustered at different levels of sequence identity).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1766': {'id': 'EDAM_data:1766',\n",
       "  'name': 'CATH domain sequences (ATOM)',\n",
       "  'def': '\"FASTA sequence database for all CATH domains (based on PDB ATOM records).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1767': {'id': 'EDAM_data:1767',\n",
       "  'name': 'CATH domain sequences (COMBS)',\n",
       "  'def': '\"FASTA sequence database for all CATH domains (based on COMBS sequence data).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1771': {'id': 'EDAM_data:1771',\n",
       "  'name': 'Sequence version information',\n",
       "  'def': '\"Information on an molecular sequence version.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1772': {'id': 'EDAM_data:1772',\n",
       "  'name': 'Score or penalty',\n",
       "  'def': '\"A numerical value, either some type of scored value arising for example from a prediction method or a value used in a scoring scheme, which might reduce the final score (penalty).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1776': {'id': 'EDAM_data:1776',\n",
       "  'name': 'Protein report (function)',\n",
       "  'def': '\"Report on general functional properties of specific protein(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1783': {'id': 'EDAM_data:1783',\n",
       "  'name': 'Gene name (ASPGD)',\n",
       "  'def': '\"Name of a gene from Aspergillus Genome Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1784': {'id': 'EDAM_data:1784',\n",
       "  'name': 'Gene name (CGD)',\n",
       "  'def': '\"Name of a gene from Candida Genome Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1785': {'id': 'EDAM_data:1785',\n",
       "  'name': 'Gene name (dictyBase)',\n",
       "  'def': '\"Name of a gene from dictyBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1786': {'id': 'EDAM_data:1786',\n",
       "  'name': 'Gene name (EcoGene primary)',\n",
       "  'def': '\"Primary name of a gene from EcoGene Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1787': {'id': 'EDAM_data:1787',\n",
       "  'name': 'Gene name (MaizeGDB)',\n",
       "  'def': '\"Name of a gene from MaizeGDB (maize genes) database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1788': {'id': 'EDAM_data:1788',\n",
       "  'name': 'Gene name (SGD)',\n",
       "  'def': '\"Name of a gene from Saccharomyces Genome Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1789': {'id': 'EDAM_data:1789',\n",
       "  'name': 'Gene name (TGD)',\n",
       "  'def': '\"Name of a gene from Tetrahymena Genome Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1790': {'id': 'EDAM_data:1790',\n",
       "  'name': 'Gene name (CGSC)',\n",
       "  'def': '\"Symbol of a gene from E.coli Genetic Stock Center.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1791': {'id': 'EDAM_data:1791',\n",
       "  'name': 'Gene name (HGNC)',\n",
       "  'def': '\"Symbol of a gene approved by the HUGO Gene Nomenclature Committee.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1792': {'id': 'EDAM_data:1792',\n",
       "  'name': 'Gene name (MGD)',\n",
       "  'def': '\"Symbol of a gene from the Mouse Genome Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1793': {'id': 'EDAM_data:1793',\n",
       "  'name': 'Gene name (Bacillus subtilis)',\n",
       "  'def': '\"Symbol of a gene from Bacillus subtilis Genome Sequence Project.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1794': {'id': 'EDAM_data:1794',\n",
       "  'name': 'Gene ID (PlasmoDB)',\n",
       "  'def': '\"Identifier of a gene from PlasmoDB Plasmodium Genome Resource.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1795': {'id': 'EDAM_data:1795',\n",
       "  'name': 'Gene ID (EcoGene)',\n",
       "  'def': '\"Identifier of a gene from EcoGene Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1796': {'id': 'EDAM_data:1796',\n",
       "  'name': 'Gene ID (FlyBase)',\n",
       "  'def': '\"Gene identifier from FlyBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1797': {'id': 'EDAM_data:1797',\n",
       "  'name': 'Gene ID (GeneDB Glossina morsitans)',\n",
       "  'def': '\"Gene identifier from Glossina morsitans GeneDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1798': {'id': 'EDAM_data:1798',\n",
       "  'name': 'Gene ID (GeneDB Leishmania major)',\n",
       "  'def': '\"Gene identifier from Leishmania major GeneDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1799': {'id': 'EDAM_data:1799',\n",
       "  'name': 'Gene ID (GeneDB Plasmodium falciparum)',\n",
       "  'def': '\"Gene identifier from Plasmodium falciparum GeneDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1800': {'id': 'EDAM_data:1800',\n",
       "  'name': 'Gene ID (GeneDB Schizosaccharomyces pombe)',\n",
       "  'def': '\"Gene identifier from Schizosaccharomyces pombe GeneDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1801': {'id': 'EDAM_data:1801',\n",
       "  'name': 'Gene ID (GeneDB Trypanosoma brucei)',\n",
       "  'def': '\"Gene identifier from Trypanosoma brucei GeneDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1802': {'id': 'EDAM_data:1802',\n",
       "  'name': 'Gene ID (Gramene)',\n",
       "  'def': '\"Gene identifier from Gramene database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1803': {'id': 'EDAM_data:1803',\n",
       "  'name': 'Gene ID (Virginia microbial)',\n",
       "  'def': '\"Gene identifier from Virginia Bioinformatics Institute microbial database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1804': {'id': 'EDAM_data:1804',\n",
       "  'name': 'Gene ID (SGN)',\n",
       "  'def': '\"Gene identifier from Sol Genomics Network.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1805': {'id': 'EDAM_data:1805',\n",
       "  'name': 'Gene ID (WormBase)',\n",
       "  'def': '\"Gene identifier used by WormBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1806': {'id': 'EDAM_data:1806',\n",
       "  'name': 'Gene synonym',\n",
       "  'def': '\"Any name (other than the recommended one) for a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1807': {'id': 'EDAM_data:1807',\n",
       "  'name': 'ORF name',\n",
       "  'def': '\"The name of an open reading frame attributed by a sequencing project.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1852': {'id': 'EDAM_data:1852',\n",
       "  'name': 'Sequence assembly component',\n",
       "  'def': '\"A component of a larger sequence assembly.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1853': {'id': 'EDAM_data:1853',\n",
       "  'name': 'Chromosome annotation (aberration)',\n",
       "  'def': '\"A report on a chromosome aberration such as abnormalities in chromosome structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1855': {'id': 'EDAM_data:1855',\n",
       "  'name': 'Clone ID',\n",
       "  'def': '\"An identifier of a clone (cloned molecular sequence) from a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1856': {'id': 'EDAM_data:1856',\n",
       "  'name': 'PDB insertion code',\n",
       "  'def': '\"An insertion code (part of the residue number) for an amino acid residue from a PDB file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1857': {'id': 'EDAM_data:1857',\n",
       "  'name': 'Atomic occupancy',\n",
       "  'def': '\"The fraction of an atom type present at a site in a molecular structure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1858': {'id': 'EDAM_data:1858',\n",
       "  'name': 'Isotropic B factor',\n",
       "  'def': '\"Isotropic B factor (atomic displacement parameter) for an atom from a PDB file.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1859': {'id': 'EDAM_data:1859',\n",
       "  'name': 'Deletion map',\n",
       "  'def': '\"A cytogenetic map showing chromosome banding patterns in mutant cell lines relative to the wild type.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1860': {'id': 'EDAM_data:1860',\n",
       "  'name': 'QTL map',\n",
       "  'def': '\"A genetic map which shows the approximate location of quantitative trait loci (QTL) between two or more markers.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1863': {'id': 'EDAM_data:1863',\n",
       "  'name': 'Haplotype map',\n",
       "  'def': '\"A map of haplotypes in a genome or other sequence, describing common patterns of genetic variation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1864': {'id': 'EDAM_data:1864',\n",
       "  'name': 'Map set',\n",
       "  'def': '\"Data describing a set of multiple genetic or physical maps, typically sharing a common set of features which are mapped.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1865': {'id': 'EDAM_data:1865',\n",
       "  'name': 'Map feature',\n",
       "  'def': '\"A feature which may mapped (positioned) on a genetic or other type of map.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1866': {'id': 'EDAM_data:1866',\n",
       "  'name': 'Map type',\n",
       "  'def': '\"A designation of the type of map (genetic map, physical map, sequence map etc) or map set.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1867': {'id': 'EDAM_data:1867',\n",
       "  'name': 'Protein fold name',\n",
       "  'def': '\"The name of a protein fold.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1868': {'id': 'EDAM_data:1868',\n",
       "  'name': 'Taxon',\n",
       "  'def': '\"The name of a group of organisms belonging to the same taxonomic rank.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1869': {'id': 'EDAM_data:1869',\n",
       "  'name': 'Organism identifier',\n",
       "  'def': '\"A unique identifier of a (group of) organisms.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1870': {'id': 'EDAM_data:1870',\n",
       "  'name': 'Genus name',\n",
       "  'def': '\"The name of a genus of organism.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1872': {'id': 'EDAM_data:1872',\n",
       "  'name': 'Taxonomic classification',\n",
       "  'def': '\"The full name for a group of organisms, reflecting their biological classification and (usually) conforming to a standard nomenclature.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1873': {'id': 'EDAM_data:1873',\n",
       "  'name': 'iHOP organism ID',\n",
       "  'def': '\"A unique identifier for an organism used in the iHOP database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1874': {'id': 'EDAM_data:1874',\n",
       "  'name': 'Genbank common name',\n",
       "  'def': '\"Common name for an organism as used in the GenBank database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1875': {'id': 'EDAM_data:1875',\n",
       "  'name': 'NCBI taxon',\n",
       "  'def': '\"The name of a taxon from the NCBI taxonomy database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1877': {'id': 'EDAM_data:1877',\n",
       "  'name': 'Synonym',\n",
       "  'def': '\"An alternative for a word.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1878': {'id': 'EDAM_data:1878',\n",
       "  'name': 'Misspelling',\n",
       "  'def': '\"A common misspelling of a word.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1879': {'id': 'EDAM_data:1879',\n",
       "  'name': 'Acronym',\n",
       "  'def': '\"An abbreviation of a phrase or word.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1880': {'id': 'EDAM_data:1880',\n",
       "  'name': 'Misnomer',\n",
       "  'def': '\"A term which is likely to be misleading of its meaning.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1881': {'id': 'EDAM_data:1881',\n",
       "  'name': 'Author ID',\n",
       "  'def': '\"Information on the authors of a published work.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1882': {'id': 'EDAM_data:1882',\n",
       "  'name': 'DragonDB author identifier',\n",
       "  'def': '\"An identifier representing an author in the DragonDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1883': {'id': 'EDAM_data:1883',\n",
       "  'name': 'Annotated URI',\n",
       "  'def': '\"A URI along with annotation describing the data found at the address.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1884': {'id': 'EDAM_data:1884',\n",
       "  'name': 'UniProt keywords',\n",
       "  'def': '\"A controlled vocabulary for words and phrases that can appear in the keywords field (KW line) of entries from the UniProt database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1885': {'id': 'EDAM_data:1885',\n",
       "  'name': 'Gene ID (GeneFarm)',\n",
       "  'def': '\"Identifier of a gene from the GeneFarm database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1886': {'id': 'EDAM_data:1886',\n",
       "  'name': 'Blattner number',\n",
       "  'def': '\"The blattner identifier for a gene.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1887': {'id': 'EDAM_data:1887',\n",
       "  'name': 'Gene ID (MIPS Maize)',\n",
       "  'def': '\"Identifier for genetic elements in MIPS Maize database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1888': {'id': 'EDAM_data:1888',\n",
       "  'name': 'Gene ID (MIPS Medicago)',\n",
       "  'def': '\"Identifier for genetic elements in MIPS Medicago database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1889': {'id': 'EDAM_data:1889',\n",
       "  'name': 'Gene name (DragonDB)',\n",
       "  'def': '\"The name of an Antirrhinum Gene from the DragonDB database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1890': {'id': 'EDAM_data:1890',\n",
       "  'name': 'Gene name (Arabidopsis)',\n",
       "  'def': '\"A unique identifier for an Arabidopsis gene, which is an acronym or abbreviation of the gene name.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1891': {'id': 'EDAM_data:1891',\n",
       "  'name': 'iHOP symbol',\n",
       "  'def': '\"A unique identifier of a protein or gene used in the iHOP database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1892': {'id': 'EDAM_data:1892',\n",
       "  'name': 'Gene name (GeneFarm)',\n",
       "  'def': '\"Name of a gene from the GeneFarm database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1893': {'id': 'EDAM_data:1893',\n",
       "  'name': 'Locus ID',\n",
       "  'def': '\"A unique name or other identifier of a genetic locus, typically conforming to a scheme that names loci (such as predicted genes) depending on their position in a molecular sequence, for example a completely sequenced genome or chromosome.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1895': {'id': 'EDAM_data:1895',\n",
       "  'name': 'Locus ID (AGI)',\n",
       "  'def': '\"Locus identifier for Arabidopsis Genome Initiative (TAIR, TIGR and MIPS databases)\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1896': {'id': 'EDAM_data:1896',\n",
       "  'name': 'Locus ID (ASPGD)',\n",
       "  'def': '\"Identifier for loci from ASPGD (Aspergillus Genome Database).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1897': {'id': 'EDAM_data:1897',\n",
       "  'name': 'Locus ID (MGG)',\n",
       "  'def': '\"Identifier for loci from Magnaporthe grisea Database at the Broad Institute.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1898': {'id': 'EDAM_data:1898',\n",
       "  'name': 'Locus ID (CGD)',\n",
       "  'def': '\"Identifier for loci from CGD (Candida Genome Database).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1899': {'id': 'EDAM_data:1899',\n",
       "  'name': 'Locus ID (CMR)',\n",
       "  'def': '\"Locus identifier for Comprehensive Microbial Resource at the J. Craig Venter Institute.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1900': {'id': 'EDAM_data:1900',\n",
       "  'name': 'NCBI locus tag',\n",
       "  'def': '\"Identifier for loci from NCBI database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1901': {'id': 'EDAM_data:1901',\n",
       "  'name': 'Locus ID (SGD)',\n",
       "  'def': '\"Identifier for loci from SGD (Saccharomyces Genome Database).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1902': {'id': 'EDAM_data:1902',\n",
       "  'name': 'Locus ID (MMP)',\n",
       "  'def': '\"Identifier of loci from Maize Mapping Project.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1903': {'id': 'EDAM_data:1903',\n",
       "  'name': 'Locus ID (DictyBase)',\n",
       "  'def': '\"Identifier of locus from DictyBase (Dictyostelium discoideum).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1904': {'id': 'EDAM_data:1904',\n",
       "  'name': 'Locus ID (EntrezGene)',\n",
       "  'def': '\"Identifier of a locus from EntrezGene database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1905': {'id': 'EDAM_data:1905',\n",
       "  'name': 'Locus ID (MaizeGDB)',\n",
       "  'def': '\"Identifier of locus from MaizeGDB (Maize genome database).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1906': {'id': 'EDAM_data:1906',\n",
       "  'name': 'Quantitative trait locus',\n",
       "  'def': '\"A stretch of DNA that is closely linked to the genes underlying a quantitative trait (a phenotype that varies in degree and depends upon the interactions between multiple genes and their environment).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1907': {'id': 'EDAM_data:1907',\n",
       "  'name': 'Gene ID (KOME)',\n",
       "  'def': '\"Identifier of a gene from the KOME database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1908': {'id': 'EDAM_data:1908',\n",
       "  'name': 'Locus ID (Tropgene)',\n",
       "  'def': '\"Identifier of a locus from the Tropgene database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:1916': {'id': 'EDAM_data:1916',\n",
       "  'name': 'Alignment',\n",
       "  'def': '\"An alignment of molecular sequences, structures or profiles derived from them.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:1917': {'id': 'EDAM_data:1917',\n",
       "  'name': 'Atomic property',\n",
       "  'def': '\"Data for an atom (in a molecular structure).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2007': {'id': 'EDAM_data:2007',\n",
       "  'name': 'UniProt keyword',\n",
       "  'def': '\"A word or phrase that can appear in the keywords field (KW line) of entries from the UniProt database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2009': {'id': 'EDAM_data:2009',\n",
       "  'name': 'Ordered locus name',\n",
       "  'def': '\"A name for a genetic locus conforming to a scheme that names loci (such as predicted genes) depending on their position in a molecular sequence, for example a completely sequenced genome or chromosome.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2012': {'id': 'EDAM_data:2012',\n",
       "  'name': 'Map position',\n",
       "  'def': '\"A position in a map (for example a genetic map), either a single position (point) or a region / interval.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2016': {'id': 'EDAM_data:2016',\n",
       "  'name': 'Amino acid property',\n",
       "  'def': '\"Data concerning the intrinsic physical (e.g. structural) or chemical properties of one, more or all amino acids.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2018': {'id': 'EDAM_data:2018',\n",
       "  'name': 'Annotation',\n",
       "  'def': '\"A human-readable collection of information which (typically) is generated or collated by hand and which describes a biological entity, phenomena or associated primary (e.g. sequence or structural) data, as distinct from the primary data itself and computer-generated reports derived from it.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2019': {'id': 'EDAM_data:2019',\n",
       "  'name': 'Map attribute',\n",
       "  'def': '\"An attribute of a molecular map (genetic or physical).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2022': {'id': 'EDAM_data:2022',\n",
       "  'name': 'Vienna RNA structural data',\n",
       "  'def': '\"Data used by the Vienna RNA analysis package.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2023': {'id': 'EDAM_data:2023',\n",
       "  'name': 'Sequence mask parameter',\n",
       "  'def': '\"Data used to replace (mask) characters in a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2024': {'id': 'EDAM_data:2024',\n",
       "  'name': 'Enzyme kinetics data',\n",
       "  'def': '\"Data concerning chemical reaction(s) catalysed by enzyme(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2025': {'id': 'EDAM_data:2025',\n",
       "  'name': 'Michaelis Menten plot',\n",
       "  'def': '\"A plot giving an approximation of the kinetics of an enzyme-catalysed reaction, assuming simple kinetics (i.e. no intermediate or product inhibition, allostericity or cooperativity). It plots initial reaction rate to the substrate concentration (S) from which the maximum rate (vmax) is apparent.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2026': {'id': 'EDAM_data:2026',\n",
       "  'name': 'Hanes Woolf plot',\n",
       "  'def': '\"A plot based on the Michaelis Menten equation of enzyme kinetics plotting the ratio of the initial substrate concentration (S) against the reaction velocity (v).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2028': {'id': 'EDAM_data:2028',\n",
       "  'name': 'Experimental data',\n",
       "  'def': '\"Raw data from or annotation on laboratory experiments.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2041': {'id': 'EDAM_data:2041',\n",
       "  'name': 'Genome version information',\n",
       "  'def': '\"Information on a genome version.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2042': {'id': 'EDAM_data:2042',\n",
       "  'name': 'Evidence',\n",
       "  'def': '\"Typically a statement about some data or results, including evidence or the source of a statement, which may include computational prediction, laboratory experiment, literature reference etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2043': {'id': 'EDAM_data:2043',\n",
       "  'name': 'Sequence record lite',\n",
       "  'def': '\"A molecular sequence and minimal metadata, typically an identifier of the sequence and/or a comment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2044': {'id': 'EDAM_data:2044',\n",
       "  'name': 'Sequence',\n",
       "  'def': '\"One or more molecular sequences, possibly with associated annotation.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2046': {'id': 'EDAM_data:2046',\n",
       "  'name': 'Sequence record lite (nucleic acid)',\n",
       "  'def': '\"A nucleic acid sequence and minimal metadata, typically an identifier of the sequence and/or a comment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2047': {'id': 'EDAM_data:2047',\n",
       "  'name': 'Sequence record lite (protein)',\n",
       "  'def': '\"A protein sequence and minimal metadata, typically an identifier of the sequence and/or a comment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2048': {'id': 'EDAM_data:2048',\n",
       "  'name': 'Report',\n",
       "  'def': '\"A human-readable collection of information including annotation on a biological entity or phenomena, computer-generated reports of analysis of primary (e.g. sequence or structural) data, metadata about the primary data, and any free (essentially unformatted) text, as distinct from the primary data itself.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2050': {'id': 'EDAM_data:2050',\n",
       "  'name': 'Molecular property (general)',\n",
       "  'def': '\"General data for a molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2053': {'id': 'EDAM_data:2053',\n",
       "  'name': 'Structural data',\n",
       "  'def': '\"Data concerning molecular structural data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2070': {'id': 'EDAM_data:2070',\n",
       "  'name': 'Sequence motif (nucleic acid)',\n",
       "  'def': '\"A nucleotide sequence motif.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2071': {'id': 'EDAM_data:2071',\n",
       "  'name': 'Sequence motif (protein)',\n",
       "  'def': '\"An amino acid sequence motif.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2079': {'id': 'EDAM_data:2079',\n",
       "  'name': 'Search parameter',\n",
       "  'def': '\"Some simple value controlling a search operation, typically a search of a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2080': {'id': 'EDAM_data:2080',\n",
       "  'name': 'Database hits',\n",
       "  'def': '\"A report of hits from searching a database of some type.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2081': {'id': 'EDAM_data:2081',\n",
       "  'name': 'Secondary structure',\n",
       "  'def': '\"The secondary structure assignment (predicted or real) of a nucleic acid or protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2082': {'id': 'EDAM_data:2082',\n",
       "  'name': 'Matrix',\n",
       "  'def': '\"An array of numerical values where (typically) the rows and columns correspond to molecular entities and the values are comparative data, for example, distances between molecular sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2083': {'id': 'EDAM_data:2083',\n",
       "  'name': 'Alignment report',\n",
       "  'def': '\"An informative report about a molecular alignment of some type, including alignment-derived data or metadata.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2084': {'id': 'EDAM_data:2084',\n",
       "  'name': 'Nucleic acid report',\n",
       "  'def': '\"An informative report about one or more specific nucleic acid molecules, derived from analysis of primary (sequence or structural) data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2085': {'id': 'EDAM_data:2085',\n",
       "  'name': 'Structure report',\n",
       "  'def': '\"An informative report on general information, properties or features of one or more molecular tertiary (3D) structures.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2086': {'id': 'EDAM_data:2086',\n",
       "  'name': 'Nucleic acid structure report',\n",
       "  'def': '\"A report on nucleic acid structure-derived data, describing structural properties of a DNA molecule, or any other annotation or information about specific nucleic acid 3D structure(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2087': {'id': 'EDAM_data:2087',\n",
       "  'name': 'Molecular property',\n",
       "  'def': '\"A report on the physical (e.g. structural) or chemical properties of molecules, or parts of a molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2088': {'id': 'EDAM_data:2088',\n",
       "  'name': 'DNA base structural data',\n",
       "  'def': '\"Structural data for DNA base pairs or runs of bases, such as energy or angle data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2090': {'id': 'EDAM_data:2090',\n",
       "  'name': 'Database entry version information',\n",
       "  'def': '\"Information on a database (or ontology) entry version, such as name (or other identifier) or parent database, unique identifier of entry, data, author and so on.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2091': {'id': 'EDAM_data:2091',\n",
       "  'name': 'Accession',\n",
       "  'def': '\"A persistent (stable) and unique identifier, typically identifying an object (entry) from a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2092': {'id': 'EDAM_data:2092',\n",
       "  'name': 'Nucleic acid features (SNP)',\n",
       "  'def': '\"Annotation on a single nucleotide polymorphism (SNP) in a DNA sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2093': {'id': 'EDAM_data:2093',\n",
       "  'name': 'Data reference',\n",
       "  'def': '\"Reference to a dataset (or a cross-reference between two datasets), typically one or more entries in a biological database or ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2098': {'id': 'EDAM_data:2098',\n",
       "  'name': 'Job identifier',\n",
       "  'def': '\"An identifier of a submitted job.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2099': {'id': 'EDAM_data:2099',\n",
       "  'name': 'Name',\n",
       "  'def': '\"A name of a thing, which need not necessarily uniquely identify it.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2100': {'id': 'EDAM_data:2100',\n",
       "  'name': 'Type',\n",
       "  'def': '\"A label (text token) describing the type of a thing, typically an enumerated string (a string with one of a limited set of values).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2101': {'id': 'EDAM_data:2101',\n",
       "  'name': 'User ID',\n",
       "  'def': '\"An identifier of a software end-user (typically a person).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2102': {'id': 'EDAM_data:2102',\n",
       "  'name': 'KEGG organism code',\n",
       "  'def': '\"A three-letter code used in the KEGG databases to uniquely identify organisms.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2103': {'id': 'EDAM_data:2103',\n",
       "  'name': 'Gene name (KEGG GENES)',\n",
       "  'def': '\"Name of an entry (gene) from the KEGG GENES database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2104': {'id': 'EDAM_data:2104',\n",
       "  'name': 'BioCyc ID',\n",
       "  'def': '\"Identifier of an object from one of the BioCyc databases.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2105': {'id': 'EDAM_data:2105',\n",
       "  'name': 'Compound ID (BioCyc)',\n",
       "  'def': '\"Identifier of a compound from the BioCyc chemical compounds database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2106': {'id': 'EDAM_data:2106',\n",
       "  'name': 'Reaction ID (BioCyc)',\n",
       "  'def': '\"Identifier of a biological reaction from the BioCyc reactions database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2107': {'id': 'EDAM_data:2107',\n",
       "  'name': 'Enzyme ID (BioCyc)',\n",
       "  'def': '\"Identifier of an enzyme from the BioCyc enzymes database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2108': {'id': 'EDAM_data:2108',\n",
       "  'name': 'Reaction ID',\n",
       "  'def': '\"Identifier of a biological reaction from a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2109': {'id': 'EDAM_data:2109',\n",
       "  'name': 'Identifier (hybrid)',\n",
       "  'def': '\"An identifier that is re-used for data objects of fundamentally different types (typically served from a single database).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2110': {'id': 'EDAM_data:2110',\n",
       "  'name': 'Molecular property identifier',\n",
       "  'def': '\"Identifier of a molecular property.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2111': {'id': 'EDAM_data:2111',\n",
       "  'name': 'Codon usage table identifier',\n",
       "  'def': '\"Identifier of a codon usage table, for example a genetic code.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2112': {'id': 'EDAM_data:2112',\n",
       "  'name': 'FlyBase primary identifier',\n",
       "  'def': '\"Primary identifier of an object from the FlyBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2113': {'id': 'EDAM_data:2113',\n",
       "  'name': 'WormBase identifier',\n",
       "  'def': '\"Identifier of an object from the WormBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2114': {'id': 'EDAM_data:2114',\n",
       "  'name': 'WormBase wormpep ID',\n",
       "  'def': '\"Protein identifier used by WormBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2116': {'id': 'EDAM_data:2116',\n",
       "  'name': 'Nucleic acid features (codon)',\n",
       "  'def': '\"An informative report on a trinucleotide sequence that encodes an amino acid including the triplet sequence, the encoded amino acid or whether it is a start or stop codon.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2117': {'id': 'EDAM_data:2117',\n",
       "  'name': 'Map identifier',\n",
       "  'def': '\"An identifier of a map of a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2118': {'id': 'EDAM_data:2118',\n",
       "  'name': 'Person identifier',\n",
       "  'def': '\"An identifier of a software end-user (typically a person).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2119': {'id': 'EDAM_data:2119',\n",
       "  'name': 'Nucleic acid identifier',\n",
       "  'def': '\"Name or other identifier of a nucleic acid molecule.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2126': {'id': 'EDAM_data:2126',\n",
       "  'name': 'Translation frame specification',\n",
       "  'def': '\"Frame for translation of DNA (3 forward and 3 reverse frames relative to a chromosome).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2127': {'id': 'EDAM_data:2127',\n",
       "  'name': 'Genetic code identifier',\n",
       "  'def': '\"An identifier of a genetic code.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2128': {'id': 'EDAM_data:2128',\n",
       "  'name': 'Genetic code name',\n",
       "  'def': '\"Informal name for a genetic code, typically an organism name.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2129': {'id': 'EDAM_data:2129',\n",
       "  'name': 'File format name',\n",
       "  'def': '\"Name of a file format such as HTML, PNG, PDF, EMBL, GenBank and so on.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2130': {'id': 'EDAM_data:2130',\n",
       "  'name': 'Sequence profile type',\n",
       "  'def': '\"A label (text token) describing a type of sequence profile such as frequency matrix, Gribskov profile, hidden Markov model etc.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2131': {'id': 'EDAM_data:2131',\n",
       "  'name': 'Operating system name',\n",
       "  'def': '\"Name of a computer operating system such as Linux, PC or Mac.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2132': {'id': 'EDAM_data:2132',\n",
       "  'name': 'Mutation type',\n",
       "  'def': '\"A type of point or block mutation, including insertion, deletion, change, duplication and moves.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2133': {'id': 'EDAM_data:2133',\n",
       "  'name': 'Logical operator',\n",
       "  'def': '\"A logical operator such as OR, AND, XOR, and NOT.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2134': {'id': 'EDAM_data:2134',\n",
       "  'name': 'Results sort order',\n",
       "  'def': '\"A control of the order of data that is output, for example the order of sequences in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2135': {'id': 'EDAM_data:2135',\n",
       "  'name': 'Toggle',\n",
       "  'def': '\"A simple parameter that is a toggle (boolean value), typically a control for a modal tool.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2136': {'id': 'EDAM_data:2136',\n",
       "  'name': 'Sequence width',\n",
       "  'def': '\"The width of an output sequence or alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2137': {'id': 'EDAM_data:2137',\n",
       "  'name': 'Gap penalty',\n",
       "  'def': '\"A penalty for introducing or extending a gap in an alignment.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2139': {'id': 'EDAM_data:2139',\n",
       "  'name': 'Nucleic acid melting temperature',\n",
       "  'def': '\"A temperature concerning nucleic acid denaturation, typically the temperature at which the two strands of a hybridized or double stranded nucleic acid (DNA or RNA/DNA) molecule separate.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2140': {'id': 'EDAM_data:2140',\n",
       "  'name': 'Concentration',\n",
       "  'def': '\"The concentration of a chemical compound.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2141': {'id': 'EDAM_data:2141',\n",
       "  'name': 'Window step size',\n",
       "  'def': '\"Size of the incremental \\'step\\' a sequence window is moved over a sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2142': {'id': 'EDAM_data:2142',\n",
       "  'name': 'EMBOSS graph',\n",
       "  'def': '\"An image of a graph generated by the EMBOSS suite.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2143': {'id': 'EDAM_data:2143',\n",
       "  'name': 'EMBOSS report',\n",
       "  'def': '\"An application report generated by the EMBOSS suite.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2145': {'id': 'EDAM_data:2145',\n",
       "  'name': 'Sequence offset',\n",
       "  'def': '\"An offset for a single-point sequence position.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2146': {'id': 'EDAM_data:2146',\n",
       "  'name': 'Threshold',\n",
       "  'def': '\"A value that serves as a threshold for a tool (usually to control scoring or output).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2147': {'id': 'EDAM_data:2147',\n",
       "  'name': 'Protein report (transcription factor)',\n",
       "  'def': '\"An informative report on a transcription factor protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2149': {'id': 'EDAM_data:2149',\n",
       "  'name': 'Database category name',\n",
       "  'def': '\"The name of a category of biological or bioinformatics database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2150': {'id': 'EDAM_data:2150',\n",
       "  'name': 'Sequence profile name',\n",
       "  'def': '\"Name of a sequence profile.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2151': {'id': 'EDAM_data:2151',\n",
       "  'name': 'Color',\n",
       "  'def': '\"Specification of one or more colors.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2152': {'id': 'EDAM_data:2152',\n",
       "  'name': 'Rendering parameter',\n",
       "  'def': '\"A parameter that is used to control rendering (drawing) to a device or image.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2154': {'id': 'EDAM_data:2154',\n",
       "  'name': 'Sequence name',\n",
       "  'def': '\"Any arbitrary name of a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2156': {'id': 'EDAM_data:2156',\n",
       "  'name': 'Date',\n",
       "  'def': '\"A temporal date.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2157': {'id': 'EDAM_data:2157',\n",
       "  'name': 'Word composition',\n",
       "  'def': '\"Word composition data for a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2160': {'id': 'EDAM_data:2160',\n",
       "  'name': 'Fickett testcode plot',\n",
       "  'def': '\"A plot of Fickett testcode statistic (identifying protein coding regions) in a nucleotide sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2161': {'id': 'EDAM_data:2161',\n",
       "  'name': 'Sequence similarity plot',\n",
       "  'def': '\"A plot of sequence similarities identified from word-matching or character comparison.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2162': {'id': 'EDAM_data:2162',\n",
       "  'name': 'Helical wheel',\n",
       "  'def': '\"An image of peptide sequence sequence looking down the axis of the helix for highlighting amphipathicity and other properties.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2163': {'id': 'EDAM_data:2163',\n",
       "  'name': 'Helical net',\n",
       "  'def': '\"An image of peptide sequence sequence in a simple 3,4,3,4 repeating pattern that emulates at a simple level the arrangement of residues around an alpha helix.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2164': {'id': 'EDAM_data:2164',\n",
       "  'name': 'Protein sequence properties plot',\n",
       "  'def': '\"A plot of general physicochemical properties of a protein sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2165': {'id': 'EDAM_data:2165',\n",
       "  'name': 'Protein ionization curve',\n",
       "  'def': '\"A plot of pK versus pH for a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2166': {'id': 'EDAM_data:2166',\n",
       "  'name': 'Sequence composition plot',\n",
       "  'def': '\"A plot of character or word composition / frequency of a molecular sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2167': {'id': 'EDAM_data:2167',\n",
       "  'name': 'Nucleic acid density plot',\n",
       "  'def': '\"Density plot (of base composition) for a nucleotide sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2168': {'id': 'EDAM_data:2168',\n",
       "  'name': 'Sequence trace image',\n",
       "  'def': '\"Image of a sequence trace (nucleotide sequence versus probabilities of each of the 4 bases).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2169': {'id': 'EDAM_data:2169',\n",
       "  'name': 'Nucleic acid features (siRNA)',\n",
       "  'def': '\"A report on siRNA duplexes in mRNA.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2173': {'id': 'EDAM_data:2173',\n",
       "  'name': 'Sequence set (stream)',\n",
       "  'def': '\"A collection of multiple molecular sequences and (typically) associated metadata that is intended for sequential processing.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2174': {'id': 'EDAM_data:2174',\n",
       "  'name': 'FlyBase secondary identifier',\n",
       "  'def': '\"Secondary identifier of an object from the FlyBase database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2176': {'id': 'EDAM_data:2176',\n",
       "  'name': 'Cardinality',\n",
       "  'def': '\"The number of a certain thing.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2177': {'id': 'EDAM_data:2177',\n",
       "  'name': 'Exactly 1',\n",
       "  'def': '\"A single thing.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2178': {'id': 'EDAM_data:2178',\n",
       "  'name': '1 or more',\n",
       "  'def': '\"One or more things.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2179': {'id': 'EDAM_data:2179',\n",
       "  'name': 'Exactly 2',\n",
       "  'def': '\"Exactly two things.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2180': {'id': 'EDAM_data:2180',\n",
       "  'name': '2 or more',\n",
       "  'def': '\"Two or more things.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2190': {'id': 'EDAM_data:2190',\n",
       "  'name': 'Sequence checksum',\n",
       "  'def': '\"A fixed-size datum calculated (by using a hash function) for a molecular sequence, typically for purposes of error detection or indexing.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2191': {'id': 'EDAM_data:2191',\n",
       "  'name': 'Protein features (chemical modification)',\n",
       "  'def': '\"Data on a chemical modification of a protein.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2192': {'id': 'EDAM_data:2192',\n",
       "  'name': 'Error',\n",
       "  'def': '\"Data on an error generated by computer system or tool.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2193': {'id': 'EDAM_data:2193',\n",
       "  'name': 'Database entry metadata',\n",
       "  'def': '\"Basic information on any arbitrary database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2198': {'id': 'EDAM_data:2198',\n",
       "  'name': 'Gene cluster',\n",
       "  'def': '\"A cluster of similar genes.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2201': {'id': 'EDAM_data:2201',\n",
       "  'name': 'Sequence record full',\n",
       "  'def': '\"A molecular sequence and comprehensive metadata (such as a feature table), typically corresponding to a full entry from a molecular sequence database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2208': {'id': 'EDAM_data:2208',\n",
       "  'name': 'Plasmid identifier',\n",
       "  'def': '\"An identifier of a plasmid in a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2209': {'id': 'EDAM_data:2209',\n",
       "  'name': 'Mutation ID',\n",
       "  'def': '\"A unique identifier of a specific mutation catalogued in a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2212': {'id': 'EDAM_data:2212',\n",
       "  'name': 'Mutation annotation (basic)',\n",
       "  'def': '\"Information describing the mutation itself, the organ site, tissue and type of lesion where the mutation has been identified, description of the patient origin and life-style.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2213': {'id': 'EDAM_data:2213',\n",
       "  'name': 'Mutation annotation (prevalence)',\n",
       "  'def': '\"An informative report on the prevalence of mutation(s), including data on samples and mutation prevalence (e.g. by tumour type)..\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2214': {'id': 'EDAM_data:2214',\n",
       "  'name': 'Mutation annotation (prognostic)',\n",
       "  'def': '\"An informative report on mutation prognostic data, such as information on patient cohort, the study settings and the results of the study.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2215': {'id': 'EDAM_data:2215',\n",
       "  'name': 'Mutation annotation (functional)',\n",
       "  'def': '\"An informative report on the functional properties of mutant proteins including transcriptional activities, promotion of cell growth and tumorigenicity, dominant negative effects, capacity to induce apoptosis, cell-cycle arrest or checkpoints in human cells and so on.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2216': {'id': 'EDAM_data:2216',\n",
       "  'name': 'Codon number',\n",
       "  'def': '\"The number of a codon, for instance, at which a mutation is located.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2217': {'id': 'EDAM_data:2217',\n",
       "  'name': 'Tumor annotation',\n",
       "  'def': '\"An informative report on a specific tumor including nature and origin of the sample, anatomic site, organ or tissue, tumor type, including morphology and/or histologic type, and so on.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2218': {'id': 'EDAM_data:2218',\n",
       "  'name': 'Server metadata',\n",
       "  'def': '\"Basic information about a server on the web, such as an SRS server.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2219': {'id': 'EDAM_data:2219',\n",
       "  'name': 'Database field name',\n",
       "  'def': '\"The name of a field in a database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2220': {'id': 'EDAM_data:2220',\n",
       "  'name': 'Sequence cluster ID (SYSTERS)',\n",
       "  'def': '\"Unique identifier of a sequence cluster from the SYSTERS database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2223': {'id': 'EDAM_data:2223',\n",
       "  'name': 'Ontology metadata',\n",
       "  'def': '\"Data concerning a biological ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2235': {'id': 'EDAM_data:2235',\n",
       "  'name': 'Raw SCOP domain classification',\n",
       "  'def': '\"Raw SCOP domain classification data files.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2236': {'id': 'EDAM_data:2236',\n",
       "  'name': 'Raw CATH domain classification',\n",
       "  'def': '\"Raw CATH domain classification data files.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2240': {'id': 'EDAM_data:2240',\n",
       "  'name': 'Heterogen annotation',\n",
       "  'def': '\"An informative report on the types of small molecules or \\'heterogens\\' (non-protein groups) that are represented in PDB files.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2242': {'id': 'EDAM_data:2242',\n",
       "  'name': 'Phylogenetic property values',\n",
       "  'def': '\"Phylogenetic property values data.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2245': {'id': 'EDAM_data:2245',\n",
       "  'name': 'Sequence set (bootstrapped)',\n",
       "  'def': '\"A collection of sequences output from a bootstrapping (resampling) procedure.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2247': {'id': 'EDAM_data:2247',\n",
       "  'name': 'Phylogenetic consensus tree',\n",
       "  'def': '\"A consensus phylogenetic tree derived from comparison of multiple trees.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2248': {'id': 'EDAM_data:2248',\n",
       "  'name': 'Schema',\n",
       "  'def': '\"A data schema for organising or transforming data of some type.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2249': {'id': 'EDAM_data:2249',\n",
       "  'name': 'DTD',\n",
       "  'def': '\"A DTD (document type definition).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2250': {'id': 'EDAM_data:2250',\n",
       "  'name': 'XML Schema',\n",
       "  'def': '\"An XML Schema.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2251': {'id': 'EDAM_data:2251',\n",
       "  'name': 'Relax-NG schema',\n",
       "  'def': '\"A relax-NG schema.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2252': {'id': 'EDAM_data:2252',\n",
       "  'name': 'XSLT stylesheet',\n",
       "  'def': '\"An XSLT stylesheet.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2253': {'id': 'EDAM_data:2253',\n",
       "  'name': 'Data resource definition name',\n",
       "  'def': '\"The name of a data type.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2254': {'id': 'EDAM_data:2254',\n",
       "  'name': 'OBO file format name',\n",
       "  'def': '\"Name of an OBO file format such as OBO-XML, plain and so on.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2285': {'id': 'EDAM_data:2285',\n",
       "  'name': 'Gene ID (MIPS)',\n",
       "  'def': '\"Identifier for genetic elements in MIPS database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2288': {'id': 'EDAM_data:2288',\n",
       "  'name': 'Sequence identifier (protein)',\n",
       "  'def': '\"An identifier of protein sequence(s) or protein sequence database entries.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2289': {'id': 'EDAM_data:2289',\n",
       "  'name': 'Sequence identifier (nucleic acid)',\n",
       "  'def': '\"An identifier of nucleotide sequence(s) or nucleotide sequence database entries.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2290': {'id': 'EDAM_data:2290',\n",
       "  'name': 'EMBL accession',\n",
       "  'def': '\"An accession number of an entry from the EMBL sequence database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2291': {'id': 'EDAM_data:2291',\n",
       "  'name': 'UniProt ID',\n",
       "  'def': '\"An identifier of a polypeptide in the UniProt database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2292': {'id': 'EDAM_data:2292',\n",
       "  'name': 'GenBank accession',\n",
       "  'def': '\"Accession number of an entry from the GenBank sequence database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2293': {'id': 'EDAM_data:2293',\n",
       "  'name': 'Gramene secondary identifier',\n",
       "  'def': '\"Secondary (internal) identifier of a Gramene database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2294': {'id': 'EDAM_data:2294',\n",
       "  'name': 'Sequence variation ID',\n",
       "  'def': '\"An identifier of an entry from a database of molecular sequence variation.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2295': {'id': 'EDAM_data:2295',\n",
       "  'name': 'Gene ID',\n",
       "  'def': '\"A unique (and typically persistent) identifier of a gene in a database, that is (typically) different to the gene name/symbol.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2296': {'id': 'EDAM_data:2296',\n",
       "  'name': 'Gene name (AceView)',\n",
       "  'def': '\"Name of an entry (gene) from the AceView genes database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2297': {'id': 'EDAM_data:2297',\n",
       "  'name': 'Gene ID (ECK)',\n",
       "  'def': '\"Identifier of an E. coli K-12 gene from EcoGene Database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2298': {'id': 'EDAM_data:2298',\n",
       "  'name': 'Gene ID (HGNC)',\n",
       "  'def': '\"Identifier for a gene approved by the HUGO Gene Nomenclature Committee.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2299': {'id': 'EDAM_data:2299',\n",
       "  'name': 'Gene name',\n",
       "  'def': '\"The name of a gene, (typically) assigned by a person and/or according to a naming scheme. It may contain white space characters and is typically more intuitive and readable than a gene symbol. It (typically) may be used to identify similar genes in different species and to derive a gene symbol.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2300': {'id': 'EDAM_data:2300',\n",
       "  'name': 'Gene name (NCBI)',\n",
       "  'def': '\"Name of an entry (gene) from the NCBI genes database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2301': {'id': 'EDAM_data:2301',\n",
       "  'name': 'SMILES string',\n",
       "  'def': '\"A specification of a chemical structure in SMILES format.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2302': {'id': 'EDAM_data:2302',\n",
       "  'name': 'STRING ID',\n",
       "  'def': '\"Unique identifier of an entry from the STRING database of protein-protein interactions.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2307': {'id': 'EDAM_data:2307',\n",
       "  'name': 'Virus annotation',\n",
       "  'def': '\"An informative report on a specific virus.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2308': {'id': 'EDAM_data:2308',\n",
       "  'name': 'Virus annotation (taxonomy)',\n",
       "  'def': '\"An informative report on the taxonomy of a specific virus.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2309': {'id': 'EDAM_data:2309',\n",
       "  'name': 'Reaction ID (SABIO-RK)',\n",
       "  'def': '\"Identifier of a biological reaction from the SABIO-RK reactions database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2313': {'id': 'EDAM_data:2313',\n",
       "  'name': 'Carbohydrate structure report',\n",
       "  'def': '\"Annotation on or information derived from one or more specific carbohydrate 3D structure(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2314': {'id': 'EDAM_data:2314',\n",
       "  'name': 'GI number',\n",
       "  'def': '\"A series of digits that are assigned consecutively to each sequence record processed by NCBI. The GI number bears no resemblance to the Accession number of the sequence record.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2315': {'id': 'EDAM_data:2315',\n",
       "  'name': 'NCBI version',\n",
       "  'def': '\"An identifier assigned to sequence records processed by NCBI, made of the accession number of the database record followed by a dot and a version number.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2316': {'id': 'EDAM_data:2316',\n",
       "  'name': 'Cell line name',\n",
       "  'def': '\"The name of a cell line.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2317': {'id': 'EDAM_data:2317',\n",
       "  'name': 'Cell line name (exact)',\n",
       "  'def': '\"The name of a cell line.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2318': {'id': 'EDAM_data:2318',\n",
       "  'name': 'Cell line name (truncated)',\n",
       "  'def': '\"The name of a cell line.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2319': {'id': 'EDAM_data:2319',\n",
       "  'name': 'Cell line name (no punctuation)',\n",
       "  'def': '\"The name of a cell line.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2320': {'id': 'EDAM_data:2320',\n",
       "  'name': 'Cell line name (assonant)',\n",
       "  'def': '\"The name of a cell line.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2321': {'id': 'EDAM_data:2321',\n",
       "  'name': 'Enzyme ID',\n",
       "  'def': '\"A unique, persistent identifier of an enzyme.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2325': {'id': 'EDAM_data:2325',\n",
       "  'name': 'REBASE enzyme number',\n",
       "  'def': '\"Identifier of an enzyme from the REBASE enzymes database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2326': {'id': 'EDAM_data:2326',\n",
       "  'name': 'DrugBank ID',\n",
       "  'def': '\"Unique identifier of a drug from the DrugBank database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2327': {'id': 'EDAM_data:2327',\n",
       "  'name': 'GI number (protein)',\n",
       "  'def': '\"A unique identifier assigned to NCBI protein sequence records.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2335': {'id': 'EDAM_data:2335',\n",
       "  'name': 'Bit score',\n",
       "  'def': '\"A score derived from the alignment of two sequences, which is then normalized with respect to the scoring system.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2336': {'id': 'EDAM_data:2336',\n",
       "  'name': 'Translation phase specification',\n",
       "  'def': '\"Phase for translation of DNA (0, 1 or 2) relative to a fragment of the coding sequence.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2337': {'id': 'EDAM_data:2337',\n",
       "  'name': 'Metadata',\n",
       "  'def': '\"Data concerning or describing some core data, as distinct from the primary data that is being described.  This includes metadata on the origin, source, history, ownership or location of some thing.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2338': {'id': 'EDAM_data:2338',\n",
       "  'name': 'Ontology identifier',\n",
       "  'def': '\"Any arbitrary identifier of an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2339': {'id': 'EDAM_data:2339',\n",
       "  'name': 'Ontology concept name',\n",
       "  'def': '\"The name of a concept in an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2340': {'id': 'EDAM_data:2340',\n",
       "  'name': 'Genome build identifier',\n",
       "  'def': '\"An identifier of a build of a particular genome.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2342': {'id': 'EDAM_data:2342',\n",
       "  'name': 'Pathway or network name',\n",
       "  'def': '\"The name of a biological pathway or network.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2343': {'id': 'EDAM_data:2343',\n",
       "  'name': 'Pathway ID (KEGG)',\n",
       "  'def': '\"Identifier of a pathway from the KEGG pathway database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2344': {'id': 'EDAM_data:2344',\n",
       "  'name': 'Pathway ID (NCI-Nature)',\n",
       "  'def': '\"Identifier of a pathway from the NCI-Nature pathway database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2345': {'id': 'EDAM_data:2345',\n",
       "  'name': 'Pathway ID (ConsensusPathDB)',\n",
       "  'def': '\"Identifier of a pathway from the ConsensusPathDB pathway database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2346': {'id': 'EDAM_data:2346',\n",
       "  'name': 'Sequence cluster ID (UniRef)',\n",
       "  'def': '\"Unique identifier of an entry from the UniRef database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2347': {'id': 'EDAM_data:2347',\n",
       "  'name': 'Sequence cluster ID (UniRef100)',\n",
       "  'def': '\"Unique identifier of an entry from the UniRef100 database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2348': {'id': 'EDAM_data:2348',\n",
       "  'name': 'Sequence cluster ID (UniRef90)',\n",
       "  'def': '\"Unique identifier of an entry from the UniRef90 database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2349': {'id': 'EDAM_data:2349',\n",
       "  'name': 'Sequence cluster ID (UniRef50)',\n",
       "  'def': '\"Unique identifier of an entry from the UniRef50 database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2353': {'id': 'EDAM_data:2353',\n",
       "  'name': 'Ontological data',\n",
       "  'def': '\"Data concerning an ontology.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2354': {'id': 'EDAM_data:2354',\n",
       "  'name': 'RNA family annotation',\n",
       "  'def': '\"An informative report on a specific RNA family or other group of classified RNA sequences.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2355': {'id': 'EDAM_data:2355',\n",
       "  'name': 'RNA family identifier',\n",
       "  'def': '\"Identifier of an RNA family, typically an entry from a RNA sequence classification database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2356': {'id': 'EDAM_data:2356',\n",
       "  'name': 'RFAM accession',\n",
       "  'def': '\"Stable accession number of an entry (RNA family) from the RFAM database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2357': {'id': 'EDAM_data:2357',\n",
       "  'name': 'Protein signature type',\n",
       "  'def': '\"A label (text token) describing a type of protein family signature (sequence classifier) from the InterPro database.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2358': {'id': 'EDAM_data:2358',\n",
       "  'name': 'Domain-nucleic acid interaction',\n",
       "  'def': '\"Data on protein domain-DNA/RNA interaction(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2359': {'id': 'EDAM_data:2359',\n",
       "  'name': 'Domain-domain interaction',\n",
       "  'def': '\"Data on protein domain-protein domain interaction(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2360': {'id': 'EDAM_data:2360',\n",
       "  'name': 'Domain-domain interaction (indirect)',\n",
       "  'def': '\"Data on indirect protein domain-protein domain interaction(s).\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2362': {'id': 'EDAM_data:2362',\n",
       "  'name': 'Sequence accession (hybrid)',\n",
       "  'def': '\"Accession number of a nucleotide or protein sequence database entry.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2363': {'id': 'EDAM_data:2363',\n",
       "  'name': '2D PAGE data',\n",
       "  'def': '\"Data concerning two-dimensional polygel electrophoresis.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2364': {'id': 'EDAM_data:2364',\n",
       "  'name': 'Experiment annotation (2D PAGE)',\n",
       "  'def': '\"An informative report on a two-dimensional gel electrophoresis experiment, gel or spots in a gel.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2365': {'id': 'EDAM_data:2365',\n",
       "  'name': 'Pathway or network accession',\n",
       "  'def': '\"A persistent, unique identifier of a biological pathway or network (typically a database entry).\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2366': {'id': 'EDAM_data:2366',\n",
       "  'name': 'Secondary structure alignment',\n",
       "  'def': '\"Alignment of the (1D representations of) secondary structure of two or more molecules.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2367': {'id': 'EDAM_data:2367',\n",
       "  'name': 'ASTD ID',\n",
       "  'def': '\"Identifier of an object from the ASTD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2368': {'id': 'EDAM_data:2368',\n",
       "  'name': 'ASTD ID (exon)',\n",
       "  'def': '\"Identifier of an exon from the ASTD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2369': {'id': 'EDAM_data:2369',\n",
       "  'name': 'ASTD ID (intron)',\n",
       "  'def': '\"Identifier of an intron from the ASTD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2370': {'id': 'EDAM_data:2370',\n",
       "  'name': 'ASTD ID (polya)',\n",
       "  'def': '\"Identifier of a polyA signal from the ASTD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2371': {'id': 'EDAM_data:2371',\n",
       "  'name': 'ASTD ID (tss)',\n",
       "  'def': '\"Identifier of a transcription start site from the ASTD database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2372': {'id': 'EDAM_data:2372',\n",
       "  'name': '2D PAGE spot (annotated)',\n",
       "  'def': '\"An informative report on individual spot(s) from a two-dimensional (2D PAGE) gel.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2373': {'id': 'EDAM_data:2373',\n",
       "  'name': 'Spot ID',\n",
       "  'def': '\"Unique identifier of a spot from a two-dimensional (protein) gel.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2374': {'id': 'EDAM_data:2374',\n",
       "  'name': 'Spot serial number',\n",
       "  'def': '\"Unique identifier of a spot from a two-dimensional (protein) gel in the SWISS-2DPAGE database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2375': {'id': 'EDAM_data:2375',\n",
       "  'name': 'Spot ID (HSC-2DPAGE)',\n",
       "  'def': '\"Unique identifier of a spot from a two-dimensional (protein) gel from a HSC-2DPAGE database.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " 'EDAM_data:2378': {'id': 'EDAM_data:2378',\n",
       "  'name': 'Protein-motif interaction',\n",
       "  'def': '\"Data on the interaction of a protein (or protein domain) with specific structural (3D) and/or sequence motifs.\" [http://edamontology.org]',\n",
       "  'namespace': 'data'},\n",
       " 'EDAM_data:2379': {'id': 'EDAM_data:2379',\n",
       "  'name': 'Strain identifier',\n",
       "  'def': '\"Identifier of a strain of an organism variant, typically a plant, virus or bacterium.\" [http://edamontology.org]',\n",
       "  'namespace': 'identifier'},\n",
       " ...}"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1df1f35c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Test concept extract\n",
    "from bio_mcp.cache.embeddings import extract_concepts_from_biotools_yaml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6b95bfd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "concepts = extract_concepts_from_biotools_yaml(yaml_path=Path(\"/home/ubuntu/cache/johan.yaml\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a6b174eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concepts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f65b3f2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bio-mcp (3.10.12)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
