{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4ca7e700",
   "metadata": {},
   "outputs": [],
   "source": [
    "from bio_mcp.cache.load import load_biotools, load_galaxy_singularity\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03258a05",
   "metadata": {},
   "source": [
    "## Loading snapshots\n",
    "\n",
    "Galaxy containers from the CVMFS and tool metadata from e.g. biotools are indexed by lowercased tool names so they can be mapped together."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e4034b2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "bt = load_biotools(Path(\"/home/ubuntu/bio-mcp/data/scrnaseq_biotools.yaml\"))\n",
    "gx = load_galaxy_singularity(Path(\"/home/ubuntu/bio-mcp/data/scrnaseq_galaxy_cvmfs.json\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "467ed279",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'dict' object has no attribute 'head'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mgx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhead\u001b[49m()\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'dict' object has no attribute 'head'"
     ]
    }
   ],
   "source": [
    "gx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fff19532",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'cellbender': {'biocontainers': None,\n",
       "  'biotools': 'CellBender',\n",
       "  'bunya': ['0.3.0'],\n",
       "  'description': 'a deep generative model for unsupervised removal of background noise from scRNA-seq datasets.\\n\\nCellBender is a software package for eliminating technical artifacts from high-throughput single-cell RNA sequencing (scRNA-seq) data.\\n\\nWelcome to CellBender’s documentation! — CellBender documentation.\\n\\nFree document hosting provided by Read the Docs.\\n\\nStephen J Fleming, John C Marioni, and Mehrtash Babadi. CellBender remove-background: a deep generative model for unsupervised removal of background noise from scRNA-seq datasets. bioRxiv 791699; doi: https://doi.org/10.1101/791699.',\n",
       "  'edam-inputs': [],\n",
       "  'edam-operations': ['Quantification',\n",
       "   'Data retrieval',\n",
       "   'Expression analysis'],\n",
       "  'edam-outputs': [],\n",
       "  'edam-topics': ['Gene expression', 'RNA', 'RNA-Seq'],\n",
       "  'galaxy': None,\n",
       "  'homepage': 'https://github.com/broadinstitute/CellBender',\n",
       "  'id': 'cellbender',\n",
       "  'license': 'BSD-3-Clause',\n",
       "  'name': 'CellBender',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': None,\n",
       "  'pawsey': None,\n",
       "  'publications': [{'title': '10.1101/791699',\n",
       "    'url': 'https://doi.org/10.1101/791699'}],\n",
       "  'resources': [{'title': nan, 'url': nan}]},\n",
       " 'cellranger': {'biocontainers': None,\n",
       "  'biotools': None,\n",
       "  'bunya': ['2.0.2', '7.1.0'],\n",
       "  'description': None,\n",
       "  'edam-inputs': None,\n",
       "  'edam-operations': None,\n",
       "  'edam-outputs': None,\n",
       "  'edam-topics': None,\n",
       "  'galaxy': None,\n",
       "  'homepage': 'https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger',\n",
       "  'id': 'cellranger',\n",
       "  'license': None,\n",
       "  'name': 'cellranger',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': ['6.1.2'],\n",
       "  'pawsey': None,\n",
       "  'publications': None,\n",
       "  'resources': [{'title': nan, 'url': nan}]},\n",
       " 'fastqc': {'biocontainers': 'fastqc',\n",
       "  'biotools': 'fastqc',\n",
       "  'bunya': ['0.11.9-java-11'],\n",
       "  'description': 'This tool aims to provide a QC report which can spot problems or biases which originate either in the sequencer or in the starting library material. It can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.',\n",
       "  'edam-inputs': [{'formats': ['FASTQ-like format (text)',\n",
       "     'SAM',\n",
       "     'FASTQ',\n",
       "     'BAM'],\n",
       "    'term': 'BAM'}],\n",
       "  'edam-operations': ['Sequence composition calculation',\n",
       "   'Sequencing quality control',\n",
       "   'Statistical calculation'],\n",
       "  'edam-outputs': [{'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam-topics': ['Sequencing',\n",
       "   'Data quality management',\n",
       "   'Sequence analysis'],\n",
       "  'galaxy': [{'description': 'FastQC: Read Quality reports',\n",
       "    'title': 'FastQC 0.74+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastqc%2Ffastqc%2F0.74%2Bgalaxy1'}],\n",
       "  'homepage': 'http://www.bioinformatics.babraham.ac.uk/projects/fastqc/',\n",
       "  'id': 'fastqc',\n",
       "  'license': 'GPL-3.0',\n",
       "  'name': 'FASTQC',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': ['0.12.1'],\n",
       "  'pawsey': ['0.11.9--hdfd78af_1'],\n",
       "  'publications': [{'title': '10.7490/f1000research.1114334.1',\n",
       "    'url': 'https://doi.org/10.7490/f1000research.1114334.1'}],\n",
       "  'resources': [{'title': nan, 'url': nan}]},\n",
       " 'multiqc': {'biocontainers': 'multiqc',\n",
       "  'biotools': 'multiqc',\n",
       "  'bunya': ['1.11-foss-2021a', '1.14-foss-2022a                   (D)'],\n",
       "  'description': \"MultiQC aggregates results from multiple bioinformatics analyses across many samples into a single report. It searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools.\",\n",
       "  'edam-inputs': [],\n",
       "  'edam-operations': ['Validation', 'Sequencing quality control'],\n",
       "  'edam-outputs': [{'formats': ['CSV', 'JSON', 'HTML', 'YAML', 'TSV'],\n",
       "    'term': 'TSV'},\n",
       "   {'formats': ['HTML'], 'term': 'HTML'}],\n",
       "  'edam-topics': ['Sequencing',\n",
       "   'Bioinformatics',\n",
       "   'Sequence analysis',\n",
       "   'Genomics'],\n",
       "  'galaxy': [{'description': 'MultiQC: aggregate results from bioinformatics analyses into a single report',\n",
       "    'title': 'MultiQC 1.33+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmultiqc%2Fmultiqc%2F1.33%2Bgalaxy0'}],\n",
       "  'homepage': 'http://multiqc.info/',\n",
       "  'id': 'multiqc',\n",
       "  'license': 'GPL-3.0',\n",
       "  'name': 'MultiQC',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': None,\n",
       "  'pawsey': None,\n",
       "  'publications': [{'title': 'MultiQC: Summarize analysis results for multiple tools and samples in a single report',\n",
       "    'url': 'https://doi.org/10.1093/bioinformatics/btw354'}],\n",
       "  'resources': [{'title': nan, 'url': nan}]},\n",
       " 'scanpy': {'biocontainers': None,\n",
       "  'biotools': 'scanpy',\n",
       "  'bunya': None,\n",
       "  'description': 'Scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.',\n",
       "  'edam-inputs': [{'formats': ['HDF5'], 'term': 'HDF5'}],\n",
       "  'edam-operations': ['Differential gene expression analysis',\n",
       "   'Principal component visualisation',\n",
       "   'Expression profile clustering'],\n",
       "  'edam-outputs': [{'formats': ['HDF5'], 'term': 'HDF5'}],\n",
       "  'edam-topics': ['Gene expression',\n",
       "   'Cell biology',\n",
       "   'Genetics',\n",
       "   'Single-cell sequencing'],\n",
       "  'galaxy': [{'description': 'Scanpy normalize: and impute',\n",
       "    'title': 'Scanpy normalize 1.10.2+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fscanpy_normalize%2Fscanpy_normalize%2F1.10.2%2Bgalaxy3'},\n",
       "   {'description': 'Scanpy remove confounders: with scanpy',\n",
       "    'title': 'Scanpy remove confounders 1.10.2+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fscanpy_remove_confounders%2Fscanpy_remove_confounders%2F1.10.2%2Bgalaxy3'},\n",
       "   {'description': 'Scanpy plot: ',\n",
       "    'title': 'Scanpy plot 1.10.2+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fscanpy_plot%2Fscanpy_plot%2F1.10.2%2Bgalaxy3'},\n",
       "   {'description': 'Scanpy cluster, embed: and infer trajectories',\n",
       "    'title': 'Scanpy cluster, embed 1.10.2+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fscanpy_cluster_reduce_dimension%2Fscanpy_cluster_reduce_dimension%2F1.10.2%2Bgalaxy3'},\n",
       "   {'description': 'Scanpy Inspect and manipulate: ',\n",
       "    'title': 'Scanpy Inspect and manipulate 1.10.2+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fscanpy_inspect%2Fscanpy_inspect%2F1.10.2%2Bgalaxy3'},\n",
       "   {'description': 'Scanpy filter: mark and subsample',\n",
       "    'title': 'Scanpy filter 1.10.2+galaxy3',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fscanpy_filter%2Fscanpy_filter%2F1.10.2%2Bgalaxy3'},\n",
       "   {'description': 'AnnData Operations: modifies metadata and flags genes',\n",
       "    'title': 'AnnData Operations 1.8.1+galaxy93',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fanndata_ops%2Fanndata_ops%2F1.8.1%2Bgalaxy93'},\n",
       "   {'description': 'Scanpy ScaleData: to make expression variance the same for all genes',\n",
       "    'title': 'Scanpy ScaleData 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_scale_data%2Fscanpy_scale_data%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy RunUMAP: visualise cell clusters using UMAP',\n",
       "    'title': 'Scanpy RunUMAP 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_umap%2Fscanpy_run_umap%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy RunTSNE: visualise cell clusters using tSNE',\n",
       "    'title': 'Scanpy RunTSNE 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_tsne%2Fscanpy_run_tsne%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy RunPCA: for dimensionality reduction',\n",
       "    'title': 'Scanpy RunPCA 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_pca%2Fscanpy_run_pca%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy RegressOut: variables that might introduce batch effect',\n",
       "    'title': 'Scanpy RegressOut 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_regress_variable%2Fscanpy_regress_variable%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy PlotEmbed: visualise cell embeddings',\n",
       "    'title': 'Scanpy PlotEmbed 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_embed%2Fscanpy_plot_embed%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy NormaliseData: to make all cells having the same total expression',\n",
       "    'title': 'Scanpy NormaliseData 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_normalise_data%2Fscanpy_normalise_data%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy FindVariableGenes: based on normalised dispersion of expression',\n",
       "    'title': 'Scanpy FindVariableGenes 1.9.3+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_variable_genes%2Fscanpy_find_variable_genes%2F1.9.3%2Bgalaxy1'},\n",
       "   {'description': 'Scanpy FindMarkers: to find differentially expressed genes between groups',\n",
       "    'title': 'Scanpy FindMarkers 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_markers%2Fscanpy_find_markers%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy FindCluster: based on community detection on KNN graph',\n",
       "    'title': 'Scanpy FindCluster 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_find_cluster%2Fscanpy_find_cluster%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy FilterGenes: based on counts and numbers of cells expressed',\n",
       "    'title': 'Scanpy FilterGenes 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_filter_genes%2Fscanpy_filter_genes%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy FilterCells: based on counts and numbers of genes expressed',\n",
       "    'title': 'Scanpy FilterCells 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_filter_cells%2Fscanpy_filter_cells%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy ComputeGraph: to derive kNN graph',\n",
       "    'title': 'Scanpy ComputeGraph 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_compute_graph%2Fscanpy_compute_graph%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy PAGA: trajectory inference',\n",
       "    'title': 'Scanpy PAGA 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_paga%2Fscanpy_run_paga%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy RunFDG: visualise cell clusters using force-directed graph',\n",
       "    'title': 'Scanpy RunFDG 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_fdg%2Fscanpy_run_fdg%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy DPT: diffusion pseudotime inference',\n",
       "    'title': 'Scanpy DPT 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_dpt%2Fscanpy_run_dpt%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy DiffusionMap: calculate diffusion components',\n",
       "    'title': 'Scanpy DiffusionMap 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_run_diffmap%2Fscanpy_run_diffmap%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy Read10x: into hdf5 object handled by scanpy',\n",
       "    'title': 'Scanpy Read10x 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_read_10x%2Fscanpy_read_10x%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy PlotTrajectory: visualise cell trajectories',\n",
       "    'title': 'Scanpy PlotTrajectory 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_trajectory%2Fscanpy_plot_trajectory%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy ParameterIterator: produce an iteration over a defined parameter',\n",
       "    'title': 'Scanpy ParameterIterator 0.0.1+galaxy9',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_parameter_iterator%2Fscanpy_parameter_iterator%2F0.0.1%2Bgalaxy9'},\n",
       "   {'description': 'Scanpy MNN: correct batch effects by matching mutual nearest neighbors',\n",
       "    'title': 'Scanpy MNN 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_mnn%2Fscanpy_integrate_mnn%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy ComBat: adjust expression for variables that might introduce batch effect',\n",
       "    'title': 'Scanpy ComBat 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_combat%2Fscanpy_integrate_combat%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'AnnData Operations: is a Swiss army knife for AnnData files',\n",
       "    'title': 'AnnData Operations 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fanndata_ops%2Fanndata_ops%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy BBKNN: batch-balanced K-nearest neighbours',\n",
       "    'title': 'Scanpy BBKNN 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_bbknn%2Fscanpy_integrate_bbknn%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy Harmony: adjust principal components for variables that might introduce batch effect',\n",
       "    'title': 'Scanpy Harmony 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_integrate_harmony%2Fscanpy_integrate_harmony%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy Scrublet: remove multiplets from annData objects with Scrublet',\n",
       "    'title': 'Scanpy Scrublet 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_multiplet_scrublet%2Fscanpy_multiplet_scrublet%2F1.9.3%2Bgalaxy0'},\n",
       "   {'description': 'Scanpy Plot Scrublet: visualise multiplet scoring distribution',\n",
       "    'title': 'Scanpy Plot Scrublet 1.9.3+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fscanpy_plot_scrublet%2Fscanpy_plot_scrublet%2F1.9.3%2Bgalaxy0'}],\n",
       "  'homepage': 'https://github.com/theislab/Scanpy',\n",
       "  'id': 'scanpy',\n",
       "  'license': 'BSD-3-Clause',\n",
       "  'name': 'scanpy',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': None,\n",
       "  'pawsey': None,\n",
       "  'publications': [{'title': 'SCANPY: Large-scale single-cell gene expression data analysis',\n",
       "    'url': 'https://doi.org/10.1186/s13059-017-1382-0'}],\n",
       "  'resources': [{'title': nan, 'url': nan}]},\n",
       " 'seurat': {'biocontainers': None,\n",
       "  'biotools': 'seurat',\n",
       "  'bunya': None,\n",
       "  'description': 'Seurat is an R package designed for QC, analysis, and exploration of single-cell RNA-seq data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data.',\n",
       "  'edam-inputs': None,\n",
       "  'edam-operations': None,\n",
       "  'edam-outputs': None,\n",
       "  'edam-topics': ['RNA-Seq', 'Transcriptomics'],\n",
       "  'galaxy': [{'description': 'Seurat: - toolkit for exploration of single-cell RNA-seq data',\n",
       "    'title': 'Seurat 4.3.0.1+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat%2Fseurat%2F4.3.0.1%2Bgalaxy1'},\n",
       "   {'description': 'Seurat Run Dimensional Reduction: - PCA, tSNE or UMAP',\n",
       "    'title': 'Seurat Run Dimensional Reduction 5.0+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_reduce_dimension%2Fseurat_reduce_dimension%2F5.0%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Preprocessing: - Normalize, Find Variable Features, Scale and Regress',\n",
       "    'title': 'Seurat Preprocessing 5.0+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_preprocessing%2Fseurat_preprocessing%2F5.0%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Integrate: and manipulate layers',\n",
       "    'title': 'Seurat Integrate 5.0+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_integrate%2Fseurat_integrate%2F5.0%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Data Management: - Inspect and Manipulate',\n",
       "    'title': 'Seurat Data Management 5.0+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_data%2Fseurat_data%2F5.0%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Create: - Prepare data for the pipeline',\n",
       "    'title': 'Seurat Create 5.0+galaxy1',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_create%2Fseurat_create%2F5.0%2Bgalaxy1'},\n",
       "   {'description': 'Seurat Find Clusters: - Neighbors and Markers',\n",
       "    'title': 'Seurat Find Clusters 5.0+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_clustering%2Fseurat_clustering%2F5.0%2Bgalaxy0'},\n",
       "   {'description': 'Seurat ScaleData: scale and center genes',\n",
       "    'title': 'Seurat ScaleData 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_scale_data%2Fseurat_scale_data%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat RunTSNE: run t-SNE dimensionality reduction',\n",
       "    'title': 'Seurat RunTSNE 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_tsne%2Fseurat_run_tsne%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat RunPCA: run a PCA dimensionality reduction',\n",
       "    'title': 'Seurat RunPCA 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_pca%2Fseurat_run_pca%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Read10x: Loads Tabular or 10x data into a serialized seurat R object',\n",
       "    'title': 'Seurat Read10x 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_read10x%2Fseurat_read10x%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Plot: with Seurat',\n",
       "    'title': 'Plot 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_plot%2Fseurat_plot%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat FindVariableGenes: identify variable genes',\n",
       "    'title': 'Seurat FindVariableGenes 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_variable_genes%2Fseurat_find_variable_genes%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat FindNeighbours: constructs a Shared Nearest Neighbor (SNN) Graph',\n",
       "    'title': 'Seurat FindNeighbours 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_neighbours%2Fseurat_find_neighbours%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat FindMarkers: find markers (differentially expressed genes)',\n",
       "    'title': 'Seurat FindMarkers 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_markers%2Fseurat_find_markers%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat FilterCells: filter cells in a Seurat object',\n",
       "    'title': 'Seurat FilterCells 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_filter_cells%2Fseurat_filter_cells%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Export2CellBrowser: produces files for UCSC CellBrowser import.',\n",
       "    'title': 'Seurat Export2CellBrowser 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_export_cellbrowser%2Fseurat_export_cellbrowser%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat CreateSeuratObject: create a Seurat object',\n",
       "    'title': 'Seurat CreateSeuratObject 2.3.1+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_create_seurat_object%2Fseurat_create_seurat_object%2F2.3.1%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Plot dimension reduction: graphs the output of a dimensional reduction technique (PCA by default). Cells are colored by their identity class.',\n",
       "    'title': 'Seurat Plot dimension reduction 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_dim_plot%2Fseurat_dim_plot%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat FindClusters: find clusters of cells',\n",
       "    'title': 'Seurat FindClusters 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_find_clusters%2Fseurat_find_clusters%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat NormaliseData: normalise data',\n",
       "    'title': 'Seurat NormaliseData 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_normalise_data%2Fseurat_normalise_data%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat UMAP: dimensionality reduction',\n",
       "    'title': 'Seurat UMAP 4.0.4+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Febi-gxa%2Fseurat_run_umap%2Fseurat_run_umap%2F4.0.4%2Bgalaxy0'},\n",
       "   {'description': 'Seurat Visualize: - Plot cells, features and dimensions',\n",
       "    'title': 'Seurat Visualize 5.0+galaxy0',\n",
       "    'url': 'https://usegalaxy.org.au/root?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fseurat_plot%2Fseurat_plot%2F5.0%2Bgalaxy0'}],\n",
       "  'homepage': 'https://satijalab.org/seurat/',\n",
       "  'id': 'seurat',\n",
       "  'license': 'MIT',\n",
       "  'name': 'seurat',\n",
       "  'nci-gadi': None,\n",
       "  'nci-if89': None,\n",
       "  'pawsey': None,\n",
       "  'publications': [{'title': 'Integrated analysis of multimodal single-cell data',\n",
       "    'url': 'https://doi.org/10.1016/j.cell.2021.04.048'}],\n",
       "  'resources': [{'title': nan, 'url': nan}]}}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ff9f1c6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "376b6ea0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bio-mcp (3.10.12)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
